// @guestyci/foundation/FlatButton/FlatButton.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.useStyles = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _noop = _interopRequireDefault(require("lodash/noop"));

var _utils = require("../theme/utils");

var _colors = require("../theme/colors");

var _createStyles = _interopRequireDefault(require("../createStyles"));

var _Button = _interopRequireDefault(require("../Button"));

var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    root: {
      minWidth: 95
    },
    secondary: {
      color: theme.palette.secondary["default"],
      boxShadow: '0 0 3px 0 rgba(0, 2, 3, 0.25)',
      background: _colors.white,
      '&:hover': {
        color: theme.palette.primary["default"]
      },
      '&:disabled': {
        color: (0, _utils.fade)(theme.palette.secondary["default"], 0.3)
      }
    },
    error: {
      color: _colors.white,
      backgroundColor: theme.palette.error["default"],
      '&:not([disabled]):hover': {
        backgroundColor: function backgroundColor(_ref) {
          var isLoading = _ref.isLoading;
          return (0, _utils.fade)(theme.palette.error["default"], isLoading ? 1 : 0.85);
        }
      },
      '&:disabled': {
        backgroundColor: (0, _utils.fade)(theme.palette.error["default"], 0.7),
        color: (0, _utils.fade)(_colors.white, 0.8)
      }
    },
    success: {
      color: _colors.white,
      backgroundColor: theme.palette.success["default"],
      '&:not([disabled]):hover': {
        backgroundColor: function backgroundColor(_ref2) {
          var isLoading = _ref2.isLoading;
          return (0, _utils.fade)(theme.palette.success["default"], isLoading ? 1 : 0.85);
        }
      },
      '&:disabled': {
        backgroundColor: (0, _utils.fade)(theme.palette.success["default"], 0.7),
        color: (0, _utils.fade)(_colors.white, 0.8)
      }
    }
  };
}, {
  name: 'FlatButton'
});
exports.useStyles = useStyles;

var FlatButton = _react["default"].forwardRef(function FlatButton(_ref3, ref) {
  var _cn;

  var children = _ref3.children,
      size = _ref3.size,
      isLoading = _ref3.isLoading,
      onClick = _ref3.onClick,
      disabled = _ref3.disabled,
      type = _ref3.type,
      className = _ref3.className,
      jss = _ref3.jss,
      props = (0, _objectWithoutProperties2["default"])(_ref3, ["children", "size", "isLoading", "onClick", "disabled", "type", "className", "jss"]);
  var classes = useStyles({
    isLoading: isLoading,
    jss: jss
  });
  var spinnerColor = type !== null ? 'white' : 'blue';
  var rippleColor = type !== null ? _colors.white : _colors.gray400;
  return /*#__PURE__*/_react["default"].createElement(_Button["default"], (0, _extends2["default"])({}, props, {
    disabled: disabled,
    isLoading: isLoading,
    onClick: onClick,
    size: size,
    rippleColor: rippleColor,
    spinnerColor: spinnerColor,
    className: (0, _classnames["default"])(classes.root, (_cn = {}, (0, _defineProperty2["default"])(_cn, classes[type], type !== null), (0, _defineProperty2["default"])(_cn, classes.secondary, type === null), _cn), className),
    ref: ref
  }), children);
});

FlatButton.defaultProps = {
  size: 'md',
  type: null,
  onClick: _noop["default"],
  isLoading: false,
  disabled: false
};
FlatButton.propTypes = {
  /** Child element to locate inside of button */
  children: _propTypes["default"].node.isRequired,

  /** callback to handle click event for button (evnt) => {} */
  onClick: _propTypes["default"].func,

  /** Button size - values =['md', 'lg', 'xl'] */
  size: _propTypes["default"].oneOf(['md', 'lg', 'xl']),

  /** Button type- values =['primary', 'secondary'] */
  type: _propTypes["default"].oneOf(['error', 'success']),

  /** Indicator whether or not to show loader in button */
  isLoading: _propTypes["default"].bool,

  /** Indicator whether or not to button is disabled */
  disabled: _propTypes["default"].bool,

  /** jss override object to customize the jss classes */
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func])
};
FlatButton.displayName = 'FlatButton';
var _default = FlatButton;
exports["default"] = _default;
