// @guestyci/foundation/esm/legacy/Popover/withOverflowPopoverState.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _possibleConstructorReturn from "@babel/runtime/helpers/esm/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/esm/getPrototypeOf";

function _createSuper(Derived) { return function () { var Super = _getPrototypeOf(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

import React from 'react';

var withOverflowPopoverState = function withOverflowPopoverState(WrappedComponent) {
  var WithOverflowState = /*#__PURE__*/function (_React$Component) {
    _inherits(WithOverflowState, _React$Component);

    var _super = _createSuper(WithOverflowState);

    function WithOverflowState(props) {
      var _this;

      _classCallCheck(this, WithOverflowState);

      _this = _super.call(this, props);

      _this.onMouseEnter = function () {
        var _this$ref$current = _this.ref.current,
            clientWidth = _this$ref$current.clientWidth,
            scrollWidth = _this$ref$current.scrollWidth;

        if (scrollWidth > clientWidth) {
          _this.setState({
            showPopover: true
          });
        }
      };

      _this.onMouseLeave = function () {
        _this.setState({
          showPopover: false
        });
      };

      _this.state = {
        showPopover: false
      };
      _this.ref = React.createRef();
      return _this;
    }

    _createClass(WithOverflowState, [{
      key: "render",
      value: function render() {
        var showPopover = this.state.showPopover;
        return /*#__PURE__*/React.createElement("div", {
          className: "w-fill",
          ref: this.ref,
          onMouseEnter: this.onMouseEnter,
          onMouseLeave: this.onMouseLeave,
          "data-qa": "with-overflow-state"
        }, /*#__PURE__*/React.createElement(WrappedComponent, _extends({
          showPopover: showPopover
        }, this.props)));
      }
    }]);

    return WithOverflowState;
  }(React.Component);

  WithOverflowState.displayName = "WithOverflowState(".concat(WrappedComponent.displayName || WrappedComponent.name || 'Component', ")");
  return WithOverflowState;
};

export default withOverflowPopoverState;
