// @guestyci/foundation/legacy/Input/Input.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _noop = _interopRequireDefault(require("lodash/noop"));

var _classnames = _interopRequireDefault(require("classnames"));

var _InputLabel = _interopRequireDefault(require("../InputLabel/InputLabel"));

function _createSuper(Derived) { return function () { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

var containerStyle = {
  height: 46,
  borderRadius: 2,
  outline: 'none'
};

var Input = /*#__PURE__*/function (_Component) {
  (0, _inherits2["default"])(Input, _Component);

  var _super = _createSuper(Input);

  function Input() {
    var _this;

    (0, _classCallCheck2["default"])(this, Input);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _super.call.apply(_super, [this].concat(args));

    _this.handleInputChange = function (e) {
      var target = e.target;
      var _this$props = _this.props,
          id = _this$props.id,
          name = _this$props.name,
          onChange = _this$props.onChange;

      if (onChange) {
        onChange(target.value, id || name);
      }
    };

    return _this;
  }

  (0, _createClass2["default"])(Input, [{
    key: "render",
    value: function render() {
      var _this$props2 = this.props,
          value = _this$props2.value,
          className = _this$props2.className,
          style = _this$props2.style,
          placeholder = _this$props2.placeholder,
          disabled = _this$props2.disabled,
          suffix = _this$props2.suffix,
          containerClassName = _this$props2.containerClassName,
          label = _this$props2.label,
          tooltip = _this$props2.tooltip,
          type = _this$props2.type,
          onChange = _this$props2.onChange,
          required = _this$props2.required,
          props = (0, _objectWithoutProperties2["default"])(_this$props2, ["value", "className", "style", "placeholder", "disabled", "suffix", "containerClassName", "label", "tooltip", "type", "onChange", "required"]);
      return /*#__PURE__*/_react["default"].createElement("div", {
        className: "d-flex-fill flex-column input-wrapper",
        "data-qa": "input"
      }, label && /*#__PURE__*/_react["default"].createElement(_InputLabel["default"], {
        label: label,
        tooltip: tooltip
      }), /*#__PURE__*/_react["default"].createElement("div", {
        className: (0, _classnames["default"])('d-flex-fill flex-start-center border-gray-lighter b-1', "bg-gray-".concat(disabled ? 'lighter' : 'lightest'), containerClassName),
        style: containerStyle
      }, /*#__PURE__*/_react["default"].createElement("input", (0, _extends2["default"])({
        onChange: this.handleInputChange,
        type: type,
        value: value,
        className: (0, _classnames["default"])('py-2', 'px-4', 'd-flex-fill', 'bg-transparent', 'b-0', 'font-size-md', className),
        disabled: disabled,
        style: style,
        placeholder: placeholder
      }, props)), suffix && /*#__PURE__*/_react["default"].createElement("div", {
        className: "font-size-md input-suffix cursor-regular pr-4 h-fill flex-center"
      }, suffix)));
    }
  }]);
  return Input;
}(_react.Component);

Input.defaultProps = {
  onChange: _noop["default"],
  value: undefined,
  placeholder: '',
  className: '',
  style: undefined,
  suffix: null,
  label: undefined,
  tooltip: undefined,
  type: 'text',
  disabled: false,
  containerClassName: ''
};
Input.propTypes = {
  /** Controlled input value to be set to input */
  value: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].number]),

  /** Function to be called on input value change -
   params value: {String}, id|name :{String|number}
   */
  onChange: _propTypes["default"].func,

  /** Input type */
  type: _propTypes["default"].string,

  /** Input Placeholder */
  placeholder: _propTypes["default"].string,

  /** Input class */
  className: _propTypes["default"].string,

  /** Input style */
  style: _propTypes["default"].shape(),

  /** String to be shown after the input */
  suffix: _propTypes["default"].node,

  /** A label to show on top of the input */
  label: _propTypes["default"].string,

  /** Add tooltip help icon and message - tooltip - String help message */
  tooltip: _propTypes["default"].string,

  /** Indicator for disabled status */
  disabled: _propTypes["default"].bool,

  /** Additional class for input wrapper level */
  containerClassName: _propTypes["default"].string
};
var _default = Input;
exports["default"] = _default;
