// @guestyci/foundation/Layout/Col/Col.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.useStyles = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _createStyles = _interopRequireDefault(require("../../createStyles"));

var _utils = require("../../utils");

var _styles = _interopRequireWildcard(require("../styles"));

var useStyles = (0, _createStyles["default"])(function (theme) {
  var breakpoints = (0, _styles.generateMediaSupport)(theme.breakpoints);
  var spacing = (0, _styles.generateSpacing)(theme.spacer, _styles.spacing, 'bottom');
  return (0, _extends2["default"])({
    root: {
      display: 'flex'
    },
    flexDirectionColumn: {
      flexDirection: 'column'
    },
    flexDirectionColumnReverse: {
      flexDirection: 'column-reverse'
    },
    lastChildSpacing: {
      '& > *:last-child': {
        marginBottom: 0
      }
    }
  }, _styles["default"], {}, breakpoints, {}, spacing);
}, {
  name: 'Col'
});
exports.useStyles = useStyles;

var Col = _react["default"].forwardRef(function Col(_ref, ref) {
  var _cn;

  var className = _ref.className,
      style = _ref.style,
      children = _ref.children,
      wrap = _ref.wrap,
      breakpoints = _ref.breakpoints,
      span = _ref.span,
      reverse = _ref.reverse,
      align = _ref.align,
      justify = _ref.justify,
      fullWidth = _ref.fullWidth,
      spacing = _ref.spacing,
      _ref$jss = _ref.jss,
      jss = _ref$jss === void 0 ? {} : _ref$jss,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["className", "style", "children", "wrap", "breakpoints", "span", "reverse", "align", "justify", "fullWidth", "spacing", "jss"]);
  var classes = useStyles({
    jss: jss
  });

  var _ref2 = breakpoints || {},
      xs = _ref2.xs,
      sm = _ref2.sm,
      md = _ref2.md,
      lg = _ref2.lg,
      xl = _ref2.xl,
      xlg = _ref2.xlg;

  var xsSize = span || xs;
  return /*#__PURE__*/_react["default"].createElement("div", (0, _extends2["default"])({}, props, {
    ref: ref,
    className: (0, _classnames["default"])(classes.root, (_cn = {}, (0, _defineProperty2["default"])(_cn, classes.flexWrap, wrap), (0, _defineProperty2["default"])(_cn, classes["spacing".concat(spacing)], spacing), (0, _defineProperty2["default"])(_cn, classes.lastChildSpacing, spacing), (0, _defineProperty2["default"])(_cn, classes.fullWidth, fullWidth), (0, _defineProperty2["default"])(_cn, classes.flexDirectionColumn, !reverse), (0, _defineProperty2["default"])(_cn, classes.flexDirectionColumnReverse, reverse), (0, _defineProperty2["default"])(_cn, classes["justify".concat((0, _utils.firstLetterUppercase)(justify))], justify !== 'start'), (0, _defineProperty2["default"])(_cn, classes["alignItems".concat((0, _utils.firstLetterUppercase)(align))], align !== 'stretch'), (0, _defineProperty2["default"])(_cn, classes["col-xs-".concat(xsSize)], xsSize), (0, _defineProperty2["default"])(_cn, classes["col-sm-".concat(sm)], sm), (0, _defineProperty2["default"])(_cn, classes["col-md-".concat(md)], md), (0, _defineProperty2["default"])(_cn, classes["col-lg-".concat(lg)], lg), (0, _defineProperty2["default"])(_cn, classes["col-xl-".concat(xl)], xl), (0, _defineProperty2["default"])(_cn, classes["col-xlg-".concat(xlg)], xlg), _cn), 'col-wrapper', className),
    style: style
  }), children);
});

Col.defaultProps = {
  className: '',
  style: {},
  children: null,
  align: 'inherit',
  justify: 'inherit',
  spacing: 0,
  span: 0,
  breakpoints: {},
  fullWidth: false
};
Col.propTypes = {
  /** Additional class for top level */
  className: _propTypes["default"].string,

  /** Indicator whether or not to render flex wrap */
  wrap: _propTypes["default"].bool,

  /** Indicator whether or not to render flex direction reverse */
  reverse: _propTypes["default"].bool,

  /** Additional styling for root level */
  style: _propTypes["default"].shape(),

  /** set spacing between child components - available value - [0,1,2,3,4,5,6,7,8,9,10, 11 ,12] will be determined by the theme spacer (default 5) */
  spacing: _propTypes["default"].oneOf([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]),

  /** Child components inside a section */
  children: _propTypes["default"].oneOfType([_propTypes["default"].node, _propTypes["default"].arrayOf(_propTypes["default"].node)]),

  /** Set flex alignment - available values = ['start', 'center', 'end', 'stretch', 'baseline']  */
  align: _propTypes["default"].oneOf(['start', 'center', 'end', 'stretch', 'baseline', 'inherit']),

  /** Set flex justify content - available values = ['start','center','end','between','around','evenly'] */
  justify: _propTypes["default"].oneOf(['start', 'center', 'end', 'between', 'around', 'evenly', 'inherit']),

  /** Set the column width to fit parent */
  fullWidth: _propTypes["default"].bool,

  /** Set the column span width - available values = [auto,1,2,3,4,5,6,7,8,9,10,11,12] */
  span: _propTypes["default"].oneOf(['auto', 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]),

  /** specify the media query breakpoints for the column - available values = {xs,sm,md,lg,xl,xlg} */
  breakpoints: _propTypes["default"].shape(),

  /** JSS overrides */
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func])
}; // Hack for withInfo plugin

Col.displayName = 'Col';
var _default = Col;
exports["default"] = _default;
