// @guestyci/foundation/Backdrop/Backdrop.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _noop = _interopRequireDefault(require("lodash/noop"));

var _createStyles = _interopRequireDefault(require("../createStyles"));

var _utils = require("../theme/utils");

var _colors = require("../theme/colors");

var useStyles = (0, _createStyles["default"])({
  root: {
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    zIndex: -1,
    position: 'fixed',
    touchAction: 'none',
    backgroundColor: (0, _utils.fade)(_colors.blueGray700, 0.85)
  }
}, {
  name: 'Backdrop'
});

var Backdrop = function Backdrop(_ref) {
  var _ref$className = _ref.className,
      className = _ref$className === void 0 ? '' : _ref$className,
      _ref$jss = _ref.jss,
      jss = _ref$jss === void 0 ? {} : _ref$jss,
      _ref$onClick = _ref.onClick,
      onClick = _ref$onClick === void 0 ? _noop["default"] : _ref$onClick,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["className", "jss", "onClick"]);
  var classes = useStyles({
    jss: jss
  });
  return /*#__PURE__*/_react["default"].createElement("div", (0, _extends2["default"])({
    role: "presentation",
    className: (0, _classnames["default"])(classes.root, className),
    onClick: onClick
  }, props, {
    "data-qa": "backdrop"
  }));
};

Backdrop.propTypes = {
  className: _propTypes["default"].string,
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func]),
  onClick: _propTypes["default"].func
};
var _default = Backdrop;
exports["default"] = _default;
