// @guestyci/foundation/es/legacy/Divider/Divider.js
import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
const Divider = forwardRef(function Divider({
  vertical,
  horizontal,
  className,
  style,
  color,
  width
}, ref) {
  const isVertical = vertical || !horizontal;
  return /*#__PURE__*/React.createElement("div", {
    ref: ref,
    className: cn(className, {
      'bt w-fill': !isVertical
    }, {
      [`bw-${width}`]: width
    }, {
      'bl height-inherit': isVertical
    }, {
      [`border-${color}`]: color
    }),
    style: style
  });
});
Divider.defaultProps = {
  color: 'gray-light',
  vertical: false,
  horizontal: true,
  className: '',
  style: {},
  width: 1
};
Divider.propTypes = {
  color: PropTypes.string,

  /** Set divider as vertical by default is horizontal */
  vertical: PropTypes.bool,

  /** Set divider as horizontal - DEPRECATED - USE VERTICAL horizontal is true by default */
  horizontal: PropTypes.bool,

  /** Set additional class names to divider */
  className: PropTypes.string,

  /** Set style to divider */
  style: PropTypes.shape(),

  /** Border width */
  width: PropTypes.number
}; // Hack for withInfo plugin

Divider.displayName = 'Divider';
export default Divider;
