// @guestyci/foundation/legacy/Divider/Divider.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var Divider = (0, _react.forwardRef)(function Divider(_ref, ref) {
  var vertical = _ref.vertical,
      horizontal = _ref.horizontal,
      className = _ref.className,
      style = _ref.style,
      color = _ref.color,
      width = _ref.width;
  var isVertical = vertical || !horizontal;
  return /*#__PURE__*/_react["default"].createElement("div", {
    ref: ref,
    className: (0, _classnames["default"])(className, {
      'bt w-fill': !isVertical
    }, (0, _defineProperty2["default"])({}, "bw-".concat(width), width), {
      'bl height-inherit': isVertical
    }, (0, _defineProperty2["default"])({}, "border-".concat(color), color)),
    style: style
  });
});
Divider.defaultProps = {
  color: 'gray-light',
  vertical: false,
  horizontal: true,
  className: '',
  style: {},
  width: 1
};
Divider.propTypes = {
  color: _propTypes["default"].string,

  /** Set divider as vertical by default is horizontal */
  vertical: _propTypes["default"].bool,

  /** Set divider as horizontal - DEPRECATED - USE VERTICAL horizontal is true by default */
  horizontal: _propTypes["default"].bool,

  /** Set additional class names to divider */
  className: _propTypes["default"].string,

  /** Set style to divider */
  style: _propTypes["default"].shape(),

  /** Border width */
  width: _propTypes["default"].number
}; // Hack for withInfo plugin

Divider.displayName = 'Divider';
var _default = Divider;
exports["default"] = _default;
