// @guestyci/foundation/DatePicker/NavButton.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _Button = _interopRequireDefault(require("../Button"));

var _Icon = _interopRequireDefault(require("../Icon"));

var _createStyles = _interopRequireDefault(require("../createStyles"));

var BtnLeftSmall = function BtnLeftSmall(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M.137 4.172L4.172.137a.488.488 0 0 1 .689.69L1.156 4.53l3.705 3.705a.488.488 0 0 1-.69.69L.138 4.89A.477.477 0 0 1 0 4.53a.474.474 0 0 1 .136-.357z"
  }));
};

BtnLeftSmall.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 4.9977479 9.0618038"
};

var BtnRightSmall = function BtnRightSmall(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M4.997 4.529a.477.477 0 0 1-.136.361L.826 8.925a.489.489 0 0 1-.69-.69l3.706-3.704L.137.826a.489.489 0 0 1 .69-.69L4.86 4.173a.474.474 0 0 1 .136.357z"
  }));
};

BtnRightSmall.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 4.9977479 9.0618038"
};
var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    button: {
      position: 'absolute',
      top: '20px',
      width: 40,
      height: 40,
      border: "solid 1px ".concat(theme.palette.border),
      borderRadius: 0
    },
    nextButton: {
      right: '22px'
    },
    prevButton: {
      left: '22px'
    }
  };
});

var NavButton = function NavButton(_ref) {
  var direction = _ref.direction;
  var classes = useStyles();
  return /*#__PURE__*/_react["default"].createElement(_Button["default"], {
    type: "button",
    className: (0, _classnames["default"])(classes.button, (0, _defineProperty2["default"])({}, classes.prevButton, direction === 'Left'), (0, _defineProperty2["default"])({}, classes.nextButton, direction === 'Right')),
    "data-qa": "nav-button"
  }, /*#__PURE__*/_react["default"].createElement(_Icon["default"], {
    height: 11,
    width: 6,
    svg: direction === 'Right' ? BtnRightSmall : BtnLeftSmall
  }));
};

var _default = NavButton;
exports["default"] = _default;
