// @guestyci/foundation/DatePicker/styles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _styles = _interopRequireDefault(require("../Input/styles"));

var _default = {
  root: function root(theme) {
    return (0, _extends2["default"])({}, _styles["default"].shared(theme), {
      height: 46,
      paddingRight: 0,
      paddingLeft: 0,
      width: '100%',
      '&:hover': _styles["default"].hover(theme)
    });
  },
  disabled: function disabled(theme) {
    return _styles["default"].disabled(theme);
  },
  error: function error(theme) {
    return (0, _extends2["default"])({}, _styles["default"].error(theme), {
      '& .DateInput_input': {
        backgroundColor: theme.palette.error.background
      },
      '& .DateRangePickerInput': {
        backgroundColor: theme.palette.error.background
      }
    });
  },
  focused: function focused(theme) {
    return _styles["default"].focused(theme);
  },
  inputContainer: {
    display: 'flex',
    flex: 1,
    width: 'calc(100% - 85px)',
    alignItems: 'center'
  }
};
exports["default"] = _default;
