// @guestyci/redux-tools/index.js
/** @license Guesty v1.1.4
 *
 * This source code is licensed under the ISC license found in the
 * LICENSE file in the root directory of this source tree.
 */
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "actionSplitterMiddleware", {
  enumerable: true,
  get: function get() {
    return _actionSplitter.actionSplitterMiddleware;
  }
});
Object.defineProperty(exports, "apiMiddleware", {
  enumerable: true,
  get: function get() {
    return _api2.apiMiddleware;
  }
});
Object.defineProperty(exports, "loggerMiddleware", {
  enumerable: true,
  get: function get() {
    return _logger.loggerMiddleware;
  }
});
Object.defineProperty(exports, "stateFreezer", {
  enumerable: true,
  get: function get() {
    return _stateFreezer.stateFreezer;
  }
});
Object.defineProperty(exports, "undoable", {
  enumerable: true,
  get: function get() {
    return _undoable.undoable;
  }
});
Object.defineProperty(exports, "makeAsyncActionCreator", {
  enumerable: true,
  get: function get() {
    return _reduxUtility.makeAsyncActionCreator;
  }
});
Object.defineProperty(exports, "makeActionCreator", {
  enumerable: true,
  get: function get() {
    return _reduxUtility.makeActionCreator;
  }
});
exports.apiActions = exports["default"] = void 0;

var apiActions = _interopRequireWildcard(require("./actions/api"));

exports.apiActions = apiActions;

var _actionSplitter = require("./middleware/actionSplitter");

var _api2 = require("./middleware/api");

var _logger = require("./middleware/logger");

var _stateFreezer = require("./reducerEnhancers/stateFreezer");

var _undoable = require("./reducerEnhancers/undoable");

var _reduxUtility = require("./utils/reduxUtility");

var redux = {
  apiActions: apiActions,
  actionSplitterMiddleware: _actionSplitter.actionSplitterMiddleware,
  apiMiddleware: _api2.apiMiddleware,
  loggerMiddleware: _logger.loggerMiddleware,
  stateFreezer: _stateFreezer.stateFreezer,
  undoable: _undoable.undoable,
  makeAsyncActionCreator: _reduxUtility.makeAsyncActionCreator,
  makeActionCreator: _reduxUtility.makeActionCreator
};
var _default = redux;
exports["default"] = _default;
