// @guestyci/foundation/Tag/Tag.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.useStyles = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _TextField = _interopRequireDefault(require("../TextField"));

var _createStyles = _interopRequireDefault(require("../createStyles"));

var _Paper = _interopRequireDefault(require("../Paper"));

var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    root: {
      padding: "2px ".concat(theme.spacer(1), "px"),
      minWidth: 37
    },
    bgDefault: {
      backgroundColor: theme.palette.background.system
    },
    bgBold: {
      backgroundColor: theme.palette.background.secondary
    }
  };
}, {
  name: 'Tag'
});
exports.useStyles = useStyles;

var Tag = _react["default"].forwardRef(function Tag(_ref, ref) {
  var _cn;

  var children = _ref.children,
      _ref$bold = _ref.bold,
      bold = _ref$bold === void 0 ? false : _ref$bold,
      _ref$jss = _ref.jss,
      jss = _ref$jss === void 0 ? {} : _ref$jss,
      _ref$className = _ref.className,
      className = _ref$className === void 0 ? '' : _ref$className,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["children", "bold", "jss", "className"]);
  var classes = useStyles({
    jss: jss
  });
  return /*#__PURE__*/_react["default"].createElement(_Paper["default"], (0, _extends2["default"])({
    elevation: 0,
    ref: ref,
    className: (0, _classnames["default"])(classes.root, (_cn = {}, (0, _defineProperty2["default"])(_cn, classes.bgBold, bold), (0, _defineProperty2["default"])(_cn, classes.bgDefault, !bold), _cn), className)
  }, props), /*#__PURE__*/_react["default"].createElement(_TextField["default"], {
    variant: "h5",
    overflowEllipsis: true,
    align: "center"
  }, children));
});

Tag.propTypes = {
  /** Child component to render */
  children: _propTypes["default"].node,

  /** Jss override to create custom styles */
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func]),

  /** Boolean indicator whether or not to render in emphasized mode  */
  bold: _propTypes["default"].bool,

  /** Additional classname  */
  className: _propTypes["default"].string
};
Tag.displayName = 'Tag';
var _default = Tag;
exports["default"] = _default;
