// @guestyci/foundation/Chip/Chip.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _IconButton = _interopRequireDefault(require("../IconButton"));

var _TextField = _interopRequireDefault(require("../TextField"));

var _Row = _interopRequireDefault(require("../Layout/Row"));

var _createStyles = _interopRequireDefault(require("../createStyles"));

var _colors = require("../theme/colors");

var BtnDelete = function BtnDelete(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M8.485.707L7.778 0 4.243 3.536.707 0 0 .707l3.536 3.536L0 7.778l.707.707L4.243 4.95l3.535 3.535.707-.707L4.95 4.243z"
  }));
};

BtnDelete.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 8.4854 8.4854"
};
var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    root: {
      height: 35,
      marginBottom: theme.spacer(1),
      marginTop: theme.spacer(1),
      padding: "0px ".concat(theme.spacer(3), "px"),
      backgroundColor: theme.palette.secondary["default"],
      borderRadius: 2,
      color: _colors.white
    },
    icon: {
      color: _colors.white,
      '&:not([disabled]):hover': {
        backgroundColor: 'transparent'
      }
    }
  };
}, {
  name: 'Chip'
});

var Chip = _react["default"].forwardRef(function Chip(_ref, ref) {
  var value = _ref.value,
      onClear = _ref.onClear,
      noClear = _ref.noClear,
      jss = _ref.jss,
      className = _ref.className,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["value", "onClear", "noClear", "jss", "className"]);
  var classes = useStyles({
    jss: jss
  });

  var handleClearClick = function handleClearClick(e) {
    onClear(e, value);
  };

  return /*#__PURE__*/_react["default"].createElement(_Row["default"], (0, _extends2["default"])({
    ref: ref,
    spacing: 2,
    className: (0, _classnames["default"])(classes.root, className),
    justify: "between",
    align: "center",
    wrap: false
  }, props), /*#__PURE__*/_react["default"].createElement(_TextField["default"], null, value), !noClear && /*#__PURE__*/_react["default"].createElement(_IconButton["default"], {
    value: value,
    position: "suffix",
    component: "span",
    className: classes.icon,
    size: "xs",
    onClick: handleClearClick
  }, /*#__PURE__*/_react["default"].createElement(BtnDelete, {
    width: 12,
    height: 12
  })));
});

Chip.propTypes = {
  /** Child component to show in chip */
  value: _propTypes["default"].string.isRequired,

  /** CB for clear click */
  onClear: _propTypes["default"].func,

  /** Jss override to create custom styles */
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func]),

  /** Boolean indicator whether or not to show clear button */
  noClear: _propTypes["default"].bool,

  /** Additional classname  */
  className: _propTypes["default"].string
};
Chip.displayName = 'Chip';
var _default = Chip;
exports["default"] = _default;
