// @guestyci/localize/utils/localeData.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MomentFormatTypes = exports.timeIdsMap = exports.dateIdsMap = exports.currencyIdsMap = exports.currencyTypes = exports.localeDropdownMenu = void 0;
var localeDropdownMenu = [{
  value: 'en-US',
  label: 'English'
}, {
  value: 'fr',
  label: 'French (Français)'
}, {
  value: 'de',
  label: 'German (Deutsch)'
}, {
  value: 'it',
  label: 'Italian (Italiano)'
}, {
  value: 'es-ES',
  label: 'Spanish (Español)'
}, {
  value: 'nl',
  label: 'Dutch (Nederlands)'
}, {
  value: 'ja',
  label: 'Japanese (日本語)'
}, {
  value: 'cs',
  label: 'Czech (Čeština)'
}, {
  value: 'da',
  label: 'Danish (Dansk)'
}, {
  value: 'el',
  label: 'Greek (ελληνικά)'
}, {
  value: 'hr',
  label: 'Croatian (Hrvatski jezik)'
}, {
  value: 'hu',
  label: 'Hungarian (Magyar)'
}, {
  value: 'pl',
  label: 'Polish (Polszczyzna)'
}, {
  value: 'ru',
  label: 'Russian (русский)'
}, {
  value: 'sk',
  label: 'Slovak (Slovenčina)'
}, {
  value: 'pt-BR',
  label: 'Portuguese Brazil (Português)'
}, {
  value: 'zh-CN',
  label: 'Chinese (简体中文)'
}];
exports.localeDropdownMenu = localeDropdownMenu;
var currencyTypes = ['EUR', 'USD', 'GBP', 'ILS', 'KRW', 'JPY', 'HKD', 'BRL', 'CZK', 'CNY', 'NGN', 'PHP', 'RUB', 'PLN', 'PYG', 'THB', 'UAH', 'VND', 'TRY', 'AFN', 'AED', 'SAR', 'BGN', 'IDR', 'ARS', 'CLP', 'COP', 'NZD', 'SGD', 'AUD', 'JMD', 'INR', 'TWD', 'MXN', 'CAD', 'HRK', 'SEK', 'RON', 'DKK', 'NOK', 'CHF', 'HUF', 'MYR', 'ZAR', 'PEN', 'MAD', 'CRC'];
exports.currencyTypes = currencyTypes;
var currencyIdsMap = new Map(currencyTypes.map(function (currency) {
  var curr = currency.toLowerCase();
  return [curr, "common:formatted_currency.".concat(curr)];
}));
exports.currencyIdsMap = currencyIdsMap;
var dateIdsMap = new Map([['short', 'common:formatted_date.short'], ['medium', 'common:formatted_date.medium'], ['long', 'common:formatted_date.long'], ['full', 'common:formatted_date.full']]);
exports.dateIdsMap = dateIdsMap;
var timeIdsMap = new Map([['short', 'common:formatted_time.short'], ['medium', 'common:formatted_time.medium'], ['long', 'common:formatted_time.long'], ['full', 'common:formatted_time.full']]);
exports.timeIdsMap = timeIdsMap;
var MomentFormatTypes = {
  ShortDate: 'shortDate',
  FullDate: 'fullDate',
  Date: 'date',
  MMMM_YYYY: 'mmmm yyyy',
  MMM_YYYY: 'mmm yyyy',
  ShortMonthYear: 'mmm yy',
  Time: 'time'
};
exports.MomentFormatTypes = MomentFormatTypes;
