// @guestyci/foundation/legacy/ThreadMessage/ThreadMessageBody.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _reactJss = _interopRequireDefault(require("react-jss"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _size = _interopRequireDefault(require("lodash/size"));

var _classnames = _interopRequireDefault(require("classnames"));

var _ThreadAttachment = _interopRequireDefault(require("./ThreadAttachment"));

var threadMessageStyle = {
  automatedMessage: {
    backgroundImage: 'repeating-linear-gradient(-67.5deg, transparent, transparent 7px, rgba(255, 255, 255, 0.4) 7px, rgba(255, 255, 255, 0.4) 10px)'
  }
};

var ThreadMessageBody = function ThreadMessageBody(_ref) {
  var body = _ref.body,
      attachments = _ref.attachments,
      isAutoMessage = _ref.isAutoMessage,
      byUs = _ref.byUs,
      errorMessage = _ref.errorMessage,
      bgColor = _ref.bgColor,
      onAttachmentClick = _ref.onAttachmentClick,
      classes = _ref.classes;
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])('align-items-center mb-2 px-4 py-3 font-size-md', {
      'by-guest': !byUs
    }, {
      'by-us': byUs
    }, {
      'border border-red': !!errorMessage
    }, "bg-".concat(bgColor), isAutoMessage ? "autoMsg ".concat(classes.automatedMessage) : null),
    "data-qa": "thread-message-body"
  }, /*#__PURE__*/_react["default"].createElement("span", {
    dangerouslySetInnerHTML: {
      __html: body ? body.replace(/\n/g, '<br/>') : null
    }
  }), !!(0, _size["default"])(attachments) && /*#__PURE__*/_react["default"].createElement("div", {
    className: "mt-4"
  }, attachments.map(function (attachment) {
    return /*#__PURE__*/_react["default"].createElement(_ThreadAttachment["default"], (0, _extends2["default"])({
      key: attachment.url,
      attachment: attachment,
      onAttachmentClick: onAttachmentClick
    }, attachment));
  })));
};

ThreadMessageBody.defaultProps = {
  isAutoMessage: false,
  byUs: true,
  bgColor: null,
  errorMessage: false
};
ThreadMessageBody.propTypes = {
  body: _propTypes["default"].node.isRequired,
  attachments: _propTypes["default"].arrayOf(_propTypes["default"].shape()).isRequired,
  isAutoMessage: _propTypes["default"].bool,
  byUs: _propTypes["default"].bool,
  bgColor: _propTypes["default"].string,
  onAttachmentClick: _propTypes["default"].func,
  errorMessage: _propTypes["default"].bool
};

var _default = (0, _reactJss["default"])(threadMessageStyle)(ThreadMessageBody);

exports["default"] = _default;
