// @guestyci/foundation/legacy/ThreadMessage/ThreadAttachment.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Icon = _interopRequireDefault(require("../Icon"));

var _Spinner = _interopRequireDefault(require("../Spinner/Spinner"));

var _FlatButton = _interopRequireDefault(require("../FlatButton/FlatButton"));

var ThreadAttachment = function ThreadAttachment(_ref) {
  var contentName = _ref.contentName,
      url = _ref.url,
      icon = _ref.icon,
      name = _ref.name,
      isLoading = _ref.isLoading,
      onAttachmentClick = _ref.onAttachmentClick;
  var handleAttachment = (0, _react.useCallback)(function () {
    if (onAttachmentClick) {
      onAttachmentClick({
        contentName: contentName,
        url: url
      });
    }
  }, [onAttachmentClick, contentName, url]);
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: "attachment d-flex align-items-center mb-2",
    "data-qa": "thread-attachment"
  }, /*#__PURE__*/_react["default"].createElement(_FlatButton["default"], {
    onClick: handleAttachment
  }, /*#__PURE__*/_react["default"].createElement(_Icon["default"], {
    icon: icon,
    height: 30,
    width: 26,
    noFill: true,
    className: "mr-2",
    color: "gray-dark"
  }), name), isLoading && /*#__PURE__*/_react["default"].createElement("div", {
    "data-testid": "loader",
    className: "ml-2"
  }, /*#__PURE__*/_react["default"].createElement(_Spinner["default"], {
    size: 20
  })));
};

ThreadAttachment.propTypes = {
  /** url of the attachment */
  url: _propTypes["default"].string.isRequired,

  /** icon name of the attachment */
  icon: _propTypes["default"].string.isRequired,

  /** name of the file */
  name: _propTypes["default"].string.isRequired,

  /** Indicates if we should show a spinner to the side of the attachment */
  isLoading: _propTypes["default"].bool
};
var _default = ThreadAttachment;
exports["default"] = _default;
