// @guestyci/foundation/legacy/ThreadMessage/ThreadMessage.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _get = _interopRequireDefault(require("lodash/get"));

var _size = _interopRequireDefault(require("lodash/size"));

var _ThreadMessageIcon = _interopRequireDefault(require("./ThreadMessageIcon"));

var _ThreadMessageBody = _interopRequireDefault(require("./ThreadMessageBody"));

var _ThreadMessageError = _interopRequireDefault(require("./ThreadMessageError"));

var _ThreadMessageInfo = _interopRequireDefault(require("./ThreadMessageInfo"));

function _createSuper(Derived) { return function () { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

var fileTypes = ['Png', 'Pdf', 'Xls', 'Gif', 'Doc', 'Jpg', 'Mp4'];

var ThreadMessage = /*#__PURE__*/function (_Component) {
  (0, _inherits2["default"])(ThreadMessage, _Component);

  var _super = _createSuper(ThreadMessage);

  function ThreadMessage(props) {
    var _this;

    (0, _classCallCheck2["default"])(this, ThreadMessage);
    _this = _super.call(this, props);

    _this.getThumbnail = function (sentBy) {
      var _this$props = _this.props,
          isAutomatic = _this$props.isAutomatic,
          from = _this$props.from,
          module = _this$props.module,
          guest = _this$props.guest;
      if (isAutomatic) return 'BtnAutomsg';
      if (sentBy === 'guest') return (0, _get["default"])(guest, 'picture.thumbnail');
      if (sentBy === 'host' || (0, _get["default"])(module, 'type') === 'note') return (0, _get["default"])(from, 'picture.thumbnail');
      if (sentBy === 'guesty') return 'gseIcon';
      if ((0, _get["default"])(module, 'action')) return 'airBnbIcon';
      return null;
    };

    _this.getBgColor = function (sentBy, module) {
      if (module.type === 'note') return 'yellow-light';
      return 'gray-lightest';
    };

    _this.getModuleType = function (module) {
      var moduleDisplayType = (0, _get["default"])(module, 'displayType');
      return moduleDisplayType || (0, _get["default"])(module, 'type');
    };

    _this.getFullName = function (sentBy) {
      var _this$props2 = _this.props,
          from = _this$props2.from,
          isAutomatic = _this$props2.isAutomatic,
          module = _this$props2.module,
          guest = _this$props2.guest;
      if (sentBy === 'guest') return (0, _get["default"])(guest, 'fullName');
      if (sentBy === 'guesty') return 'Guesty’s communication service';
      if (sentBy === 'channel') return 'Airbnb';
      if (sentBy === 'host' && !isAutomatic) return (0, _get["default"])(from, 'fullName');
      if (!sentBy && (0, _get["default"])(module, 'action')) return 'airbnb';
      if (isAutomatic) return 'Automated message';
      return (0, _get["default"])(from, 'fullName') || 'Not Available';
    };

    _this.getIconType = function (item) {
      if (!item) return 'Jpg';
      var itemType = item.split('.')[1];
      return fileTypes.indexOf(itemType) > -1 ? itemType : 'Jpg';
    };

    _this.prepareAttachments = function (attachments) {
      if (!(0, _size["default"])(attachments)) return [];
      return attachments.map(function (item) {
        return {
          url: item.attachmentUrl,
          name: item.origFileName || item.contentName || 'unnamed file',
          icon: _this.getIconType(item.origFileName),
          contentName: item.contentName,
          isLoading: item.isLoading
        };
      });
    };

    _this.isByUs = function () {
      var _this$props3 = _this.props,
          sentBy = _this$props3.sentBy,
          module = _this$props3.module;
      return sentBy !== 'guest' && sentBy !== 'third party' && !(0, _get["default"])(module, 'action');
    };

    _this.retrySendMessage = function () {
      var _this$props4 = _this.props,
          id = _this$props4.id,
          onRetry = _this$props4.onRetry;
      onRetry(id);
    };

    _this.deleteMessage = function () {
      var _this$props5 = _this.props,
          id = _this$props5.id,
          onDelete = _this$props5.onDelete;
      onDelete(id);
    };

    _this.state = {
      bgColor: _this.getBgColor(props.sentBy, props.module),
      byGuest: props.sentBy === 'guest',
      byGSE: props.sentBy === 'guesty',
      byChannel: (!props.sentBy || props.sentBy === 'channel') && !!(0, _get["default"])(props.module, 'action'),
      thumbnail: _this.getThumbnail(props.sentBy),
      byUs: _this.isByUs(),
      fullName: _this.getFullName(props.sentBy),
      moduleType: _this.getModuleType(props.module)
    };
    return _this;
  }

  (0, _createClass2["default"])(ThreadMessage, [{
    key: "componentDidUpdate",
    value: function componentDidUpdate(prevProps) {
      var _this$props6 = this.props,
          id = _this$props6.id,
          guest = _this$props6.guest;
      var dataUpdated = prevProps.guest._id !== guest._id || prevProps.id !== id;

      if (dataUpdated) {
        this.init();
      }
    }
  }, {
    key: "init",
    value: function init() {
      var _this$props7 = this.props,
          sentBy = _this$props7.sentBy,
          module = _this$props7.module;
      this.setState({
        bgColor: this.getBgColor(sentBy, module),
        byGuest: sentBy === 'guest',
        byGSE: sentBy === 'guesty',
        byChannel: (!sentBy || sentBy === 'channel') && !!(0, _get["default"])(module, 'action'),
        thumbnail: this.getThumbnail(sentBy),
        byUs: this.isByUs(),
        fullName: this.getFullName(sentBy),
        moduleType: this.getModuleType(module)
      });
    }
  }, {
    key: "render",
    value: function render() {
      var _this$state = this.state,
          bgColor = _this$state.bgColor,
          byGuest = _this$state.byGuest,
          byGSE = _this$state.byGSE,
          byChannel = _this$state.byChannel,
          thumbnail = _this$state.thumbnail,
          byUs = _this$state.byUs,
          fullName = _this$state.fullName,
          moduleType = _this$state.moduleType;
      var _this$props8 = this.props,
          attachments = _this$props8.attachments,
          isAutomatic = _this$props8.isAutomatic,
          className = _this$props8.className,
          body = _this$props8.body,
          error = _this$props8.error,
          deleteTxt = _this$props8.deleteTxt,
          retryTxt = _this$props8.retryTxt,
          reasonTxt = _this$props8.reasonTxt,
          sentAt = _this$props8.sentAt,
          createdAt = _this$props8.createdAt,
          onAttachmentClick = _this$props8.onAttachmentClick;
      return /*#__PURE__*/_react["default"].createElement("div", {
        className: (0, _classnames["default"])('threadMessageBubble d-flex', {
          'flex-row-reverse by-us': byUs
        }, className),
        "data-qa": "thread-message"
      }, /*#__PURE__*/_react["default"].createElement(_ThreadMessageIcon["default"], {
        avatarName: fullName,
        isAutoMessage: isAutomatic,
        thumbnail: thumbnail,
        byChannel: byChannel,
        byUs: byUs,
        byGSE: byGSE,
        byGuest: byGuest
      }), /*#__PURE__*/_react["default"].createElement("div", {
        className: "d-flex flex-column flex-1"
      }, /*#__PURE__*/_react["default"].createElement(_ThreadMessageBody["default"], {
        errorMessage: error,
        body: body,
        attachments: this.prepareAttachments(attachments),
        isAutoMessage: isAutomatic,
        byUs: byUs,
        bgColor: bgColor,
        onAttachmentClick: onAttachmentClick
      }), error ? /*#__PURE__*/_react["default"].createElement(_ThreadMessageError["default"], {
        messageReason: reasonTxt,
        tooltip: /*#__PURE__*/_react["default"].createElement(_ThreadMessageInfo["default"], {
          fullName: fullName,
          sentAt: sentAt,
          createdAt: createdAt,
          byUs: byUs,
          moduleType: moduleType,
          fromErrorInfo: true
        }),
        retrySendMessage: this.retrySendMessage,
        onDelete: this.deleteMessage,
        retryTxt: retryTxt,
        deleteTxt: deleteTxt
      }) : /*#__PURE__*/_react["default"].createElement(_ThreadMessageInfo["default"], {
        fullName: fullName,
        sentAt: sentAt,
        createdAt: createdAt,
        byUs: byUs,
        moduleType: moduleType
      })));
    }
  }]);
  return ThreadMessage;
}(_react.Component);

ThreadMessage.defaultProps = {
  attachments: null,
  isAutomatic: false,
  guest: {}
};
ThreadMessage.propTypes = {
  /** The message body, can be a string or an html node. */
  body: _propTypes["default"].node.isRequired,

  /** Array of objects that contain the imgs */
  attachments: _propTypes["default"].arrayOf(_propTypes["default"].shape()),

  /** Bool, flag for automatic messages */
  isAutomatic: _propTypes["default"].bool,

  /** String, name if the module which the message sent  */
  module: _propTypes["default"].shape().isRequired,

  /** Object, metadata about the message sender. */
  from: _propTypes["default"].shape().isRequired,

  /** String, by who message was sent */
  sentBy: _propTypes["default"].string.isRequired,

  /** String, time that message was sent */
  sentAt: _propTypes["default"].string.isRequired,

  /** String, item id */
  id: _propTypes["default"].string.isRequired,

  /** object, item id */
  guest: _propTypes["default"].shape(),

  /** Bool, if the message is with error */
  error: _propTypes["default"].bool.isRequired,

  /** function, retry send the specific message */
  onRetry: _propTypes["default"].func.isRequired,

  /** function, delete the unsent message */
  onDelete: _propTypes["default"].func.isRequired,

  /** string, txt for delete btn */
  deleteTxt: _propTypes["default"].string.isRequired,

  /** string, txt for retry btn */
  retryTxt: _propTypes["default"].string.isRequired,

  /** String, err reason */
  reasonTxt: _propTypes["default"].string.isRequired,

  /** Open attachment event */
  onAttachmentClick: _propTypes["default"].func
};
var _default = ThreadMessage;
exports["default"] = _default;
