// @guestyci/history/NavTabs/Tab.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _isObject = _interopRequireDefault(require("lodash/isObject"));

var _classnames = _interopRequireDefault(require("classnames"));

var _createStyles = _interopRequireDefault(require("@guestyci/foundation/createStyles"));

var _Link = _interopRequireDefault(require("../Link"));

var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    root: {
      textTransform: 'uppercase',
      padding: 0,
      minHeight: 50,
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      '&:hover': {
        color: theme.palette.brand.primary
      }
    },
    active: {
      color: theme.palette.brand.primary,
      fontWeight: 'bold'
    }
  };
}, {
  name: 'Tab'
});

var Tab = function Tab(_ref) {
  var onClick = _ref.onClick,
      tabIndex = _ref.tabIndex,
      to = _ref.to,
      value = _ref.value,
      className = _ref.className,
      disabled = _ref.disabled,
      isActive = _ref.isActive,
      children = _ref.children;
  var classes = useStyles();
  var pathname = (0, _react.useMemo)(function () {
    return (0, _isObject["default"])(to) ? to.pathname : to;
  }, [to]);
  var onTabClick = (0, _react.useCallback)(function () {
    onClick({
      tabIndex: tabIndex,
      to: to,
      value: value
    });
  }, [onClick, tabIndex, to, value]);
  return /*#__PURE__*/_react["default"].createElement(_Link["default"], {
    "data-cy": isActive ? 'active' : '',
    href: pathname,
    className: (0, _classnames["default"])('tab-item', classes.root, (0, _defineProperty2["default"])({}, classes.active, isActive), className),
    onClick: onTabClick,
    disabled: disabled
  }, children);
};

Tab.defaultProps = {
  children: null,
  disabled: false,
  to: undefined
};
Tab.propTypes = {
  /** Tab name */
  children: _propTypes["default"].node,

  /** Indicator whether or not the tab is disabled */
  disabled: _propTypes["default"].bool,

  /** The tab value to be passed when clicked */
  to: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].shape()]),

  /** value to pass on click  */
  value: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].number, _propTypes["default"].shape()])
};
var _default = Tab;
exports["default"] = _default;
