// @guestyci/foundation/esm/legacy/baseDropdown/withMultiSelect.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _possibleConstructorReturn from "@babel/runtime/helpers/esm/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/esm/getPrototypeOf";

function _createSuper(Derived) { return function () { var Super = _getPrototypeOf(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

import React, { Component } from 'react';
import filter from 'lodash/filter';
import head from 'lodash/head';
import isString from 'lodash/isString';
import map from 'lodash/map';
import isEqual from 'lodash/isEqual';
import find from 'lodash/find';
import size from 'lodash/size';
/**
 * Create a wrapper compose to support multi-select functionality
 * @param ComposedComponent
 * @returns {{Component}}
 */

var withMultiSelect = function withMultiSelect(ComposedComponent) {
  var _temp;

  return _temp = /*#__PURE__*/function (_Component) {
    _inherits(MultiSelectWrapper, _Component);

    var _super = _createSuper(MultiSelectWrapper);

    function MultiSelectWrapper(props) {
      var _this;

      _classCallCheck(this, MultiSelectWrapper);

      _this = _super.call(this, props);

      _this.getSelectedItems = function (options, selected) {
        if (!size(selected)) return [];
        var checkedItemsSet = new Set(selected);
        return filter(options, function (option) {
          return checkedItemsSet.has(option.value);
        });
      };

      _this.getUnselectedItems = function (options, selected) {
        if (!size(selected)) return options;
        var checkedItemsSet = new Set(selected);
        return filter(options, function (option) {
          return !checkedItemsSet.has(option.value);
        });
      };

      _this.getPlaceholder = function (options) {
        var single = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 'more';
        var plural = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : 'more';
        if (!options || !size(options)) return '';
        var tmpOptions = options;
        var firstItem;
        firstItem = head(options).label;

        if (!isString(firstItem)) {
          tmpOptions = map(options, 'name');
          firstItem = head(tmpOptions);
        }

        if (tmpOptions.length === 1) return firstItem;
        var type = tmpOptions.length === 2 ? single : plural;
        return "".concat(firstItem, " + ").concat(tmpOptions.length - 1, " ").concat(type);
      };

      _this.setValues = function (values) {
        var _this$props = _this.props,
            single = _this$props.single,
            plural = _this$props.plural,
            options = _this$props.options;

        var unselectedOptions = _this.getUnselectedItems(options, values);

        var selectedOptions = _this.getSelectedItems(options, values);

        var searchInputValue = _this.getPlaceholder(selectedOptions, single, plural);

        _this.setState({
          searchInputValue: searchInputValue,
          selectedOptions: selectedOptions,
          unselectedOptions: unselectedOptions
        });
      };

      _this.setNewOptions = function (options, selectedValues) {
        var selectedOptions = _this.state.selectedOptions;
        var selectedItems = selectedValues || selectedOptions.map(function (option) {
          return option.value;
        });

        var unselectedOptions = _this.getUnselectedItems(options, selectedItems);

        _this.setState({
          unselectedOptions: unselectedOptions
        });
      };

      _this.clearSelected = function () {
        var _this$props2 = _this.props,
            onChange = _this$props2.onChange,
            id = _this$props2.id,
            name = _this$props2.name,
            options = _this$props2.options;

        _this.setState({
          searchInputValue: '',
          selectedOptions: [],
          unselectedOptions: options
        });

        onChange([], id || name);
      };

      _this.handleItemChange = function (item) {
        var _this$state = _this.state,
            selectedOptions = _this$state.selectedOptions,
            unselectedOptions = _this$state.unselectedOptions;
        var _this$props3 = _this.props,
            single = _this$props3.single,
            plural = _this$props3.plural,
            id = _this$props3.id,
            name = _this$props3.name,
            onChange = _this$props3.onChange;
        var isChecked = find(selectedOptions, {
          value: item.value
        });

        if (isChecked) {
          unselectedOptions.push(item);
          selectedOptions = filter(selectedOptions, function (option) {
            return option.value !== item.value;
          });
        } else {
          selectedOptions.push(item);
          unselectedOptions = filter(unselectedOptions, function (option) {
            return option.value !== item.value;
          });
        }

        var searchInputValue = _this.getPlaceholder(selectedOptions, single, plural);

        _this.setState({
          searchInputValue: searchInputValue,
          selectedOptions: selectedOptions,
          unselectedOptions: unselectedOptions
        });

        onChange(map(selectedOptions, 'value'), id || name);
      };

      var intialValues = props.value || props.values;

      var initialSelectedItems = _this.getSelectedItems(props.options, intialValues);

      _this.state = {
        unselectedOptions: _this.getUnselectedItems(props.options, intialValues),
        selectedOptions: initialSelectedItems,
        searchInputValue: _this.getPlaceholder(initialSelectedItems, props.single, props.plural)
      };
      return _this;
    }

    _createClass(MultiSelectWrapper, [{
      key: "componentDidUpdate",
      value: function componentDidUpdate(prevProps) {
        var _this$props4 = this.props,
            value = _this$props4.value,
            values = _this$props4.values,
            options = _this$props4.options;
        var prevValue = prevProps.value,
            prevValues = prevProps.values,
            prevOptions = prevProps.options;
        var dropdownValues = value || values;
        var prevDropdownValues = prevValue || prevValues;

        if (options.length !== prevOptions.length && !isEqual(options, prevOptions)) {
          this.setNewOptions(options, dropdownValues);
        }

        if (!isEqual(dropdownValues, prevDropdownValues)) {
          this.setValues(dropdownValues);
        }
      }
    }, {
      key: "render",
      value: function render() {
        var _this$state2 = this.state,
            searchInputValue = _this$state2.searchInputValue,
            unselectedOptions = _this$state2.unselectedOptions,
            selectedOptions = _this$state2.selectedOptions;
        return /*#__PURE__*/React.createElement(ComposedComponent, _extends({}, this.props, {
          unselectedOptions: unselectedOptions,
          selectedOptions: selectedOptions,
          searchInputValue: searchInputValue,
          onChange: this.handleItemChange,
          getSelectedItems: this.getSelectedItems,
          getUnselectedItems: this.getUnselectedItems,
          clearSelected: this.clearSelected,
          setNewOptions: this.setNewOptions,
          setValues: this.setValues
        }));
      }
    }]);

    return MultiSelectWrapper;
  }(Component), _temp;
};

export default withMultiSelect;
