// @guestyci/foundation/legacy/Steps/StepDivider.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.StepDividerComponent = void 0;

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactJss = _interopRequireDefault(require("react-jss"));

var styles = {
  stepDivider: {
    position: 'relative'
  },
  stepDividerLine: function stepDividerLine(props) {
    return {
      height: '1px',
      width: "".concat(props.width, "px")
    };
  },
  stepDividerText: {
    position: 'absolute',
    bottom: '0',
    left: '50%',
    transform: 'translateX(-50%)',
    'text-align': 'center'
  }
};

var StepDivider = function StepDivider(_ref) {
  var text = _ref.text,
      color = _ref.color,
      classes = _ref.classes;
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: ('flex-1', classes.stepDivider),
    "data-qa": "step-divider"
  }, /*#__PURE__*/_react["default"].createElement("span", {
    className: (0, _classnames["default"])(classes.stepDividerText, "text-".concat(color))
  }, text), /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])(classes.stepDividerLine, "bg-".concat(color))
  }));
};

StepDivider.defaultProps = {
  text: '',
  color: 'gray',
  width: 19
};
StepDivider.propTypes = {
  text: _propTypes["default"].string,
  color: _propTypes["default"].string,
  width: _propTypes["default"].number
};
/** Work around to fix storybook HOC propTypes bug  - DO NOT IMPORT THIS */

var StepDividerComponent = StepDivider;
exports.StepDividerComponent = StepDividerComponent;

var _default = (0, _reactJss["default"])(styles)(StepDivider);

exports["default"] = _default;
