// @guestyci/foundation/legacy/Steps/Step.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.StepComponent = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactJss = _interopRequireDefault(require("react-jss"));

var styles = {
  step: {
    minWidth: '22px',
    minHeight: '22px',
    borderRadius: '50%'
  }
};

var Step = function Step(_ref) {
  var _cn;

  var children = _ref.children,
      active = _ref.active,
      color = _ref.color,
      classes = _ref.classes;
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])('flex-center', 'steps-step', 'font-size-md', 'b-1', "text-".concat(color), classes.step, (_cn = {}, (0, _defineProperty2["default"])(_cn, "bg-".concat(color), active), (0, _defineProperty2["default"])(_cn, 'text-white', active), _cn)),
    "data-qa": "step"
  }, children);
};

Step.defaultProps = {
  color: 'blue'
};
Step.propTypes = {
  /** active will show if current stip is active */
  active: _propTypes["default"].bool.isRequired,

  /** children will be the data inside the step circle */
  children: _propTypes["default"].string.isRequired,

  /** color for the text content and background when it is current */
  color: _propTypes["default"].string
};
/** Work around to fix storybook HOC propTypes bug  - DO NOT IMPORT THIS */

var StepComponent = Step;
exports.StepComponent = StepComponent;

var _default = (0, _reactJss["default"])(styles)(Step);

exports["default"] = _default;
