// @guestyci/foundation/es/legacy/Accordion/AccordionChevron.js
import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import SvgIcon from "../SvgIcon/SvgIcon";

var BtnArrowDown = function BtnArrowDown(props) {
  return /*#__PURE__*/React.createElement("svg", props, /*#__PURE__*/React.createElement("path", {
    d: "M4.533 4.997c.13.006.26-.039.357-.136L8.925.826a.488.488 0 0 0-.69-.69L4.532 3.843.826.137a.488.488 0 0 0-.69.69L4.172 4.86c.097.097.23.143.362.136z"
  }));
};

BtnArrowDown.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 9.0618038 4.9977479"
};

const AccordionChevron = ({
  isOpen,
  isActive,
  className
}) => /*#__PURE__*/React.createElement(SvgIcon, {
  svg: BtnArrowDown,
  className: cn('relative', className, {
    'transform-180': isOpen
  }),
  height: 9,
  width: 9,
  color: isOpen || isActive ? 'blue' : 'gray-dark',
  "data-qa": "accordion-chevron"
});

AccordionChevron.defaultProps = {
  isActive: false,
  className: ''
};
AccordionChevron.propTypes = {
  /** Indicator whether or not collapse should be open */
  isOpen: PropTypes.bool.isRequired,

  /** Indicator whether or not we are in active state */
  isActive: PropTypes.bool,

  /** Additional class name to chevron */
  className: PropTypes.string
};
export default AccordionChevron;
