// @guestyci/foundation/es/legacy/Avatar/ImageAvatar.js
import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import { createUseStyles } from 'react-jss';
export const getClass = size => {
  switch (size) {
    case 'xl':
      return '80px';

    case 'md':
      return '40px';

    case 'lg':
      return '60px';

    case 'sm':
      return '25px';

    default:
      return '40px';
  }
};
const useStyles = createUseStyles({
  icon: {
    height: ({
      size
    }) => getClass(size),
    width: ({
      size
    }) => getClass(size)
  }
});

const ImageAvatar = ({
  text,
  src,
  size = 'md',
  bordered,
  disabled
}) => {
  const classes = useStyles({
    size
  });
  return /*#__PURE__*/React.createElement("img", {
    className: cn(`img-avatar avatar`, 'd-inline-block', {
      'avatar-border': bordered
    }, {
      disabled
    }, classes.icon),
    src: src,
    alt: text ? text.trim() : '',
    "data-qa": "image-avatar"
  });
};

ImageAvatar.propTypes = {
  src: PropTypes.string.isRequired,
  size: PropTypes.oneOf(['sm', 'md', 'lg', 'xl']),
  disabled: PropTypes.bool,
  text: PropTypes.string,
  bordered: PropTypes.bool
};
export default ImageAvatar;
