// @guestyci/foundation/legacy/LabelDropdown/LabelDropdownInputField.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _AccordionChevron = _interopRequireDefault(require("../Accordion/AccordionChevron"));

var _Row = _interopRequireDefault(require("../Row/Row"));

var LabelDropdownInputField = function LabelDropdownInputField(_ref) {
  var label = _ref.label,
      isOpen = _ref.isOpen,
      placeholder = _ref.placeholder,
      onClick = _ref.onClick;
  return /*#__PURE__*/_react["default"].createElement("div", {
    role: "presentation",
    className: "py-2 clickable",
    onClick: onClick,
    "data-qa": "label-dropdown-input-field"
  }, /*#__PURE__*/_react["default"].createElement(_Row["default"], {
    spacing: "sm",
    align: "center"
  }, label ? /*#__PURE__*/_react["default"].createElement("div", {
    className: "text-blue overflow-ellipsis label-selected-item"
  }, label) : /*#__PURE__*/_react["default"].createElement("div", {
    className: "text-gray-dark overflow-ellipsis"
  }, placeholder), /*#__PURE__*/_react["default"].createElement(_AccordionChevron["default"], {
    isOpen: isOpen
  })));
};

LabelDropdownInputField.defaultProps = {
  label: null,
  placeholder: 'Select'
};
LabelDropdownInputField.propTypes = {
  label: _propTypes["default"].node,
  placeholder: _propTypes["default"].string,
  isOpen: _propTypes["default"].bool.isRequired,
  onClick: _propTypes["default"].func.isRequired
};
var _default = LabelDropdownInputField;
exports["default"] = _default;
