// @guestyci/foundation/legacy/ExpandingSearchBar/ExpandingSearchBar.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _noop = _interopRequireDefault(require("lodash/noop"));

var _SearchBar = _interopRequireDefault(require("../SearchBar/SearchBar"));

var _IconButton = _interopRequireDefault(require("../IconButton"));

var _Fade = _interopRequireDefault(require("../Fade/Fade"));

function _createSuper(Derived) { return function () { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

var BtnSearch = function BtnSearch(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M19.96 20.243l-5.995-5.995c.049-.046.1-.086.149-.134A8.269 8.269 0 0 0 2.422 2.422a8.267 8.267 0 0 0 0 11.692c2.939 2.939 7.538 3.193 10.775.779l6.057 6.057.707-.707zM3.13 13.407A7.268 7.268 0 1 1 13.408 3.129a7.268 7.268 0 0 1-10.28 10.278z"
  }));
};

BtnSearch.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 19.9607 20.9498"
};

var ExpandingSearchBar = /*#__PURE__*/function (_Component) {
  (0, _inherits2["default"])(ExpandingSearchBar, _Component);

  var _super = _createSuper(ExpandingSearchBar);

  function ExpandingSearchBar(props) {
    var _this;

    (0, _classCallCheck2["default"])(this, ExpandingSearchBar);
    _this = _super.call(this, props);

    _this.closeSearchBar = function () {
      _this.setState({
        isShowSearch: false
      });

      var onToggleSearch = _this.props.onToggleSearch;

      if (onToggleSearch) {
        onToggleSearch(false);
      }
    };

    _this.showSearch = function () {
      _this.setState({
        isShowSearch: true
      });

      var onToggleSearch = _this.props.onToggleSearch;

      if (onToggleSearch) {
        onToggleSearch(true);
      }

      setTimeout(function () {
        _this.inputWrapper.current.getElementsByClassName('search-bar-input')[0].focus();
      }, 300);
    };

    _this.inputWrapper = _react["default"].createRef();
    _this.state = {
      isShowSearch: props.initialIsOpen
    };
    return _this;
  }

  (0, _createClass2["default"])(ExpandingSearchBar, [{
    key: "render",
    value: function render() {
      var isShowSearch = this.state.isShowSearch;
      var _this$props = this.props,
          defaultValue = _this$props.defaultValue,
          onSubmit = _this$props.onSubmit,
          onChange = _this$props.onChange,
          className = _this$props.className,
          style = _this$props.style,
          noBorder = _this$props.noBorder,
          containerClassName = _this$props.containerClassName,
          containerStyle = _this$props.containerStyle,
          placeholder = _this$props.placeholder;
      return /*#__PURE__*/_react["default"].createElement("div", {
        className: (0, _classnames["default"])('d-flex-fill', 'align-items-center', 'relative', 'expandable-search-bar-wrapper', containerClassName),
        style: containerStyle,
        "data-qa": "expanding-search-bar"
      }, /*#__PURE__*/_react["default"].createElement("div", {
        ref: this.inputWrapper,
        className: "d-flex-fill expandable-search-bar slide-open ".concat(isShowSearch ? 'search-expand' : 'search-collapse')
      }, /*#__PURE__*/_react["default"].createElement(_SearchBar["default"], {
        showDelete: true,
        defaultValue: defaultValue,
        placeholder: placeholder,
        onSubmit: onSubmit,
        onChange: onChange,
        onClear: this.closeSearchBar,
        style: style,
        className: (0, _classnames["default"])('mt-1', className),
        noBorder: noBorder
      })), /*#__PURE__*/_react["default"].createElement("div", {
        className: "absolute right-0"
      }, /*#__PURE__*/_react["default"].createElement(_Fade["default"], {
        timeout: {
          enter: 225,
          exit: 225
        },
        speed: "slow",
        show: !isShowSearch
      }, /*#__PURE__*/_react["default"].createElement(_IconButton["default"], {
        iconHeight: 18,
        iconWidth: 17,
        svg: BtnSearch,
        onClick: this.showSearch
      }))));
    }
  }]);
  return ExpandingSearchBar;
}(_react.Component);

ExpandingSearchBar.defaultProps = {
  onChange: _noop["default"],
  onToggleSearch: _noop["default"],
  onSubmit: _noop["default"],
  className: '',
  style: {},
  noBorder: false,
  containerClassName: '',
  containerStyle: {},
  placeholder: '',
  initialIsOpen: false
};
ExpandingSearchBar.propTypes = {
  /** Callback function to handle search bar submit */
  onSubmit: _propTypes["default"].func,

  /** Callback function to handle search bar open and close toggle */
  onToggleSearch: _propTypes["default"].func,

  /** Callback function to handle searchbar value change */
  onChange: _propTypes["default"].func,

  /** Initial input content */
  defaultValue: _propTypes["default"].string,

  /** Additional class for searchbar root */
  className: _propTypes["default"].string,

  /** Additional style for searchbar root */
  style: _propTypes["default"].shape(),

  /** Indicator whether or not to show search border bottom */
  noBorder: _propTypes["default"].bool,

  /** Additional class or root level */
  containerClassName: _propTypes["default"].string,

  /** Additional style for root level */
  containerStyle: _propTypes["default"].shape(),

  /** placeholder for search bar */
  placeholder: _propTypes["default"].string,

  /** Initial value for isShowSearch */
  initialIsOpen: _propTypes["default"].bool
};
var _default = ExpandingSearchBar;
exports["default"] = _default;
