// @guestyci/foundation/legacy/CheckboxGroup/CheckboxGroup.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.CheckboxGroupComponent = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _withSelectionGroup = _interopRequireDefault(require("../baseSelectionControl/withSelectionGroup"));

function _createSuper(Derived) { return function () { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

var CheckboxGroup = /*#__PURE__*/function (_Component) {
  (0, _inherits2["default"])(CheckboxGroup, _Component);

  var _super = _createSuper(CheckboxGroup);

  function CheckboxGroup() {
    var _this;

    (0, _classCallCheck2["default"])(this, CheckboxGroup);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _super.call.apply(_super, [this].concat(args));

    _this.changeCheckboxStatus = function (value, childName, checkedRadioId) {
      var onOptionClicked = _this.props.onOptionClicked;
      onOptionClicked(checkedRadioId);
    };

    _this.addPropsToChild = function (child, hasRightBorder) {
      return _react["default"].cloneElement(child.child, {
        key: child.uid,
        onChange: _this.changeCheckboxStatus,
        checked: child.checked,
        id: child.uid,
        className: child.child.props.className,
        borderRight: hasRightBorder
      });
    };

    return _this;
  }

  (0, _createClass2["default"])(CheckboxGroup, [{
    key: "componentDidUpdate",
    value: function componentDidUpdate() {
      var _this$props = this.props,
          onChange = _this$props.onChange,
          selectedCheckboxesWhenMulti = _this$props.selectedCheckboxesWhenMulti;
      onChange(selectedCheckboxesWhenMulti);
    }
  }, {
    key: "wrapCheckedChildrenWithDivs",
    value: function wrapCheckedChildrenWithDivs(childrenStructure) {
      var _this2 = this;

      var indexes = childrenStructure.map(function (child) {
        return child.checked ? 1 : 0;
      });
      var checkedChildren = [];
      var fineStructure = [];
      indexes.forEach(function (value, index) {
        if (checkedChildren.length > 0 && value === 0) {
          fineStructure.push( /*#__PURE__*/_react["default"].createElement("div", {
            className: (0, _classnames["default"])('d-flex', 'box-shadow', 'z-index-1'),
            key: "CheckboxGroup_".concat(childrenStructure[index].uid)
          }, checkedChildren));
          fineStructure.push(_this2.addPropsToChild(childrenStructure[index], index !== indexes.length - 1));
          checkedChildren = [];
        } else if (value === 0) {
          fineStructure.push(_this2.addPropsToChild(childrenStructure[index], index !== indexes.length - 1));
        } else {
          checkedChildren.push(_this2.addPropsToChild(childrenStructure[index], index !== indexes.length - 1));

          if (indexes.length - 1 === index) {
            fineStructure.push( /*#__PURE__*/_react["default"].createElement("div", {
              className: (0, _classnames["default"])('d-flex', 'box-shadow', 'z-index-1'),
              key: "CheckboxGroup_".concat(childrenStructure[index].uid)
            }, checkedChildren));
          }
        }
      });
      return fineStructure;
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props2 = this.props,
          options = _this$props2.options,
          className = _this$props2.className;
      var children = Array.from(options.values());
      return /*#__PURE__*/_react["default"].createElement("div", {
        className: className,
        "data-qa": "checkbox-group"
      }, /*#__PURE__*/_react["default"].createElement("div", {
        className: (0, _classnames["default"])('row', 'border', 'border-radius-2')
      }, this.wrapCheckedChildrenWithDivs(children)));
    }
  }]);
  return CheckboxGroup;
}(_react.Component);

CheckboxGroup.propTypes = {
  /** children will be rendered inside the CheckboxGroup */
  children: _propTypes["default"].node.isRequired,

  /** classes to be added to the component */
  className: _propTypes["default"].string,

  /** callback for on checkboxGroup change */
  onChange: _propTypes["default"].func.isRequired
};
var CheckboxGroupComponent = CheckboxGroup;
exports.CheckboxGroupComponent = CheckboxGroupComponent;

var _default = (0, _withSelectionGroup["default"])(CheckboxGroup);

exports["default"] = _default;
