// @guestyci/foundation/esm/legacy/DynamicField/DynamicField.js
import React from 'react';
import PropTypes from 'prop-types';
import noop from 'lodash/noop';
import memoize from 'lodash/memoize';
import Input from "../Input/Input";
import Checkbox from "../Checkbox/Checkbox";
import NumberPicker from "../NumberPicker/NumberPicker";
import TimePicker from "../TimePicker/TimePicker";
import Dropdown from "../Dropdown/Dropdown";
import TextArea from "../TextArea/TextArea";
import Person from "../Person/Person";
import InputLabel from "../InputLabel/InputLabel";
import DatePicker from "../DatePicker/DatePicker";
import { DatePickerType } from "../../enums/enums";

var createPerson = function createPerson(data) {
  return /*#__PURE__*/React.createElement(Person, {
    name: data.fullName,
    size: "sm",
    "data-qa": "create-person"
  });
};

var getPersonOptions = function getPersonOptions() {
  var options = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : [];
  return options.map(function (option) {
    return {
      label: createPerson(option),
      value: option._id,
      name: option.fullName
    };
  });
};

var getOptions = function getOptions() {
  var options = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : [];
  return options.map(function (option) {
    return {
      label: option,
      value: option
    };
  });
};

var memoizeGetOptions = memoize(getOptions);
var memoizeGetPersonOptions = memoize(getPersonOptions);

var getLabel = function getLabel(type, label) {
  return type === 'boolean' ? 'A yes no issue' : label;
};

var getField = function getField(props) {
  var type = props.type,
      _onChange = props.onChange,
      value = props.value,
      name = props.name,
      id = props.id,
      label = props.label,
      options = props.options,
      onLoadMore = props.onLoadMore,
      pageCount = props.pageCount,
      floatingPoint = props.floatingPoint,
      isFetching = props.isFetching,
      searchable = props.searchable,
      onSearchQueryChange = props.onSearchQueryChange;

  switch (type) {
    case 'boolean':
      return /*#__PURE__*/React.createElement("div", {
        className: "row w-100 pl-3"
      }, /*#__PURE__*/React.createElement(Checkbox, {
        name: name,
        id: id,
        onChange: function onChange(val, nameVal) {
          return _onChange(!val, nameVal);
        },
        value: value,
        checked: !!value
      }, label));

    case 'number':
      return /*#__PURE__*/React.createElement("div", {
        className: "row w-100 pl-3 d-block"
      }, /*#__PURE__*/React.createElement(NumberPicker, {
        floatingPoint: floatingPoint,
        value: value,
        name: name,
        id: id,
        onChange: _onChange
      }));

    case 'time':
      return /*#__PURE__*/React.createElement("div", {
        className: "row w-100 pl-3 d-block"
      }, /*#__PURE__*/React.createElement(TimePicker, {
        onChange: _onChange,
        name: name,
        id: id,
        value: value
      }));

    case 'longtext':
      return /*#__PURE__*/React.createElement("div", {
        className: "row w-100 pl-3 d-block"
      }, /*#__PURE__*/React.createElement(TextArea, {
        onChange: _onChange,
        name: name,
        id: id,
        value: value
      }));

    case 'enum':
      {
        var dropDownOptions = memoizeGetOptions(options);
        return /*#__PURE__*/React.createElement("div", {
          className: "row w-100 pl-3 d-block"
        }, /*#__PURE__*/React.createElement(Dropdown, {
          options: dropDownOptions,
          onSelect: _onChange,
          name: name,
          id: id,
          defaultValue: value,
          onLoadMore: onLoadMore,
          pageCount: pageCount,
          isFetching: isFetching,
          searchable: searchable,
          onSearchQueryChange: onSearchQueryChange
        }));
      }

    case 'user':
    case 'contact':
      {
        var _dropDownOptions = memoizeGetPersonOptions(options);

        return /*#__PURE__*/React.createElement("div", {
          className: "row w-100 pl-3 d-block"
        }, /*#__PURE__*/React.createElement(Dropdown, {
          options: _dropDownOptions,
          onSelect: _onChange,
          name: name,
          id: id,
          defaultValue: value,
          onLoadMore: onLoadMore,
          pageCount: pageCount,
          isFetching: isFetching,
          searchable: searchable,
          onSearchQueryChange: onSearchQueryChange
        }));
      }

    case 'date':
      return /*#__PURE__*/React.createElement(DatePicker, {
        type: DatePickerType.Single,
        value: value,
        onChange: _onChange,
        id: id
      });

    case 'text':
    default:
      return /*#__PURE__*/React.createElement("div", {
        className: "row w-100 d-block pl-3"
      }, /*#__PURE__*/React.createElement(Input, {
        value: value,
        name: name,
        id: id,
        onChange: _onChange,
        className: "pr-3"
      }));
  }
};

var DynamicField = function DynamicField(props) {
  var type = props.type,
      label = props.label;
  var fieldElement = getField(props);
  var inputLabel = getLabel(type, label);
  return /*#__PURE__*/React.createElement("div", {
    className: "form-group-body",
    "data-qa": "dynamic-field"
  }, /*#__PURE__*/React.createElement("div", {
    className: "mb-6 form-field"
  }, /*#__PURE__*/React.createElement(InputLabel, {
    className: "dynamic-field-label",
    label: inputLabel
  }), fieldElement));
};

DynamicField.defaultProps = {
  onChange: noop,
  value: undefined,
  label: undefined,
  options: undefined,
  type: 'text',
  id: undefined,
  name: '',
  pageCount: 0,
  onLoadMore: noop,
  isFetching: false,
  searchable: false,
  onSearchQueryChange: null
};
DynamicField.propTypes = {
  /** Function to be called on value change */
  onChange: PropTypes.func,

  /** Element type to render */
  type: PropTypes.string,

  /** A label to show on top of the input */
  label: PropTypes.string,

  /** Controlled input value to be set to input */
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool, PropTypes.instanceOf(Date), PropTypes.shape()]),

  /** The dropdown options to show Can be array of strings or Objects Structure ({
   _id: user/contact id,
   fullName: user/contact full Name,
  }) */
  options: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.string, PropTypes.object])),

  /** Name indicator for field */
  name: PropTypes.string,

  /** id indicator for field */
  id: PropTypes.string,

  /** Callback to handle pagination fetch more */
  onLoadMore: PropTypes.func,

  /** total pages for pagination  */
  pageCount: PropTypes.number,

  /** Indicator whether, or not we are in fetching mode in pagination */
  isFetching: PropTypes.bool,

  /** Set if or not support dropdown search */
  searchable: PropTypes.bool,

  /** Callback to handle search query change */
  onSearchQueryChange: PropTypes.func
};
export default DynamicField;
