// @guestyci/foundation/legacy/Table/PersonCell/Placeholder.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactJss = require("react-jss");

var _Placeholder = _interopRequireDefault(require("../../Placeholder"));

var useStyles = (0, _reactJss.createUseStyles)({
  placeholderBase: {
    height: 25,
    minWidth: 25,
    borderRadius: '50%',
    marginRight: '10px'
  }
});

var Placeholder = function Placeholder() {
  var classes = useStyles();
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: "d-flex align-items-center",
    "data-qa": "placeholder"
  }, /*#__PURE__*/_react["default"].createElement(_Placeholder["default"], {
    className: classes.placeholderBase
  }), /*#__PURE__*/_react["default"].createElement(_Placeholder["default"], {
    className: "height-10 w-fill"
  }));
};

var _default = _react["default"].memo(Placeholder);

exports["default"] = _default;
