// @guestyci/foundation/NumberPicker/NumberPickerArrows.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.useStyles = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _createStyles = _interopRequireDefault(require("../createStyles"));

var _Icon = _interopRequireDefault(require("../legacy/Icon"));

var _Button = _interopRequireDefault(require("../Button"));

var _Col = _interopRequireDefault(require("../Layout/Col"));

var _colors = require("../theme/colors");

var BtnArrowUp = function BtnArrowUp(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M4.529 0a.474.474 0 0 0-.357.137L.137 4.172a.488.488 0 0 0 .69.689L4.53 1.156l3.705 3.705a.488.488 0 0 0 .69-.69L4.89.138A.477.477 0 0 0 4.53 0z"
  }));
};

BtnArrowUp.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 9.0618038 4.9977479"
};

var BtnArrowDown = function BtnArrowDown(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M4.533 4.997c.13.006.26-.039.357-.136L8.925.826a.488.488 0 0 0-.69-.69L4.532 3.843.826.137a.488.488 0 0 0-.69.69L4.172 4.86c.097.097.23.143.362.136z"
  }));
};

BtnArrowDown.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 9.0618038 4.9977479"
};
var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    arrowsContainer: {
      height: '100%',
      flex: '0 0 46px'
    },
    arrow: {
      flexGrow: 1,
      borderRadius: 0,
      fill: theme.palette.secondary["default"],
      transition: theme.transition.create('fill'),
      '&:hover': {
        fill: theme.palette.primary["default"]
      }
    },
    disabled: {
      fill: _colors.blueGray100,
      '&:hover': {
        fill: _colors.blueGray100
      }
    },
    upArrow: {
      borderLeft: "solid 1px ".concat(theme.palette.border),
      borderBottom: "solid 1px ".concat(theme.palette.border)
    },
    downArrow: {
      borderLeft: "solid 1px ".concat(theme.palette.border)
    }
  };
});
exports.useStyles = useStyles;

var NumberPickerArrows = function NumberPickerArrows(_ref) {
  var disableDec = _ref.disableDec,
      disableInc = _ref.disableInc,
      onInc = _ref.onInc,
      onDec = _ref.onDec;
  var classes = useStyles();
  return /*#__PURE__*/_react["default"].createElement(_Col["default"], {
    className: classes.arrowsContainer,
    "data-qa": "number-picker-arrows"
  }, /*#__PURE__*/_react["default"].createElement(_Button["default"], {
    className: (0, _classnames["default"])(classes.arrow, classes.upArrow, (0, _defineProperty2["default"])({}, classes.disabled, disableInc)),
    disabled: disableInc,
    type: "button",
    onClick: onInc,
    size: "xs"
  }, /*#__PURE__*/_react["default"].createElement(_Icon["default"], {
    svg: BtnArrowUp,
    width: 10,
    height: 10
  })), /*#__PURE__*/_react["default"].createElement(_Button["default"], {
    className: (0, _classnames["default"])(classes.arrow, classes.downArrow, (0, _defineProperty2["default"])({}, classes.disabled, disableDec)),
    disabled: disableDec,
    type: "button",
    onClick: onDec,
    size: "xs"
  }, /*#__PURE__*/_react["default"].createElement(_Icon["default"], {
    svg: BtnArrowDown,
    width: 10,
    height: 10
  })));
};

var _default = NumberPickerArrows;
exports["default"] = _default;
