// @guestyci/foundation/legacy/ButtonDropdown/DropdownButtonField.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _enums = require("../../enums/enums");

var _OutlineButton = _interopRequireDefault(require("../OutlineButton/OutlineButton"));

var _SvgIcon = _interopRequireDefault(require("../SvgIcon/SvgIcon"));

var BtnArrowDown = function BtnArrowDown(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M4.533 4.997c.13.006.26-.039.357-.136L8.925.826a.488.488 0 0 0-.69-.69L4.532 3.843.826.137a.488.488 0 0 0-.69.69L4.172 4.86c.097.097.23.143.362.136z"
  }));
};

BtnArrowDown.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 9.0618038 4.9977479"
};

var DropdownButtonField = function DropdownButtonField(_ref) {
  var toggleCollapse = _ref.toggleCollapse,
      isCollapsed = _ref.isCollapsed,
      className = _ref.className,
      type = _ref.type,
      title = _ref.title,
      disabled = _ref.disabled,
      inDialog = _ref.inDialog;
  return /*#__PURE__*/_react["default"].createElement(_OutlineButton["default"], {
    onClick: toggleCollapse,
    color: type,
    disabled: disabled,
    className: (0, _classnames["default"])(className, 'width-125 height-25'),
    childrenClassName: inDialog ? '' : 'w-fill',
    "data-qa": "dropdown-button-field"
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: "d-flex-fill flex-space-between-center"
  }, title, !disabled && /*#__PURE__*/_react["default"].createElement(_SvgIcon["default"], {
    svg: BtnArrowDown,
    height: 9,
    width: 9,
    color: _enums.DropdownButtonColors[type],
    className: (0, _classnames["default"])('transition-transform ml-2', {
      'transform-180': !isCollapsed
    })
  })));
};

DropdownButtonField.defaultProps = {
  className: '',
  isCollapsed: true,
  onClick: null,
  type: 'plain',
  disabled: false
};
DropdownButtonField.propTypes = {
  /** Button click callback function */
  onClick: _propTypes["default"].func,

  /** Button additional classes */
  className: _propTypes["default"].string,

  /** boolean indicates whether the dropdown menu is open or not */
  isCollapsed: _propTypes["default"].bool,

  /** button type (sets the color) */
  type: _propTypes["default"].oneOf(Object.keys(_enums.DropdownButtonColors)),

  /** button title */
  title: _propTypes["default"].string.isRequired,

  /** button disable indicator */
  disabled: _propTypes["default"].bool
};
var _default = DropdownButtonField;
exports["default"] = _default;
