// @guestyci/foundation/legacy/Table/HeaderCell/HeaderCell.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _noop = _interopRequireDefault(require("lodash/noop"));

var _useHover3 = _interopRequireDefault(require("../../../useHover"));

var _useComposeRefs = _interopRequireDefault(require("../../../useComposeRefs"));

var _Row = _interopRequireDefault(require("../../../Layout/Row"));

var _Fade = _interopRequireDefault(require("../../../Fade"));

var _Icon = _interopRequireDefault(require("../../../Icon"));

var _TextField = _interopRequireDefault(require("../../../TextField"));

var _Divider = _interopRequireDefault(require("../../../Divider"));

var _SortIndicator = _interopRequireDefault(require("./SortIndicator"));

var _useDrag3 = _interopRequireDefault(require("../../../useDrag"));

var _useResize3 = _interopRequireDefault(require("../../../useResize"));

var _Popover = _interopRequireDefault(require("../../Popover"));

var _commonUtility = require("../../../utils/commonUtility");

var _createStyles = _interopRequireDefault(require("../../../createStyles"));

var _colors = require("../../../theme/colors");

var IcoDraggable = function IcoDraggable(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("g", {
    fill: "#707070",
    fillRule: "evenodd",
    opacity: ".53"
  }, /*#__PURE__*/_react["default"].createElement("circle", {
    cx: "1.5",
    cy: "1.5",
    r: "1.5"
  }), /*#__PURE__*/_react["default"].createElement("circle", {
    cx: "5.5",
    cy: "1.5",
    r: "1.5"
  }), /*#__PURE__*/_react["default"].createElement("circle", {
    cx: "1.5",
    cy: "5.5",
    r: "1.5"
  }), /*#__PURE__*/_react["default"].createElement("circle", {
    cx: "5.5",
    cy: "5.5",
    r: "1.5"
  }), /*#__PURE__*/_react["default"].createElement("circle", {
    cx: "1.5",
    cy: "9.5",
    r: "1.5"
  }), /*#__PURE__*/_react["default"].createElement("circle", {
    cx: "5.5",
    cy: "9.5",
    r: "1.5"
  })));
};

IcoDraggable.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  width: "7",
  height: "11",
  viewBox: "0 0 7 11"
};

var getHeaderCursor = function getHeaderCursor(draggable, isDragging, sortable) {
  if (isDragging) return 'grabbing';
  if (draggable || sortable) return 'pointer';
  return 'default';
};

var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    root: {
      borderBottom: "1px solid ".concat(theme.palette.text.primary),
      paddingTop: theme.spacer(2),
      paddingBottom: theme.spacer(2),
      textTransform: 'uppercase',
      backgroundColor: _colors.white,
      display: 'table-cell',
      transition: theme.transition.create('color'),
      zIndex: 10,
      color: theme.palette.placeholder,
      position: 'sticky',
      top: 0,
      width: function width(_ref) {
        var resizedWidth = _ref.resizedWidth;
        return resizedWidth;
      },
      left: function left(_ref2) {
        var positionLeft = _ref2.positionLeft;
        return positionLeft;
      },
      right: function right(_ref3) {
        var positionRight = _ref3.positionRight;
        return positionRight;
      },
      cursor: function cursor(_ref4) {
        var draggable = _ref4.draggable,
            isDragging = _ref4.isDragging,
            sortable = _ref4.sortable;
        return getHeaderCursor(draggable, isDragging, sortable);
      }
    },
    hint: {
      '&:hover': {
        color: theme.palette.text.secondary
      }
    },
    fixedCell: {
      borderTop: "1px solid ".concat(theme.palette.border),
      zIndex: 15
    },
    alignCenter: {
      textAlign: 'center'
    },
    alignLeft: {
      textAlign: 'left'
    },
    alignRight: {
      textAlign: 'right'
    },
    active: {
      color: theme.palette.text.primary
    },
    draggableIcon: {
      marginBottom: 3
    },
    resizable: {
      cursor: 'col-resize'
    },
    contentWrapper: {
      paddingLeft: theme.spacer(3),
      height: 30
    },
    popover: {
      width: '100%',
      overflow: 'hidden',
      textOverflow: 'ellipsis',
      whiteSpace: 'nowrap'
    },
    divider: {
      transition: theme.transition.create('border-color')
    },
    draggable: {
      borderColor: theme.palette.text.primary
    },
    title: {
      overflow: 'hidden',
      textOverflow: 'ellipsis'
    }
  };
}, {
  name: 'HeaderCell'
});

var HeaderCell = function HeaderCell(_ref5) {
  var _contentRef$current, _cn, _cn2;

  var children = _ref5.children,
      id = _ref5.id,
      width = _ref5.width,
      align = _ref5.align,
      fixed = _ref5.fixed,
      className = _ref5.className,
      positionLeft = _ref5.positionLeft,
      positionRight = _ref5.positionRight,
      draggable = _ref5.draggable,
      onSort = _ref5.onSort,
      sortable = _ref5.sortable,
      sortDirection = _ref5.sortDirection,
      style = _ref5.style,
      onDragDrop = _ref5.onDragDrop,
      resizable = _ref5.resizable,
      onResize = _ref5.onResize;

  var _useHover = (0, _useHover3["default"])(),
      _useHover2 = (0, _slicedToArray2["default"])(_useHover, 2),
      isHovering = _useHover2[0],
      hoverRef = _useHover2[1];

  var contentRef = (0, _react.useRef)(null);

  var _useResize = (0, _useResize3["default"])({
    startWidth: width,
    isResizable: resizable,
    id: id,
    onResizeEnd: onResize
  }),
      _useResize2 = (0, _slicedToArray2["default"])(_useResize, 2),
      resizedWidth = _useResize2[0],
      resizeRef = _useResize2[1];

  var _useDrag = (0, _useDrag3["default"])(id, onDragDrop, draggable),
      _useDrag2 = (0, _slicedToArray2["default"])(_useDrag, 2),
      isDragging = _useDrag2[0].isDragging,
      dragRef = _useDrag2[1];

  var classes = useStyles({
    positionLeft: positionLeft,
    positionRight: positionRight,
    draggable: draggable,
    isDragging: isDragging,
    resizedWidth: resizedWidth,
    sortable: sortable
  });
  var draggableRef = draggable ? dragRef : null;
  var headerCellRef = (0, _useComposeRefs["default"])(hoverRef, draggableRef);
  var handleCellClick = (0, _react.useCallback)(function (e) {
    e.stopPropagation();

    if (sortable && onSort) {
      var newOrder = sortDirection === 'desc' ? '' : '-';
      onSort("".concat(newOrder).concat(id));
    }
  }, [onSort, sortable, id, sortDirection]);
  var showSortHint = sortable && !sortDirection && isHovering;
  var showResizeIndicator = resizable && isHovering;
  var isOverflowed = ((_contentRef$current = contentRef.current) === null || _contentRef$current === void 0 ? void 0 : _contentRef$current.scrollWidth) > resizedWidth;
  return /*#__PURE__*/_react["default"].createElement("div", {
    ref: headerCellRef,
    role: "presentation",
    onClick: handleCellClick,
    className: (0, _classnames["default"])('header-cell', classes.root, (_cn = {}, (0, _defineProperty2["default"])(_cn, classes.active, sortDirection), (0, _defineProperty2["default"])(_cn, classes.hint, !sortDirection), (0, _defineProperty2["default"])(_cn, classes.fixedCell, fixed), (0, _defineProperty2["default"])(_cn, classes["align".concat((0, _commonUtility.capitalize)(align))], align), _cn), className),
    style: style,
    "data-qa": "header-cell"
  }, /*#__PURE__*/_react["default"].createElement(_Row["default"], {
    spacing: 2,
    align: "center",
    className: (0, _classnames["default"])('header-cell-content', classes.contentWrapper),
    fullWidth: true
  }, /*#__PURE__*/_react["default"].createElement(_Fade["default"], {
    show: draggable && isHovering,
    unmountOnExit: !draggable,
    className: (0, _classnames["default"])(classes.draggableIcon, 'drag-icon')
  }, /*#__PURE__*/_react["default"].createElement(_Icon["default"], {
    svg: IcoDraggable,
    height: 11,
    width: 7
  })), /*#__PURE__*/_react["default"].createElement(_Popover["default"], {
    isOpen: isOverflowed && isHovering,
    containerClassName: classes.popover,
    body: children
  }, /*#__PURE__*/_react["default"].createElement(_TextField["default"], {
    className: classes.title,
    wrap: false,
    ref: contentRef
  }, children)), /*#__PURE__*/_react["default"].createElement(_Fade["default"], {
    mountOnEnter: !sortable,
    show: sortDirection || showSortHint
  }, /*#__PURE__*/_react["default"].createElement(_SortIndicator["default"], {
    direction: sortDirection,
    isHint: showSortHint
  })), /*#__PURE__*/_react["default"].createElement(_Divider["default"], {
    ref: resizeRef,
    className: (0, _classnames["default"])('header-cell-divider', classes.divider, (_cn2 = {}, (0, _defineProperty2["default"])(_cn2, classes.resizable, resizable), (0, _defineProperty2["default"])(_cn2, classes.draggable, showResizeIndicator), _cn2)),
    orientation: "vertical",
    thickness: 1
  })));
};

HeaderCell.defaultProps = {
  sortable: false,
  resizable: false,
  onSort: undefined,
  className: '',
  style: {},
  onDragDrop: _noop["default"],
  onResize: _noop["default"],
  draggable: false
};
HeaderCell.propTypes = {
  children: _propTypes["default"].node.isRequired,
  width: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].number]),
  align: _propTypes["default"].string,
  resizable: _propTypes["default"].bool,
  sortable: _propTypes["default"].bool,
  sortDirection: _propTypes["default"].oneOf(['asc', 'desc']),
  onSort: _propTypes["default"].func,
  className: _propTypes["default"].string,
  style: _propTypes["default"].shape(),
  id: _propTypes["default"].string.isRequired,
  onDragDrop: _propTypes["default"].func,
  draggable: _propTypes["default"].bool,
  onResize: _propTypes["default"].func
};
var _default = HeaderCell;
exports["default"] = _default;
