// @guestyci/foundation/esm/legacy/SearchBar/SearchBar.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _possibleConstructorReturn from "@babel/runtime/helpers/esm/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/esm/getPrototypeOf";

function _createSuper(Derived) { return function () { var Super = _getPrototypeOf(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import noop from 'lodash/noop';
import debounce from 'lodash/debounce';

var BtnSearch = function BtnSearch(props) {
  return /*#__PURE__*/React.createElement("svg", props, /*#__PURE__*/React.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M19.96 20.243l-5.995-5.995c.049-.046.1-.086.149-.134A8.269 8.269 0 0 0 2.422 2.422a8.267 8.267 0 0 0 0 11.692c2.939 2.939 7.538 3.193 10.775.779l6.057 6.057.707-.707zM3.13 13.407A7.268 7.268 0 1 1 13.408 3.129a7.268 7.268 0 0 1-10.28 10.278z"
  }));
};

BtnSearch.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 19.9607 20.9498"
};

var BtnDelete = function BtnDelete(props) {
  return /*#__PURE__*/React.createElement("svg", props, /*#__PURE__*/React.createElement("path", {
    d: "M8.485.707L7.778 0 4.243 3.536.707 0 0 .707l3.536 3.536L0 7.778l.707.707L4.243 4.95l3.535 3.535.707-.707L4.95 4.243z"
  }));
};

BtnDelete.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 8.4854 8.4854"
};
import IconButton from "../IconButton";
import SvgIcon from "../SvgIcon";
import Row from "../Row/Row";
import Fade from "../Fade/Fade";

var SearchBar = /*#__PURE__*/function (_PureComponent) {
  _inherits(SearchBar, _PureComponent);

  var _super = _createSuper(SearchBar);

  function SearchBar(props) {
    var _this;

    _classCallCheck(this, SearchBar);

    _this = _super.call(this, props);
    _this.onInputChange = debounce(function (value) {
      var onChange = _this.props.onChange;
      onChange(value);
    }, 500);

    _this.handleInputChange = function (e) {
      var value = e.currentTarget.value;

      _this.setState({
        searchValue: value
      });

      var onChange = _this.props.onChange;

      if (onChange) {
        _this.onInputChange(value);
      }
    };

    _this.submit = function (event) {
      event.preventDefault();
      var onSubmit = _this.props.onSubmit;
      var searchValue = _this.state.searchValue;
      onSubmit(searchValue);
    };

    _this.clearSearch = function () {
      _this.setState({
        searchValue: ''
      });

      var _this$props = _this.props,
          onSubmit = _this$props.onSubmit,
          onClear = _this$props.onClear;
      onSubmit('');

      if (onClear) {
        onClear();
      }
    };

    _this.state = {
      searchValue: props.defaultValue
    };
    return _this;
  }

  _createClass(SearchBar, [{
    key: "render",
    value: function render() {
      var _this$props2 = this.props,
          className = _this$props2.className,
          showDelete = _this$props2.showDelete,
          noBorder = _this$props2.noBorder,
          placeholder = _this$props2.placeholder;
      var searchValue = this.state.searchValue;
      return /*#__PURE__*/React.createElement("form", {
        onSubmit: this.submit,
        className: cn('search-bar', 'w-fill', 'pb-1', 'border-gray-light', 'relative', {
          'bb-1': !noBorder
        }, className),
        "data-qa": "search-bar"
      }, /*#__PURE__*/React.createElement(Row, {
        spacing: "sm",
        align: "center",
        justify: "start",
        fullWidth: true
      }, /*#__PURE__*/React.createElement(SvgIcon, {
        svg: BtnSearch,
        fillColor: searchValue ? 'black' : 'gray-dark',
        height: 18,
        width: 17
      }), /*#__PURE__*/React.createElement("input", {
        className: "b-0 font-size-lg search-bar-input",
        placeholder: placeholder,
        onChange: this.handleInputChange,
        value: searchValue
      }), /*#__PURE__*/React.createElement(Fade, {
        mountOnEnter: true,
        show: !!(searchValue || showDelete)
      }, /*#__PURE__*/React.createElement(IconButton, {
        svg: BtnDelete,
        iconWidth: 10,
        iconHeight: 10,
        onClick: this.clearSearch,
        className: "p-1"
      }))));
    }
  }]);

  return SearchBar;
}(PureComponent);

SearchBar.defaultProps = {
  onChange: noop,
  onClear: noop,
  className: '',
  style: {},
  noBorder: false,
  placeholder: 'Type in to Search...',
  defaultValue: ''
};
SearchBar.propTypes = {
  /** Callback function to handle search bar submit */
  onSubmit: PropTypes.func.isRequired,

  /** Callback function to handle searchbar value change */
  onChange: PropTypes.func,

  /** Initial input content */
  defaultValue: PropTypes.string,

  /** Additional class for searchbar root */
  className: PropTypes.string,

  /** Additional style for earchbar root */
  style: PropTypes.shape(),

  /** Callback method to handle clear search */
  onClear: PropTypes.func,

  /** Indicator whether or not to show search border bottom */
  noBorder: PropTypes.bool,

  /** Placeholder for search  */
  placeholder: PropTypes.string
};
export default SearchBar;
