// @guestyci/foundation/esm/legacy/SearchBar/withSearch.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _possibleConstructorReturn from "@babel/runtime/helpers/esm/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/esm/getPrototypeOf";

function _createSuper(Derived) { return function () { var Super = _getPrototypeOf(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

import React, { Component } from 'react';
/**
 * Create a wrapper compose to support searchBar functionality
 * @param ComposedComponent
 * @returns {{Component}}
 */

function withSearch(ComposedComponent) {
  var SearchWrapper = /*#__PURE__*/function (_Component) {
    _inherits(SearchWrapper, _Component);

    var _super = _createSuper(SearchWrapper);

    function SearchWrapper(props) {
      var _this;

      _classCallCheck(this, SearchWrapper);

      _this = _super.call(this, props);

      _this.onSearchClose = function () {
        _this.setState({
          isSearchOpen: false
        });
      };

      _this.onSearchOpen = function () {
        _this.setState({
          isSearchOpen: true
        });
      };

      _this.onSearchToggle = function (val) {
        _this.setState({
          isSearchOpen: val
        });
      };

      _this.state = {
        isSearchOpen: false
      };
      return _this;
    }

    _createClass(SearchWrapper, [{
      key: "render",
      value: function render() {
        var isSearchOpen = this.state.isSearchOpen;
        return /*#__PURE__*/React.createElement(ComposedComponent, _extends({}, this.props, {
          isSearchOpen: isSearchOpen,
          onSearchClose: this.onSearchClose,
          onSearchOpen: this.onSearchOpen,
          onSearchToggle: this.onSearchToggle,
          "data-qa": "search-wrapper"
        }));
      }
    }]);

    return SearchWrapper;
  }(Component);

  return SearchWrapper;
}

export default withSearch;
