// @guestyci/foundation/legacy/SearchBar/SearchBar.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _noop = _interopRequireDefault(require("lodash/noop"));

var _debounce = _interopRequireDefault(require("lodash/debounce"));

var _IconButton = _interopRequireDefault(require("../IconButton"));

var _SvgIcon = _interopRequireDefault(require("../SvgIcon"));

var _Row = _interopRequireDefault(require("../Row/Row"));

var _Fade = _interopRequireDefault(require("../Fade/Fade"));

function _createSuper(Derived) { return function () { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

var BtnSearch = function BtnSearch(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M19.96 20.243l-5.995-5.995c.049-.046.1-.086.149-.134A8.269 8.269 0 0 0 2.422 2.422a8.267 8.267 0 0 0 0 11.692c2.939 2.939 7.538 3.193 10.775.779l6.057 6.057.707-.707zM3.13 13.407A7.268 7.268 0 1 1 13.408 3.129a7.268 7.268 0 0 1-10.28 10.278z"
  }));
};

BtnSearch.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 19.9607 20.9498"
};

var BtnDelete = function BtnDelete(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M8.485.707L7.778 0 4.243 3.536.707 0 0 .707l3.536 3.536L0 7.778l.707.707L4.243 4.95l3.535 3.535.707-.707L4.95 4.243z"
  }));
};

BtnDelete.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 8.4854 8.4854"
};

var SearchBar = /*#__PURE__*/function (_PureComponent) {
  (0, _inherits2["default"])(SearchBar, _PureComponent);

  var _super = _createSuper(SearchBar);

  function SearchBar(props) {
    var _this;

    (0, _classCallCheck2["default"])(this, SearchBar);
    _this = _super.call(this, props);
    _this.onInputChange = (0, _debounce["default"])(function (value) {
      var onChange = _this.props.onChange;
      onChange(value);
    }, 500);

    _this.handleInputChange = function (e) {
      var value = e.currentTarget.value;

      _this.setState({
        searchValue: value
      });

      var onChange = _this.props.onChange;

      if (onChange) {
        _this.onInputChange(value);
      }
    };

    _this.submit = function (event) {
      event.preventDefault();
      var onSubmit = _this.props.onSubmit;
      var searchValue = _this.state.searchValue;
      onSubmit(searchValue);
    };

    _this.clearSearch = function () {
      _this.setState({
        searchValue: ''
      });

      var _this$props = _this.props,
          onSubmit = _this$props.onSubmit,
          onClear = _this$props.onClear;
      onSubmit('');

      if (onClear) {
        onClear();
      }
    };

    _this.state = {
      searchValue: props.defaultValue
    };
    return _this;
  }

  (0, _createClass2["default"])(SearchBar, [{
    key: "render",
    value: function render() {
      var _this$props2 = this.props,
          className = _this$props2.className,
          showDelete = _this$props2.showDelete,
          noBorder = _this$props2.noBorder,
          placeholder = _this$props2.placeholder;
      var searchValue = this.state.searchValue;
      return /*#__PURE__*/_react["default"].createElement("form", {
        onSubmit: this.submit,
        className: (0, _classnames["default"])('search-bar', 'w-fill', 'pb-1', 'border-gray-light', 'relative', {
          'bb-1': !noBorder
        }, className),
        "data-qa": "search-bar"
      }, /*#__PURE__*/_react["default"].createElement(_Row["default"], {
        spacing: "sm",
        align: "center",
        justify: "start",
        fullWidth: true
      }, /*#__PURE__*/_react["default"].createElement(_SvgIcon["default"], {
        svg: BtnSearch,
        fillColor: searchValue ? 'black' : 'gray-dark',
        height: 18,
        width: 17
      }), /*#__PURE__*/_react["default"].createElement("input", {
        className: "b-0 font-size-lg search-bar-input",
        placeholder: placeholder,
        onChange: this.handleInputChange,
        value: searchValue
      }), /*#__PURE__*/_react["default"].createElement(_Fade["default"], {
        mountOnEnter: true,
        show: !!(searchValue || showDelete)
      }, /*#__PURE__*/_react["default"].createElement(_IconButton["default"], {
        svg: BtnDelete,
        iconWidth: 10,
        iconHeight: 10,
        onClick: this.clearSearch,
        className: "p-1"
      }))));
    }
  }]);
  return SearchBar;
}(_react.PureComponent);

SearchBar.defaultProps = {
  onChange: _noop["default"],
  onClear: _noop["default"],
  className: '',
  style: {},
  noBorder: false,
  placeholder: 'Type in to Search...',
  defaultValue: ''
};
SearchBar.propTypes = {
  /** Callback function to handle search bar submit */
  onSubmit: _propTypes["default"].func.isRequired,

  /** Callback function to handle searchbar value change */
  onChange: _propTypes["default"].func,

  /** Initial input content */
  defaultValue: _propTypes["default"].string,

  /** Additional class for searchbar root */
  className: _propTypes["default"].string,

  /** Additional style for earchbar root */
  style: _propTypes["default"].shape(),

  /** Callback method to handle clear search */
  onClear: _propTypes["default"].func,

  /** Indicator whether or not to show search border bottom */
  noBorder: _propTypes["default"].bool,

  /** Placeholder for search  */
  placeholder: _propTypes["default"].string
};
var _default = SearchBar;
exports["default"] = _default;
