// @guestyci/foundation/legacy/ProxyEmailInput/ProxyEmailInput.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.ProxyEmailInputComponent = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _react = _interopRequireWildcard(require("react"));

var _trim = _interopRequireDefault(require("lodash/trim"));

var _debounce = _interopRequireDefault(require("lodash/debounce"));

var _classnames = _interopRequireDefault(require("classnames"));

var _reactJss = _interopRequireDefault(require("react-jss"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Input = _interopRequireDefault(require("../Input/Input"));

var _IndicationIcon = _interopRequireDefault(require("./IndicationIcon"));

function _createSuper(Derived) { return function () { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

var style = {
  fadeOut: {
    opacity: 0,
    width: 0,
    transition: 'width 0.45s 0.45s, height 0.45s 0.45s, opacity 0.5s',
    borderLeft: 0
  },
  fadeIn: {
    opacity: 1,
    width: '46px',
    transition: 'width 0.45s, height 0.45s, opacity 0.5s'
  }
};
var DEBOUNCE_DELAY = 500;

var ProxyEmailInput = /*#__PURE__*/function (_Component) {
  (0, _inherits2["default"])(ProxyEmailInput, _Component);

  var _super = _createSuper(ProxyEmailInput);

  function ProxyEmailInput(props) {
    var _this;

    (0, _classCallCheck2["default"])(this, ProxyEmailInput);
    _this = _super.call(this, props);

    _this.onInputValueChange = function (value) {
      var inputField = _this.state.inputField;
      var trimmedValue = (0, _trim["default"])(value);

      if (inputField === trimmedValue) {
        return;
      }

      _this.setState({
        inputField: trimmedValue
      });

      _this.handleValueChange();
    };

    _this.state = {
      inputField: props.defaultInputValue
    };
    _this.handleValueChange = (0, _debounce["default"])(function () {
      var onChange = _this.props.onChange;
      var inputField = _this.state.inputField;
      onChange(inputField);
    }, DEBOUNCE_DELAY);
    return _this;
  }

  (0, _createClass2["default"])(ProxyEmailInput, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          placeholder = _this$props.placeholder,
          _this$props$classes = _this$props.classes,
          fadeIn = _this$props$classes.fadeIn,
          fadeOut = _this$props$classes.fadeOut,
          required = _this$props.required,
          maxLength = _this$props.maxLength,
          minLength = _this$props.minLength,
          domain = _this$props.domain,
          isValid = _this$props.isValid,
          isValidating = _this$props.isValidating,
          name = _this$props.name;
      var inputField = this.state.inputField;
      return /*#__PURE__*/_react["default"].createElement("div", {
        className: (0, _classnames["default"])('d-flex-fill', 'border', 'border-gray-lighter', 'bg-gray-lightest'),
        "data-qa": "proxy-email-input"
      }, /*#__PURE__*/_react["default"].createElement(_Input["default"], {
        onChange: this.onInputValueChange,
        name: name,
        value: inputField,
        containerClassName: "border-0",
        placeholder: placeholder,
        minLength: minLength,
        maxLength: maxLength,
        required: required
      }), /*#__PURE__*/_react["default"].createElement("div", {
        className: (0, _classnames["default"])('label', 'flex-center', 'bg-gray-lightest', 'px-4', 'font-size-md', 'letter-spacing-sm', 'bl-1', 'border-gray-light')
      }, "@".concat(domain)), /*#__PURE__*/_react["default"].createElement("div", {
        className: (0, _classnames["default"])(inputField.length > 0 ? fadeIn : fadeOut, 'd-flex-fill', 'bl-1', 'border-gray-light')
      }, /*#__PURE__*/_react["default"].createElement(_IndicationIcon["default"], {
        isValidating: isValidating,
        isValid: isValid
      })));
    }
  }], [{
    key: "getDerivedStateFromProps",
    value: function getDerivedStateFromProps(props, state) {
      var newState = null;

      if (props.value && props.value !== state.inputField) {
        newState = {
          inputField: props.value
        };
      }

      return newState;
    }
  }]);
  return ProxyEmailInput;
}(_react.Component);

ProxyEmailInput.defaultProps = {
  defaultInputValue: '',
  value: '',
  minLength: undefined,
  maxLength: undefined,
  placeholder: '5-30 characters, with no spaces',
  name: ''
};
ProxyEmailInput.propTypes = {
  /** An indicator for the icon indicator */
  isValid: _propTypes["default"].bool.isRequired,

  /** A default input value */
  defaultInputValue: _propTypes["default"].string,

  /** A value for controlled input field */
  value: _propTypes["default"].string,

  /** An indicator if the req is validating */
  isValidating: _propTypes["default"].bool.isRequired,

  /** Function to toggle the input changes */
  onChange: _propTypes["default"].func.isRequired,

  /** A placeholder */
  placeholder: _propTypes["default"].string,

  /** A domain placeholder */
  domain: _propTypes["default"].string.isRequired,

  /** Min input length */
  minLength: _propTypes["default"].number,

  /** Max input length */
  maxLength: _propTypes["default"].number,

  /** A unique name */
  name: _propTypes["default"].string
};
var ProxyEmailInputComponent = ProxyEmailInput;
exports.ProxyEmailInputComponent = ProxyEmailInputComponent;

var _default = (0, _reactJss["default"])(style)(ProxyEmailInput);

exports["default"] = _default;
