// @guestyci/foundation/EmailInput/EmailInput.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _noop = _interopRequireDefault(require("lodash/noop"));

var _InputActions = _interopRequireDefault(require("../Input/InputActions"));

var _Input = _interopRequireDefault(require("../Input"));

var _useHover3 = _interopRequireDefault(require("../useHover"));

var EmailInput = _react["default"].forwardRef(function EmailInput(_ref, ref) {
  var _ref$value = _ref.value,
      value = _ref$value === void 0 ? '' : _ref$value,
      _ref$disabled = _ref.disabled,
      disabled = _ref$disabled === void 0 ? false : _ref$disabled,
      _ref$name = _ref.name,
      name = _ref$name === void 0 ? '' : _ref$name,
      _ref$withActions = _ref.withActions,
      withActions = _ref$withActions === void 0 ? true : _ref$withActions,
      _ref$className = _ref.className,
      className = _ref$className === void 0 ? '' : _ref$className,
      _ref$placeholder = _ref.placeholder,
      placeholder = _ref$placeholder === void 0 ? '' : _ref$placeholder,
      _ref$onChange = _ref.onChange,
      onChange = _ref$onChange === void 0 ? _noop["default"] : _ref$onChange,
      _ref$onStar = _ref.onStar,
      onStar = _ref$onStar === void 0 ? _noop["default"] : _ref$onStar,
      _ref$onRemove = _ref.onRemove,
      onRemove = _ref$onRemove === void 0 ? _noop["default"] : _ref$onRemove,
      _ref$primary = _ref.primary,
      primary = _ref$primary === void 0 ? false : _ref$primary,
      _ref$style = _ref.style,
      style = _ref$style === void 0 ? {} : _ref$style,
      _ref$jss = _ref.jss,
      jss = _ref$jss === void 0 ? {} : _ref$jss,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["value", "disabled", "name", "withActions", "className", "placeholder", "onChange", "onStar", "onRemove", "primary", "style", "jss"]);

  var _useHover = (0, _useHover3["default"])(),
      _useHover2 = (0, _slicedToArray2["default"])(_useHover, 2),
      isHovering = _useHover2[0],
      hoverRef = _useHover2[1];

  return /*#__PURE__*/_react["default"].createElement("div", {
    ref: hoverRef
  }, /*#__PURE__*/_react["default"].createElement(_Input["default"], (0, _extends2["default"])({}, props, {
    ref: ref,
    onChange: onChange,
    style: style,
    className: className,
    disabled: disabled,
    value: value,
    type: "email",
    placeholder: placeholder,
    name: name,
    jss: jss,
    suffix: withActions && /*#__PURE__*/_react["default"].createElement(_InputActions["default"], {
      disabled: disabled,
      show: isHovering,
      primary: primary,
      onRemove: onRemove,
      onStar: onStar,
      name: name
    })
  })));
});

EmailInput.propTypes = {
  /** Cb for input change
   *  passing : value: {String}, target: {id or name defined}
   *  */
  onChange: _propTypes["default"].func,

  /**  Cb for clicking on the trash icon */
  onRemove: _propTypes["default"].func,

  /** Cb for clicking on the star (set primary) icon */
  onStar: _propTypes["default"].func,

  /** Indicator whether to show the star/trash actions */
  withActions: _propTypes["default"].bool,

  /** Boolean setting the email as primary */
  primary: _propTypes["default"].bool,

  /** Input level name property */
  name: _propTypes["default"].string,

  /** Email value - setting as controlled */
  value: _propTypes["default"].string,

  /** Placeholder for the email */
  placeholder: _propTypes["default"].string,

  /** Boolean to set the email and component as disabled */
  disabled: _propTypes["default"].bool,

  /** Additional class to the top component level */
  className: _propTypes["default"].string,

  /** Additional style to the top component level */
  style: _propTypes["default"].shape(),

  /** JSS override object */
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func])
};
EmailInput.displayName = 'EmailInput';
var _default = EmailInput;
exports["default"] = _default;
