// @guestyci/foundation/esm/legacy/Steps/StepDivider.js
import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import withStyle from 'react-jss';
var styles = {
  stepDivider: {
    position: 'relative'
  },
  stepDividerLine: function stepDividerLine(props) {
    return {
      height: '1px',
      width: "".concat(props.width, "px")
    };
  },
  stepDividerText: {
    position: 'absolute',
    bottom: '0',
    left: '50%',
    transform: 'translateX(-50%)',
    'text-align': 'center'
  }
};

var StepDivider = function StepDivider(_ref) {
  var text = _ref.text,
      color = _ref.color,
      classes = _ref.classes;
  return /*#__PURE__*/React.createElement("div", {
    className: ('flex-1', classes.stepDivider),
    "data-qa": "step-divider"
  }, /*#__PURE__*/React.createElement("span", {
    className: cn(classes.stepDividerText, "text-".concat(color))
  }, text), /*#__PURE__*/React.createElement("div", {
    className: cn(classes.stepDividerLine, "bg-".concat(color))
  }));
};

StepDivider.defaultProps = {
  text: '',
  color: 'gray',
  width: 19
};
StepDivider.propTypes = {
  text: PropTypes.string,
  color: PropTypes.string,
  width: PropTypes.number
};
/** Work around to fix storybook HOC propTypes bug  - DO NOT IMPORT THIS */

export var StepDividerComponent = StepDivider;
export default withStyle(styles)(StepDivider);
