// @guestyci/foundation/esm/legacy/Steps/Step.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import withStyle from 'react-jss';
var styles = {
  step: {
    minWidth: '22px',
    minHeight: '22px',
    borderRadius: '50%'
  }
};

var Step = function Step(_ref) {
  var _cn;

  var children = _ref.children,
      active = _ref.active,
      color = _ref.color,
      classes = _ref.classes;
  return /*#__PURE__*/React.createElement("div", {
    className: cn('flex-center', 'steps-step', 'font-size-md', 'b-1', "text-".concat(color), classes.step, (_cn = {}, _defineProperty(_cn, "bg-".concat(color), active), _defineProperty(_cn, 'text-white', active), _cn)),
    "data-qa": "step"
  }, children);
};

Step.defaultProps = {
  color: 'blue'
};
Step.propTypes = {
  /** active will show if current stip is active */
  active: PropTypes.bool.isRequired,

  /** children will be the data inside the step circle */
  children: PropTypes.string.isRequired,

  /** color for the text content and background when it is current */
  color: PropTypes.string
};
/** Work around to fix storybook HOC propTypes bug  - DO NOT IMPORT THIS */

export var StepComponent = Step;
export default withStyle(styles)(Step);
