// @guestyci/foundation/legacy/OutlineButton/OutlineButton.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _BaseButton = _interopRequireDefault(require("../BaseButton/BaseButton"));

var _enums = require("../../enums/enums");

var lightThemeClasses = 'text-white border-white text-hover-blue-dark-theme border-hover-blue-dark-theme';

var OutlineButton = function OutlineButton(_ref) {
  var children = _ref.children,
      onClick = _ref.onClick,
      className = _ref.className,
      style = _ref.style,
      disabled = _ref.disabled,
      isLoading = _ref.isLoading,
      theme = _ref.theme,
      childrenClassName = _ref.childrenClassName,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["children", "onClick", "className", "style", "disabled", "isLoading", "theme", "childrenClassName"]);
  return /*#__PURE__*/_react["default"].createElement(_BaseButton["default"], (0, _extends2["default"])({
    onClick: onClick,
    className: (0, _classnames["default"])('font-size-sm', className, (0, _defineProperty2["default"])({}, lightThemeClasses, theme === _enums.Theme.Light)),
    variant: "outline",
    style: style,
    disabled: disabled,
    isLoading: isLoading,
    loaderColor: "blue",
    childrenClassName: childrenClassName
  }, props, {
    "data-qa": "outline-button"
  }), children);
};

OutlineButton.defaultProps = {
  className: '',
  style: {},
  disabled: false,
  isLoading: false,
  theme: 'dark'
};
OutlineButton.propTypes = {
  /** Button click callback function */
  onClick: _propTypes["default"].func.isRequired,

  /** Button inner children */
  children: _propTypes["default"].node.isRequired,

  /** Button indicator whether disabled or not */
  disabled: _propTypes["default"].bool,

  /** Button additional classes */
  className: _propTypes["default"].string,

  /** Button additional styling */
  style: _propTypes["default"].shape(),

  /** Loader indicator for button */
  isLoading: _propTypes["default"].bool,

  /** Defined the theme for the button component */
  theme: _propTypes["default"].oneOf(['dark', 'light'])
};
var _default = OutlineButton;
exports["default"] = _default;
