// @guestyci/foundation/CSSTransition/CSSTransition.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactTransitionGroup = require("react-transition-group");

var CSSTransition = function CSSTransition(_ref) {
  var _children$props;

  var animation = _ref.animation,
      className = _ref.className,
      timeout = _ref.timeout,
      children = _ref.children,
      show = _ref.show,
      style = _ref.style,
      asGroup = _ref.asGroup,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["animation", "className", "timeout", "children", "show", "style", "asGroup"]);
  var transitionProps = !asGroup ? (0, _extends2["default"])({}, props, {
    "in": show
  }) : props;
  return animation ? /*#__PURE__*/_react["default"].createElement(_reactTransitionGroup.CSSTransition, (0, _extends2["default"])({}, transitionProps, {
    classNames: animation,
    timeout: timeout,
    style: style,
    className: (0, _classnames["default"])('css-transition', className, children === null || children === void 0 ? void 0 : (_children$props = children.props) === null || _children$props === void 0 ? void 0 : _children$props.className)
  }), children) : children;
};

CSSTransition.defaultProps = {
  className: '',
  timeout: 255,
  animation: '',
  show: undefined
};
CSSTransition.propTypes = {
  /* boolean flag whether or not to show the children */
  show: _propTypes["default"].bool,

  /* Animation for css transition */
  animation: _propTypes["default"].oneOfType([_propTypes["default"].shape({
    appear: _propTypes["default"].string,
    appearActive: _propTypes["default"].string,
    enter: _propTypes["default"].string.isRequired,
    enterActive: _propTypes["default"].string.isRequired,
    enterDone: _propTypes["default"].string,
    exitActive: _propTypes["default"].string.isRequired,
    exit: _propTypes["default"].string.isRequired,
    exitDone: _propTypes["default"].string
  }), _propTypes["default"].string]),

  /* Additional class name for css transition root */
  className: _propTypes["default"].string,

  /* Child component for css transition */
  children: _propTypes["default"].node.isRequired,

  /* timeout between animations - values: number or {enter, exit} */
  timeout: _propTypes["default"].oneOfType([_propTypes["default"].shape({
    enter: _propTypes["default"].number.isRequired,
    exit: _propTypes["default"].number.isRequired
  }), _propTypes["default"].number])
};
var _default = CSSTransition;
exports["default"] = _default;
