// @guestyci/foundation/esm/legacy/FormGroup/FormGroup.js
import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';

var FormGroup = function FormGroup(_ref) {
  var title = _ref.title,
      children = _ref.children,
      inline = _ref.inline,
      className = _ref.className,
      style = _ref.style;
  return /*#__PURE__*/React.createElement("div", {
    className: cn('form-group-wrapper d-flex-fill flex-column', className),
    style: style,
    "data-qa": "form-group"
  }, title && /*#__PURE__*/React.createElement("div", {
    className: "text-muted font-size-md form-group-title mb-4"
  }, title), /*#__PURE__*/React.createElement("div", {
    className: cn('form-group-body', 'd-flex-fill', {
      'flex-column': !inline
    })
  }, children));
};

FormGroup.defaultProps = {
  inline: false,
  className: '',
  style: {}
};
FormGroup.propTypes = {
  /** Form Group inner children */
  children: PropTypes.node.isRequired,

  /** Form Group Title */
  title: PropTypes.string.isRequired,

  /** Indicator whether form should be column or row */
  inline: PropTypes.bool,

  /** Class  to be added to FormGroup */
  className: PropTypes.string,

  /** Style to add to FormGroup */
  style: PropTypes.shape()
};
export default FormGroup;
