// @guestyci/foundation/legacy/VirtualTable/Column.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Placeholder = _interopRequireDefault(require("../Placeholder"));

var _HeaderRenderer = _interopRequireDefault(require("./HeaderRenderer"));

var Column = function Column() {};

Column.defaultProps = {
  placeholder: function placeholder(loading) {
    return /*#__PURE__*/_react["default"].createElement(_Placeholder["default"], {
      className: "height-10 w-100",
      animate: loading
    });
  },
  headerRenderer: _HeaderRenderer["default"]
};
Column.propTypes = {
  /** render function for header */
  headerRenderer: _propTypes["default"].func,

  /** render function for placeholder */
  placeholder: _propTypes["default"].func,

  /** Show/hide left border for the cell  */
  borderLeft: _propTypes["default"].bool,

  /** Show/hide right border for the cell */
  borderRight: _propTypes["default"].bool
};
var _default = Column;
exports["default"] = _default;
