// @guestyci/foundation/legacy/VirtualTable/HeaderRenderer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _SortDirection = _interopRequireDefault(require("react-virtualized/dist/commonjs/Table/SortDirection"));

var _classnames = _interopRequireDefault(require("classnames"));

var _reactJss = require("react-jss");

var _SvgIcon = _interopRequireDefault(require("../SvgIcon"));

var BtnSort = function BtnSort(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M0 0h8L4 4z"
  }));
};

BtnSort.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 8 4"
};
var useStyles = (0, _reactJss.createUseStyles)({
  sort: {
    opacity: 0
  },
  hover: {
    display: 'flex',
    alignItems: 'center',
    userSelect: 'none',
    '&:hover $sort': {
      opacity: 1,
      transition: 'opacity 0.2s ease-in'
    }
  }
});

var defaultHeaderRenderer = function defaultHeaderRenderer(props) {
  return /*#__PURE__*/_react["default"].createElement(HeaderRenderer, (0, _extends2["default"])({}, props, {
    "data-qa": "default-header-renderer"
  }));
};

var HeaderRenderer = function HeaderRenderer(_ref) {
  var dataKey = _ref.dataKey,
      disableSort = _ref.disableSort,
      label = _ref.label,
      sortBy = _ref.sortBy,
      sortDirection = _ref.sortDirection;
  var classes = useStyles();
  var showSortIndicator = sortBy === dataKey;
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: classes.hover,
    "data-qa": "header-renderer"
  }, /*#__PURE__*/_react["default"].createElement("span", {
    className: (0, _classnames["default"])('ReactVirtualized__Table__headerTruncatedText', {
      'text-black': showSortIndicator,
      'font-weight-bold': showSortIndicator
    }),
    title: label
  }, label), !disableSort && /*#__PURE__*/_react["default"].createElement(_SvgIcon["default"], {
    svg: BtnSort,
    className: (0, _classnames["default"])('relative', 'ml-1', (0, _defineProperty2["default"])({
      'transform-180': showSortIndicator && sortDirection === _SortDirection["default"].ASC
    }, classes.sort, !showSortIndicator)),
    height: 9,
    width: 9,
    color: "black"
  }));
};

var _default = defaultHeaderRenderer;
exports["default"] = _default;
