// @guestyci/foundation/legacy/VirtualTable/VirtualTable.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactJss = require("react-jss");

var _classnames = _interopRequireDefault(require("classnames"));

var _Table = _interopRequireDefault(require("react-virtualized/dist/commonjs/Table/Table"));

var _Column = _interopRequireDefault(require("react-virtualized/dist/commonjs/Table/Column"));

var _InfiniteLoader = _interopRequireDefault(require("react-virtualized/dist/commonjs/InfiniteLoader"));

var _AutoSizer = _interopRequireDefault(require("react-virtualized/dist/commonjs/AutoSizer"));

var _SortDirection = _interopRequireDefault(require("react-virtualized/dist/commonjs/Table/SortDirection"));

var _getCellRenderer = _interopRequireDefault(require("./getCellRenderer"));

var _colors = require("../../theme/colors");

var UNIT_SIZE = 5;

var units = function units(size) {
  return UNIT_SIZE * size;
};

var useStyles = (0, _reactJss.createUseStyles)({
  table: {
    '& .ReactVirtualized__Table__row': {
      borderBottom: "1px solid ".concat(_colors.grayLight),
      display: 'flex',
      flexDirection: 'row',
      alignItems: 'center'
    },
    '& .ReactVirtualized__Table__headerRow': {
      borderBottom: "1px solid ".concat(_colors.grayLight),
      fontWeight: 'normal',
      textTransform: 'uppercase',
      display: 'flex',
      alignItems: 'center'
    },
    '& .ReactVirtualized__Table__headerColumn:first-of-type, & .ReactVirtualized__Table__rowColumn:first-of-type ': {
      paddingLeft: units(2)
    },
    '& .ReactVirtualized__Table__headerColumn, & .ReactVirtualized__Table__rowColumn': {
      minWidth: 0,
      paddingRight: units(4),
      paddingLeft: units(4)
    },
    '& .ReactVirtualized__Table__headerTruncatedText': {
      display: 'inline-block',
      maxWidth: '100%',
      whiteSpace: 'nowrap',
      textOverflow: 'ellipsis',
      overflow: 'hidden'
    },
    '& .ReactVirtualized__Table__sortableHeaderColumn': {
      cursor: 'pointer'
    },
    '& .ReactVirtualized__Table__rowColumn': {
      textOverflow: 'ellipsis',
      whiteSpace: 'nowrap'
    },
    '& .ReactVirtualized__Table__rowColumn:last-child, ReactVirtualized__Table__headerColumn:last-child': {
      paddingRight: units(2)
    },
    '& .ReactVirtualized__Table__rowColumn.borderLeft': {
      borderLeft: "1px solid ".concat(_colors.grayLight)
    },
    '& .ReactVirtualized__Table__rowColumn.borderRight': {
      borderRight: "1px solid ".concat(_colors.grayLight)
    },
    '& .ReactVirtualized__Table__rowColumn.borderLeft, & .ReactVirtualized__Table__rowColumn.borderRight': {
      display: 'flex',
      alignItems: 'center',
      height: units(8)
    }
  }
});

var Table = function Table(_ref) {
  var className = _ref.className,
      rowClassName = _ref.rowClassName,
      style = _ref.style,
      onSort = _ref.onSort,
      sortBy = _ref.sortBy,
      sortDirection = _ref.sortDirection,
      hasNextPage = _ref.hasNextPage,
      isLoading = _ref.isLoading,
      onLoadNextPage = _ref.onLoadNextPage,
      items = _ref.items,
      children = _ref.children,
      onRowOver = _ref.onRowOver,
      onRowOut = _ref.onRowOut,
      placeholderRowCount = _ref.placeholderRowCount,
      pageSize = _ref.pageSize,
      restOfProps = (0, _objectWithoutProperties2["default"])(_ref, ["className", "rowClassName", "style", "onSort", "sortBy", "sortDirection", "hasNextPage", "isLoading", "onLoadNextPage", "items", "children", "onRowOver", "onRowOut", "placeholderRowCount", "pageSize"]);
  var classes = useStyles();
  var ref = (0, _react.useRef)(null);
  var promiseRef = (0, _react.useRef)(null);
  var loadMoreRows = (0, _react.useCallback)(function () {
    return isLoading ? null : new Promise(function (resolve) {
      promiseRef.current = resolve;
      onLoadNextPage();
    });
  }, [isLoading, onLoadNextPage]);
  var rowCount = hasNextPage ? Math.max(pageSize, items.length + placeholderRowCount) : items.length;
  var rowGetter = (0, _react.useCallback)(function (_ref2) {
    var index = _ref2.index;
    return items[index] || {};
  }, [items]);
  var isRowLoaded = (0, _react.useCallback)(function (_ref3) {
    var index = _ref3.index;
    return !hasNextPage || index < items.length;
  }, [hasNextPage, items]);
  (0, _react.useEffect)(function () {
    if (items.length === 0) {
      // Workaround to fix table not being re-rendered if data changed and scroll position is on top
      var forceRender = ref.current._lastRenderedStartIndex === 0;
      ref.current.resetLoadMoreRowsCache(forceRender);

      ref.current._registeredChild.scrollToRow(0);
    }
  }, [items]);
  (0, _react.useEffect)(function () {
    if (!isLoading && promiseRef) {
      if (promiseRef.current) {
        promiseRef.current();
        promiseRef.current = null;
      }
    }
  }, [isLoading]);
  return /*#__PURE__*/_react["default"].createElement(_InfiniteLoader["default"], {
    ref: ref,
    rowCount: rowCount,
    loadMoreRows: loadMoreRows,
    isRowLoaded: isRowLoaded,
    "data-qa": "table"
  }, function (_ref4) {
    var onRowsRendered = _ref4.onRowsRendered,
        registerChild = _ref4.registerChild;
    return /*#__PURE__*/_react["default"].createElement(_AutoSizer["default"], null, function (_ref5) {
      var width = _ref5.width,
          height = _ref5.height;
      return /*#__PURE__*/_react["default"].createElement(_Table["default"], (0, _extends2["default"])({
        className: (0, _classnames["default"])(className, classes.table),
        style: style,
        ref: registerChild,
        onRowsRendered: onRowsRendered,
        headerClassName: "font-size-sm text-gray-dark",
        rowClassName: rowClassName,
        headerHeight: 50,
        rowHeight: 40,
        width: width,
        height: height,
        sort: onSort,
        sortBy: sortBy,
        sortDirection: sortDirection,
        rowCount: rowCount,
        onRowMouseOver: onRowOver,
        onRowMouseOut: onRowOut,
        rowGetter: rowGetter
      }, restOfProps), _react["default"].Children.map(children, function (child) {
        var _child$props = child.props,
            borderLeft = _child$props.borderLeft,
            borderRight = _child$props.borderRight,
            placeholder = _child$props.placeholder,
            cellRenderer = _child$props.cellRenderer,
            headerRenderer = _child$props.headerRenderer,
            columnClassName = _child$props.className,
            rest = (0, _objectWithoutProperties2["default"])(_child$props, ["borderLeft", "borderRight", "placeholder", "cellRenderer", "headerRenderer", "className"]);
        return /*#__PURE__*/_react["default"].createElement(_Column["default"], (0, _extends2["default"])({}, rest, {
          className: (0, _classnames["default"])('font-size-md', columnClassName, {
            borderLeft: borderLeft,
            borderRight: borderRight
          }),
          headerRenderer: headerRenderer,
          cellRenderer: (0, _getCellRenderer["default"])({
            isLoading: isLoading,
            placeholder: placeholder,
            cellRenderer: cellRenderer
          })
        }));
      }));
    });
  });
};

Table.defaultProps = {
  className: '',
  style: {},
  sortBy: '',
  placeholderRowCount: 3,
  pageSize: 40
};
Table.propTypes = {
  /** Additional class name */
  className: _propTypes["default"].string,

  /**	Additional style for table element  */
  style: _propTypes["default"].shape(),

  /** Sort function called when you click on the headers */
  onSort: _propTypes["default"].func,

  /** Sort key, should be the same as current sort column dataKey */
  sortBy: _propTypes["default"].string,

  /** Sort direction: ASC, DESC */
  sortDirection: _propTypes["default"].oneOf([_SortDirection["default"].ASC, _SortDirection["default"].DESC]),

  /** boolean indicating if there is a next page */
  hasNextPage: _propTypes["default"].bool,

  /** function invoked automatically to fetch more data */
  onLoadNextPage: _propTypes["default"].func.isRequired,

  /** Items array repesenting the data */
  items: _propTypes["default"].arrayOf(_propTypes["default"].object).isRequired,

  /** How many extra rows to render as placeholders at the end of the table */
  placeholderRowCount: _propTypes["default"].number,

  /** How many placeholders to show on initial load */
  pageSize: _propTypes["default"].number,

  /** Event for row over */
  onRowOver: _propTypes["default"].func
};
var _default = Table;
exports["default"] = _default;
