// @guestyci/foundation/esm/Dialog/DialogHeader.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React from 'react';
import PropTypes from 'prop-types';
import TextField from "../TextField";
import Section from "../Layout/Section";
import createStyles from "../createStyles";
var useStyles = createStyles(function (theme) {
  return {
    text: {
      color: theme.palette.title
    }
  };
}, {
  name: 'DialogHeader'
});

var DialogHeader = function DialogHeader(_ref) {
  var children = _ref.children,
      _ref$jss = _ref.jss,
      jss = _ref$jss === void 0 ? {} : _ref$jss,
      props = _objectWithoutProperties(_ref, ["children", "jss"]);

  var classes = useStyles({
    jss: jss
  });
  return /*#__PURE__*/React.createElement(Section, _extends({
    gutter: "6 8 0 6"
  }, props, {
    "data-qa": "dialog-header"
  }), /*#__PURE__*/React.createElement(TextField, {
    variant: "h2",
    className: classes.text
  }, children));
};

DialogHeader.propTypes = {
  /** Child elements to render in header */
  children: PropTypes.node.isRequired,

  /** JSS override object  */
  jss: PropTypes.oneOfType([PropTypes.shape(), PropTypes.func])
};
export default DialogHeader;
