// @guestyci/foundation/esm/Dialog/DialogContent.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React from 'react';
import PropTypes from 'prop-types';
import Section from "../Layout/Section";
import TextField from "../TextField";
import createStyles from "../createStyles";
var useStyles = createStyles(function (theme) {
  return {
    text: {
      color: theme.palette.text.secondary
    }
  };
}, {
  name: 'DialogContent'
});

var DialogContent = function DialogContent(_ref) {
  var children = _ref.children,
      _ref$jss = _ref.jss,
      jss = _ref$jss === void 0 ? {} : _ref$jss,
      className = _ref.className,
      props = _objectWithoutProperties(_ref, ["children", "jss", "className"]);

  var classes = useStyles({
    jss: jss
  });
  return /*#__PURE__*/React.createElement(Section, _extends({
    gutter: "6 8 6 6",
    className: className
  }, props, {
    "data-qa": "dialog-content"
  }), /*#__PURE__*/React.createElement(TextField, {
    className: classes.text
  }, children));
};

DialogContent.propTypes = {
  /** Child elements to render in header */
  children: PropTypes.node.isRequired,

  /** JSS override object */
  jss: PropTypes.oneOfType([PropTypes.shape(), PropTypes.func]),

  /** Additional classname to root component */
  className: PropTypes.string
};
export default DialogContent;
