// @guestyci/foundation/legacy/Table/DateCell/DateCell.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "MomentFormatTypes", {
  enumerable: true,
  get: function get() {
    return _localeData.MomentFormatTypes;
  }
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = require("prop-types");

var _classnames = _interopRequireDefault(require("classnames"));

var _FormattedDate = _interopRequireDefault(require("@guestyci/localize/FormattedDate"));

var _localeData = require("@guestyci/localize/utils/localeData");

var _TextCell = _interopRequireDefault(require("../TextCell"));

var DateCell = function DateCell(_ref) {
  var date = _ref.date,
      format = _ref.format,
      isLoading = _ref.isLoading,
      className = _ref.className,
      style = _ref.style,
      rest = (0, _objectWithoutProperties2["default"])(_ref, ["date", "format", "isLoading", "className", "style"]);
  return /*#__PURE__*/_react["default"].createElement(_TextCell["default"], (0, _extends2["default"])({}, rest, {
    isLoading: isLoading,
    className: (0, _classnames["default"])('date-cell', className),
    style: style,
    "data-qa": "date-cell"
  }), date && /*#__PURE__*/_react["default"].createElement(_FormattedDate["default"], {
    date: date,
    format: format
  }));
};

DateCell.propTypes = {
  /** date value in ISO format */
  date: _propTypes.string,

  /** indicator that cell is yet to receive data */
  isLoading: _propTypes.bool,

  /** Date format to custom show moment (shortDate, date, fullDate, mmmm yyyyy, mmm yyyyy) */
  format: (0, _propTypes.oneOf)([_localeData.MomentFormatTypes.ShortDate, _localeData.MomentFormatTypes.Date, _localeData.MomentFormatTypes.FullDate, _localeData.MomentFormatTypes.MMMM_YYYY, _localeData.MomentFormatTypes.MMM_YYYY]),

  /** class name of cell */
  className: _propTypes.string,

  /** styles applied to a cell */
  style: (0, _propTypes.shape)()
};
DateCell.defaultProps = {
  date: null,
  format: _localeData.MomentFormatTypes.Date,
  isLoading: false
};
var _default = DateCell;
exports["default"] = _default;
