// @guestyci/testify/esm/utils.js
import { mount as enzymeMount, shallow as enzymeShallow } from 'enzyme';
/**
 * Create  a new container and append to document body
 * @param id  {String}  Container id
 * @return {HTMLElement}
 */

export var createContainer = function createContainer(id) {
  var container = document.createElement('div');
  container.setAttribute('id', id);
  document.body.appendChild(container);
  return container;
};
/**
 * Generate mock function for unit tests
 * @returns {*}
 */

export var mockFn = jest.fn;
/**
 * Clear all mocks
 * @returns {*}
 */

export var clearMocks = jest.clearAllMocks;
/**
 * Clear all mocks
 * @returns {*}
 */

var _jest = jest,
    clearAllMocks = _jest.clearAllMocks;
/**
 * Helper function to create context switch to await async operations
 * @returns {Promise<any>}
 */

export { clearAllMocks };
export var asyncFlush = function asyncFlush() {
  var count = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 0;
  return new Promise(function (resolve) {
    return setTimeout(resolve, count);
  });
};
export var mount = enzymeMount;
export var shallow = enzymeShallow;
