// @guestyci/foundation/withActive/withActive.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _react = _interopRequireWildcard(require("react"));

function _createSuper(Derived) { return function () { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

/**
 * Create a wrapper compose to support active functionality
 * @param ComposedComponent
 * @returns {{Component}}
 */
function withActive(ComposedComponent) {
  var ActiveWrapper = /*#__PURE__*/function (_Component) {
    (0, _inherits2["default"])(ActiveWrapper, _Component);

    var _super = _createSuper(ActiveWrapper);

    function ActiveWrapper(props) {
      var _this;

      (0, _classCallCheck2["default"])(this, ActiveWrapper);
      _this = _super.call(this, props);

      _this.activeOn = function () {
        _this.setState({
          isActive: true
        });
      };

      _this.activeOff = function () {
        _this.setState({
          isActive: false
        });
      };

      _this.toggleActive = function () {
        _this.setState(function (prevState) {
          return {
            isActive: !prevState.isActive
          };
        });
      };

      _this.state = {
        isActive: false
      };
      return _this;
    }

    (0, _createClass2["default"])(ActiveWrapper, [{
      key: "render",
      value: function render() {
        var isActive = this.state.isActive;
        return /*#__PURE__*/_react["default"].createElement(ComposedComponent, (0, _extends2["default"])({}, this.props, {
          isActive: isActive,
          toggleActive: this.toggleActive,
          activeOn: this.activeOn,
          activeOff: this.activeOff,
          "data-qa": "active-wrapper"
        }));
      }
    }]);
    return ActiveWrapper;
  }(_react.Component);

  return ActiveWrapper;
}

var _default = withActive;
exports["default"] = _default;
