// @guestyci/foundation/es/legacy/Tabs/Tab.js
import React, { useCallback } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import FlatButton from "../FlatButton";
import createStyles from "../../createStyles";
const useStyles = createStyles({
  root: {
    boxShadow: 'none !important'
  }
}, {
  name: 'Tab'
});

const Tab = ({
  onClick,
  tabIndex,
  value,
  title,
  className,
  disabled,
  isActive
}) => {
  const classes = useStyles();
  const onTabClick = useCallback(e => {
    e.stopPropagation();
    onClick(tabIndex, value);
  }, [tabIndex, value]);
  return /*#__PURE__*/React.createElement(FlatButton, {
    value: value,
    className: cn(className, classes.root, 'text-uppercase', 'p-0', 'tab-item', {
      'text-blue': isActive
    }),
    onClick: onTabClick,
    disabled: disabled,
    "data-qa": "tab"
  }, title);
};

Tab.defaultProps = {
  children: null,
  disabled: false,
  value: undefined
};
Tab.propTypes = {
  /** The tab title */
  title: PropTypes.node.isRequired,

  /** Set tab content to show when active */
  children: PropTypes.node,

  /** Indicator whether or not the tab is disabled */
  disabled: PropTypes.bool,

  /** The tab value to be passed when clicked */
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.shape()])
};
export default Tab;
