// @guestyci/foundation/legacy/ActionTooltip/ActionTooltipBody.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _enums = require("../../enums/enums");

var _ActionTooltipButton = _interopRequireDefault(require("./ActionTooltipButton"));

var _Col = _interopRequireDefault(require("../Col/Col"));

var ActionTooltipBody = function ActionTooltipBody(_ref) {
  var actionType = _ref.actionType,
      actionText = _ref.actionText,
      actionHref = _ref.actionHref,
      onAction = _ref.onAction,
      body = _ref.body;
  return /*#__PURE__*/_react["default"].createElement(_Col["default"], {
    spacing: 2,
    "data-qa": "action-tooltip-body"
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: "action-tooltip-text letter-spacing-lg text-left"
  }, body), /*#__PURE__*/_react["default"].createElement("div", {
    className: "flex-end"
  }, /*#__PURE__*/_react["default"].createElement(_ActionTooltipButton["default"], {
    actionType: actionType,
    actionText: actionText,
    actionHref: actionHref,
    onAction: onAction
  })));
};

ActionTooltipBody.defaultProps = {
  actionHref: '',
  onAction: function onAction() {}
};
ActionTooltipBody.propTypes = {
  /** Sets the type of the action */
  actionType: _propTypes["default"].oneOf([_enums.ActionTooltipType.Button, _enums.ActionTooltipType.Link]).isRequired,

  /** Sets the text of the action */
  actionText: _propTypes["default"].node.isRequired,

  /** Sets the href of the action */
  actionHref: _propTypes["default"].string,

  /** Callback function - called when clicking the tooltip action */
  onAction: _propTypes["default"].func,

  /** Sets the body (content) of the tooltip F.E: text to be diplayed */
  body: _propTypes["default"].node.isRequired
};
var _default = ActionTooltipBody;
exports["default"] = _default;
