// @guestyci/foundation/legacy/Dialog/DialogXHeader.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _IconButton = _interopRequireDefault(require("../IconButton"));

var BtnDelete = function BtnDelete(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M8.485.707L7.778 0 4.243 3.536.707 0 0 .707l3.536 3.536L0 7.778l.707.707L4.243 4.95l3.535 3.535.707-.707L4.95 4.243z"
  }));
};

BtnDelete.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 8.4854 8.4854"
};

var DialogXHeader = function DialogXHeader(_ref) {
  var onClick = _ref.onClick,
      className = _ref.className,
      style = _ref.style,
      setAbsoluteRight = _ref.setAbsoluteRight;
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])(setAbsoluteRight ? 'absolute top-1 right-2' : 'dialog-x-header flex-end-center height-10 w-fill', className),
    style: style,
    "data-qa": "dialog-x-header"
  }, /*#__PURE__*/_react["default"].createElement(_IconButton["default"], {
    onClick: onClick,
    svg: BtnDelete,
    iconHeight: 10,
    iconWidth: 10
  }));
};

DialogXHeader.defaultProps = {
  className: '',
  style: {},
  setAbsoluteRight: false
};
DialogXHeader.propTypes = {
  onClick: _propTypes["default"].func.isRequired,
  className: _propTypes["default"].string,
  style: _propTypes["default"].shape(),
  setAbsoluteRight: _propTypes["default"].bool
};
var _default = DialogXHeader;
exports["default"] = _default;
