// @guestyci/foundation/legacy/Dialog/DialogFooter.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _RaisedButton = _interopRequireDefault(require("../RaisedButton/RaisedButton"));

var _FlatButton = _interopRequireDefault(require("../FlatButton/FlatButton"));

var _Row = _interopRequireDefault(require("../Row/Row"));

var DialogFooter = function DialogFooter(_ref) {
  var onConfirm = _ref.onConfirm,
      onCancel = _ref.onCancel,
      confirmTitle = _ref.confirmTitle,
      cancelTitle = _ref.cancelTitle,
      leftComponent = _ref.leftComponent,
      className = _ref.className,
      style = _ref.style,
      disabled = _ref.disabled,
      showCancel = _ref.showCancel,
      showConfirm = _ref.showConfirm,
      isSubmitting = _ref.isSubmitting;
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])('flex-space-between-center w-fill py-6', className),
    style: style,
    "data-qa": "dialog-footer"
  }, leftComponent && leftComponent, /*#__PURE__*/_react["default"].createElement(_Row["default"], {
    justify: "end",
    align: "center",
    className: "flex-1"
  }, showCancel && /*#__PURE__*/_react["default"].createElement(_FlatButton["default"], {
    className: "dialog-footer-cancel",
    onClick: onCancel
  }, cancelTitle), showConfirm && /*#__PURE__*/_react["default"].createElement(_RaisedButton["default"], {
    className: "dialog-footer-confirm",
    disabled: disabled,
    onClick: onConfirm,
    isLoading: isSubmitting
  }, confirmTitle)));
};

DialogFooter.defaultProps = {
  cancelTitle: 'Cancel',
  confirmTitle: 'Save',
  onConfirm: undefined,
  onCancel: undefined,
  leftComponent: null,
  className: '',
  style: {},
  disabled: false,
  showCancel: true,
  showConfirm: true,
  isSubmitting: false
};
DialogFooter.propTypes = {
  cancelTitle: _propTypes["default"].string,
  confirmTitle: _propTypes["default"].string,
  onConfirm: _propTypes["default"].func,
  onCancel: _propTypes["default"].func,
  leftComponent: _propTypes["default"].node,
  className: _propTypes["default"].string,
  style: _propTypes["default"].shape(),
  disabled: _propTypes["default"].bool,
  showCancel: _propTypes["default"].bool,
  showConfirm: _propTypes["default"].bool,
  isSubmitting: _propTypes["default"].bool
};
var _default = DialogFooter;
exports["default"] = _default;
