// @guestyci/foundation/legacy/Dialog/withDialog.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _react = _interopRequireWildcard(require("react"));

function _createSuper(Derived) { return function () { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

/**
 * Create a wrapper compose to support Dialog functionality
 * @param ComposedComponent
 * @returns {{Component}}
 */
function withDialog(ComposedComponent) {
  var DialogWrapper = /*#__PURE__*/function (_Component) {
    (0, _inherits2["default"])(DialogWrapper, _Component);

    var _super = _createSuper(DialogWrapper);

    function DialogWrapper(props) {
      var _this;

      (0, _classCallCheck2["default"])(this, DialogWrapper);
      _this = _super.call(this, props);

      _this.closeDialog = function () {
        _this.setState({
          isOpen: false
        });
      };

      _this.openDialog = function () {
        _this.setState({
          isOpen: true
        });
      };

      _this.toggleDialog = function () {
        _this.setState(function (prevState) {
          return {
            isOpen: !prevState.isOpen
          };
        });
      };

      _this.state = {
        isOpen: false
      };
      return _this;
    }

    (0, _createClass2["default"])(DialogWrapper, [{
      key: "render",
      value: function render() {
        var isOpen = this.state.isOpen;
        return /*#__PURE__*/_react["default"].createElement(ComposedComponent, (0, _extends2["default"])({}, this.props, {
          isDialogOpen: isOpen,
          closeDialog: this.closeDialog,
          openDialog: this.openDialog,
          toggleDialog: this.toggleDialog,
          "data-qa": "dialog-wrapper"
        }));
      }
    }]);
    return DialogWrapper;
  }(_react.Component);

  return DialogWrapper;
}

var _default = withDialog;
exports["default"] = _default;
