// @guestyci/ebisu/bin/deploy-s3.js
#!/usr/bin/env node

const fs = require('fs');
const { execSync } = require('child_process');
const { getConfig } = require('../util/files');
const git = require('../util/git');

const {
  getBuildFolder,
  getS3DeployConfig,
  getS3Targets,
  deployFolder,
  fillPublicURL,
} = require('../scripts/deploy-s3');

async function run({
  AWS_ACCESS_KEY: awsAccessKey,
  AWS_SECRET_ACCESS_KEY: awsSecretAccessKey,
  CLUSTER: cluster,
  COMMIT: commit,
  REPO_NAME: repoName,
}) {
  const customDeploymentScript = getConfig('s3_scripts');

  if (customDeploymentScript && customDeploymentScript[cluster]) {
    execSync(
      `AWS_ACCESS_KEY=${awsAccessKey} CLUSTER=${cluster} ` +
        `AWS_SECRET_ACCESS_KEY=${awsSecretAccessKey} ` +
        `yarn ${customDeploymentScript[cluster]}`
    );
    return;
  }

  const customConfig = getConfig('s3_config');
  const src = getBuildFolder(customConfig);

  const deployTargets = getS3Targets({ cluster, commit, repoName });

  fillPublicURL({ cluster, commit, repoName });

  const deployConfig = getS3DeployConfig(customConfig);

  return Promise.all(
    Object.values(deployTargets).map(target =>
      deployFolder(src, target, deployConfig)
    )
  );
}

module.exports = run;
