// @guestyci/ebisu/bin/invalidate-cf-manifest.js
const aws = require('aws-sdk');
const ora = require('ora');

const {
  AWS_ACCESS_KEY_ID,
  AWS_SECRET_ACCESS_KEY,
  AWS_CF_MANIFEST_DISTRIBUTION_ID,
  AWS_CF_WAIT_FOR_INVALIDATION_COMPLETION = false,
  CALLER_REFERENCE,
  CIRCLE_PROJECT_REPONAME,
} = process.env;

const run = async () => {
  const invalidationPath = `/production/${CIRCLE_PROJECT_REPONAME}/static/asset-manifest.json`;
  const spinner = ora();
  const cloudfront = new aws.CloudFront({
    accessKeyId: AWS_ACCESS_KEY_ID,
    secretAccessKey: AWS_SECRET_ACCESS_KEY,
  });

  spinner.start(`Invalidating cache for ${invalidationPath}`);

  const data = await cloudfront
    .createInvalidation({
      DistributionId: AWS_CF_MANIFEST_DISTRIBUTION_ID,
      InvalidationBatch: {
        CallerReference: CALLER_REFERENCE,
        Paths: {
          Quantity: 1,
          Items: [invalidationPath],
        },
      },
    })
    .promise();

  spinner.succeed();
  const id = data.Invalidation.Id;
  console.log(`invalidation id: ${id}`);
  const shouldWait = AWS_CF_WAIT_FOR_INVALIDATION_COMPLETION === 'true';
  if (shouldWait) {
    spinner.start(
      'Waiting for invalidation to complete. This might take a while'
    );
    await cloudfront
      .waitFor('invalidationCompleted', {
        DistributionId: AWS_CF_MANIFEST_DISTRIBUTION_ID,
        Id: id,
      })
      .promise();
    spinner.succeed();
  }
};

module.exports = run;
