// @guestyci/foundation/TextArea/TextArea.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.useStyles = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _noop = _interopRequireDefault(require("lodash/noop"));

var _reactTextareaAutosize = _interopRequireDefault(require("react-textarea-autosize"));

var _Col = _interopRequireDefault(require("../Layout/Col"));

var _createStyles = _interopRequireDefault(require("../createStyles"));

var _styles = _interopRequireDefault(require("../Input/styles"));

var _colors = require("../theme/colors");

var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    root: (0, _extends2["default"])({}, theme.typography.primary),
    textArea: (0, _extends2["default"])({}, _styles["default"].shared(theme), {
      '&::placeholder': _styles["default"].placeholder(theme),
      '&:hover': _styles["default"].hover(theme),
      '&:focus': _styles["default"].focused(theme),
      '&:disabled': _styles["default"].disabled(theme),
      paddingTop: theme.spacer(3),
      paddingBottom: theme.spacer(3),
      minHeight: '60px'
    }),
    error: _styles["default"].error(theme),
    charCount: {
      marginTop: theme.spacer(1),
      color: _colors.blueGray500
    }
  };
}, {
  name: 'TextArea'
});
exports.useStyles = useStyles;

var TextArea = _react["default"].forwardRef(function TextArea(_ref, ref) {
  var name = _ref.name,
      value = _ref.value,
      onChange = _ref.onChange,
      disabled = _ref.disabled,
      placeholder = _ref.placeholder,
      className = _ref.className,
      maxLength = _ref.maxLength,
      minRows = _ref.minRows,
      maxRows = _ref.maxRows,
      error = _ref.error,
      jss = _ref.jss,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["name", "value", "onChange", "disabled", "placeholder", "className", "maxLength", "minRows", "maxRows", "error", "jss"]);
  var classes = useStyles({
    jss: jss
  });
  var charCount = value.length;
  return /*#__PURE__*/_react["default"].createElement(_Col["default"], {
    className: (0, _classnames["default"])(classes.root, className)
  }, /*#__PURE__*/_react["default"].createElement(_reactTextareaAutosize["default"], (0, _extends2["default"])({
    value: value,
    onChange: onChange,
    name: name,
    placeholder: placeholder,
    disabled: disabled,
    maxRows: maxRows,
    minRows: minRows,
    maxLength: maxLength,
    className: (0, _classnames["default"])(classes.textArea, (0, _defineProperty2["default"])({}, classes.error, error)),
    ref: ref
  }, props)), maxLength && /*#__PURE__*/_react["default"].createElement("span", {
    className: classes.charCount
  }, "".concat(charCount, "/").concat(maxLength)));
});

TextArea.defaultProps = {
  value: '',
  name: '',
  onChange: _noop["default"],
  placeholder: '',
  disabled: false,
  minRows: 1,
  maxRows: 5,
  maxLength: undefined
};
TextArea.propTypes = {
  /** text area value */
  value: _propTypes["default"].string,

  /** text area name */
  name: _propTypes["default"].string,

  /** On change trigger function to update on text change */
  onChange: _propTypes["default"].func,

  /** The max length for the text area input */
  maxLength: _propTypes["default"].number,

  /** TextArea Placeholder */
  placeholder: _propTypes["default"].string,

  /** Indicator for disabled status */
  disabled: _propTypes["default"].bool,

  /** Minimum height according by units of rows */
  minRows: _propTypes["default"].number,

  /** Maximum height according by units of rows */
  maxRows: _propTypes["default"].number,

  /** jss override object to customize the jss classes */
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func])
};
TextArea.displayName = 'TextArea';
var _default = TextArea;
exports["default"] = _default;
