// @guestyci/foundation/legacy/Form/Form.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.scrollToFirstError = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _enums = require("../../enums");

var _FormContext = _interopRequireDefault(require("./FormContext"));

var scrollToElement = function scrollToElement(element, offSet) {
  var position = element.getBoundingClientRect().top + document.documentElement.scrollTop;
  element.focus({
    preventScroll: true
  });
  window.scrollTo({
    top: position - offSet,
    behavior: 'smooth'
  });
};

var scrollToFirstError = function scrollToFirstError(formRef, offSet) {
  var el = formRef && formRef.current && formRef.current.querySelector('.invalid-field');

  if (el) {
    scrollToElement(el, offSet);
  }
};

exports.scrollToFirstError = scrollToFirstError;

var Form = function Form(_ref) {
  var onSubmit = _ref.onSubmit,
      children = _ref.children,
      scrollOffset = _ref.scrollOffset,
      className = _ref.className,
      style = _ref.style,
      provider = _ref.provider,
      fieldInstance = _ref.fieldInstance;
  var formRef = (0, _react.useRef)(null);
  var handleSubmit = (0, _react.useCallback)(function (e) {
    e.preventDefault();
    onSubmit();
    scrollToFirstError(formRef, scrollOffset);
  }, [formRef]);
  return /*#__PURE__*/_react["default"].createElement("form", {
    className: className,
    style: style,
    ref: formRef,
    onSubmit: handleSubmit,
    "data-qa": "form"
  }, /*#__PURE__*/_react["default"].createElement(_FormContext["default"].Provider, {
    value: {
      provider: provider,
      fieldInstance: fieldInstance
    }
  }, children));
};

Form.defaultProps = {
  scrollOffset: 50,
  provider: _enums.FormProvider.Redux,
  fieldInstance: null
};
Form.propTypes = {
  /** Form field instance to pass */
  fieldInstance: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func]),

  /** String - the form provider to render * */
  provider: _propTypes["default"].oneOf(['redux', 'final']),

  /** Callback function to handle form submit */
  onSubmit: _propTypes["default"].func.isRequired,

  /** Additional scrollOffset from top off the selected element */
  scrollOffset: _propTypes["default"].number,

  /** Form child element to render */
  children: _propTypes["default"].node.isRequired,

  /** Form root level additional class */
  className: _propTypes["default"].string,

  /** Form root level additional style */
  style: _propTypes["default"].shape()
};
var _default = Form;
exports["default"] = _default;
