// @guestyci/foundation/CSSTransitionGroup/CSSTransitionGroup.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _reactTransitionGroup = require("react-transition-group");

var _isEmpty = _interopRequireDefault(require("lodash/isEmpty"));

var _CSSTransition = _interopRequireDefault(require("../CSSTransition"));

var CSSTransitionGroup = function CSSTransitionGroup(_ref) {
  var animation = _ref.animation,
      className = _ref.className,
      timeout = _ref.timeout,
      children = _ref.children,
      style = _ref.style,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["animation", "className", "timeout", "children", "style"]);
  var transitionChildren = children.filter(function (child) {
    return Boolean(child);
  });

  if ((0, _isEmpty["default"])(animation)) {
    return children;
  }

  return /*#__PURE__*/_react["default"].createElement(_reactTransitionGroup.TransitionGroup, {
    component: null,
    className: (0, _classnames["default"])('css-transition-group', className),
    style: style,
    "data-qa": "css-transition-group"
  }, _react["default"].Children.map(transitionChildren, function (child) {
    return /*#__PURE__*/_react["default"].createElement(_CSSTransition["default"], (0, _extends2["default"])({
      key: child === null || child === void 0 ? void 0 : child.key,
      asGroup: true,
      unmountOnExit: true,
      animation: animation,
      timeout: timeout
    }, props), child);
  }));
};

CSSTransitionGroup.defaultProps = {
  className: '',
  timeout: 225,
  animation: ''
};
CSSTransitionGroup.propTypes = {
  /* Animation for css transition */
  animation: _propTypes["default"].oneOfType([_propTypes["default"].shape({
    appear: _propTypes["default"].string,
    appearActive: _propTypes["default"].string,
    enter: _propTypes["default"].string.isRequired,
    enterActive: _propTypes["default"].string.isRequired,
    enterDone: _propTypes["default"].string,
    exitActive: _propTypes["default"].string.isRequired,
    exit: _propTypes["default"].string.isRequired,
    exitDone: _propTypes["default"].string
  }), _propTypes["default"].string]),

  /* Additional class name for css transition root */
  className: _propTypes["default"].string,

  /* Child components for css transition */
  children: _propTypes["default"].arrayOf(_propTypes["default"].node).isRequired,

  /* timeout between animations - values: number or {enter, exit} */
  timeout: _propTypes["default"].oneOfType([_propTypes["default"].shape({
    enter: _propTypes["default"].number.isRequired,
    exit: _propTypes["default"].number.isRequired
  }), _propTypes["default"].number])
};
var _default = CSSTransitionGroup;
exports["default"] = _default;
