// @guestyci/foundation/legacy/FixedRaisedButton/FixedRaisedButton.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _BaseButton = _interopRequireDefault(require("../BaseButton/BaseButton"));

/* eslint-disable react/jsx-max-props-per-line */
var FixedRaisedButton = function FixedRaisedButton(_ref) {
  var className = _ref.className,
      children = _ref.children,
      onClick = _ref.onClick,
      style = _ref.style,
      disabled = _ref.disabled,
      color = _ref.color,
      isLoading = _ref.isLoading,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["className", "children", "onClick", "style", "disabled", "color", "isLoading"]);
  return /*#__PURE__*/_react["default"].createElement(_BaseButton["default"], (0, _extends2["default"])({
    className: (0, _classnames["default"])(className, 'overflow-ellipsis', 'min-width-200', 'btn-raised-lg', 'font-size-lg'),
    variant: "raised",
    onClick: onClick,
    style: style,
    disabled: disabled,
    color: color,
    isLoading: isLoading
  }, props, {
    "data-qa": "fixed-raised-button"
  }), children);
};

FixedRaisedButton.defaultProps = {
  className: '',
  color: 'primary',
  style: {},
  disabled: false,
  isLoading: false
};
FixedRaisedButton.propTypes = {
  /** Button click callback function */
  onClick: _propTypes["default"].func,

  /** Button inner children */
  children: _propTypes["default"].node.isRequired,

  /** Button color - allowing primary, danger and success */
  color: _propTypes["default"].oneOf(['primary', 'success', 'danger']),

  /** Button indicator whether disabled or not */
  disabled: _propTypes["default"].bool,

  /** Button additional classes */
  className: _propTypes["default"].string,

  /** Button additional styling */
  style: _propTypes["default"].shape(),

  /** Indicator for  button loading state  */
  isLoading: _propTypes["default"].bool
};
var _default = FixedRaisedButton;
exports["default"] = _default;
