// @guestyci/foundation/legacy/Col/Col.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _reactJss = require("react-jss");

var spacer = (0, _toConsumableArray2["default"])(Array(7).keys()).map(function (i) {
  return i * 5;
});
var useStyles = (0, _reactJss.createUseStyles)({
  col: {
    '& > * ': {
      marginBottom: function marginBottom(_ref) {
        var spacing = _ref.spacing;
        return spacer[spacing];
      }
    },
    '& > *:last-child ': {
      marginBottom: 0
    }
  }
});

var Col = _react["default"].forwardRef(function Col(_ref2, ref) {
  var className = _ref2.className,
      style = _ref2.style,
      children = _ref2.children,
      dense = _ref2.dense,
      fullWidth = _ref2.fullWidth,
      align = _ref2.align,
      justify = _ref2.justify,
      spacing = _ref2.spacing,
      onClick = _ref2.onClick,
      props = (0, _objectWithoutProperties2["default"])(_ref2, ["className", "style", "children", "dense", "fullWidth", "align", "justify", "spacing", "onClick"]);
  var classes = useStyles({
    spacing: spacing
  });
  return /*#__PURE__*/_react["default"].createElement("div", (0, _extends2["default"])({}, props, {
    ref: ref,
    role: "presentation",
    onClick: onClick,
    className: (0, _classnames["default"])({
      clickable: onClick
    }, 'd-flex flex-column', (0, _defineProperty2["default"])({}, "col-wrapper ".concat(classes.col), !dense), {
      'w-fill': fullWidth
    }, (0, _defineProperty2["default"])({}, "align-items-".concat(align), align), (0, _defineProperty2["default"])({}, "justify-content-".concat(justify), justify), className),
    style: style
  }), children);
});

Col.defaultProps = {
  className: '',
  style: {},
  children: null,
  dense: false,
  fullWidth: false,
  align: 'inherit',
  justify: 'inherit',
  spacing: 6,
  onClick: null
};
Col.propTypes = {
  /** Additional classes for row */
  className: _propTypes["default"].string,

  /** Additional classes for row */
  align: _propTypes["default"].string,

  /** Additional classes for row */
  justify: _propTypes["default"].string,

  /** Additional styles for row */
  style: _propTypes["default"].shape(),

  /** Children for row wrapper */
  children: _propTypes["default"].oneOfType([_propTypes["default"].node, _propTypes["default"].arrayOf(_propTypes["default"].node)]),

  /** Boolean whether or not we are in dense (no margins) mode */
  dense: _propTypes["default"].bool,

  /** Boolean whether or not we are in full width mode */
  fullWidth: _propTypes["default"].bool,

  /** Spacing between components [0, 1, 2, 3, 4, 5, 6] -
   *  multiplies of 5 (1 => 5, 2=> 10). - defaults to 6
   */
  spacing: _propTypes["default"].oneOf([0, 1, 2, 3, 4, 5, 6]),

  /** handle click event for Col */
  onClick: _propTypes["default"].func
}; // Hack for withInfo plugin

Col.displayName = 'Col';
var _default = Col;
exports["default"] = _default;
