// @guestyci/foundation/Listing/ConfirmedBadge.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _createStyles = _interopRequireDefault(require("../createStyles"));

var BtnApprove = function BtnApprove(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M5.745 11.282L.935 6.933a.546.546 0 0 0-.766 0 .557.557 0 0 0 0 .799l5.191 4.881c.097.095.237.15.385.15a.543.543 0 0 0 .35-.122L15.468.944a.559.559 0 0 0 0-.8.587.587 0 0 0-.727-.034L5.745 11.282z"
  }));
};

BtnApprove.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 15.636672 12.7622719"
};
var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    root: {
      height: 20,
      width: 20,
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      fill: theme.palette.success["default"],
      backgroundColor: 'white',
      borderRadius: '50%',
      border: "1.5px solid ".concat(theme.palette.success["default"])
    }
  };
}, {
  name: 'ConfirmedBadge'
});

var ConfirmedBadge = function ConfirmedBadge(_ref) {
  var _ref$className = _ref.className,
      className = _ref$className === void 0 ? '' : _ref$className,
      _ref$style = _ref.style,
      style = _ref$style === void 0 ? {} : _ref$style,
      _ref$jss = _ref.jss,
      jss = _ref$jss === void 0 ? {} : _ref$jss,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["className", "style", "jss"]);
  var classes = useStyles({
    jss: jss
  });
  return /*#__PURE__*/_react["default"].createElement("div", (0, _extends2["default"])({}, props, {
    className: (0, _classnames["default"])(classes.root, className),
    style: style,
    "data-qa": "confirmed-badge"
  }), /*#__PURE__*/_react["default"].createElement(BtnApprove, {
    strokeWidth: 2,
    height: 8,
    width: 9
  }));
};

ConfirmedBadge.propTypes = {
  /** Additional class for badge */
  className: _propTypes["default"].string,

  /** Additional style for badge */
  style: _propTypes["default"].shape(),

  /** jss override object to customize the jss classes */
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func])
};
var _default = ConfirmedBadge;
exports["default"] = _default;
