// @guestyci/foundation/Listing/Listing.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.useStyles = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _TextField = _interopRequireDefault(require("../TextField"));

var _Row = _interopRequireDefault(require("../Layout/Row"));

var _Avatar = _interopRequireDefault(require("../Avatar"));

var _createStyles = _interopRequireDefault(require("../createStyles"));

var _Col = _interopRequireDefault(require("../Layout/Col"));

var _ConfirmedBadge = _interopRequireDefault(require("./ConfirmedBadge"));

var IcoComplex = function IcoComplex(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("g", {
    fill: "#FFF",
    fillRule: "nonzero"
  }, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M24.5 19a.75.75 0 0 1-.75-.75 3.5 3.5 0 0 0-3.5-3.5h-.5a3.5 3.5 0 0 0-3.5 3.5.75.75 0 0 1-.75.75H9.512a.5.5 0 0 0-.5.5V31a1 1 0 0 0 1 1H18a.5.5 0 0 0 .5-.5v-3a1.5 1.5 0 0 1 3 0v3a.5.5 0 0 0 .5.5h8.011a1 1 0 0 0 1-1V19.5a.5.5 0 0 0-.5-.5H24.5zm-13 3a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-2zm4.012 7.5a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-2a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 .5.5v2zM21.5 22a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1-.5-.5v-3a1.5 1.5 0 0 1 3 0v3zm6.988 7.5a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-2a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 .5.5v2zM28.5 24a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-2a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 .5.5v2z"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    d: "M31 17.5a.999.999 0 0 0 .912-1.409l-.012-.023-3.823-7.477A.999.999 0 0 0 27.166 8H12.834a.989.989 0 0 0-.9.567l-3.846 7.524A.999.999 0 0 0 9 17.5h5.394a.5.5 0 0 0 .486-.386 5.01 5.01 0 0 1 4.87-3.864h.5a5.01 5.01 0 0 1 4.87 3.864.5.5 0 0 0 .486.386H31z"
  })));
};

IcoComplex.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  width: "40",
  height: "40",
  viewBox: "0 0 40 40"
};
var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    root: {},
    description: {
      color: theme.palette.description
    },
    badge: {
      position: 'absolute',
      bottom: -2,
      right: -6
    },
    complex: {
      backgroundColor: theme.palette.secondary["default"]
    },
    multiUnit: {
      position: 'relative',
      border: "2px solid ".concat(theme.palette.secondary["default"]),
      '&:before': {
        borderRadius: '50%',
        content: '""',
        display: 'block',
        position: 'absolute',
        top: 1,
        bottom: 1,
        left: 1,
        right: 1,
        border: '2px solid white'
      }
    }
  };
}, {
  name: 'Listing'
});
exports.useStyles = useStyles;

var Listing = _react["default"].forwardRef(function Person(_ref, ref) {
  var _cn;

  var _ref$img = _ref.img,
      img = _ref$img === void 0 ? '' : _ref$img,
      name = _ref.name,
      _ref$description = _ref.description,
      description = _ref$description === void 0 ? '' : _ref$description,
      _ref$multiUnit = _ref.multiUnit,
      multiUnit = _ref$multiUnit === void 0 ? false : _ref$multiUnit,
      _ref$complex = _ref.complex,
      complex = _ref$complex === void 0 ? false : _ref$complex,
      _ref$confirmed = _ref.confirmed,
      confirmed = _ref$confirmed === void 0 ? false : _ref$confirmed,
      _ref$className = _ref.className,
      className = _ref$className === void 0 ? '' : _ref$className,
      _ref$style = _ref.style,
      style = _ref$style === void 0 ? {} : _ref$style,
      _ref$jss = _ref.jss,
      jss = _ref$jss === void 0 ? {} : _ref$jss,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["img", "name", "description", "multiUnit", "complex", "confirmed", "className", "style", "jss"]);
  var classes = useStyles({
    jss: jss
  });
  return /*#__PURE__*/_react["default"].createElement(_Row["default"], (0, _extends2["default"])({}, props, {
    style: style,
    className: (0, _classnames["default"])(classes.root, className),
    ref: ref,
    spacing: 2,
    fullWidth: true,
    justify: "start",
    align: "center"
  }), /*#__PURE__*/_react["default"].createElement(_Avatar["default"], {
    src: img,
    className: (0, _classnames["default"])((_cn = {}, (0, _defineProperty2["default"])(_cn, classes.complex, complex), (0, _defineProperty2["default"])(_cn, classes.multiUnit, !complex && multiUnit), _cn)),
    badge: confirmed ? /*#__PURE__*/_react["default"].createElement(_ConfirmedBadge["default"], {
      className: classes.badge
    }) : null
  }, !img && complex && /*#__PURE__*/_react["default"].createElement(IcoComplex, null)), /*#__PURE__*/_react["default"].createElement(_Col["default"], {
    justify: "center",
    spacing: 1
  }, /*#__PURE__*/_react["default"].createElement(_TextField["default"], null, name), description && /*#__PURE__*/_react["default"].createElement(_TextField["default"], {
    variant: "h5",
    className: classes.description
  }, description)));
});

Listing.propTypes = {
  /** Image url for listing */
  img: _propTypes["default"].string,

  /** Listing name or nickname */
  name: _propTypes["default"].node,

  /** Subtitle for listing ( address etc..) */
  description: _propTypes["default"].node,

  /** Additional class for listing */
  className: _propTypes["default"].string,

  /** Additional style for listing */
  style: _propTypes["default"].shape(),

  /** jss override object to customize the jss classes */
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func]),

  /** Indicator for listing complex */
  complex: _propTypes["default"].bool,

  /** Indicator for confirmed listing */
  confirmed: _propTypes["default"].bool,

  /** Indicator for multi unit listing */
  multiUnit: _propTypes["default"].bool
};
Listing.displayName = 'Listing';
var _default = Listing;
exports["default"] = _default;
