// @guestyci/foundation/legacy/TimePicker/ItemSelector.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _IconButton = _interopRequireDefault(require("../IconButton/IconButton"));

var _TextField = _interopRequireDefault(require("../TextField"));

var _Col = _interopRequireDefault(require("../Col"));

function _createSuper(Derived) { return function () { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

var BtnArrowUp = function BtnArrowUp(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M4.529 0a.474.474 0 0 0-.357.137L.137 4.172a.488.488 0 0 0 .69.689L4.53 1.156l3.705 3.705a.488.488 0 0 0 .69-.69L4.89.138A.477.477 0 0 0 4.53 0z"
  }));
};

BtnArrowUp.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 9.0618038 4.9977479"
};

var BtnArrowDown = function BtnArrowDown(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M4.533 4.997c.13.006.26-.039.357-.136L8.925.826a.488.488 0 0 0-.69-.69L4.532 3.843.826.137a.488.488 0 0 0-.69.69L4.172 4.86c.097.097.23.143.362.136z"
  }));
};

BtnArrowDown.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 9.0618038 4.9977479"
};

var ItemSelector = /*#__PURE__*/function (_Component) {
  (0, _inherits2["default"])(ItemSelector, _Component);

  var _super = _createSuper(ItemSelector);

  function ItemSelector(props) {
    var _this;

    (0, _classCallCheck2["default"])(this, ItemSelector);
    _this = _super.call(this, props);

    _this.onIncrease = function () {
      var _this$props = _this.props,
          step = _this$props.step,
          values = _this$props.values,
          onChange = _this$props.onChange;
      var stateIndex = _this.state.index;
      var index = (stateIndex + step) % values.length;

      _this.setState({
        index: index
      });

      onChange(index);
    };

    _this.onDecrease = function () {
      var _this$props2 = _this.props,
          step = _this$props2.step,
          values = _this$props2.values,
          onChange = _this$props2.onChange;
      var stateIndex = _this.state.index;
      var index = (stateIndex - step + values.length) % values.length;

      _this.setState({
        index: index
      });

      onChange(index);
    };

    _this.state = {
      index: props.index || 0
    };
    return _this;
  }

  (0, _createClass2["default"])(ItemSelector, [{
    key: "render",
    value: function render() {
      var index = this.state.index;
      var _this$props3 = this.props,
          renderValue = _this$props3.renderValue,
          values = _this$props3.values;
      var rawValue = values[index];
      var value = renderValue ? renderValue(rawValue) : rawValue;
      return /*#__PURE__*/_react["default"].createElement(_Col["default"], {
        spacing: 0,
        align: "center",
        justify: "center",
        className: "fill-hover-blue width-35",
        "data-qa": "item-selector"
      }, /*#__PURE__*/_react["default"].createElement(_IconButton["default"], {
        onClick: this.onIncrease,
        svg: BtnArrowUp,
        iconWidth: 9,
        iconHeight: 5
      }), /*#__PURE__*/_react["default"].createElement(_TextField["default"], {
        color: "gray-dark"
      }, value), /*#__PURE__*/_react["default"].createElement(_IconButton["default"], {
        onClick: this.onDecrease,
        svg: BtnArrowDown,
        iconWidth: 9,
        iconHeight: 5
      }));
    }
  }]);
  return ItemSelector;
}(_react.Component);

ItemSelector.defaultProps = {
  onChange: function onChange() {},
  step: 1,
  values: [],
  renderValue: null
};
ItemSelector.propTypes = {
  onChange: _propTypes["default"].func,
  step: _propTypes["default"].number,
  values: _propTypes["default"].arrayOf(_propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].number])),
  renderValue: _propTypes["default"].func
};
var _default = ItemSelector;
exports["default"] = _default;
