// @guestyci/foundation/legacy/TimePicker/TimePickerInputField.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _noop = _interopRequireDefault(require("lodash/noop"));

var _reactJss = require("react-jss");

var _ActionButton = _interopRequireDefault(require("../ActionButton"));

var _IconButton = _interopRequireDefault(require("../IconButton/IconButton"));

var _PickerResetWrapper = _interopRequireDefault(require("../basePicker/PickerResetWrapper"));

var _Fade = _interopRequireDefault(require("../Fade"));

var BtnTime = function BtnTime(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M10.5 0C4.71 0 0 4.71 0 10.5S4.71 21 10.5 21 21 16.29 21 10.5 16.29 0 10.5 0zm0 20C5.262 20 1 15.738 1 10.5S5.262 1 10.5 1 20 5.262 20 10.5 15.738 20 10.5 20z"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    d: "M10 5H9v7h7v-1h-6z"
  }));
};

BtnTime.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 21 21"
};

var BtnDelete = function BtnDelete(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M8.485.707L7.778 0 4.243 3.536.707 0 0 .707l3.536 3.536L0 7.778l.707.707L4.243 4.95l3.535 3.535.707-.707L4.95 4.243z"
  }));
};

BtnDelete.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 8.4854 8.4854"
};
var useStyles = (0, _reactJss.createUseStyles)({
  timeBtn: {
    borderLeftWidth: '0 !important',
    height: 46
  },
  input: {
    height: 46
  }
});

var TimePickerInputField = function TimePickerInputField(_ref) {
  var onReset = _ref.onReset,
      disabled = _ref.disabled,
      resetable = _ref.resetable,
      className = _ref.className,
      style = _ref.style,
      onClick = _ref.onClick,
      id = _ref.id,
      isOpen = _ref.isOpen,
      value = _ref.value,
      placeholder = _ref.placeholder,
      onClear = _ref.onClear,
      defaultValue = _ref.defaultValue,
      name = _ref.name,
      noClear = _ref.noClear;
  var classes = useStyles();
  return /*#__PURE__*/_react["default"].createElement(_PickerResetWrapper["default"], {
    id: id,
    onReset: onReset,
    disabled: disabled || defaultValue === value,
    resetable: resetable,
    className: (0, _classnames["default"])('flex-end-center', className),
    style: style,
    "data-qa": "time-picker-input-field"
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])('d-flex flex-1 b-1 bg-white pl-4 border-gray-light align-items-center', classes.input),
    onClick: disabled ? null : onClick,
    role: "presentation"
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: "flex-1 h-fill flex-start-center font-size-md",
    name: name,
    id: id
  }, value ? /*#__PURE__*/_react["default"].createElement("span", {
    className: "time-value"
  }, value) : /*#__PURE__*/_react["default"].createElement("span", {
    className: "time-placeholder text-gray-dark"
  }, placeholder)), !noClear && /*#__PURE__*/_react["default"].createElement(_Fade["default"], {
    show: !!value && !disabled,
    className: "h-fill px-2 flex-center"
  }, /*#__PURE__*/_react["default"].createElement(_IconButton["default"], {
    key: 1,
    onClick: onClear,
    svg: BtnDelete,
    iconHeight: 10,
    iconWidth: 10
  }))), /*#__PURE__*/_react["default"].createElement(_ActionButton["default"], {
    iconHeight: 21,
    className: classes.timeBtn,
    iconColor: isOpen ? 'blue' : 'gray-dark',
    iconWidth: 21,
    onClick: onClick,
    disabled: disabled,
    id: id,
    svg: BtnTime,
    tooltip: "Select time"
  }));
};

TimePickerInputField.defaultProps = {
  onReset: _noop["default"],
  disabled: false,
  resetable: false,
  className: '',
  style: {},
  value: '',
  placeholder: '',
  defaultValue: '',
  name: '',
  id: '',
  noClear: false,
  isOpen: false
};
TimePickerInputField.propTypes = {
  onReset: _propTypes["default"].func,
  disabled: _propTypes["default"].bool,
  resetable: _propTypes["default"].bool,
  noClear: _propTypes["default"].bool,
  className: _propTypes["default"].string,
  style: _propTypes["default"].shape(),
  onClick: _propTypes["default"].func.isRequired,
  value: _propTypes["default"].string,
  placeholder: _propTypes["default"].string,
  onClear: _propTypes["default"].func.isRequired,
  defaultValue: _propTypes["default"].string,
  name: _propTypes["default"].string,
  id: _propTypes["default"].string,
  isOpen: _propTypes["default"].bool
};
var _default = TimePickerInputField;
exports["default"] = _default;
