// @guestyci/foundation/createStyles/createStyles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = require("react");

var _isEmpty = _interopRequireDefault(require("lodash/isEmpty"));

var _memoize = _interopRequireDefault(require("@guestyci/memoize"));

var _reactJss = require("react-jss");

var _sheetCounter = require("./sheetCounter");

var _useTheme = _interopRequireDefault(require("../useTheme"));

var generateUseStyles = function generateUseStyles(options) {
  return function (sheet) {
    return (0, _reactJss.createUseStyles)(sheet, options);
  };
};

var getSheet = (0, _memoize["default"])(function (styles, theme) {
  return typeof styles === 'function' ? styles(theme) : styles;
});

var getClasses = function getClasses(sheetClasses, overrideClasses) {
  if (!(0, _isEmpty["default"])(overrideClasses)) {
    return (0, _extends2["default"])({}, sheetClasses, {}, overrideClasses);
  }

  return sheetClasses;
};

var generateOverrideSheet = function generateOverrideSheet(_ref, jssOverrides) {
  var jssIndex = _ref.jssIndex,
      options = _ref.options,
      useStylesOverrides = _ref.useStylesOverrides;
  var jss = {
    useStylesOverrides: useStylesOverrides
  };

  if ((0, _isEmpty["default"])(jssOverrides)) {
    return {};
  }

  if (jssIndex) {
    return jss;
  }

  jss.jssIndex = (0, _sheetCounter.increment)();
  var createUseStylesOverride = generateUseStyles((0, _extends2["default"])({}, options, {
    index: jss.jssIndex
  }));
  jss.useStylesOverrides = createUseStylesOverride(jssOverrides);
  return jss;
};

function createStyles(styles) {
  var options = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};
  var index = (0, _sheetCounter.increment)();
  var instance = {
    index: index,
    options: options,
    jssIndex: null,
    useStylesOverrides: function useStylesOverrides() {}
  };
  var createUseStyles = generateUseStyles((0, _extends2["default"])({}, options, {
    index: index
  }));
  return function useStyles() {
    var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
    var jss = props.jss,
        otherProps = (0, _objectWithoutProperties2["default"])(props, ["jss"]);
    var theme = (0, _useTheme["default"])(); // create basic style sheet for component instance

    var useStyleSheet = (0, _react.useMemo)(function () {
      return createUseStyles(getSheet(styles, theme));
    }, [theme]);
    (0, _react.useEffect)(function () {
      return function () {
        instance.jssIndex = null;
      };
    }, []); // cache base sheet classes

    var classes = useStyleSheet(otherProps); // Create Overriding sheet in case jss is passed

    var _useMemo = (0, _react.useMemo)(function () {
      return generateOverrideSheet(instance, getSheet(jss, theme));
    }, [theme, jss]),
        _useMemo$useStylesOve = _useMemo.useStylesOverrides,
        useStylesOverrides = _useMemo$useStylesOve === void 0 ? function () {} : _useMemo$useStylesOve,
        jssInstance = (0, _objectWithoutProperties2["default"])(_useMemo, ["useStylesOverrides"]);

    instance = (0, _extends2["default"])({}, instance, {}, jssInstance); // cache jss override sheet classes

    var jssOverrideClasses = useStylesOverrides(otherProps);

    if (!(0, _isEmpty["default"])(jssOverrideClasses)) {
      instance.useStylesOverrides = useStylesOverrides;
    }

    return getClasses(classes, jssOverrideClasses);
  };
}

var _default = createStyles;
exports["default"] = _default;
