// @guestyci/foundation/legacy/Row/Row.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactJss = require("react-jss");

var _classnames = _interopRequireDefault(require("classnames"));

var spacer = {
  sm: 7,
  md: 20,
  lg: 45
};
var useStyles = (0, _reactJss.createUseStyles)({
  row: {
    '& > * ': {
      marginRight: function marginRight(_ref) {
        var spacing = _ref.spacing;
        return spacer[spacing];
      }
    },
    '& > *:last-child': {
      marginRight: 0,
      marginLeft: function marginLeft(_ref2) {
        var pushLastChildToEnd = _ref2.pushLastChildToEnd;
        return pushLastChildToEnd ? 'auto' : 'initial';
      }
    }
  }
});

var Row = _react["default"].forwardRef(function Row(_ref3, ref) {
  var className = _ref3.className,
      style = _ref3.style,
      children = _ref3.children,
      dense = _ref3.dense,
      fullWidth = _ref3.fullWidth,
      spacing = _ref3.spacing,
      pushLastChildToEnd = _ref3.pushLastChildToEnd,
      align = _ref3.align,
      justify = _ref3.justify,
      wrap = _ref3.wrap,
      onClick = _ref3.onClick,
      props = (0, _objectWithoutProperties2["default"])(_ref3, ["className", "style", "children", "dense", "fullWidth", "spacing", "pushLastChildToEnd", "align", "justify", "wrap", "onClick"]);
  var classes = useStyles({
    spacing: spacing,
    pushLastChildToEnd: pushLastChildToEnd
  });
  return /*#__PURE__*/_react["default"].createElement("div", (0, _extends2["default"])({}, props, {
    ref: ref,
    role: "presentation",
    onClick: onClick,
    className: (0, _classnames["default"])({
      clickable: onClick
    }, 'd-flex', (0, _defineProperty2["default"])({}, "row-wrapper ".concat(classes.row), !dense), {
      'w-fill': fullWidth
    }, (0, _defineProperty2["default"])({}, "align-items-".concat(align), align), (0, _defineProperty2["default"])({}, "justify-content-".concat(justify), justify), {
      'flex-row-wrap': wrap
    }, className),
    style: style
  }), children);
});

Row.defaultProps = {
  className: '',
  spacing: 'md',
  style: {},
  children: null,
  onClick: null,
  dense: false,
  fullWidth: false,
  align: 'inherit',
  justify: 'inherit',
  wrap: false,
  pushLastChildToEnd: false
};
Row.propTypes = {
  /** Additional classes for row */
  className: _propTypes["default"].string,

  /** Additional classes for row */
  align: _propTypes["default"].string,

  /** Additional classes for row */
  justify: _propTypes["default"].string,

  /** Additional styles for row */
  style: _propTypes["default"].shape(),

  /** Children for row wrapper */
  children: _propTypes["default"].oneOfType([_propTypes["default"].node, _propTypes["default"].arrayOf(_propTypes["default"].node)]),

  /** Boolean whether or not we are in dense (no margins) mode */
  dense: _propTypes["default"].bool,

  /** Boolean whether or not we are in full width mode */
  fullWidth: _propTypes["default"].bool,

  /** Amount of spacing between child elements -
   *  sm - 7 md - 20. By default in forms 20 is enforced.
   *  generally:
   *  sm - used for spacing inner components
   *  md - used for default spacing structures
   *  lg - used for spacing larger structures
   *  md is set by default
   */
  spacing: _propTypes["default"].oneOf(['sm', 'md', 'lg']),

  /** Set row as wrap to fit content */
  wrap: _propTypes["default"].bool,

  /** Push the last row child to the end of the row */
  pushLastChildToEnd: _propTypes["default"].bool,

  /** handle click event for Col */
  onClick: _propTypes["default"].func
}; // Hack for withInfo plugin

Row.displayName = 'Row';
var _default = Row;
exports["default"] = _default;
