// @guestyci/foundation/es/Layout/Row/Row.js
import _objectWithoutPropertiesLoose from "@babel/runtime/helpers/esm/objectWithoutPropertiesLoose";
import _extends from "@babel/runtime/helpers/esm/extends";
import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { firstLetterUppercase as capitalize } from "../../utils";
import createStyles from "../../createStyles";
import styles, { generateMediaSupport, generateSpacing, spacing as spacingArray } from "../styles";
export const useStyles = createStyles(theme => _extends({
  root: {
    display: 'flex'
  },
  flexDirectionReverse: {
    flexDirection: 'row-reverse'
  },
  lastChildSpacing: {
    '& > *:last-child': {
      marginRight: 0
    }
  }
}, styles, {}, generateMediaSupport(theme.breakpoints), {}, generateSpacing(theme.spacer, spacingArray, 'right')), {
  name: 'Row'
});
const Row = React.forwardRef(function Row(_ref, ref) {
  let {
    className,
    style,
    children,
    spacing,
    align,
    justify,
    wrap,
    fullWidth,
    reverse,
    breakpoints,
    span,
    jss = {}
  } = _ref,
      props = _objectWithoutPropertiesLoose(_ref, ["className", "style", "children", "spacing", "align", "justify", "wrap", "fullWidth", "reverse", "breakpoints", "span", "jss"]);

  const classes = useStyles({
    jss
  });
  const {
    xs,
    sm,
    md,
    lg,
    xl,
    xlg
  } = breakpoints || {};
  const xsSize = span || xs;
  return /*#__PURE__*/React.createElement("div", _extends({}, props, {
    ref: ref,
    className: cn(classes.root, {
      [classes.flexWrap]: wrap,
      [classes[`spacing${spacing}`]]: spacing,
      [classes.lastChildSpacing]: spacing,
      [classes.fullWidth]: fullWidth,
      [classes.flexDirectionReverse]: reverse,
      [classes[`justify${capitalize(justify)}`]]: justify !== 'start',
      [classes[`alignItems${capitalize(align)}`]]: align !== 'stretch',
      [classes[`col-xs-${xsSize}`]]: xsSize,
      [classes[`col-sm-${sm}`]]: sm,
      [classes[`col-md-${md}`]]: md,
      [classes[`col-lg-${lg}`]]: lg,
      [classes[`col-xl-${xl}`]]: xl,
      [classes[`col-xlg-${xlg}`]]: xlg
    }, 'row-wrapper', className),
    style: style
  }), children);
});
Row.defaultProps = {
  className: '',
  spacing: 0,
  style: {},
  children: null,
  justify: 'start',
  align: 'stretch',
  reverse: false,
  wrap: false,
  fullWidth: false,
  span: 0,
  breakpoints: null
};
Row.propTypes = {
  /** Additional class for top level */
  className: PropTypes.string,

  /** Indicator whether or not to render flex wrap */
  wrap: PropTypes.bool,

  /** Indicator whether or not to render flex direction reverse */
  reverse: PropTypes.bool,

  /** Additional styling for root level */
  style: PropTypes.shape(),

  /** set spacing between child components - available value - [0,1,2,3,4,5,6,7,8,9,10, 11 ,12] will be determined by the theme spacer (default 5) */
  spacing: PropTypes.oneOf([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]),

  /** Child components inside a section */
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.arrayOf(PropTypes.node)]),

  /** Set flex alignment - available values = ['start', 'center', 'end', 'stretch', 'baseline']  */
  align: PropTypes.oneOf(['start', 'center', 'end', 'stretch', 'baseline', 'inherit']),

  /** Set flex justify content - available values = ['start','center','end','between','around','evenly'] */
  justify: PropTypes.oneOf(['start', 'center', 'end', 'between', 'around', 'evenly', 'inherit']),

  /** Set the row width to fit parent */
  fullWidth: PropTypes.bool,

  /** Set the row span width - available values = [auto,1,2,3,4,5,6,7,8,9,10,11,12] */
  span: PropTypes.oneOf(['auto', 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]),

  /** specify the media query breakpoints for the row - available values = {xs,sm,md,lg,xl,xlg} */
  breakpoints: PropTypes.shape(),

  /** JSS overrides */
  jss: PropTypes.oneOfType([PropTypes.shape(), PropTypes.func])
}; // Hack for withInfo plugin

Row.displayName = 'Row';
export default Row;
