// @guestyci/foundation/legacy/Avatar/ImageAvatar.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.getClass = void 0;

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactJss = require("react-jss");

var getClass = function getClass(size) {
  switch (size) {
    case 'xl':
      return '80px';

    case 'md':
      return '40px';

    case 'lg':
      return '60px';

    case 'sm':
      return '25px';

    default:
      return '40px';
  }
};

exports.getClass = getClass;
var useStyles = (0, _reactJss.createUseStyles)({
  icon: {
    height: function height(_ref) {
      var size = _ref.size;
      return getClass(size);
    },
    width: function width(_ref2) {
      var size = _ref2.size;
      return getClass(size);
    }
  }
});

var ImageAvatar = function ImageAvatar(_ref3) {
  var text = _ref3.text,
      src = _ref3.src,
      _ref3$size = _ref3.size,
      size = _ref3$size === void 0 ? 'md' : _ref3$size,
      bordered = _ref3.bordered,
      disabled = _ref3.disabled;
  var classes = useStyles({
    size: size
  });
  return /*#__PURE__*/_react["default"].createElement("img", {
    className: (0, _classnames["default"])("img-avatar avatar", 'd-inline-block', {
      'avatar-border': bordered
    }, {
      disabled: disabled
    }, classes.icon),
    src: src,
    alt: text ? text.trim() : '',
    "data-qa": "image-avatar"
  });
};

ImageAvatar.propTypes = {
  src: _propTypes["default"].string.isRequired,
  size: _propTypes["default"].oneOf(['sm', 'md', 'lg', 'xl']),
  disabled: _propTypes["default"].bool,
  text: _propTypes["default"].string,
  bordered: _propTypes["default"].bool
};
var _default = ImageAvatar;
exports["default"] = _default;
