// @guestyci/foundation/legacy/Avatar/TextAvatar.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.getClass = void 0;

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactJss = require("react-jss");

var _utils = require("../../utils");

var getClass = function getClass(size) {
  switch (size) {
    case 'xl':
      return '80px';

    case 'md':
      return '40px';

    case 'lg':
      return '60px';

    case 'sm':
      return '25px';

    default:
      return '40px';
  }
};

exports.getClass = getClass;
var useStyles = (0, _reactJss.createUseStyles)({
  icon: {
    height: function height(_ref) {
      var size = _ref.size;
      return getClass(size);
    },
    width: function width(_ref2) {
      var size = _ref2.size;
      return getClass(size);
    }
  }
});

var convertText = function convertText() {
  var text = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : '';
  var avatarText = text.replace(/[\W_][ ]+/g, '');
  var words = avatarText.split(' ');

  if (avatarText.length < 2 || avatarText === '') {
    return '- -';
  }

  return words.length > 1 ? (0, _utils.nameToInitials)(avatarText) : avatarText;
};

var TextAvatar = function TextAvatar(_ref3) {
  var text = _ref3.text,
      isUser = _ref3.isUser,
      _ref3$size = _ref3.size,
      size = _ref3$size === void 0 ? 'md' : _ref3$size,
      disabled = _ref3.disabled;
  var avatarClasses = useStyles({
    size: size
  });
  var avatarText = (0, _react.useMemo)(function () {
    return convertText(text);
  }, [text]);
  var textClasses = 'text-avatar text-uppercase text-white d-flex align-items-center justify-content-center';
  var bgColor = isUser ? 'blue-dark' : 'gray-darkest';
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])(textClasses, 'flex-center', "avatar", "bg-".concat(bgColor), {
      disabled: disabled
    }, avatarClasses.icon, {
      'font-size-3xl': size === 'xl'
    }),
    "data-qa": "text-avatar"
  }, /*#__PURE__*/_react["default"].createElement("span", null, avatarText));
};

TextAvatar.propTypes = {
  text: _propTypes["default"].string.isRequired,
  isUser: _propTypes["default"].bool,
  size: _propTypes["default"].oneOf(['sm', 'md', 'lg']),
  disabled: _propTypes["default"].bool
};
var _default = TextAvatar;
exports["default"] = _default;
