// @guestyci/foundation/legacy/Table/StatusCell/StatusCell.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = require("prop-types");

var _classnames = _interopRequireDefault(require("classnames"));

var _Pill = _interopRequireDefault(require("../../Pill"));

var _useTranslations3 = _interopRequireDefault(require("@guestyci/localize/useTranslations"));

var _utils = require("../../../utils");

var _Cell = _interopRequireDefault(require("../Cell"));

var _statuses = _interopRequireDefault(require("./statuses"));

var _statusColors;

var translations = [{
  id: 'smartviews_v3:status_cell.status.pending',
  d: 'Pending'
}, {
  id: 'smartviews_v3:status_cell.status.approved',
  d: 'Approved'
}, {
  id: 'smartviews_v3:status_cell.status.flagged',
  d: 'Flagged'
}, {
  id: 'smartviews_v3:status_cell.status.archived',
  d: 'Archived'
}, {
  id: 'smartviews_v3:status_cell.status.regenerating',
  d: 'Regenerating'
}, {
  id: 'smartviews_v3:status_cell.status.regeneration_failed',
  d: 'Regeneration Failed'
}, {
  id: 'smartviews_v3:status_cell.status.regeneration_required',
  d: 'Regeneration required'
}, {
  id: 'smartviews_v3:status_cell.status.unknown',
  d: 'Unknown'
}];
var statusColors = (_statusColors = {}, (0, _defineProperty2["default"])(_statusColors, _statuses["default"].Pending, 'yellow'), (0, _defineProperty2["default"])(_statusColors, _statuses["default"].Approved, 'green'), (0, _defineProperty2["default"])(_statusColors, _statuses["default"].Flagged, 'red'), (0, _defineProperty2["default"])(_statusColors, _statuses["default"].Archived, 'gray'), (0, _defineProperty2["default"])(_statusColors, _statuses["default"].Regenerating, 'blue'), (0, _defineProperty2["default"])(_statusColors, _statuses["default"].RegenerationFailed, 'red'), (0, _defineProperty2["default"])(_statusColors, _statuses["default"].RegenerationRequired, 'gray'), (0, _defineProperty2["default"])(_statusColors, "unknown", 'black'), _statusColors);

function StatusCell(_ref) {
  var _statusTitles;

  var status = _ref.status,
      isLoading = _ref.isLoading,
      overflowEllipsis = _ref.overflowEllipsis,
      overflowEllipsisTooltip = _ref.overflowEllipsisTooltip,
      className = _ref.className,
      contentClassName = _ref.contentClassName,
      style = _ref.style,
      rest = (0, _objectWithoutProperties2["default"])(_ref, ["status", "isLoading", "overflowEllipsis", "overflowEllipsisTooltip", "className", "contentClassName", "style"]);

  var _useTranslations = (0, _useTranslations3["default"])(translations),
      _useTranslations2 = (0, _slicedToArray2["default"])(_useTranslations, 8),
      pending = _useTranslations2[0],
      approved = _useTranslations2[1],
      flagged = _useTranslations2[2],
      archived = _useTranslations2[3],
      regenerating = _useTranslations2[4],
      regenerationFailed = _useTranslations2[5],
      regenerationRequired = _useTranslations2[6],
      unknown = _useTranslations2[7];

  var statusTitles = (_statusTitles = {}, (0, _defineProperty2["default"])(_statusTitles, _statuses["default"].Pending, pending), (0, _defineProperty2["default"])(_statusTitles, _statuses["default"].Approved, approved), (0, _defineProperty2["default"])(_statusTitles, _statuses["default"].Flagged, flagged), (0, _defineProperty2["default"])(_statusTitles, _statuses["default"].Archived, archived), (0, _defineProperty2["default"])(_statusTitles, _statuses["default"].Regenerating, regenerating), (0, _defineProperty2["default"])(_statusTitles, _statuses["default"].RegenerationFailed, regenerationFailed), (0, _defineProperty2["default"])(_statusTitles, _statuses["default"].RegenerationRequired, regenerationRequired), _statusTitles);
  var displayStatus = statusTitles[status] || unknown;
  return /*#__PURE__*/_react["default"].createElement(_Cell["default"], (0, _extends2["default"])({}, rest, {
    isLoading: isLoading,
    overflowEllipsisTooltip: overflowEllipsisTooltip || displayStatus,
    overflowEllipsis: overflowEllipsis,
    className: (0, _classnames["default"])('status-cell', className),
    contentClassName: contentClassName,
    style: style,
    "data-qa": "status-cell"
  }), !(0, _utils.isEmpty)(status) && /*#__PURE__*/_react["default"].createElement(_Pill["default"], {
    bgColor: statusColors[status] || statusColors.unknown,
    className: "text-center"
  }, displayStatus));
}

StatusCell.propTypes = {
  /** value of a status */
  status: _propTypes.string,

  /** indicator that cell is yet to receive data */
  isLoading: _propTypes.bool,

  /** indicator that cell should trim cell content and show tooltip with full data */
  overflowEllipsis: _propTypes.bool,

  /** tooltip body of an overflowed cell content */
  overflowEllipsisTooltip: _propTypes.node,

  /** class name of cell */
  className: _propTypes.string,

  /** class name of cell content wrapper */
  contentClassName: _propTypes.string,

  /** styles applied to a cell */
  style: (0, _propTypes.shape)()
};
StatusCell.defaultProps = {
  status: '',
  isLoading: false,
  className: '',
  contentClassName: '',
  style: {},
  overflowEllipsis: true,
  overflowEllipsisTooltip: null
};
var _default = StatusCell;
exports["default"] = _default;
