// @guestyci/foundation/esm/legacy/ActionPopover/ActionPopoverBody.js
import React from 'react';
import PropTypes from 'prop-types';
import Col from "../Col/Col";
import RaisedButton from "../RaisedButton/RaisedButton";

var ActionPopoverBody = function ActionPopoverBody(_ref) {
  var buttonText = _ref.buttonText,
      onClick = _ref.onClick,
      body = _ref.body;
  return /*#__PURE__*/React.createElement(Col, {
    spacing: 2,
    "data-qa": "action-popover-body"
  }, /*#__PURE__*/React.createElement("div", {
    className: "text-left"
  }, body), /*#__PURE__*/React.createElement("div", {
    className: "flex-end"
  }, /*#__PURE__*/React.createElement(RaisedButton, {
    onClick: onClick
  }, buttonText)));
};

ActionPopoverBody.propTypes = {
  /** Sets the text of the button */
  buttonText: PropTypes.string.isRequired,

  /** Callback function - called when clicking the tooltip action */
  onClick: PropTypes.func.isRequired,

  /** Sets the body (content) of the tooltip F.E: text to be displayed */
  body: PropTypes.node.isRequired
};
export default ActionPopoverBody;
