// @guestyci/foundation/Avatar/Avatar.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.useStyles = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _createStyles = _interopRequireDefault(require("../createStyles"));

var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    root: {
      position: 'relative',
      backgroundColor: theme.palette.background.system,
      borderRadius: '50%',
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center'
    },
    icon: {
      padding: theme.spacer(2),
      textTransform: 'uppercase'
    },
    small: {
      width: 24,
      height: 24,
      '& $icon': {
        padding: theme.spacer(1)
      }
    },
    large: {
      width: 40,
      height: 40
    },
    disabled: {
      opacity: 0.3
    },
    img: {
      objectFit: 'cover',
      width: '100%',
      height: '100%',
      borderRadius: 'inherit'
    }
  };
}, {
  name: 'Avatar'
});
exports.useStyles = useStyles;

var Avatar = _react["default"].forwardRef(function Avatar(_ref, ref) {
  var _cn;

  var _ref$alt = _ref.alt,
      alt = _ref$alt === void 0 ? '' : _ref$alt,
      _ref$src = _ref.src,
      src = _ref$src === void 0 ? '' : _ref$src,
      _ref$children = _ref.children,
      children = _ref$children === void 0 ? null : _ref$children,
      _ref$className = _ref.className,
      className = _ref$className === void 0 ? '' : _ref$className,
      _ref$style = _ref.style,
      style = _ref$style === void 0 ? {} : _ref$style,
      _ref$jss = _ref.jss,
      jss = _ref$jss === void 0 ? {} : _ref$jss,
      _ref$disabled = _ref.disabled,
      disabled = _ref$disabled === void 0 ? false : _ref$disabled,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? 'large' : _ref$size,
      _ref$badge = _ref.badge,
      badge = _ref$badge === void 0 ? null : _ref$badge,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["alt", "src", "children", "className", "style", "jss", "disabled", "size", "badge"]);
  var classes = useStyles({
    jss: jss
  });
  var showChildren = !src && children;
  return /*#__PURE__*/_react["default"].createElement("div", (0, _extends2["default"])({}, props, {
    ref: ref,
    className: (0, _classnames["default"])(classes.root, (_cn = {}, (0, _defineProperty2["default"])(_cn, classes.disabled, disabled), (0, _defineProperty2["default"])(_cn, classes.small, size === 'small'), (0, _defineProperty2["default"])(_cn, classes.large, size === 'large'), _cn), className),
    style: style
  }), src && /*#__PURE__*/_react["default"].createElement("img", {
    className: classes.img,
    alt: alt,
    src: src
  }), showChildren && /*#__PURE__*/_react["default"].createElement("div", {
    className: classes.icon
  }, children), badge && badge);
});

Avatar.propTypes = {
  /** Badge component */
  badge: _propTypes["default"].node,

  /** Avatar size - values: [small , large] */
  size: _propTypes["default"].oneOf(['small', 'large']),

  /** Image alt */
  alt: _propTypes["default"].string,

  /** Image src in case we want to render an image as avatar */
  src: _propTypes["default"].string,

  /** In case we render icon or children components pass them as children */
  children: _propTypes["default"].node,

  /** Additional root level classes */
  className: _propTypes["default"].string,

  /** Additional root level styles */
  style: _propTypes["default"].shape(),

  /** Disabled mode for avatars */
  disabled: _propTypes["default"].bool,

  /** jss override object to customize the jss classes */
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func])
};
Avatar.displayName = 'Avatar';
var _default = Avatar;
exports["default"] = _default;
