// @guestyci/foundation/esm/legacy/Drawer/withDrawer.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _possibleConstructorReturn from "@babel/runtime/helpers/esm/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/esm/getPrototypeOf";

function _createSuper(Derived) { return function () { var Super = _getPrototypeOf(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

import React, { Component } from 'react';
/**
 * Create a wrapper compose to support Drawer functionality
 * @param ComposedComponent
 * @returns {{Component}}
 */

var withDrawer = function withDrawer(ComposedComponent) {
  var _temp;

  return _temp = /*#__PURE__*/function (_Component) {
    _inherits(_temp, _Component);

    var _super = _createSuper(_temp);

    function _temp(props) {
      var _this;

      _classCallCheck(this, _temp);

      _this = _super.call(this, props);

      _this.defaultState = function (isOpen) {
        _this.setState({
          show: isOpen
        });
      };

      _this.closeDrawer = function () {
        _this.setState({
          show: false
        });
      };

      _this.openDrawer = function (event) {
        if (event) {
          event.stopPropagation();
        }

        _this.setState({
          show: true
        });
      };

      _this.toggleDrawer = function (event) {
        if (event) {
          event.stopPropagation();
        }

        _this.setState(function (prevState) {
          return {
            show: !prevState.show
          };
        });
      };

      _this.state = {
        show: false
      };
      return _this;
    }

    _createClass(_temp, [{
      key: "render",
      value: function render() {
        var show = this.state.show;
        return /*#__PURE__*/React.createElement(ComposedComponent, _extends({}, this.props, {
          isDrawerOpen: show,
          closeDrawer: this.closeDrawer,
          openDrawer: this.openDrawer,
          toggleDrawer: this.toggleDrawer,
          drawerDefaultState: this.defaultState
        }));
      }
    }]);

    return _temp;
  }(Component), _temp;
};

export default withDrawer;
