// @guestyci/foundation/es/legacy/TimePicker/ItemSelector.js
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import IconButton from "../IconButton/IconButton";

var BtnArrowUp = function BtnArrowUp(props) {
  return /*#__PURE__*/React.createElement("svg", props, /*#__PURE__*/React.createElement("path", {
    d: "M4.529 0a.474.474 0 0 0-.357.137L.137 4.172a.488.488 0 0 0 .69.689L4.53 1.156l3.705 3.705a.488.488 0 0 0 .69-.69L4.89.138A.477.477 0 0 0 4.53 0z"
  }));
};

BtnArrowUp.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 9.0618038 4.9977479"
};

var BtnArrowDown = function BtnArrowDown(props) {
  return /*#__PURE__*/React.createElement("svg", props, /*#__PURE__*/React.createElement("path", {
    d: "M4.533 4.997c.13.006.26-.039.357-.136L8.925.826a.488.488 0 0 0-.69-.69L4.532 3.843.826.137a.488.488 0 0 0-.69.69L4.172 4.86c.097.097.23.143.362.136z"
  }));
};

BtnArrowDown.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 9.0618038 4.9977479"
};
import TextField from "../TextField";
import Col from "../Col";

class ItemSelector extends Component {
  constructor(props) {
    super(props);

    this.onIncrease = () => {
      const {
        step,
        values,
        onChange
      } = this.props;
      const {
        index: stateIndex
      } = this.state;
      const index = (stateIndex + step) % values.length;
      this.setState({
        index
      });
      onChange(index);
    };

    this.onDecrease = () => {
      const {
        step,
        values,
        onChange
      } = this.props;
      const {
        index: stateIndex
      } = this.state;
      const index = (stateIndex - step + values.length) % values.length;
      this.setState({
        index
      });
      onChange(index);
    };

    this.state = {
      index: props.index || 0
    };
  }

  render() {
    const {
      index
    } = this.state;
    const {
      renderValue,
      values
    } = this.props;
    const rawValue = values[index];
    const value = renderValue ? renderValue(rawValue) : rawValue;
    return /*#__PURE__*/React.createElement(Col, {
      spacing: 0,
      align: "center",
      justify: "center",
      className: "fill-hover-blue width-35",
      "data-qa": "item-selector"
    }, /*#__PURE__*/React.createElement(IconButton, {
      onClick: this.onIncrease,
      svg: BtnArrowUp,
      iconWidth: 9,
      iconHeight: 5
    }), /*#__PURE__*/React.createElement(TextField, {
      color: "gray-dark"
    }, value), /*#__PURE__*/React.createElement(IconButton, {
      onClick: this.onDecrease,
      svg: BtnArrowDown,
      iconWidth: 9,
      iconHeight: 5
    }));
  }

}

ItemSelector.defaultProps = {
  onChange() {},

  step: 1,
  values: [],
  renderValue: null
};
ItemSelector.propTypes = {
  onChange: PropTypes.func,
  step: PropTypes.number,
  values: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.string, PropTypes.number])),
  renderValue: PropTypes.func
};
export default ItemSelector;
