// @guestyci/foundation/legacy/List/List.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.ListComponent = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _reactJss = require("react-jss");

var _CSSTransiton = _interopRequireDefault(require("../CSSTransiton"));

var useStyles = (0, _reactJss.createUseStyles)({
  list: {
    width: '100%',
    overflowX: function overflowX(_ref) {
      var row = _ref.row,
          scrollable = _ref.scrollable;
      return row && scrollable ? 'auto' : 'hidden';
    },
    overflowY: function overflowY(_ref2) {
      var noScroll = _ref2.noScroll,
          scrollable = _ref2.scrollable,
          row = _ref2.row;
      return !row && scrollable || !noScroll ? 'auto' : 'hidden';
    },
    padding: 0,
    listStyleType: 'none',
    height: '100%'
  }
});

var List = function List(_ref3) {
  var children = _ref3.children,
      className = _ref3.className,
      style = _ref3.style,
      maxHeight = _ref3.maxHeight,
      transition = _ref3.transition,
      row = _ref3.row,
      innerClassName = _ref3.innerClassName,
      noScroll = _ref3.noScroll,
      scrollable = _ref3.scrollable,
      id = _ref3.id;
  var classes = useStyles({
    row: row,
    maxHeight: maxHeight,
    noScroll: noScroll,
    scrollable: scrollable
  });
  return /*#__PURE__*/_react["default"].createElement("ul", {
    className: (0, _classnames["default"])(classes.list, className, 'mb-0'),
    style: style,
    id: id,
    "data-qa": "list"
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])('d-flex-fill', !row ? 'flex-column' : 'flex-row-wrap', innerClassName)
  }, /*#__PURE__*/_react["default"].createElement(_CSSTransiton["default"], {
    animation: transition,
    show: true,
    className: "inherit-all"
  }, children)));
};

List.defaultProps = {
  className: '',
  style: {},
  maxHeight: null,
  transition: null,
  row: false,
  innerClassName: '',
  scrollable: true,
  id: ''
};
List.propTypes = {
  /** List of elements under the list item */
  children: _propTypes["default"].node.isRequired,

  /** Additional class for list component */
  className: _propTypes["default"].string,

  /** Additional style for the list component */
  style: _propTypes["default"].shape(),

  /** Maximum height for the component */
  maxHeight: _propTypes["default"].number,

  /** Transition for add/remove child elements */
  transition: _propTypes["default"].string,

  /** Indicator whether or not we want list as row or column */
  row: _propTypes["default"].bool,

  /** List inner class */
  innerClassName: _propTypes["default"].string,

  /** List id */
  id: _propTypes["default"].string,

  /** Indicator whether or not to allow scroll */
  scrollable: _propTypes["default"].bool
};
/** Work around to fix storybook HOC propTypes bug  - DO NOT IMPORT THIS */

var ListComponent = List;
exports.ListComponent = ListComponent;
var _default = List;
exports["default"] = _default;
