// @guestyci/foundation/esm/legacy/BedPicker/BedPicker.js
import React from 'react';
import PropTypes from 'prop-types';
import TilesPicker from "../tiles/TilesPicker";
var optionsMap = {
  single: {
    icon: 'IcoBedSingle',
    width: 27,
    height: 29,
    text: 'Single'
  },
  "double": {
    icon: 'IcoBedDouble',
    width: 40,
    height: 29,
    text: 'Double'
  },
  queen: {
    icon: 'IcoBedDouble',
    width: 40,
    height: 29,
    text: 'Queen'
  },
  king: {
    icon: 'IcoBedDouble',
    width: 40,
    height: 29,
    text: 'King'
  },
  bunk: {
    icon: 'IcoBedBunk',
    width: 29,
    height: 34,
    text: 'Bunk'
  },
  sofa: {
    icon: 'IcoBedSofa',
    width: 40,
    height: 24,
    text: 'Sofa'
  },
  mat: {
    icon: 'IcoBedMat',
    width: 40,
    height: 24,
    text: 'Mat'
  }
};
var options = [{
  label: 'Single',
  value: 'single'
}, {
  label: 'Double',
  value: 'double'
}, {
  label: 'Queen',
  value: 'queen'
}, {
  label: 'King',
  value: 'king'
}, {
  label: 'Bunk',
  value: 'bunk'
}, {
  label: 'Sofa',
  value: 'sofa'
}, {
  label: 'Mat',
  value: 'mat'
}];

var BedPicker = function BedPicker(_ref) {
  var maxShownItems = _ref.maxShownItems,
      onChangeBeds = _ref.onChangeBeds;
  return /*#__PURE__*/React.createElement(TilesPicker, {
    maxShownItems: maxShownItems,
    optionsMap: optionsMap,
    options: options,
    onChange: onChangeBeds,
    "data-qa": "bed-picker"
  });
};

BedPicker.defaultProps = {
  onChangeBeds: function onChangeBeds() {},
  maxShownItems: 4
};
BedPicker.propTypes = {
  // sets the number of tiles what would be rendered
  maxShownItems: PropTypes.number,
  // callback to be called when a tile is changed
  onChangeBeds: PropTypes.func
};
export default BedPicker;
