// @guestyci/foundation/legacy/InfiniteScroll/withInfiniteScroll.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _react = _interopRequireWildcard(require("react"));

function _createSuper(Derived) { return function () { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

/**
 * Create a wrapper compose to support infinite scroll functionality
 * @param ComposedComponent
 * @returns {{Component}}
 */
function withInfiniteScroll(ComposedComponent) {
  var InfiniteScrollWrapper = /*#__PURE__*/function (_Component) {
    (0, _inherits2["default"])(InfiniteScrollWrapper, _Component);

    var _super = _createSuper(InfiniteScrollWrapper);

    function InfiniteScrollWrapper(props) {
      var _this;

      (0, _classCallCheck2["default"])(this, InfiniteScrollWrapper);
      _this = _super.call(this, props);

      _this.setCurrentPage = function (page) {
        _this.setState({
          page: page
        });
      };

      _this.incrementCurrentPage = function () {
        _this.setState(function (prevState) {
          return {
            page: prevState.page + 1
          };
        });
      };

      _this.initCurrentPage = function () {
        var page = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 0;

        _this.setState({
          page: page
        });
      };

      _this.state = {
        page: props.initialPage || 0
      };
      return _this;
    }

    (0, _createClass2["default"])(InfiniteScrollWrapper, [{
      key: "render",
      value: function render() {
        var page = this.state.page;
        return /*#__PURE__*/_react["default"].createElement(ComposedComponent, (0, _extends2["default"])({}, this.props, {
          infiniteScrollCurrentPage: page,
          setInfiniteScrollCurrentPage: this.setCurrentPage,
          initInfiniteScrollCurrentPage: this.initCurrentPage,
          incrementInfiniteScrollCurrentPage: this.incrementCurrentPage,
          "data-qa": "infinite-scroll-wrapper"
        }));
      }
    }]);
    return InfiniteScrollWrapper;
  }(_react.Component);

  return InfiniteScrollWrapper;
}

var _default = withInfiniteScroll;
exports["default"] = _default;
