// @guestyci/foundation/esm/enums/enums.js
/*eslint-disable */
function Enum(a) {
  var i = Object.keys(a).reduce(function (o, k) {
    return o[a[k]] = k, o;
  }, {});
  return Object.freeze(Object.keys(a).reduce(function (o, k) {
    return o[k] = a[k], o;
  }, function (v) {
    return i[v];
  }));
}

export var StoryStatus = Enum({
  Verified: "\u2713",
  Deprecated: "\u26A0",
  Failed: "\xD7"
});
export var Theme = Enum({
  Light: 'light',
  Dark: 'dark',
  Error: 'error',
  Yellow: 'yellow'
});
export var StayStatus = Enum({
  Planned: 'pre',
  Staying: 'curr',
  Complete: 'post',
  Reserved: 'reserved'
});
export var CalendarDayStatus = Enum({
  Booked: 'booked',
  Reserved: 'reserved',
  Available: 'available',
  Unavailable: 'unavailable',
  Confirmed: 'confirmed'
});
export var DropdownButtonColors = Enum({
  primary: 'blue',
  success: 'green',
  warning: 'yellow',
  danger: 'red',
  secondary: 'gray-dark'
});
export var DatePickerType = Enum({
  Single: 'single',
  Range: 'range'
});
export var DateDisplayFormat = 'MMM D, YYYY';
export var ActionTooltipType = Enum({
  Button: 'button',
  Link: 'link'
});
export var AnchorDirectionType = Enum({
  Left: 'left',
  Center: 'center',
  Right: 'right',
  EndCenter: 'end center',
  StartCenter: 'start center',
  Start: 'start',
  End: 'end'
});
export var OriginDirectionType = Enum({
  Top: 'top',
  Bottom: 'bottom',
  Left: 'left',
  Right: 'right'
});
export var MenuSlideAnchorDirectionType = Enum({
  Top: 'top',
  Bottom: 'bottom',
  Center: 'center',
  Left: 'left',
  Right: 'right'
});
export var FormProvider = Enum({
  Redux: 'redux',
  Final: 'final'
});
