// @guestyci/foundation/DatePicker/SingleDatePicker/SingleDatePicker.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.useStyles = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _constants = require("react-dates/lib/constants");

var _moment = _interopRequireDefault(require("moment"));

require("react-dates/initialize");

var _reactDates = require("react-dates");

var _classnames = _interopRequireDefault(require("classnames"));

var _createStyles = _interopRequireDefault(require("../../createStyles"));

var _styles = _interopRequireDefault(require("../styles"));

var _commonUtility = require("../../utils/commonUtility");

var _NavButton = _interopRequireDefault(require("../NavButton"));

var _PickerClearButton = _interopRequireDefault(require("../../basePicker/PickerClearButton"));

var _PickerInputButton = _interopRequireDefault(require("../../basePicker/PickerInputButton"));

var _Row = _interopRequireDefault(require("../../Layout/Row"));

var _constants2 = require("../constants");

var BtnCalendar = function BtnCalendar(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M18.5 2h-4V0h-1v2h-6V0h-1v2h-4A2.503 2.503 0 0 0 0 4.5v12C0 17.879 1.122 19 2.5 19h16c1.379 0 2.5-1.121 2.5-2.5v-12C21 3.122 19.879 2 18.5 2zm-16 1h4v.667h1V3h6v.667h1V3h4c.827 0 1.5.673 1.5 1.5V7H1V4.5C1 3.673 1.673 3 2.5 3zm16 15h-16c-.827 0-1.5-.673-1.5-1.5V8h19v8.5c0 .827-.673 1.5-1.5 1.5z"
  }));
};

BtnCalendar.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 21 19"
};
var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    root: _styles["default"].root(theme),
    inputContainer: _styles["default"].inputContainer,
    focused: _styles["default"].focused(theme),
    error: _styles["default"].error(theme),
    disabled: _styles["default"].disabled(theme)
  };
}, {
  name: 'SingleDatePicker'
});
exports.useStyles = useStyles;

var renderMonthElement = function renderMonthElement(_ref) {
  var month = _ref.month;
  return month.format(_constants2.MONTH_DISPLAY_FORMAT);
};

var SingleDatePicker = (0, _react.forwardRef)(function SingleDatePicker(_ref2, ref) {
  var id = _ref2.id,
      name = _ref2.name,
      blockedDates = _ref2.blockedDates,
      disabled = _ref2.disabled,
      initialVisibleMonth = _ref2.initialVisibleMonth,
      value = _ref2.value,
      onMonthChange = _ref2.onMonthChange,
      onChange = _ref2.onChange,
      enablePastDays = _ref2.enablePastDays,
      focusedProp = _ref2.focused,
      onFocusChange = _ref2.onFocusChange,
      _ref2$isFetching = _ref2.isFetching,
      isFetching = _ref2$isFetching === void 0 ? false : _ref2$isFetching,
      _ref2$anchorDirection = _ref2.anchorDirection,
      anchorDirection = _ref2$anchorDirection === void 0 ? 'left' : _ref2$anchorDirection,
      _ref2$showClearDate = _ref2.showClearDate,
      showClearDate = _ref2$showClearDate === void 0 ? true : _ref2$showClearDate,
      error = _ref2.error,
      className = _ref2.className,
      style = _ref2.style,
      jss = _ref2.jss,
      rest = (0, _objectWithoutProperties2["default"])(_ref2, ["id", "name", "blockedDates", "disabled", "initialVisibleMonth", "value", "onMonthChange", "onChange", "enablePastDays", "focused", "onFocusChange", "isFetching", "anchorDirection", "showClearDate", "error", "className", "style", "jss"]);
  var classes = useStyles({
    jss: jss
  });

  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      focusedState = _useState2[0],
      setFocusState = _useState2[1];

  var isFocused = focusedProp || focusedState;

  var handleFocusChange = function handleFocusChange(_ref3) {
    var focused = _ref3.focused;

    if (onFocusChange) {
      onFocusChange(focused);
    }

    setFocusState(focused);
  };

  var handleCalendarButtonClick = function handleCalendarButtonClick() {
    if (!isFocused) {
      if (onFocusChange) {
        onFocusChange(true);
      }

      setFocusState(true);
    }
  };

  var renderDayContents = function renderDayContents(day) {
    if (isFetching) {
      return '...';
    }

    return day.date();
  };

  var inputId = (0, _react.useMemo)(function () {
    return id || (0, _commonUtility.uuid)();
  }, [id]);
  var blockedDatesSet = (0, _react.useMemo)(function () {
    return new Set(blockedDates || []);
  }, [blockedDates]);

  var isDayBlocked = function isDayBlocked(day) {
    var formattedDay = day.format(_constants2.DATE_FORMAT);
    return isFetching || !!blockedDatesSet.has(formattedDay);
  };

  var getInitialVisibleMonth = function getInitialVisibleMonth() {
    return initialVisibleMonth ? (0, _moment["default"])(initialVisibleMonth) : (0, _moment["default"])();
  };

  var isOutsideRange = enablePastDays ? function () {
    return false;
  } : undefined;

  var handleDateChange = function handleDateChange(newDate) {
    if (onChange) {
      onChange(newDate, name);
    }
  };

  var handleClearDates = function handleClearDates() {
    handleDateChange(undefined);
  };

  return /*#__PURE__*/_react["default"].createElement(_Row["default"], {
    justify: "between",
    className: (0, _classnames["default"])('date-picker', classes.root, (0, _defineProperty2["default"])({}, classes.focused, isFocused), (0, _defineProperty2["default"])({}, classes.error, error), (0, _defineProperty2["default"])({}, classes.disabled, disabled), className),
    style: style,
    ref: ref
  }, /*#__PURE__*/_react["default"].createElement(_Row["default"], {
    className: classes.inputContainer
  }, /*#__PURE__*/_react["default"].createElement(_PickerInputButton["default"], {
    disabled: disabled,
    onClick: handleCalendarButtonClick,
    svg: BtnCalendar,
    active: !!isFocused
  }), /*#__PURE__*/_react["default"].createElement(_reactDates.SingleDatePicker, (0, _extends2["default"])({
    id: inputId,
    date: value ? (0, _moment["default"])(value) : value,
    onDateChange: handleDateChange,
    placeholder: "Select",
    focused: isFocused,
    disabled: disabled,
    initialVisibleMonth: getInitialVisibleMonth,
    onFocusChange: handleFocusChange,
    isDayBlocked: isDayBlocked,
    isOutsideRange: isOutsideRange,
    hideKeyboardShortcutsPanel: true,
    noBorder: true,
    openDirection: _constants.OPEN_DOWN,
    readOnly: true,
    daySize: 40,
    verticalSpacing: 4,
    numberOfMonths: 1,
    displayFormat: _constants2.DATE_DISPLAY_FORMAT,
    renderMonthElement: renderMonthElement,
    renderDayContents: renderDayContents,
    appendToBody: false,
    anchorDirection: anchorDirection,
    onNextMonthClick: onMonthChange,
    onPrevMonthClick: onMonthChange,
    navPrev: /*#__PURE__*/_react["default"].createElement(_NavButton["default"], {
      direction: "Left"
    }),
    navNext: /*#__PURE__*/_react["default"].createElement(_NavButton["default"], {
      direction: "Right"
    })
  }, rest)), /*#__PURE__*/_react["default"].createElement(_PickerClearButton["default"], {
    show: value && showClearDate,
    onClick: handleClearDates
  })));
});
SingleDatePicker.displayName = 'SingleDatePicker';
var _default = SingleDatePicker;
exports["default"] = _default;
SingleDatePicker.propTypes = {
  /** Id for date picker */
  id: _propTypes["default"].string,

  /** date picker name to be set */
  name: _propTypes["default"].string,

  /** Array of blocked dates - formatted as YYYY-MM-DD */
  blockedDates: _propTypes["default"].arrayOf(_propTypes["default"].string),

  /** Disable state in which picker interaction is not available */
  disabled: _propTypes["default"].bool,

  /** month to be show when the dialog picker opens - formatted as YYYY-MM-DD */
  initialVisibleMonth: _propTypes["default"].string,

  /** the picked date - formatted as YYYY-MM-DD */
  value: _propTypes["default"].string,

  /** callback to be called when a switching months. Called with a momnet object of the new month */
  onMonthChange: _propTypes["default"].func,

  /** Callback to be called on picker value change */
  onChange: _propTypes["default"].func,

  /** Enable past days */
  enablePastDays: _propTypes["default"].bool,

  /** Control focus state - focued => picker opens. Not required - focus state is uncontrolled by default */
  focused: _propTypes["default"].bool,

  /** Callback to be called when focus state changes. Called with a boolean  */
  onFocusChange: _propTypes["default"].func,

  /** Fetching state */
  isFetching: _propTypes["default"].bool,

  /** Anchor direction to open the picker - */
  anchorDirection: _propTypes["default"].oneOf(['left', 'right']),

  /** When dates is selected - show a clear button */
  showClearDate: _propTypes["default"].bool,

  /** Indicator for form error state  */
  error: _propTypes["default"].bool,

  /** class name to be added to the root */
  className: _propTypes["default"].string,

  /** style to be added to the root */
  style: _propTypes["default"].shape(),
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func])
};
