// @guestyci/foundation/theme/defaultTheme.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _colors = require("./colors");

var _zIndex = _interopRequireDefault(require("./zIndex"));

var _transition = _interopRequireDefault(require("./transition"));

var _boxShadow = _interopRequireDefault(require("./boxShadow"));

var _spacing = _interopRequireDefault(require("./spacing"));

var _typography = _interopRequireDefault(require("./typography"));

var _breakpoints = _interopRequireDefault(require("./breakpoints"));

var _default = {
  palette: {
    // deprecated
    title: _colors.darkBlue900,
    label: _colors.blueGray500,
    border: _colors.darkBlue10,
    description: _colors.blueGray500,
    placeholder: _colors.blueGray300,
    disabled: _colors.darkBlue10,
    disabledText: _colors.blueGray300,
    link: _colors.blue700,
    // texts
    text: {
      "default": _colors.blueGray500,
      dark: _colors.darkBlue900,
      secondary: _colors.blueGray400,
      placeholder: _colors.blueGray300,
      light: _colors.blueGray100,
      white: _colors.white,
      link: _colors.blue700,
      // deprecated
      primary: _colors.gray900,
      input: _colors.gray900
    },
    // backgrounds
    background: {
      system: _colors.gray100,
      secondary: _colors.darkBlue10,
      white: _colors.white
    },
    // system
    error: {
      "default": _colors.red400,
      text: _colors.red700,
      hover: _colors.red200,
      background: _colors.red10
    },
    alert: {
      "default": _colors.yellow400,
      text: _colors.yellow800,
      hover: _colors.yellow300,
      background: _colors.yellow10
    },
    success: {
      "default": _colors.green400,
      text: _colors.green800,
      hover: _colors.green300,
      background: _colors.green10
    },
    info: {
      "default": _colors.purple400,
      text: _colors.purple600,
      hover: _colors.purple300,
      background: _colors.purple10
    },
    primary: {
      "default": _colors.blue700,
      text: _colors.blue800,
      hover: _colors.blue500,
      light: _colors.blue200,
      background: _colors.blue10,
      disabled: _colors.gray400
    },
    secondary: {
      "default": _colors.darkBlue900,
      light: _colors.darkBlue800,
      background: _colors.darkBlue10
    },
    action: {
      selected: _colors.gray50
    },
    // Deprecated - use primary or secondary
    brand: {
      primary: _colors.blue700,
      primaryHover: _colors.blue10,
      secondary: _colors.darkBlue900
    }
  },
  zIndex: _zIndex["default"],
  transition: _transition["default"],
  typography: _typography["default"],
  boxShadow: _boxShadow["default"],
  breakpoints: _breakpoints["default"],
  spacer: (0, _spacing["default"])(5)
};
exports["default"] = _default;
