// @guestyci/foundation/theme/ThemeProvider.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireWildcard(require("react"));

var _reactJss = require("react-jss");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _ThemeContext = _interopRequireDefault(require("./ThemeContext"));

var _useTheme = _interopRequireDefault(require("../useTheme/useTheme"));

var _defaultTheme = _interopRequireDefault(require("./defaultTheme"));

var mergeThemes = function mergeThemes() {
  var theme = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  var outerTheme = arguments.length > 1 ? arguments[1] : undefined;

  if (typeof theme === 'function') {
    var mergedTheme = theme(outerTheme);
    return mergedTheme;
  }

  return (0, _extends2["default"])({}, outerTheme, {}, theme);
};

var setupJssGlobals = function setupJssGlobals(theme) {
  var _theme$typography, _theme$typography2, _theme$palette, _theme$palette$text;

  var sheetsRegistry = new _reactJss.SheetsRegistry();

  var globalStyleSheet = _reactJss.jss.createStyleSheet({
    '@global': {
      html: {
        fontSize: theme === null || theme === void 0 ? void 0 : (_theme$typography = theme.typography) === null || _theme$typography === void 0 ? void 0 : _theme$typography.htmlFontSize
      },
      body: (0, _extends2["default"])({}, theme === null || theme === void 0 ? void 0 : (_theme$typography2 = theme.typography) === null || _theme$typography2 === void 0 ? void 0 : _theme$typography2.running, {
        color: theme === null || theme === void 0 ? void 0 : (_theme$palette = theme.palette) === null || _theme$palette === void 0 ? void 0 : (_theme$palette$text = _theme$palette.text) === null || _theme$palette$text === void 0 ? void 0 : _theme$palette$text["default"]
      })
    }
  }).attach();

  sheetsRegistry.add(globalStyleSheet);
  return sheetsRegistry;
};

var ThemeProvider = function ThemeProvider(_ref) {
  var _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? _defaultTheme["default"] : _ref$theme,
      children = _ref.children;
  var outerTheme = (0, _useTheme["default"])();
  var customTheme = (0, _react.useMemo)(function () {
    return mergeThemes(theme, outerTheme);
  }, [theme, outerTheme]);
  (0, _react.useEffect)(function () {
    setupJssGlobals(customTheme);
  }, [customTheme]);
  return /*#__PURE__*/_react["default"].createElement(_ThemeContext["default"].Provider, {
    value: customTheme,
    "data-qa": "theme-provider"
  }, children);
};

ThemeProvider.propTypes = {
  /** Children to render under the theme */
  children: _propTypes["default"].node.isRequired,

  /** New theme to extend the outer theme */
  theme: _propTypes["default"].oneOfType([_propTypes["default"].object, _propTypes["default"].func])
};
var _default = ThemeProvider;
exports["default"] = _default;
