// @guestyci/foundation/legacy/Section/Section.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _reactJss = require("react-jss");

var _Col = _interopRequireDefault(require("../Col/Col"));

var _Row = _interopRequireDefault(require("../Row/Row"));

var useStyles = (0, _reactJss.createUseStyles)({
  children: {
    justifyContent: 'inherit',
    alignItems: 'inherit',
    width: '100%'
  }
});

var Section = _react["default"].forwardRef(function Section(_ref, ref) {
  var className = _ref.className,
      style = _ref.style,
      children = _ref.children,
      col = _ref.col,
      width = _ref.width,
      dense = _ref.dense,
      innerClassName = _ref.innerClassName,
      spacing = _ref.spacing,
      flex = _ref.flex,
      justify = _ref.justify,
      align = _ref.align,
      bgColor = _ref.bgColor;
  var classes = useStyles();
  return /*#__PURE__*/_react["default"].createElement("div", {
    ref: ref,
    className: (0, _classnames["default"])("p-".concat(spacing), width ? "width-".concat(width) : 'w-fill', {
      'd-flex': flex
    }, (0, _defineProperty2["default"])({}, "bg-".concat(bgColor), bgColor), (0, _defineProperty2["default"])({}, "justify-content-".concat(justify), justify), (0, _defineProperty2["default"])({}, "align-items-".concat(align), align), className),
    style: style
  }, col ? /*#__PURE__*/_react["default"].createElement(_Col["default"], {
    className: (0, _classnames["default"])(classes.children, innerClassName),
    dense: dense
  }, children) : /*#__PURE__*/_react["default"].createElement(_Row["default"], {
    className: (0, _classnames["default"])(classes.children, innerClassName),
    dense: dense
  }, children));
});

Section.defaultProps = {
  bgColor: null,
  className: '',
  style: {},
  col: false,
  children: null,
  width: null,
  spacing: 6,
  dense: false,
  innerClassName: '',
  flex: false
};
Section.propTypes = {
  /** Background color for section */
  bgColor: _propTypes["default"].string,

  /** Additional class for top level */
  className: _propTypes["default"].string,

  /** Indicator whether or not to render section as column */
  col: _propTypes["default"].bool,

  /** Additional styling for root level */
  style: _propTypes["default"].shape(),

  /** Width to specify to section - otherwise full width */
  width: _propTypes["default"].number,

  /** Child components inside a section */
  children: _propTypes["default"].oneOfType([_propTypes["default"].node, _propTypes["default"].arrayOf(_propTypes["default"].node)]),

  /** Indicator to set the children as dense (no inherited margins) */
  dense: _propTypes["default"].bool,

  /** Additional class name for Col/Row child components */
  innerClassName: _propTypes["default"].string,

  /** Boolean to set section a flex */
  flex: _propTypes["default"].bool,

  /** Set the section padding spacing 0-10 calculated in leaps of 5px */
  spacing: _propTypes["default"].number,

  /** Set flex alignment  */
  align: _propTypes["default"].string,

  /** Set flex justify content */
  justify: _propTypes["default"].string
}; // Hack for withInfo plugin

Section.displayName = 'Section';
var _default = Section;
exports["default"] = _default;
