// @guestyci/foundation/legacy/ProgressMap/ProgressMap.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _StepDivider = _interopRequireDefault(require("../Steps/StepDivider"));

var _ProgressMapStep = _interopRequireDefault(require("./ProgressMapStep"));

function _createSuper(Derived) { return function () { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

var ProgressMap = /*#__PURE__*/function (_Component) {
  (0, _inherits2["default"])(ProgressMap, _Component);

  var _super = _createSuper(ProgressMap);

  function ProgressMap() {
    (0, _classCallCheck2["default"])(this, ProgressMap);
    return _super.apply(this, arguments);
  }

  (0, _createClass2["default"])(ProgressMap, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          descriptions = _this$props.descriptions,
          className = _this$props.className,
          style = _this$props.style,
          color = _this$props.color,
          width = _this$props.width;
      return /*#__PURE__*/_react["default"].createElement("div", {
        className: (0, _classnames["default"])('flex-center', className),
        style: (0, _extends2["default"])({}, style),
        "data-qa": "progress-map"
      }, descriptions.map(function (item, index, array) {
        if (array.length - 1 !== index) {
          return [/*#__PURE__*/_react["default"].createElement(_ProgressMapStep["default"], {
            color: color,
            item: index + 1,
            key: "s-".concat(index + 1)
          }), /*#__PURE__*/_react["default"].createElement(_StepDivider["default"], {
            color: color,
            text: item,
            key: "s-".concat(index + 1),
            width: width
          })];
        }

        return /*#__PURE__*/_react["default"].createElement(_ProgressMapStep["default"], {
          color: color,
          item: index + 1,
          key: "s-".concat(index + 1),
          "data-qa": "progress-map"
        });
      }));
    }
  }]);
  return ProgressMap;
}(_react.Component);

ProgressMap.defaultProps = {
  color: 'blue',
  width: 150
};
ProgressMap.propTypes = {
  /** Array of descriptions which will define steps count also */
  descriptions: _propTypes["default"].arrayOf(_propTypes["default"].string, _propTypes["default"].node).isRequired,

  /** color for step */
  color: _propTypes["default"].oneOf(['blue', 'gray']),

  /** distance between steps */
  width: _propTypes["default"].number
};
var _default = ProgressMap;
exports["default"] = _default;
