// @guestyci/foundation/legacy/ProgressMap/ProgressMapStep.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.ProgressMapStepComponent = void 0;

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactJss = _interopRequireDefault(require("react-jss"));

var styles = {
  progressMapStep: {
    minWidth: '64px',
    minHeight: '64px',
    borderRadius: '50%'
  }
};

var ProgressMapStep = function ProgressMapStep(_ref) {
  var index = _ref.index,
      item = _ref.item,
      color = _ref.color,
      classes = _ref.classes;
  return /*#__PURE__*/_react["default"].createElement("div", {
    key: index,
    className: (0, _classnames["default"])('flex-center', 'font-size-2xl', 'b-2', 'progress-map-step', classes.progressMapStep, "text-".concat(color)),
    "data-qa": "progress-map-step"
  }, item);
};

ProgressMapStep.defaultProps = {
  color: 'blue'
};
ProgressMapStep.propTypes = {
  /** number for the range of steps */
  item: _propTypes["default"].number.isRequired,

  /** color for step */
  color: _propTypes["default"].oneOf(['blue', 'gray'])
};
var ProgressMapStepComponent = ProgressMapStep;
exports.ProgressMapStepComponent = ProgressMapStepComponent;

var _default = (0, _reactJss["default"])(styles)(ProgressMapStep);

exports["default"] = _default;
