// @guestyci/foundation/AddButton/AddButton.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _createStyles = _interopRequireDefault(require("../createStyles"));

var _Button = _interopRequireDefault(require("../Button"));

var _Row = _interopRequireDefault(require("../Layout/Row"));

var _Icon = _interopRequireDefault(require("../Icon"));

var _TextField = _interopRequireDefault(require("../TextField"));

var IcoPlusCircle = function IcoPlusCircle(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("g", {
    fill: "#206CFF",
    fillRule: "evenodd"
  }, /*#__PURE__*/_react["default"].createElement("ellipse", {
    cx: "6.491",
    cy: "6.539",
    opacity: ".114",
    rx: "6.491",
    ry: "6.539"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    d: "M6.978 7.036l2.33.005c.275 0 .496-.22.495-.495a.497.497 0 0 0-.497-.497L6.98 6.043l.004-2.329a.492.492 0 0 0-.495-.495.497.497 0 0 0-.497.497l-.006 2.325-2.333-.004a.494.494 0 0 0-.495.495c0 .274.222.496.497.497l2.329.005-.003 2.336a.492.492 0 0 0 .495.495.5.5 0 0 0 .498-.498l.002-2.33z"
  })));
};

IcoPlusCircle.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  width: "13",
  height: "14",
  viewBox: "0 0 13 14"
};
var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    root: {
      color: theme.palette.text.link,
      padding: theme.spacer(1),
      transition: theme.transition.create('color'),
      '&:hover': {
        color: theme.palette.primary.hover
      }
    }
  };
});

var AddButton = _react["default"].forwardRef(function AddButton(_ref, ref) {
  var onClick = _ref.onClick,
      text = _ref.text,
      className = _ref.className,
      jss = _ref.jss;
  var classes = useStyles({
    jss: jss
  });
  return /*#__PURE__*/_react["default"].createElement(_Button["default"], {
    ref: ref,
    className: (0, _classnames["default"])(classes.root, className),
    onClick: onClick
  }, /*#__PURE__*/_react["default"].createElement(_Row["default"], {
    spacing: 1,
    align: "center"
  }, /*#__PURE__*/_react["default"].createElement(_Icon["default"], {
    svg: IcoPlusCircle,
    width: 14,
    height: 14
  }), /*#__PURE__*/_react["default"].createElement(_TextField["default"], null, text)));
});

AddButton.displayName = 'AddButton';
AddButton.propTypes = {
  onClick: _propTypes["default"].func,
  text: _propTypes["default"].string,
  className: _propTypes["default"].string,
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func])
};
var _default = AddButton;
exports["default"] = _default;
