// @guestyci/foundation/esm/DaysPicker/DaysPicker.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React, { useMemo } from 'react';
import noop from 'lodash/noop';
import cn from 'classnames';
import PropTypes from 'prop-types';
import Row from "../Layout/Row";
import createStyles from "../createStyles";
import Button from "../Button";
import { white } from "../theme/colors";
import useTheme from "../useTheme";
import composeEvent from "../utils/composeEvent";
export var useStyles = createStyles(function (theme) {
  return {
    root: {
      border: "1px solid ".concat(theme.palette.border),
      borderRadius: 2,
      width: 'fit-content'
    },
    button: {
      height: 46,
      width: 46,
      borderRadius: 0,
      background: white,
      color: theme.palette.text.secondary,
      borderRight: "1px solid ".concat(theme.palette.border),
      '&:last-child': {
        borderRight: "0px solid ".concat(theme.palette.border)
      }
    },
    inactive: {
      '&:not([disabled]):hover': {
        backgroundColor: theme.palette.primary.background,
        color: theme.palette.primary["default"]
      }
    },
    active: {
      backgroundColor: theme.palette.primary["default"],
      color: white
    }
  };
}, {
  name: 'DaysPicker'
});
var days = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'];
var DaysPicker = React.forwardRef(function DaysPicker(_ref, ref) {
  var _ref$disabled = _ref.disabled,
      disabled = _ref$disabled === void 0 ? false : _ref$disabled,
      _ref$name = _ref.name,
      name = _ref$name === void 0 ? '' : _ref$name,
      _ref$value = _ref.value,
      value = _ref$value === void 0 ? [] : _ref$value,
      _ref$onChange = _ref.onChange,
      onChange = _ref$onChange === void 0 ? noop : _ref$onChange,
      _ref$className = _ref.className,
      className = _ref$className === void 0 ? '' : _ref$className,
      _ref$style = _ref.style,
      style = _ref$style === void 0 ? {} : _ref$style,
      _ref$jss = _ref.jss,
      jss = _ref$jss === void 0 ? {} : _ref$jss,
      props = _objectWithoutProperties(_ref, ["disabled", "name", "value", "onChange", "className", "style", "jss"]);

  var classes = useStyles({
    jss: jss
  });
  var theme = useTheme();
  var valueSet = useMemo(function () {
    return new Set(value);
  }, [value]);

  var handleDayClick = function handleDayClick(e) {
    var clickedValue = e.currentTarget.value;
    var set = new Set(valueSet);

    if (set.has(clickedValue)) {
      set["delete"](clickedValue);
    } else {
      set.add(clickedValue);
    }

    var newEvent = composeEvent(e, {
      name: name,
      value: Array.from(set)
    });
    onChange(newEvent);
  };

  return /*#__PURE__*/React.createElement(Row, _extends({
    ref: ref,
    style: style,
    wrap: false,
    className: cn(classes.root, className)
  }, props), days.map(function (day) {
    var _cn;

    return /*#__PURE__*/React.createElement(Button, {
      disabled: disabled,
      key: day,
      onClick: handleDayClick,
      value: day,
      rippleColor: theme.palette.primary["default"],
      className: cn(classes.button, (_cn = {}, _defineProperty(_cn, classes.active, valueSet.has(day)), _defineProperty(_cn, classes.inactive, !valueSet.has(day)), _cn))
    }, day);
  }));
});
DaysPicker.propTypes = {
  /** Array of selected values to pass */
  value: PropTypes.arrayOf(PropTypes.string),

  /** Callback for on item change */
  onChange: PropTypes.func,

  /** Addition class on root */
  className: PropTypes.string,

  /** Addition style on root */
  style: PropTypes.shape(),

  /** jss object override */
  jss: PropTypes.oneOfType([PropTypes.shape(), PropTypes.func]),

  /** component name */
  name: PropTypes.string,

  /** Indicator for disabled */
  disabled: PropTypes.bool
};
DaysPicker.displayName = 'DaysPicker';
export default DaysPicker;
