// @guestyci/foundation/es/Backdrop/Backdrop.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _objectWithoutPropertiesLoose from "@babel/runtime/helpers/esm/objectWithoutPropertiesLoose";
import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import noop from 'lodash/noop';
import createStyles from "../createStyles";
import { fade } from "../theme/utils";
import { blueGray700 } from "../theme/colors";
const useStyles = createStyles({
  root: {
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    zIndex: -1,
    position: 'fixed',
    touchAction: 'none',
    backgroundColor: fade(blueGray700, 0.85)
  }
}, {
  name: 'Backdrop'
});

const Backdrop = (_ref) => {
  let {
    className = '',
    jss = {},
    onClick = noop
  } = _ref,
      props = _objectWithoutPropertiesLoose(_ref, ["className", "jss", "onClick"]);

  const classes = useStyles({
    jss
  });
  return /*#__PURE__*/React.createElement("div", _extends({
    role: "presentation",
    className: cn(classes.root, className),
    onClick: onClick
  }, props, {
    "data-qa": "backdrop"
  }));
};

Backdrop.propTypes = {
  className: PropTypes.string,
  jss: PropTypes.oneOfType([PropTypes.shape(), PropTypes.func]),
  onClick: PropTypes.func
};
export default Backdrop;
