// @guestyci/foundation/es/Person/ReturningGuestBadge.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _objectWithoutPropertiesLoose from "@babel/runtime/helpers/esm/objectWithoutPropertiesLoose";
import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

var BtnStarOn = function BtnStarOn(props) {
  return /*#__PURE__*/React.createElement("svg", props, /*#__PURE__*/React.createElement("path", {
    d: "M7.5 0l2.319 4.608L15 5.347l-3.75 3.588.887 5.065-4.638-2.393L2.864 14l.886-5.065L0 5.347l5.183-.74z"
  }));
};

BtnStarOn.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 15 14"
};
import createStyles from "../createStyles";
const useStyles = createStyles(theme => ({
  root: {
    height: 20,
    width: 20,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    fill: 'white',
    backgroundColor: theme.palette.secondary.default,
    borderRadius: '50%'
  }
}), {
  name: 'ReturningGuestBadge'
});

const ReturningGuestBadge = (_ref) => {
  let {
    className = '',
    style = {},
    jss = {}
  } = _ref,
      props = _objectWithoutPropertiesLoose(_ref, ["className", "style", "jss"]);

  const classes = useStyles({
    jss
  });
  return /*#__PURE__*/React.createElement("div", _extends({}, props, {
    className: cn(classes.root, className),
    style: style,
    "data-qa": "returning-guest-badge"
  }), /*#__PURE__*/React.createElement(BtnStarOn, {
    height: 13,
    width: 13
  }));
};

ReturningGuestBadge.propTypes = {
  /** Additional class for badge */
  className: PropTypes.string,

  /** Additional style for badge */
  style: PropTypes.shape(),

  /** jss override object to customize the jss classes */
  jss: PropTypes.oneOfType([PropTypes.shape(), PropTypes.func])
};
export default ReturningGuestBadge;
