// @guestyci/foundation/es/legacy/LabelDropdown/LabelDropdownInputField.js
import React from 'react';
import PropTypes from 'prop-types';
import AccordionChevron from "../Accordion/AccordionChevron";
import Row from "../Row/Row";

const LabelDropdownInputField = ({
  label,
  isOpen,
  placeholder,
  onClick
}) => /*#__PURE__*/React.createElement("div", {
  role: "presentation",
  className: "py-2 clickable",
  onClick: onClick,
  "data-qa": "label-dropdown-input-field"
}, /*#__PURE__*/React.createElement(Row, {
  spacing: "sm",
  align: "center"
}, label ? /*#__PURE__*/React.createElement("div", {
  className: "text-blue overflow-ellipsis label-selected-item"
}, label) : /*#__PURE__*/React.createElement("div", {
  className: "text-gray-dark overflow-ellipsis"
}, placeholder), /*#__PURE__*/React.createElement(AccordionChevron, {
  isOpen: isOpen
})));

LabelDropdownInputField.defaultProps = {
  label: null,
  placeholder: 'Select'
};
LabelDropdownInputField.propTypes = {
  label: PropTypes.node,
  placeholder: PropTypes.string,
  isOpen: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired
};
export default LabelDropdownInputField;
