// @guestyci/foundation/DurationPicker/DurationPickerInput.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.useStyles = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _extends3 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _createStyles = _interopRequireDefault(require("../createStyles"));

var _Row = _interopRequireDefault(require("../Layout/Row"));

var _Button = _interopRequireDefault(require("../Button"));

var _TextField = _interopRequireDefault(require("../TextField"));

var _PickerClearButton = _interopRequireDefault(require("../basePicker/PickerClearButton"));

var _PickerInputButton = _interopRequireDefault(require("../basePicker/PickerInputButton"));

var _Input = _interopRequireDefault(require("../Input"));

var _PickerResetWrapper = _interopRequireDefault(require("../basePicker/PickerResetWrapper"));

var _styles = require("../TimePicker/styles");

var BtnIcoDuration = function BtnIcoDuration(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("title", null, "Combined Shape"), /*#__PURE__*/_react["default"].createElement("path", {
    d: "M13.625 0a.5.5 0 0 1 .09.992l-.09.008H12.55v3.37c1.89.378 3.597 1.26 4.97 2.5l1.169-1.162-.166-.166a.5.5 0 0 1 .635-.767l.07.058 1.05 1.041a.5.5 0 0 1-.636.768l-.07-.058-.173-.172-1.169 1.162a10.314 10.314 0 0 1 2.716 6.707l.004.281c0 5.743-4.691 10.396-10.475 10.396C4.69 24.958 0 20.306 0 14.563 0 9.524 3.609 5.325 8.4 4.37V1H7.325a.5.5 0 0 1-.09-.992L7.325 0h6.3zm-3.15 5.167C5.24 5.167 1 9.374 1 14.563c0 5.188 4.24 9.395 9.475 9.395 5.233 0 9.475-4.207 9.475-9.395a9.328 9.328 0 0 0-2.777-6.645 9.484 9.484 0 0 0-6.698-2.751zM6.033 9.462l.07.058 4.724 4.688a.5.5 0 0 1-.635.767l-.07-.058-4.724-4.687a.5.5 0 0 1 .635-.768zM11.55 1H9.4v3.22c.261-.026.525-.043.791-.05l.284-.003c.363 0 .722.018 1.076.054L11.55 1z",
    fillRule: "evenodd"
  }));
};

BtnIcoDuration.defaultProps = {
  width: "21",
  height: "25",
  viewBox: "0 0 21 25",
  xmlns: "http://www.w3.org/2000/svg"
};
var useStyles = (0, _createStyles["default"])(function (theme) {
  return (0, _extends3["default"])({}, (0, _styles.timePickerInputStyles)(theme));
});
exports.useStyles = useStyles;

var deserializeTimeToObject = function deserializeTimeToObject(value) {
  if (!value) {
    return {
      hours: '',
      minutes: ''
    };
  }

  var hours = Math.floor(value / 60);
  var minutes = value % 60;

  if (hours < 10) {
    hours = "0".concat(Number(hours));
  }

  if (minutes < 10) {
    minutes = "0".concat(Number(minutes));
  }

  return {
    hours: !hours && minutes ? '00' : hours,
    minutes: !minutes && hours ? '00' : minutes
  };
};

var durationToValue = function durationToValue(_ref) {
  var hours = _ref.hours,
      minutes = _ref.minutes;

  if (!hours && !minutes) {
    return '';
  }

  var numDuration = Number(hours) * 60 + Number(minutes);
  return Number.isNaN(numDuration) ? '' : numDuration;
};

var jumpToNextInput = function jumpToNextInput(ref) {
  ref.current.select();
};

var DurationPickerInput = _react["default"].forwardRef(function TimePickerInput(_ref2, ref) {
  var _cn;

  var value = _ref2.value,
      onClick = _ref2.onClick,
      onChange = _ref2.onChange,
      open = _ref2.open,
      _ref2$name = _ref2.name,
      name = _ref2$name === void 0 ? '' : _ref2$name,
      className = _ref2.className,
      style = _ref2.style,
      disabled = _ref2.disabled,
      onClear = _ref2.onClear,
      noClear = _ref2.noClear,
      defaultValue = _ref2.defaultValue,
      resetable = _ref2.resetable,
      jss = _ref2.jss,
      error = _ref2.error,
      props = (0, _objectWithoutProperties2["default"])(_ref2, ["value", "onClick", "onChange", "open", "name", "className", "style", "disabled", "onClear", "noClear", "defaultValue", "resetable", "jss", "error"]);
  var classes = useStyles({
    jss: jss
  });
  var hourRef = (0, _react.useRef)();
  var minuteRef = (0, _react.useRef)();

  var _useState = (0, _react.useState)({}),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      duration = _useState2[0],
      setDuration = _useState2[1];

  (0, _react.useEffect)(function () {
    if (open && (minuteRef === null || minuteRef === void 0 ? void 0 : minuteRef.current) !== document.activeElement) {
      hourRef.current.focus();
    }
  }, [open]);
  (0, _react.useEffect)(function () {
    var serializedDuration = durationToValue(duration);

    var _deserializeTimeToObj = deserializeTimeToObject(value),
        nextHours = _deserializeTimeToObj.hours,
        nextMinutes = _deserializeTimeToObj.minutes;

    if (Number(value) !== serializedDuration) {
      setDuration({
        hours: nextHours,
        minutes: nextMinutes
      });
    }
  }, [value]);

  var handleInputFocus = function handleInputFocus(e) {
    e.target.select();
  };

  var dispatchEvent = function dispatchEvent(e, _ref3) {
    var hours = _ref3.hours,
        minutes = _ref3.minutes;
    e.stopPropagation();
    var val = durationToValue({
      hours: hours,
      minutes: minutes
    });
    onChange(e, val);
  };

  var handleInputBlur = function handleInputBlur(inputName) {
    return function (e) {
      var inputValue = e.target.value;
      var paddedValue = inputValue.length === 1 ? "0".concat(inputValue) : inputValue;
      var newDuration = (0, _extends3["default"])({}, duration, (0, _defineProperty2["default"])({}, inputName, paddedValue));

      if (inputValue && paddedValue !== inputValue) {
        setDuration(newDuration);
      }
    };
  };

  var handleReset = function handleReset(e) {
    e.stopPropagation();
    onChange(e, defaultValue);
  };

  var handleHoursChange = function handleHoursChange(e) {
    var inputValue = e.currentTarget.value;
    var minutes = duration.minutes;

    if (inputValue < 0) {
      return;
    }

    var newDuration = {
      hours: inputValue,
      minutes: minutes || '00'
    };
    setDuration(newDuration);

    if (inputValue.length === 2) {
      jumpToNextInput(minuteRef);
    }

    dispatchEvent(e, newDuration);
  };

  var handleMinutesChange = function handleMinutesChange(e) {
    var inputValue = e.target.value;
    var hours = duration.hours;

    if (inputValue < 0 || inputValue >= 60) {
      return;
    }

    var newDuration = {
      hours: hours || '00',
      minutes: inputValue
    };
    setDuration(newDuration);
    dispatchEvent(e, newDuration);
  };

  var hours = duration.hours,
      minutes = duration.minutes;
  return /*#__PURE__*/_react["default"].createElement(_Button["default"], (0, _extends3["default"])({}, props, {
    enableRipple: false,
    ref: ref,
    value: value,
    name: name,
    className: (0, _classnames["default"])(classes.root, (_cn = {}, (0, _defineProperty2["default"])(_cn, classes.disabled, disabled), (0, _defineProperty2["default"])(_cn, classes.focus, open), (0, _defineProperty2["default"])(_cn, classes.error, error), _cn), className),
    onClick: onClick,
    disabled: disabled,
    style: style
  }), /*#__PURE__*/_react["default"].createElement(_PickerResetWrapper["default"], {
    align: "center",
    justify: "between",
    resetable: resetable,
    onReset: handleReset
  }, /*#__PURE__*/_react["default"].createElement(_PickerInputButton["default"], {
    svg: BtnIcoDuration,
    disabled: disabled,
    active: open
  }), /*#__PURE__*/_react["default"].createElement(_Row["default"], {
    className: classes.row,
    span: "auto",
    align: "center"
  }, /*#__PURE__*/_react["default"].createElement(_Input["default"], {
    ref: hourRef,
    disabled: disabled,
    maxLength: 2,
    onFocus: handleInputFocus,
    onChange: handleHoursChange,
    onBlur: handleInputBlur('hours'),
    name: "hours",
    value: hours,
    className: classes.input,
    placeholder: "HH"
  }), /*#__PURE__*/_react["default"].createElement(_TextField["default"], {
    className: classes.separator
  }, ":"), /*#__PURE__*/_react["default"].createElement(_Input["default"], {
    maxLength: 2,
    disabled: disabled,
    ref: minuteRef,
    onFocus: handleInputFocus,
    onChange: handleMinutesChange,
    onBlur: handleInputBlur('minutes'),
    name: "minutes",
    value: minutes,
    className: classes.input,
    placeholder: "MM"
  })), !noClear && !disabled && /*#__PURE__*/_react["default"].createElement(_PickerClearButton["default"], {
    show: !!value,
    onClick: onClear
  })));
});

DurationPickerInput.propTypes = {
  /** Selected value to pass to the input */
  value: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].array, _propTypes["default"].string, _propTypes["default"].number]),
  onClick: _propTypes["default"].func,
  onClear: _propTypes["default"].func,

  /** Prop to remove clear button if exists */
  noClear: _propTypes["default"].bool,

  /** Disabled boolean indicator */
  disabled: _propTypes["default"].bool,

  /** Additional className */
  className: _propTypes["default"].string,

  /** Additional style */
  style: _propTypes["default"].shape(),

  /** form level name */
  name: _propTypes["default"].string,

  /** Indicator whether or not the dropdown is open */
  open: _propTypes["default"].bool,

  /** jss override  */
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func])
};
DurationPickerInput.displayName = 'DurationPickerInput';
var _default = DurationPickerInput;
exports["default"] = _default;
