// @guestyci/foundation/DurationPicker/DurationPicker.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.useStyles = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _noop = _interopRequireDefault(require("lodash/noop"));

var _Menu = require("../Menu");

var _TextField = _interopRequireDefault(require("../TextField"));

var _MenuItem = _interopRequireDefault(require("../Menu/MenuItem"));

var _createStyles = _interopRequireDefault(require("../createStyles"));

var _DurationPickerInput = _interopRequireDefault(require("./DurationPickerInput"));

var _styles = require("../TimePicker/styles");

var useStyles = (0, _createStyles["default"])(function (theme) {
  return (0, _extends2["default"])({}, (0, _styles.timePickerStyles)(theme));
}, {
  name: 'DurationPicker'
});
exports.useStyles = useStyles;

var generateOptions = function generateOptions(duration, interval) {
  var itemCount = duration / interval;

  var padTime = function padTime(t) {
    return t < 10 ? "0".concat(t) : t;
  };

  var minutes = interval;
  var items = (0, _toConsumableArray2["default"])(Array(Math.round(itemCount)));
  return items.map(function () {
    var result = "".concat(padTime(Math.floor(minutes / 60)), " : ").concat(padTime(minutes % 60));
    var value = minutes;
    minutes += interval;
    return {
      label: result,
      value: value
    };
  });
};

var DurationPicker = _react["default"].forwardRef(function DurationPicker(_ref, ref) {
  var value = _ref.value,
      _ref$onChange = _ref.onChange,
      onChange = _ref$onChange === void 0 ? _noop["default"] : _ref$onChange,
      _ref$className = _ref.className,
      className = _ref$className === void 0 ? '' : _ref$className,
      _ref$style = _ref.style,
      style = _ref$style === void 0 ? {} : _ref$style,
      _ref$name = _ref.name,
      name = _ref$name === void 0 ? '' : _ref$name,
      defaultValue = _ref.defaultValue,
      _ref$resetable = _ref.resetable,
      resetable = _ref$resetable === void 0 ? false : _ref$resetable,
      _ref$disabled = _ref.disabled,
      disabled = _ref$disabled === void 0 ? false : _ref$disabled,
      _ref$jss = _ref.jss,
      jss = _ref$jss === void 0 ? {} : _ref$jss,
      _ref$interval = _ref.interval,
      interval = _ref$interval === void 0 ? 30 : _ref$interval,
      _ref$error = _ref.error,
      error = _ref$error === void 0 ? false : _ref$error,
      _ref$duration = _ref.duration,
      duration = _ref$duration === void 0 ? 24 * 60 : _ref$duration,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["value", "onChange", "className", "style", "name", "defaultValue", "resetable", "disabled", "jss", "interval", "error", "duration"]);
  var classes = useStyles({
    jss: jss
  });
  var anchorRef = (0, _react.useRef)();

  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      open = _useState2[0],
      setOpen = _useState2[1];

  var handleInputClick = function handleInputClick() {
    setOpen(!open);
  };

  var handleDurationPickerClose = function handleDurationPickerClose() {
    setOpen(false);
  };

  var dispatchEvent = function dispatchEvent(e, selectedValue) {
    // work around to solve - see https://github.com/guestyorg/foundation-ui/issues/284
    Object.assign(e.target, {
      name: name,
      value: selectedValue
    });
    onChange(e, selectedValue);
  };

  var handleInputChange = function handleInputChange(e, inputValue) {
    e.stopPropagation();
    dispatchEvent(e, inputValue);
    handleDurationPickerClose();
  };

  var handleItemClick = function handleItemClick(e, selectedValue) {
    dispatchEvent(e, selectedValue);
    handleDurationPickerClose();
  };

  var handleClearClick = function handleClearClick(e) {
    dispatchEvent(e);
  };

  var options = (0, _react.useMemo)(function () {
    return generateOptions(duration, interval);
  }, [duration, interval]);
  return /*#__PURE__*/_react["default"].createElement("div", (0, _extends2["default"])({
    ref: ref,
    className: (0, _classnames["default"])(classes.root, className),
    style: style
  }, props), /*#__PURE__*/_react["default"].createElement(_DurationPickerInput["default"], {
    ref: anchorRef,
    onClick: handleInputClick,
    onChange: handleInputChange,
    onClear: handleClearClick,
    open: open,
    disabled: disabled,
    defaultValue: defaultValue,
    resetable: resetable,
    value: value,
    error: error
  }), /*#__PURE__*/_react["default"].createElement(_Menu.Menu, {
    anchorDirection: "start",
    open: open,
    className: classes.menu,
    onClose: handleDurationPickerClose,
    variant: "menu",
    fitAnchor: true,
    anchor: anchorRef
  }, options.map(function (option) {
    return /*#__PURE__*/_react["default"].createElement(_MenuItem["default"], {
      key: option.value,
      value: option.value,
      onClick: handleItemClick
    }, /*#__PURE__*/_react["default"].createElement(_TextField["default"], null, option.label));
  })));
});

DurationPicker.propTypes = {
  /** Callback function to be called on value changed - arguments - (event) */
  onChange: _propTypes["default"].func.isRequired,

  /** Additional class for duration picker root */
  className: _propTypes["default"].string,

  /** Additional style for duration picker root */
  style: _propTypes["default"].shape(),

  /** form element name */
  name: _propTypes["default"].string,

  /** Controlled value to set to the duration picker */
  value: _propTypes["default"].oneOfType([_propTypes["default"].number, _propTypes["default"].string]),

  /** Indicator for disabled status */
  disabled: _propTypes["default"].bool,

  /** The interval leaps for the picker items in minutes */
  interval: _propTypes["default"].number,

  /** JSS overrides for DurationPicker */
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func]),

  /** default value to reset to */
  defaultValue: _propTypes["default"].string,

  /** indicator whether or not to allow reset */
  resetable: _propTypes["default"].bool,

  /** The max duration value to build dropdown values from  in minutes */
  duration: _propTypes["default"].number
};
DurationPicker.displayName = 'DurationPicker';
var _default = DurationPicker;
exports["default"] = _default;
