// @guestyci/foundation/legacy/BedPicker/BedPicker.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _TilesPicker = _interopRequireDefault(require("../tiles/TilesPicker"));

var optionsMap = {
  single: {
    icon: 'IcoBedSingle',
    width: 27,
    height: 29,
    text: 'Single'
  },
  "double": {
    icon: 'IcoBedDouble',
    width: 40,
    height: 29,
    text: 'Double'
  },
  queen: {
    icon: 'IcoBedDouble',
    width: 40,
    height: 29,
    text: 'Queen'
  },
  king: {
    icon: 'IcoBedDouble',
    width: 40,
    height: 29,
    text: 'King'
  },
  bunk: {
    icon: 'IcoBedBunk',
    width: 29,
    height: 34,
    text: 'Bunk'
  },
  sofa: {
    icon: 'IcoBedSofa',
    width: 40,
    height: 24,
    text: 'Sofa'
  },
  mat: {
    icon: 'IcoBedMat',
    width: 40,
    height: 24,
    text: 'Mat'
  }
};
var options = [{
  label: 'Single',
  value: 'single'
}, {
  label: 'Double',
  value: 'double'
}, {
  label: 'Queen',
  value: 'queen'
}, {
  label: 'King',
  value: 'king'
}, {
  label: 'Bunk',
  value: 'bunk'
}, {
  label: 'Sofa',
  value: 'sofa'
}, {
  label: 'Mat',
  value: 'mat'
}];

var BedPicker = function BedPicker(_ref) {
  var maxShownItems = _ref.maxShownItems,
      onChangeBeds = _ref.onChangeBeds;
  return /*#__PURE__*/_react["default"].createElement(_TilesPicker["default"], {
    maxShownItems: maxShownItems,
    optionsMap: optionsMap,
    options: options,
    onChange: onChangeBeds,
    "data-qa": "bed-picker"
  });
};

BedPicker.defaultProps = {
  onChangeBeds: function onChangeBeds() {},
  maxShownItems: 4
};
BedPicker.propTypes = {
  // sets the number of tiles what would be rendered
  maxShownItems: _propTypes["default"].number,
  // callback to be called when a tile is changed
  onChangeBeds: _propTypes["default"].func
};
var _default = BedPicker;
exports["default"] = _default;
