// @guestyci/foundation/legacy/EmptyState/EmptyState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _noop = _interopRequireDefault(require("lodash/noop"));

var _Icon = _interopRequireDefault(require("../Icon"));

var _RaisedButton = _interopRequireDefault(require("../RaisedButton/RaisedButton"));

var EmptyState = function EmptyState(_ref) {
  var className = _ref.className,
      style = _ref.style,
      iconName = _ref.iconName,
      svg = _ref.svg,
      title = _ref.title,
      description = _ref.description,
      iconWidth = _ref.iconWidth,
      iconHeight = _ref.iconHeight,
      onClick = _ref.onClick,
      buttonLabel = _ref.buttonLabel;
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])('flex-center flex-column h-fill max-width-450', className),
    style: style,
    "data-qa": "empty-state"
  }, /*#__PURE__*/_react["default"].createElement(_Icon["default"], {
    className: "mb-5",
    icon: iconName,
    svg: svg,
    height: iconHeight,
    width: iconWidth
  }), /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])('text-muted font-size-xl text-align-center overflow-ellipsis mb-2 w-fill max-width-450')
  }, title), /*#__PURE__*/_react["default"].createElement("div", {
    className: "text-muted font-size-lg text-align-center w-fill"
  }, description), buttonLabel ? /*#__PURE__*/_react["default"].createElement("div", {
    className: "mt-4"
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: "d-flex justify-content-center"
  }, /*#__PURE__*/_react["default"].createElement(_RaisedButton["default"], {
    onClick: onClick
  }, buttonLabel))) : null);
};

EmptyState.defaultProps = {
  iconWidth: 100,
  iconHeight: 100,
  className: '',
  buttonLabel: '',
  style: {},
  onClick: _noop["default"]
};
EmptyState.propTypes = {
  /** Additional class name for person */
  className: _propTypes["default"].string,

  /** Additional style name to root level */
  style: _propTypes["default"].shape(),

  /** Icon - chooses the appropriate icon by its name */
  iconName: _propTypes["default"].string,

  /** The svg icon to render */
  svg: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func]),

  /** Button click callback function */
  onClick: _propTypes["default"].func,

  /** sets the icon's width */
  iconWidth: _propTypes["default"].number,

  /** sets the icon's height */
  iconHeight: _propTypes["default"].number,

  /** Title - sets the text of the title */
  title: _propTypes["default"].string.isRequired,

  /** buttonLabel - sets the button label of the button */
  buttonLabel: _propTypes["default"].string,

  /** Description - sets the element of the description */
  description: _propTypes["default"].oneOfType([_propTypes["default"].node, _propTypes["default"].element]).isRequired
};
var _default = EmptyState;
exports["default"] = _default;
