// @guestyci/foundation/es/legacy/ThreadMessage/ThreadMessageInfo.js
import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import Bullet from "../Bullet/Bullet";
import SvgIcon from "../SvgIcon/SvgIcon";

var IcoWhatsapp = function IcoWhatsapp(props) {
  return /*#__PURE__*/React.createElement("svg", props, /*#__PURE__*/React.createElement("title", null, "whatsapp"), /*#__PURE__*/React.createElement("g", {
    fill: "#AEB6BA",
    fillRule: "nonzero"
  }, /*#__PURE__*/React.createElement("path", {
    d: "M13.394 10.698L12.01 9.642a1.193 1.193 0 0 0-1.562.103l-.54.54a2.796 2.796 0 0 1-1.124-.57l-.047-.04a2.96 2.96 0 0 1-.363-.363l-.038-.046a2.785 2.785 0 0 1-.572-1.121l.54-.54c.42-.42.464-1.085.105-1.558L7.35 4.667a1.243 1.243 0 0 0-1.488-.382l-.157.07a2.598 2.598 0 0 0-1.552 2.318 6.883 6.883 0 0 0 1.616 4.574l.07.081c.268.318.563.612.88.88l.083.07a6.922 6.922 0 0 0 4.44 1.614h.145a2.602 2.602 0 0 0 2.321-1.552l.07-.157a1.239 1.239 0 0 0-.384-1.485zm-.456 1.12v-.001l-.07.157a1.684 1.684 0 0 1-1.502 1.002h-.126a6.004 6.004 0 0 1-3.85-1.399l-.082-.07a6.296 6.296 0 0 1-.77-.768l-.07-.082a5.97 5.97 0 0 1-1.401-3.97 1.68 1.68 0 0 1 1.004-1.5l.157-.07a.329.329 0 0 1 .394.105l1.058 1.38a.272.272 0 0 1-.024.357l-.714.713a.456.456 0 0 0-.128.397 3.7 3.7 0 0 0 .82 1.786l.04.045c.143.171.302.33.474.474l.047.04c.511.428 1.13.71 1.788.818a.458.458 0 0 0 .398-.128l.714-.714a.274.274 0 0 1 .359-.023l1.383 1.059c.12.091.162.253.101.391z"
  }), /*#__PURE__*/React.createElement("path", {
    d: "M15.392 2.672A8.966 8.966 0 0 0 9.017.035C4.447.035.6 3.446.062 7.975a9.04 9.04 0 0 0 1.104 5.49l-.76 2.614a1.25 1.25 0 0 0 1.553 1.55l2.62-.758a9.082 9.082 0 0 0 4.449 1.164c4.98-.007 9.01-4.042 9.004-9.012a8.992 8.992 0 0 0-2.64-6.351zm-.647 12.081a8.022 8.022 0 0 1-4.774 2.312 8.157 8.157 0 0 1-5.098-1.08.459.459 0 0 0-.363-.046l-2.807.814a.338.338 0 0 1-.418-.418l.815-2.802a.456.456 0 0 0-.046-.362A8.118 8.118 0 0 1 .972 8.082 8.095 8.095 0 0 1 9.017.95c4.474 0 8.1 3.62 8.1 8.086a8.078 8.078 0 0 1-2.372 5.718z"
  })));
};

IcoWhatsapp.defaultProps = {
  width: "19",
  height: "19",
  viewBox: "0 0 19 19",
  xmlns: "http://www.w3.org/2000/svg"
};

var IcoAirbnb = function IcoAirbnb(props) {
  return /*#__PURE__*/React.createElement("svg", props, /*#__PURE__*/React.createElement("path", {
    d: "M13.948 13.075a2.403 2.403 0 0 1-1.47 1.891c-.39.161-.81.21-1.23.161a3.204 3.204 0 0 1-1.228-.42c-.581-.324-1.162-.825-1.841-1.569 1.066-1.308 1.712-2.505 1.955-3.57.113-.502.13-.953.081-1.375a2.63 2.63 0 0 0-.436-1.098C9.277 6.367 8.436 5.947 7.5 5.947c-.938 0-1.779.436-2.279 1.148a2.614 2.614 0 0 0-.436 1.098c-.065.422-.049.89.08 1.374.243 1.066.905 2.279 1.955 3.588-.661.743-1.26 1.245-1.841 1.567a3.143 3.143 0 0 1-1.229.42 2.575 2.575 0 0 1-1.229-.161 2.4 2.4 0 0 1-1.47-1.89 2.68 2.68 0 0 1 .147-1.261c.047-.161.128-.323.21-.517.112-.258.242-.533.37-.808l.017-.033A176.46 176.46 0 0 1 5.35 3.216l.049-.096c.13-.243.259-.502.387-.744.13-.259.275-.501.453-.711.34-.388.792-.598 1.294-.598s.952.21 1.292.598c.178.211.322.452.452.711.13.242.26.501.388.744l.049.096a214.387 214.387 0 0 1 3.539 7.274v.015c.13.258.242.55.372.808.08.194.161.356.21.517.129.42.178.824.113 1.245m-6.448-.76c-.872-1.099-1.439-2.133-1.633-3.006-.08-.372-.097-.694-.048-.985.032-.26.13-.485.259-.68.307-.435.823-.71 1.422-.71.598 0 1.131.257 1.421.71.13.195.228.42.26.68.048.29.031.63-.048.985-.195.857-.76 1.89-1.633 3.006m7.288-.84c-.08-.195-.162-.405-.242-.583-.13-.291-.26-.565-.372-.824l-.016-.016a169.892 169.892 0 0 0-3.572-7.305l-.048-.096a31.004 31.004 0 0 1-.388-.76c-.161-.291-.322-.597-.581-.889A2.613 2.613 0 0 0 7.516 0C6.707 0 5.98.356 5.448.97c-.243.29-.42.597-.583.888-.125.256-.255.51-.387.76l-.048.097A189.639 189.639 0 0 0 .858 10.02l-.017.032c-.112.259-.242.534-.371.824-.081.179-.161.371-.242.581-.21.599-.275 1.165-.194 1.746a3.435 3.435 0 0 0 2.101 2.683 3.3 3.3 0 0 0 1.309.259 4.27 4.27 0 0 0 2.036-.583c.662-.371 1.292-.903 2.004-1.68.711.777 1.357 1.309 2.003 1.68a4.27 4.27 0 0 0 2.038.583c.451 0 .904-.08 1.307-.259a3.424 3.424 0 0 0 2.102-2.683c.129-.566.063-1.13-.146-1.728"
  }));
};

IcoAirbnb.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 15 17"
};
import FormattedDateThreadMessageInfo from "./FormattedDateThreadMessageInfo";

const ThreadMessageInfo = ({
  byUs,
  fullName,
  sentAt,
  createdAt,
  moduleType,
  isErrorMessage
}) => /*#__PURE__*/React.createElement("div", {
  className: cn('d-flex align-items-center text-muted', {
    'justify-content-end': byUs,
    'pr-4': byUs && !isErrorMessage,
    'pl-4': !byUs && !isErrorMessage
  }),
  "data-qa": "thread-message-info"
}, fullName && /*#__PURE__*/React.createElement("div", {
  className: cn('text-capitalize font-size-md', {
    'text-white': isErrorMessage
  })
}, fullName), /*#__PURE__*/React.createElement(Bullet, {
  className: "mx-1"
}), /*#__PURE__*/React.createElement("div", {
  className: cn('font-size-md', {
    'text-white': isErrorMessage
  })
}, /*#__PURE__*/React.createElement(FormattedDateThreadMessageInfo, {
  then: sentAt || createdAt
})), /*#__PURE__*/React.createElement(Bullet, {
  className: "mx-1"
}), moduleType === 'airbnb' || moduleType === 'AirbnbChannel' || moduleType === 'whatsapp' ? /*#__PURE__*/React.createElement(SvgIcon, {
  svg: moduleType === 'whatsapp' ? IcoWhatsapp : IcoAirbnb,
  className: `message-info-icon ${isErrorMessage ? 'fill-white' : 'fill-gray-dark'}`,
  width: 12
}) : /*#__PURE__*/React.createElement("div", {
  className: cn('text-capitalize font-size-md', {
    'text-white': isErrorMessage
  })
}, moduleType));

ThreadMessageInfo.defaultProps = {
  fullName: 'N/A',
  date: 'N/A',
  isErrorMessage: false
};
ThreadMessageInfo.propTypes = {
  byUs: PropTypes.bool.isRequired,
  fullName: PropTypes.string,
  date: PropTypes.string,
  moduleType: PropTypes.string.isRequired,
  isErrorMessage: PropTypes.bool
};
export default ThreadMessageInfo;
