// @guestyci/foundation/legacy/NumberPicker/NumberPicker.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.NumberPickerComponent = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _reactJss = _interopRequireDefault(require("react-jss"));

var _isEmpty = _interopRequireDefault(require("lodash/isEmpty"));

var _commonUtility = require("../../utils/commonUtility");

var _Input = _interopRequireDefault(require("../Input/Input"));

var _NumberPickerActionButtons = _interopRequireDefault(require("./NumberPickerActionButtons"));

var _PickerResetWrapper = _interopRequireDefault(require("../basePicker/PickerResetWrapper"));

function _createSuper(Derived) { return function () { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

var styles = {
  numberPickerLabelWrapper: {
    alignItems: 'center',
    '& .numberPickerLabel': {
      border: 0,
      '& input': {
        padding: '10px 5px!important',
        textAlign: 'right'
      }
    }
  },
  resetBtn: {
    borderLeftWidth: '0 !important'
  },
  input: {
    borderRightWidth: '0 !important'
  },
  isReadOnly: {
    '&:disabled': {
      color: 'black'
    }
  },
  inputWrapper: {
    'border-top-left-radius': '2px !important',
    'border-bottom-left-radius': '2px !important',
    'border-right': 0
  }
};

var shouldControlChangeInputValue = function shouldControlChangeInputValue(value, prevValue, stateValue) {
  return (0, _commonUtility.isNumber)(value) && value !== prevValue && value !== stateValue;
};

var NumberPicker = /*#__PURE__*/function (_PureComponent) {
  (0, _inherits2["default"])(NumberPicker, _PureComponent);

  var _super = _createSuper(NumberPicker);

  function NumberPicker(props) {
    var _this;

    (0, _classCallCheck2["default"])(this, NumberPicker);
    _this = _super.call(this, props);

    _this.handleIncrementClick = function () {
      var _this$props = _this.props,
          max = _this$props.max,
          step = _this$props.step;
      var value = _this.state.value;
      var val = +value || 0;
      var newVal = (0, _commonUtility.parseNumber)(val + step);

      if (!(0, _commonUtility.isNumber)(max) || (0, _commonUtility.isNumber)(max) && newVal <= max) {
        _this.setState({
          value: newVal,
          disableMin: false,
          disableMax: (0, _commonUtility.isNumber)(max) && newVal >= max
        }, function () {
          return _this.handleValueChange(newVal);
        });
      }
    };

    _this.calculateValueWithText = function (value) {
      var _this$props2 = _this.props,
          innerSingularText = _this$props2.innerSingularText,
          innerPluralText = _this$props2.innerPluralText,
          suffix = _this$props2.suffix;
      var isReadOnlyLocal = _this.state.isReadOnlyLocal;

      if (!isReadOnlyLocal) {
        return value;
      }

      if (!innerPluralText) {
        return "".concat(value, " ").concat(innerSingularText);
      }

      if (innerSingularText && innerPluralText && !suffix) {
        if (parseInt(value, 10) === 1) {
          return "".concat(value, " ").concat(innerSingularText);
        }

        if (parseInt(value, 10) > 1) {
          return "".concat(value, " ").concat(innerPluralText);
        }

        if (parseInt(value, 10) === 0) {
          return "".concat(value, " ").concat(innerPluralText);
        }
      }

      return value;
    };

    _this.handleDecrementClick = function () {
      var _this$props3 = _this.props,
          min = _this$props3.min,
          step = _this$props3.step;
      var value = _this.state.value;
      var val = +value || 0;
      var newVal = (0, _commonUtility.parseNumber)(val - step);

      if (!(0, _commonUtility.isNumber)(min) || (0, _commonUtility.isNumber)(min) && newVal >= min) {
        _this.setState({
          value: newVal,
          disableMax: false,
          disableMin: (0, _commonUtility.isNumber)(min) && newVal <= min
        }, function () {
          return _this.handleValueChange(newVal);
        });
      }
    };

    _this.handleInputChange = function (value) {
      var _this$props4 = _this.props,
          min = _this$props4.min,
          max = _this$props4.max;

      if ((0, _isEmpty["default"])(value) || _this.isValid(value)) {
        _this.setState({
          value: value,
          disableMax: (0, _commonUtility.isNumber)(max) && value >= max,
          disableMin: (0, _commonUtility.isNumber)(min) && value <= min
        }, function () {
          return _this.handleValueChange(value);
        });
      }
    };

    _this.isValidFloatingPoint = function (value) {
      if (value === Math.floor(value)) {
        return true;
      }

      var floatingPoint = _this.props.floatingPoint;
      var decValue = value.toString().split('.')[1];
      var decPoints = decValue ? decValue.length : 0;
      return floatingPoint >= decPoints;
    };

    _this.reset = function () {
      var _this$props5 = _this.props,
          max = _this$props5.max,
          min = _this$props5.min,
          defaultValue = _this$props5.defaultValue;

      _this.setState({
        value: defaultValue,
        disableMax: (0, _commonUtility.isNumber)(max) && defaultValue >= max,
        disableMin: (0, _commonUtility.isNumber)(min) && defaultValue <= min
      });

      _this.handleValueChange(defaultValue);
    };

    var initialValue = (0, _commonUtility.parseNumber)((0, _commonUtility.isNumber)(props.value) ? props.value : props.defaultValue, '');
    _this.state = {
      value: initialValue,
      isReadOnlyLocal: false,
      disableMax: (0, _commonUtility.isNumber)(initialValue) && (0, _commonUtility.isNumber)(props.max) && initialValue >= props.max,
      disableMin: (0, _commonUtility.isNumber)(initialValue) && (0, _commonUtility.isNumber)(props.min) && initialValue <= props.min
    };
    return _this;
  }

  (0, _createClass2["default"])(NumberPicker, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      var _this$props6 = this.props,
          isReadOnly = _this$props6.isReadOnly,
          asLabel = _this$props6.asLabel;
      this.setState({
        isReadOnlyLocal: isReadOnly || asLabel
      });
    }
  }, {
    key: "componentDidUpdate",
    value: function componentDidUpdate(prevProps) {
      var _this$props7 = this.props,
          defaultValue = _this$props7.defaultValue,
          value = _this$props7.value;
      var stateValue = this.state.value;

      if (defaultValue !== prevProps.defaultValue) {
        this.handleInputChange(defaultValue);
      }

      if (shouldControlChangeInputValue(value, prevProps.value, stateValue)) {
        this.handleInputChange(value);
      }
    }
  }, {
    key: "handleValueChange",
    value: function handleValueChange(value) {
      var _this$props8 = this.props,
          onChange = _this$props8.onChange,
          id = _this$props8.id,
          name = _this$props8.name;
      onChange((0, _commonUtility.parseNumber)(value, null), id || name);
    }
  }, {
    key: "isValid",
    value: function isValid(value) {
      var numberRegex = RegExp('^-?\\d*\\.?\\d*$');
      return numberRegex.test(value);
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props9 = this.props,
          disabled = _this$props9.disabled,
          placeholder = _this$props9.placeholder,
          inputClassName = _this$props9.inputClassName,
          inputStyle = _this$props9.inputStyle,
          className = _this$props9.className,
          style = _this$props9.style,
          resetable = _this$props9.resetable,
          defaultValue = _this$props9.defaultValue,
          classes = _this$props9.classes,
          name = _this$props9.name,
          id = _this$props9.id,
          required = _this$props9.required,
          suffix = _this$props9.suffix,
          innerPluralText = _this$props9.innerPluralText,
          innerSingularText = _this$props9.innerSingularText,
          asLabel = _this$props9.asLabel,
          icon = _this$props9.icon;
      var isReadOnlyLocal = this.state.isReadOnlyLocal;
      var _this$state = this.state,
          value = _this$state.value,
          disableMax = _this$state.disableMax,
          disableMin = _this$state.disableMin;
      var isReadOnlyWithoutSuffixAndHasText = isReadOnlyLocal && !suffix && (innerPluralText || innerSingularText);
      return /*#__PURE__*/_react["default"].createElement(_PickerResetWrapper["default"], {
        className: (0, _classnames["default"])(className, (0, _defineProperty2["default"])({}, classes.numberPickerLabelWrapper, asLabel)),
        style: style,
        resetable: resetable,
        disabled: defaultValue === value,
        id: id,
        onReset: this.reset,
        "data-qa": "number-picker"
      }, icon && icon, /*#__PURE__*/_react["default"].createElement(_Input["default"], {
        suffix: suffix,
        required: required,
        name: name,
        id: id,
        value: this.calculateValueWithText(value),
        disabled: isReadOnlyLocal || disabled,
        placeholder: placeholder,
        onChange: this.handleInputChange,
        className: (0, _classnames["default"])(inputClassName, (0, _defineProperty2["default"])({}, classes.isReadOnly, isReadOnlyWithoutSuffixAndHasText)),
        containerClassName: (0, _classnames["default"])(classes.input, {
          'bg-white': isReadOnlyLocal
        }, 'border-radius-0', classes.inputWrapper, {
          numberPickerLabel: asLabel
        }),
        style: inputStyle
      }), /*#__PURE__*/_react["default"].createElement(_NumberPickerActionButtons["default"], {
        asLabel: asLabel,
        disableMax: disabled || disableMax,
        disableMin: disabled || disableMin,
        onIncrement: this.handleIncrementClick,
        onDecrement: this.handleDecrementClick
      }));
    }
  }]);
  return NumberPicker;
}(_react.PureComponent);

NumberPicker.defaultProps = {
  value: undefined,
  disabled: false,
  defaultValue: undefined,
  min: undefined,
  max: undefined,
  step: 1,
  floatingPoint: 0,
  placeholder: '',
  inputClassName: '',
  inputStyle: {},
  className: '',
  style: {},
  resetable: false,
  id: undefined,
  name: '',
  suffix: null,
  innerPluralText: null,
  innerSingularText: null,
  isReadOnly: false,
  asLabel: false
};
NumberPicker.propTypes = {
  /** Controlled value to be set */
  value: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].number]),

  /** Indicator whether or not the input is disabled */
  disabled: _propTypes["default"].bool,

  /** CallBack for on value change */
  onChange: _propTypes["default"].func.isRequired,

  /** Default value for input (also resets to value) */
  defaultValue: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].number]),

  /** Minimum allowed value */
  min: _propTypes["default"].number,

  /** Maximum allowed value */
  max: _propTypes["default"].number,

  /** Amount to increment / decrement on each button click
   * - also allows the support do decimal point */
  step: _propTypes["default"].number,

  /** Placeholder for input level */
  placeholder: _propTypes["default"].string,

  /** Additional class for input level */
  inputClassName: _propTypes["default"].string,

  /** Additional style for input level */
  inputStyle: _propTypes["default"].shape(),

  /** Additional class for root level */
  className: _propTypes["default"].string,

  /** additional style for root level */
  style: _propTypes["default"].shape(),

  /** Show the reset option for the input (resets to defaultValue) */
  resetable: _propTypes["default"].bool,

  /** id indicator for field */
  id: _propTypes["default"].string,

  /** name indicator for field */
  name: _propTypes["default"].string,

  /** Suffix to locate at the end of the number picker */
  suffix: _propTypes["default"].node,

  /** Floating point allowed for number picker */
  floatingPoint: _propTypes["default"].number,

  /** inner plural text locate after the number */
  innerPluralText: _propTypes["default"].string,

  /** inner singular text locate after the number */
  innerSingularText: _propTypes["default"].string,

  /** an indicator if the input is read only means you can't enter a number only use the buttons */
  isReadOnly: _propTypes["default"].bool,

  /** Component icon */
  icon: _propTypes["default"].node,

  /** Should render as label */
  asLabel: _propTypes["default"].bool
};
var NumberPickerComponent = NumberPicker;
exports.NumberPickerComponent = NumberPickerComponent;

var _default = (0, _reactJss["default"])(styles)(NumberPicker);

exports["default"] = _default;
