// @guestyci/foundation/legacy/NumberPicker/NumberPickerActionButtons.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _reactJss = require("react-jss");

var _ActionButton = _interopRequireDefault(require("../ActionButton"));

var BtnArrowUp = function BtnArrowUp(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M4.529 0a.474.474 0 0 0-.357.137L.137 4.172a.488.488 0 0 0 .69.689L4.53 1.156l3.705 3.705a.488.488 0 0 0 .69-.69L4.89.138A.477.477 0 0 0 4.53 0z"
  }));
};

BtnArrowUp.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 9.0618038 4.9977479"
};

var BtnArrowDown = function BtnArrowDown(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M4.533 4.997c.13.006.26-.039.357-.136L8.925.826a.488.488 0 0 0-.69-.69L4.532 3.843.826.137a.488.488 0 0 0-.69.69L4.172 4.86c.097.097.23.143.362.136z"
  }));
};

BtnArrowDown.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 9.0618038 4.9977479"
};
var useStyles = (0, _reactJss.createUseStyles)({
  actionButton: {
    width: 45,
    height: 23
  },
  btnTop: {
    borderRadius: '0px !important',
    borderTopRightRadius: '2px !important',
    borderWidth: '1px 1px 1px 0px !important'
  },
  btnBottom: {
    borderRadius: '0px !important',
    borderBottomRightRadius: '2px !important',
    borderWidth: '0px 1px 1px 0px !important'
  },
  btnAsLabelWrapper: {
    border: '0px'
  },
  btnAsLabel: {
    border: '0px',
    padding: '0px 5px',
    height: '15px'
  }
});

var NumberPickerActionButtons = function NumberPickerActionButtons(_ref) {
  var asLabel = _ref.asLabel,
      onIncrement = _ref.onIncrement,
      onDecrement = _ref.onDecrement,
      disableMax = _ref.disableMax,
      disableMin = _ref.disableMin;
  var classes = useStyles();
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])('flex-center bl-1 border-gray-light flex-column', (0, _defineProperty2["default"])({}, classes.btnAsLabelWrapper, asLabel)),
    "data-qa": "number-picker-action-buttons"
  }, /*#__PURE__*/_react["default"].createElement(_ActionButton["default"], {
    onClick: onIncrement,
    iconWidth: 10,
    iconHeight: 10,
    disabled: disableMax,
    svg: BtnArrowUp,
    className: (0, _classnames["default"])("".concat(classes.actionButton, " ").concat(classes.btnTop), (0, _defineProperty2["default"])({}, classes.btnAsLabel, asLabel))
  }), /*#__PURE__*/_react["default"].createElement(_ActionButton["default"], {
    onClick: onDecrement,
    iconWidth: 10,
    iconHeight: 10,
    disabled: disableMin,
    svg: BtnArrowDown,
    className: (0, _classnames["default"])("".concat(classes.actionButton, " ").concat(classes.btnBottom), (0, _defineProperty2["default"])({}, classes.btnAsLabel, asLabel))
  }));
};

NumberPickerActionButtons.defaultProps = {
  disableMin: false,
  disableMax: false
};
NumberPickerActionButtons.propTypes = {
  onIncrement: _propTypes["default"].func.isRequired,
  onDecrement: _propTypes["default"].func.isRequired,
  disableMax: _propTypes["default"].bool,
  disableMin: _propTypes["default"].bool
};
var _default = NumberPickerActionButtons;
exports["default"] = _default;
