// @guestyci/foundation/validators/validationMessages.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.lessThanMessage = exports.moreThanMessage = exports.maxMessage = exports.minMessage = exports.minLengthMessage = exports.maxLengthMessage = exports.END_DATE_REQUIRED = exports.START_DATE_REQUIRED = exports.INVALID_PHONE_NUMBER = exports.INVALID_PHONE = exports.INVALID_PHONE_10_DIGITS = exports.NOT_ZERO = exports.MAX_3_DIGITS = exports.ONLY_ALPHA_NUMERIC = exports.INVALID_EMAIL = exports.REQUIRED = void 0;
var REQUIRED = 'This field is required';
exports.REQUIRED = REQUIRED;
var INVALID_EMAIL = 'Invalid email address';
exports.INVALID_EMAIL = INVALID_EMAIL;
var ONLY_ALPHA_NUMERIC = 'Only alphanumeric characters';
exports.ONLY_ALPHA_NUMERIC = ONLY_ALPHA_NUMERIC;
var MAX_3_DIGITS = 'Max 3 digits after decimal point';
exports.MAX_3_DIGITS = MAX_3_DIGITS;
var NOT_ZERO = 'Not zero';
exports.NOT_ZERO = NOT_ZERO;
var INVALID_PHONE_10_DIGITS = 'Invalid phone number, must be 10 digits';
exports.INVALID_PHONE_10_DIGITS = INVALID_PHONE_10_DIGITS;
var INVALID_PHONE = 'Invalid phone number';
exports.INVALID_PHONE = INVALID_PHONE;
var INVALID_PHONE_NUMBER = 'Invalid phone number, check your number/country zone';
exports.INVALID_PHONE_NUMBER = INVALID_PHONE_NUMBER;
var START_DATE_REQUIRED = 'Start date is required';
exports.START_DATE_REQUIRED = START_DATE_REQUIRED;
var END_DATE_REQUIRED = 'End date is required';
exports.END_DATE_REQUIRED = END_DATE_REQUIRED;

var maxLengthMessage = function maxLengthMessage(max) {
  return "Must be ".concat(max, " characters or less");
};

exports.maxLengthMessage = maxLengthMessage;

var minLengthMessage = function minLengthMessage(min) {
  return "Must be ".concat(min, " characters or more");
};

exports.minLengthMessage = minLengthMessage;

var minMessage = function minMessage(minVal) {
  return "Must be at least ".concat(minVal);
};

exports.minMessage = minMessage;

var maxMessage = function maxMessage(maxVal) {
  return "Must be at most ".concat(maxVal);
};

exports.maxMessage = maxMessage;

var moreThanMessage = function moreThanMessage(value) {
  return "Must be more than ".concat(value);
};

exports.moreThanMessage = moreThanMessage;

var lessThanMessage = function lessThanMessage(value) {
  return "Must be less than ".concat(value);
};

exports.lessThanMessage = lessThanMessage;
