// @guestyci/foundation/legacy/Table/ActionsCell/ActionsCell.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = require("prop-types");

var _reactJss = require("react-jss");

var _noop = _interopRequireDefault(require("lodash/noop"));

var _classnames = _interopRequireDefault(require("classnames"));

var _IconDropdown = _interopRequireDefault(require("../../IconDropdown"));

var _Spinner = _interopRequireDefault(require("../../Spinner"));

var _enums = require("../../../enums");

var _Cell = _interopRequireDefault(require("../Cell"));

var BtnAc = function BtnAc(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("circle", {
    cx: "1.5",
    cy: "1.5",
    r: "1.5"
  }), /*#__PURE__*/_react["default"].createElement("circle", {
    cx: "7",
    cy: "1.5",
    r: "1.5"
  }), /*#__PURE__*/_react["default"].createElement("circle", {
    cx: "12.5",
    cy: "1.5",
    r: "1.5"
  }));
};

BtnAc.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 14 3"
};
var useStyles = (0, _reactJss.createUseStyles)({
  spinner: {
    width: '25px',
    height: '25px'
  }
});

var ActionsCell = function ActionsCell(_ref) {
  var onChange = _ref.onChange,
      options = _ref.options,
      isLoading = _ref.isLoading,
      isActionInProgress = _ref.isActionInProgress,
      className = _ref.className,
      style = _ref.style,
      rest = (0, _objectWithoutProperties2["default"])(_ref, ["onChange", "options", "isLoading", "isActionInProgress", "className", "style"]);

  var _useStyles = useStyles(),
      spinner = _useStyles.spinner;

  return /*#__PURE__*/_react["default"].createElement(_Cell["default"], (0, _extends2["default"])({}, rest, {
    isLoading: isLoading,
    className: (0, _classnames["default"])('actions-cell', className),
    style: style,
    contentClassName: "d-flex justify-content-end mr-1",
    "data-qa": "actions-cell"
  }), isActionInProgress ? /*#__PURE__*/_react["default"].createElement(_Spinner["default"], {
    size: 20,
    className: spinner
  }) : /*#__PURE__*/_react["default"].createElement(_IconDropdown["default"], {
    svg: BtnAc,
    onChange: onChange,
    options: options,
    className: (0, _classnames["default"])('d-flex', 'justify-content-center', 'width-25 height-25', 'bg-white', 'b-1 border-white border-hover-gray border-radius-25'),
    iconClassName: "p-0",
    anchorDirection: _enums.AnchorDirectionType.End,
    width: 14
  }));
};

ActionsCell.propTypes = {
  /**
   * The dropdown options to show - Structure ({
   *   label: {Node} - the label to present,
   *   value: {String|number} - unique value to be sent to the server,
   *   name: {String} - required if value to present if label is component-shown in input once selected,
   * })
   */
  options: (0, _propTypes.arrayOf)(_propTypes.object),

  /** Callback function to be called on value changed */
  onChange: _propTypes.func,

  /** indicator that cell is yet to receive data */
  isLoading: _propTypes.bool,

  /** indicator that some action is in progress, shows spinner */
  isActionInProgress: _propTypes.bool,

  /** class name of cell */
  className: _propTypes.string,

  /** styles applied to a cell */
  style: (0, _propTypes.shape)()
};
ActionsCell.defaultProps = {
  options: [],
  onChange: _noop["default"],
  isLoading: false,
  isActionInProgress: false,
  className: '',
  style: {}
};
var _default = ActionsCell;
exports["default"] = _default;
