// @guestyci/foundation/esm/legacy/ActionTooltip/ActionTooltipBody.js
import React from 'react';
import PropTypes from 'prop-types';
import { ActionTooltipType } from "../../enums/enums";
import ActionTooltipButton from "./ActionTooltipButton";
import Col from "../Col/Col";

var ActionTooltipBody = function ActionTooltipBody(_ref) {
  var actionType = _ref.actionType,
      actionText = _ref.actionText,
      actionHref = _ref.actionHref,
      onAction = _ref.onAction,
      body = _ref.body;
  return /*#__PURE__*/React.createElement(Col, {
    spacing: 2,
    "data-qa": "action-tooltip-body"
  }, /*#__PURE__*/React.createElement("div", {
    className: "action-tooltip-text letter-spacing-lg text-left"
  }, body), /*#__PURE__*/React.createElement("div", {
    className: "flex-end"
  }, /*#__PURE__*/React.createElement(ActionTooltipButton, {
    actionType: actionType,
    actionText: actionText,
    actionHref: actionHref,
    onAction: onAction
  })));
};

ActionTooltipBody.defaultProps = {
  actionHref: '',
  onAction: function onAction() {}
};
ActionTooltipBody.propTypes = {
  /** Sets the type of the action */
  actionType: PropTypes.oneOf([ActionTooltipType.Button, ActionTooltipType.Link]).isRequired,

  /** Sets the text of the action */
  actionText: PropTypes.node.isRequired,

  /** Sets the href of the action */
  actionHref: PropTypes.string,

  /** Callback function - called when clicking the tooltip action */
  onAction: PropTypes.func,

  /** Sets the body (content) of the tooltip F.E: text to be diplayed */
  body: PropTypes.node.isRequired
};
export default ActionTooltipBody;
