// @guestyci/foundation/esm/legacy/ActionTooltip/ActionTooltipButton.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _possibleConstructorReturn from "@babel/runtime/helpers/esm/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/esm/getPrototypeOf";

function _createSuper(Derived) { return function () { var Super = _getPrototypeOf(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ActionTooltipType } from "../../enums/enums";
import RaisedButton from "../RaisedButton/RaisedButton";
import Link from "../Link/Link";

var ActionTooltipButton = /*#__PURE__*/function (_Component) {
  _inherits(ActionTooltipButton, _Component);

  var _super = _createSuper(ActionTooltipButton);

  function ActionTooltipButton() {
    var _this;

    _classCallCheck(this, ActionTooltipButton);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _super.call.apply(_super, [this].concat(args));

    _this.handleClick = function (e) {
      var _this$props = _this.props,
          actionHref = _this$props.actionHref,
          onAction = _this$props.onAction;

      if (actionHref) {
        window.open(actionHref, '_blank');
      }

      onAction(e);
    };

    return _this;
  }

  _createClass(ActionTooltipButton, [{
    key: "getActionSettings",
    value: function getActionSettings() {
      var actionType = this.props.actionType;
      var buttonComponents = {
        RaisedButton: RaisedButton,
        Link: Link
      };
      var ActionComponent = buttonComponents.RaisedButton;
      var className = '';

      if (actionType !== ActionTooltipType.Button) {
        ActionComponent = buttonComponents.Link;
        className = 'text-hover-blue-dark-theme';
      }

      return {
        ActionComponent: ActionComponent,
        className: className
      };
    }
  }, {
    key: "render",
    value: function render() {
      var actionText = this.props.actionText;

      var _this$getActionSettin = this.getActionSettings(),
          ActionComponent = _this$getActionSettin.ActionComponent,
          className = _this$getActionSettin.className;

      return /*#__PURE__*/React.createElement(ActionComponent, {
        className: className,
        onClick: this.handleClick,
        "data-qa": "action-tooltip-button"
      }, /*#__PURE__*/React.createElement("div", {
        className: "flex-center"
      }, /*#__PURE__*/React.createElement("span", {
        className: ""
      }, actionText)));
    }
  }]);

  return ActionTooltipButton;
}(Component);

ActionTooltipButton.defaultProps = {
  actionHref: '',
  onAction: function onAction() {}
};
ActionTooltipButton.propTypes = {
  /** Sets the type of the action */
  actionType: PropTypes.oneOf([ActionTooltipType.Button, ActionTooltipType.Link]).isRequired,

  /** Sets the text of the action */
  actionText: PropTypes.node.isRequired,

  /** Sets the Href of the action */
  actionHref: PropTypes.string,

  /** Callback(optional) to be executed when the action is clicked */
  onAction: PropTypes.func
};
export default ActionTooltipButton;
