// @guestyci/foundation/DaysPicker/DaysPicker.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.useStyles = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _noop = _interopRequireDefault(require("lodash/noop"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Row = _interopRequireDefault(require("../Layout/Row"));

var _createStyles = _interopRequireDefault(require("../createStyles"));

var _Button = _interopRequireDefault(require("../Button"));

var _colors = require("../theme/colors");

var _useTheme = _interopRequireDefault(require("../useTheme"));

var _composeEvent = _interopRequireDefault(require("../utils/composeEvent"));

var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    root: {
      border: "1px solid ".concat(theme.palette.border),
      borderRadius: 2,
      width: 'fit-content'
    },
    button: {
      height: 46,
      width: 46,
      borderRadius: 0,
      background: _colors.white,
      color: theme.palette.text.secondary,
      borderRight: "1px solid ".concat(theme.palette.border),
      '&:last-child': {
        borderRight: "0px solid ".concat(theme.palette.border)
      }
    },
    inactive: {
      '&:not([disabled]):hover': {
        backgroundColor: theme.palette.primary.background,
        color: theme.palette.primary["default"]
      }
    },
    active: {
      backgroundColor: theme.palette.primary["default"],
      color: _colors.white
    }
  };
}, {
  name: 'DaysPicker'
});
exports.useStyles = useStyles;
var days = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'];

var DaysPicker = _react["default"].forwardRef(function DaysPicker(_ref, ref) {
  var _ref$disabled = _ref.disabled,
      disabled = _ref$disabled === void 0 ? false : _ref$disabled,
      _ref$name = _ref.name,
      name = _ref$name === void 0 ? '' : _ref$name,
      _ref$value = _ref.value,
      value = _ref$value === void 0 ? [] : _ref$value,
      _ref$onChange = _ref.onChange,
      onChange = _ref$onChange === void 0 ? _noop["default"] : _ref$onChange,
      _ref$className = _ref.className,
      className = _ref$className === void 0 ? '' : _ref$className,
      _ref$style = _ref.style,
      style = _ref$style === void 0 ? {} : _ref$style,
      _ref$jss = _ref.jss,
      jss = _ref$jss === void 0 ? {} : _ref$jss,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["disabled", "name", "value", "onChange", "className", "style", "jss"]);
  var classes = useStyles({
    jss: jss
  });
  var theme = (0, _useTheme["default"])();
  var valueSet = (0, _react.useMemo)(function () {
    return new Set(value);
  }, [value]);

  var handleDayClick = function handleDayClick(e) {
    var clickedValue = e.currentTarget.value;
    var set = new Set(valueSet);

    if (set.has(clickedValue)) {
      set["delete"](clickedValue);
    } else {
      set.add(clickedValue);
    }

    var newEvent = (0, _composeEvent["default"])(e, {
      name: name,
      value: Array.from(set)
    });
    onChange(newEvent);
  };

  return /*#__PURE__*/_react["default"].createElement(_Row["default"], (0, _extends2["default"])({
    ref: ref,
    style: style,
    wrap: false,
    className: (0, _classnames["default"])(classes.root, className)
  }, props), days.map(function (day) {
    var _cn;

    return /*#__PURE__*/_react["default"].createElement(_Button["default"], {
      disabled: disabled,
      key: day,
      onClick: handleDayClick,
      value: day,
      rippleColor: theme.palette.primary["default"],
      className: (0, _classnames["default"])(classes.button, (_cn = {}, (0, _defineProperty2["default"])(_cn, classes.active, valueSet.has(day)), (0, _defineProperty2["default"])(_cn, classes.inactive, !valueSet.has(day)), _cn))
    }, day);
  }));
});

DaysPicker.propTypes = {
  /** Array of selected values to pass */
  value: _propTypes["default"].arrayOf(_propTypes["default"].string),

  /** Callback for on item change */
  onChange: _propTypes["default"].func,

  /** Addition class on root */
  className: _propTypes["default"].string,

  /** Addition style on root */
  style: _propTypes["default"].shape(),

  /** jss object override */
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func]),

  /** component name */
  name: _propTypes["default"].string,

  /** Indicator for disabled */
  disabled: _propTypes["default"].bool
};
DaysPicker.displayName = 'DaysPicker';
var _default = DaysPicker;
exports["default"] = _default;
