// @guestyci/foundation/legacy/Table/ErrorBanner/ErrorBanner.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _noop = _interopRequireDefault(require("lodash/noop"));

var _Row = _interopRequireDefault(require("../../../Layout/Row"));

var _TextField = _interopRequireDefault(require("../../../TextField"));

var _createStyles = _interopRequireDefault(require("../../../createStyles"));

var _colors = require("../../../theme/colors");

var _CSSTransition = _interopRequireDefault(require("../../../CSSTransition"));

var _Button = _interopRequireDefault(require("../../../Button"));

var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    root: {
      backgroundColor: _colors.red,
      height: 40,
      transformOrigin: '50% 100%',
      color: _colors.white
    },
    scaleEnter: {
      opacity: 0.01,
      transform: 'scaleY(0)'
    },
    scaleEnterDone: {
      opacity: 1,
      transform: 'scaleY(1)'
    },
    scaleExit: {
      opacity: 1,
      transform: 'scaleY(1)'
    },
    scaleExitDone: {
      opacity: 0.01,
      transform: 'scaleY(0)'
    },
    scaleEnterActive: {
      transform: 'scaleY(1)',
      opacity: 1,
      transition: theme.transition.create()
    },
    scaleExitActive: {
      transform: 'scaleY(0)',
      opacity: 0.01,
      transition: theme.transition.create()
    },
    button: {
      color: [_colors.white, '!important'],
      textDecoration: 'underline',
      backgroundColor: 'transparent',
      border: 0,
      padding: [0, '!important']
    }
  };
}, {
  name: 'ErrorBanner'
});

var ErrorBanner = _react["default"].memo(function ErrorBanner(_ref) {
  var _ref$show = _ref.show,
      show = _ref$show === void 0 ? false : _ref$show,
      _ref$errorMessage = _ref.errorMessage,
      errorMessage = _ref$errorMessage === void 0 ? '' : _ref$errorMessage,
      onClick = _ref.onClick,
      actionText = _ref.actionText,
      className = _ref.className;
  var classes = useStyles();
  return /*#__PURE__*/_react["default"].createElement(_CSSTransition["default"], {
    unmountOnExit: true,
    animation: {
      enter: classes.scaleEnter,
      enterActive: classes.scaleEnterActive,
      exit: classes.scaleExit,
      exitActive: classes.scaleExitActive,
      enterDone: classes.scaleEnterDone,
      exitDone: classes.scaleExitDone
    },
    show: show
  }, /*#__PURE__*/_react["default"].createElement(_Row["default"], {
    justify: "center",
    align: "center",
    spacing: 2,
    fullWidth: true,
    className: (0, _classnames["default"])(classes.root, className)
  }, /*#__PURE__*/_react["default"].createElement(_TextField["default"], null, errorMessage), actionText && onClick && /*#__PURE__*/_react["default"].createElement(_Button["default"], {
    onClick: onClick,
    className: classes.button
  }, actionText)));
});

ErrorBanner.defaultProps = {
  show: false,
  errorMessage: '',
  onClick: _noop["default"],
  actionText: undefined,
  className: ''
};
ErrorBanner.propTypes = {
  /** Boolean flag to show the banner */
  show: _propTypes["default"].bool,

  /** String with the error message */
  errorMessage: _propTypes["default"].string,

  /** Action to happen on banner  button click */
  onClick: _propTypes["default"].func,

  /** String with the action text */
  actionText: _propTypes["default"].string,

  /** Additional class prop */
  className: _propTypes["default"].string
};
ErrorBanner.displayName = 'ErrorBanner';
var _default = ErrorBanner;
exports["default"] = _default;
