// @guestyci/foundation/legacy/Tabs/ActiveTabBottomBorder.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _reactJss = _interopRequireDefault(require("react-jss"));

var styles = {
  border: {
    transition: 'left 225ms cubic-bezier(0, 0, 0.2, 1) 0ms, width 225ms cubic-bezier(0, 0, 0.2, 1)'
  }
};

var ActiveTabBottomBorder = function ActiveTabBottomBorder(_ref) {
  var classes = _ref.classes,
      width = _ref.width,
      position = _ref.position,
      style = _ref.style;
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])('absolute bottom-0 bb-2 transition-left border-blue text-blue', classes.border),
    style: (0, _extends2["default"])({
      width: width,
      left: position
    }, style),
    "data-qa": "active-tab-bottom-border"
  });
};

var _default = (0, _reactJss["default"])(styles)(ActiveTabBottomBorder);

exports["default"] = _default;
