// @guestyci/foundation/legacy/Tabs/Tab.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _FlatButton = _interopRequireDefault(require("../FlatButton"));

var _createStyles = _interopRequireDefault(require("../../createStyles"));

var useStyles = (0, _createStyles["default"])({
  root: {
    boxShadow: 'none !important'
  }
}, {
  name: 'Tab'
});

var Tab = function Tab(_ref) {
  var onClick = _ref.onClick,
      tabIndex = _ref.tabIndex,
      value = _ref.value,
      title = _ref.title,
      className = _ref.className,
      disabled = _ref.disabled,
      isActive = _ref.isActive;
  var classes = useStyles();
  var onTabClick = (0, _react.useCallback)(function (e) {
    e.stopPropagation();
    onClick(tabIndex, value);
  }, [tabIndex, value]);
  return /*#__PURE__*/_react["default"].createElement(_FlatButton["default"], {
    value: value,
    className: (0, _classnames["default"])(className, classes.root, 'text-uppercase', 'p-0', 'tab-item', {
      'text-blue': isActive
    }),
    onClick: onTabClick,
    disabled: disabled,
    "data-qa": "tab"
  }, title);
};

Tab.defaultProps = {
  children: null,
  disabled: false,
  value: undefined
};
Tab.propTypes = {
  /** The tab title */
  title: _propTypes["default"].node.isRequired,

  /** Set tab content to show when active */
  children: _propTypes["default"].node,

  /** Indicator whether or not the tab is disabled */
  disabled: _propTypes["default"].bool,

  /** The tab value to be passed when clicked */
  value: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].number, _propTypes["default"].shape()])
};
var _default = Tab;
exports["default"] = _default;
