// @guestyci/foundation/legacy/Tabs/Tabs.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.TabsComponent = exports.shouldRecalculatePosition = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactJss = _interopRequireDefault(require("react-jss"));

var _classnames = _interopRequireDefault(require("classnames"));

var _noop = _interopRequireDefault(require("lodash/noop"));

var _isNumber = _interopRequireDefault(require("lodash/isNumber"));

var _ActiveTabBottomBorder = _interopRequireDefault(require("./ActiveTabBottomBorder"));

var _ExpandingSearchBar = _interopRequireDefault(require("../ExpandingSearchBar/ExpandingSearchBar"));

var _Fade = _interopRequireDefault(require("../Fade/Fade"));

var _utils = require("../../utils");

function _createSuper(Derived) { return function () { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

var styles = {
  expandableSearchBar: {
    top: 2
  }
};

var shouldRecalculatePosition = function shouldRecalculatePosition(prevChildren, children) {
  if (!children) {
    return false;
  }

  if (!prevChildren && children) {
    return true;
  }

  return prevChildren.length !== children.length;
};

exports.shouldRecalculatePosition = shouldRecalculatePosition;

var Tabs = /*#__PURE__*/function (_Component) {
  (0, _inherits2["default"])(Tabs, _Component);

  var _super = _createSuper(Tabs);

  function Tabs(props) {
    var _this;

    (0, _classCallCheck2["default"])(this, Tabs);
    _this = _super.call(this, props);

    _this.setControlledPosition = function (value) {
      var tabs = _this.tabsRef.current.getElementsByClassName('tab-item');

      var index = Array.from(tabs).findIndex(function (tab) {
        return tab.value === String(value);
      });

      if (index > -1) {
        _this.setPosition(index);
      }
    };

    _this.setPosition = function (activeTabIndex) {
      _this.setState({
        activeTabIndex: activeTabIndex
      }, function () {
        _this.setBorderWidth();
      });
    };

    _this.setBorderWidth = function () {
      var activeTabIndex = _this.state.activeTabIndex;

      var tabs = _this.tabsRef.current.getElementsByClassName('tab-item');

      var activeChild = tabs[activeTabIndex];

      _this.setState({
        borderWidth: activeChild.offsetWidth,
        position: activeChild.offsetLeft
      });
    };

    _this.initActiveTabIndex = function (children) {
      var activeIndex = 0;

      _react["default"].Children.map(children, function (child, index) {
        if (child.props.active && !child.props.disabled) {
          activeIndex = index;
        }
      });

      return activeIndex;
    };

    _this.changeTab = function (activeTabIndex) {
      var value = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};

      _this.setPosition(activeTabIndex);

      var _this$props = _this.props,
          onChange = _this$props.onChange,
          onChangeTab = _this$props.onChangeTab;

      if (onChange || onChangeTab) {
        var tabProps = {
          value: value,
          activeTabIndex: activeTabIndex
        };

        if (onChange) {
          onChange(tabProps);
        }

        if (onChangeTab) {
          onChangeTab(tabProps);
        }
      }
    };

    _this.toggleTabs = function (isSearchOpen) {
      _this.setState({
        showTabs: !isSearchOpen
      });
    };

    _this.isActiveIndex = function (index) {
      var activeTabIndex = _this.state.activeTabIndex;
      return index === activeTabIndex;
    };

    _this.tabsRef = _react["default"].createRef();
    _this.state = {
      activeTabIndex: !(0, _utils.isEmpty)(props.initialValue) ? props.initialValue : _this.initActiveTabIndex(props.children),
      borderWidth: 0,
      position: null,
      showTabs: true
    };
    return _this;
  }

  (0, _createClass2["default"])(Tabs, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      var _this2 = this;

      var _this$props2 = this.props,
          initialValue = _this$props2.initialValue,
          value = _this$props2.value;

      if (!(0, _utils.isEmpty)(initialValue) || !(0, _utils.isEmpty)(value)) {
        this.setControlledPosition(initialValue || value);
      } else {
        setTimeout(function () {
          return _this2.setBorderWidth();
        }, 300);
      }
    }
  }, {
    key: "componentDidUpdate",
    value: function componentDidUpdate(prevProps) {
      var _this3 = this;

      var _this$props3 = this.props,
          value = _this$props3.value,
          children = _this$props3.children;

      if (!(0, _utils.isEmpty)(value) && value !== prevProps.value) {
        this.setControlledPosition(value);
      }

      if (shouldRecalculatePosition(prevProps.children, children)) {
        setTimeout(function () {
          _this3.setBorderWidth();

          _this3.setControlledPosition(value);
        }, 155);
      }
    }
  }, {
    key: "render",
    value: function render() {
      var _children$activeTabIn,
          _children$activeTabIn2,
          _children$props,
          _this4 = this;

      var _this$props4 = this.props,
          children = _this$props4.children,
          className = _this$props4.className,
          style = _this$props4.style,
          onSearchSubmit = _this$props4.onSearchSubmit,
          withSearch = _this$props4.withSearch,
          noBorderBottom = _this$props4.noBorderBottom,
          noBorder = _this$props4.noBorder,
          classes = _this$props4.classes,
          activeTabContentClassName = _this$props4.activeTabContentClassName,
          fullWidth = _this$props4.fullWidth,
          onSearchChange = _this$props4.onSearchChange;
      var _this$state = this.state,
          activeTabIndex = _this$state.activeTabIndex,
          borderWidth = _this$state.borderWidth,
          position = _this$state.position,
          showTabs = _this$state.showTabs;
      var activeTabContent = Array.isArray(children) ? children[activeTabIndex] && ((_children$activeTabIn = children[activeTabIndex]) === null || _children$activeTabIn === void 0 ? void 0 : (_children$activeTabIn2 = _children$activeTabIn.props) === null || _children$activeTabIn2 === void 0 ? void 0 : _children$activeTabIn2.children) : children === null || children === void 0 ? void 0 : (_children$props = children.props) === null || _children$props === void 0 ? void 0 : _children$props.children;
      var borderClass = !(noBorder || noBorderBottom) ? 'bb-2 border-gray-light' : '';
      return /*#__PURE__*/_react["default"].createElement("div", {
        className: (0, _classnames["default"])('d-flex-fill', 'flex-column', className),
        ref: this.tabsRef,
        style: style,
        "data-qa": "tabs"
      }, /*#__PURE__*/_react["default"].createElement("div", {
        className: (0, _classnames["default"])('d-flex', withSearch && borderClass)
      }, /*#__PURE__*/_react["default"].createElement(_Fade["default"], {
        mountOnEnter: true,
        show: showTabs,
        className: (0, _classnames["default"])('relative', {
          'w-fill': fullWidth
        })
      }, /*#__PURE__*/_react["default"].createElement("div", {
        className: (0, _classnames["default"])('tabs-wrapper d-flex text-uppercase min-height-50 h-fill text-gray-dark', !withSearch && borderClass)
      }, _react["default"].Children.map(children, function (child, index) {
        return _react["default"].cloneElement(child, {
          onClick: _this4.changeTab,
          className: index !== children.length - 1 ? 'mr-5' : '',
          tabIndex: index,
          isActive: _this4.isActiveIndex(index)
        });
      })), !!((0, _isNumber["default"])(activeTabIndex) && (0, _isNumber["default"])(position)) && /*#__PURE__*/_react["default"].createElement(_ActiveTabBottomBorder["default"], {
        width: borderWidth,
        style: withSearch ? {
          bottom: -2
        } : null,
        position: position
      })), withSearch && /*#__PURE__*/_react["default"].createElement(_ExpandingSearchBar["default"], {
        onSubmit: onSearchSubmit,
        containerClassName: classes.expandableSearchBar,
        onToggleSearch: this.toggleTabs,
        onChange: onSearchChange
      })), activeTabContent && /*#__PURE__*/_react["default"].createElement("div", {
        className: (0, _classnames["default"])('mt-6', activeTabContentClassName)
      }, activeTabContent));
    }
  }]);
  return Tabs;
}(_react.Component);

Tabs.defaultProps = {
  onChange: _noop["default"],
  onChangeTab: _noop["default"],
  className: '',
  style: {},
  withSearch: false,
  onSearchSubmit: _noop["default"],
  onSearchChange: _noop["default"],
  noBorder: false,
  noBorderBottom: false,
  initialValue: null,
  value: null,
  activeTabContentClassName: '',
  fullWidth: false
};
Tabs.propTypes = {
  /** The tabs children array to render */
  children: _propTypes["default"].node.isRequired,

  /** Callback for on tab change payload:{ activeTabIndex, value (as passed in tab) } */
  onChange: _propTypes["default"].func,

  /** Deprecated - user onChange in stead, Callback for tab change */
  onChangeTab: _propTypes["default"].func,

  /** Additional class to tabs root */
  className: _propTypes["default"].string,

  /** Additional style to tabs root */
  style: _propTypes["default"].shape(),

  /** Indicator whether or not we want a search option on the tabs */
  withSearch: _propTypes["default"].bool,

  /** Callback function to apply on search submit */
  onSearchSubmit: _propTypes["default"].func,

  /** Callback function to apply on search input change */
  onSearchChange: _propTypes["default"].func,

  /** Indicator whether or not to show the border for the tabs - DEPRECATED */
  noBorder: _propTypes["default"].bool,

  /** Indicator whether or not to show the bottom line for the tabs */
  noBorderBottom: _propTypes["default"].bool,

  /** initial tab index value to be active */
  initialValue: _propTypes["default"].number,

  /** Controlled value for tab index - setting this will make component controlled */
  value: _propTypes["default"].oneOfType([_propTypes["default"].number, _propTypes["default"].string]),

  /** Additional class to wrapper of children */
  activeTabContentClassName: _propTypes["default"].string,

  /** Boolean indicator for parent full width */
  fullWidth: _propTypes["default"].bool
};
var TabsComponent = Tabs;
exports.TabsComponent = TabsComponent;

var _default = (0, _reactJss["default"])(styles)(Tabs);

exports["default"] = _default;
