// @guestyci/foundation/basePicker/PickerClearButton.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _IconButton = _interopRequireDefault(require("../IconButton"));

var _Fade = _interopRequireDefault(require("../Fade"));

var _createStyles = _interopRequireDefault(require("../createStyles"));

var BtnDelete = function BtnDelete(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M8.485.707L7.778 0 4.243 3.536.707 0 0 .707l3.536 3.536L0 7.778l.707.707L4.243 4.95l3.535 3.535.707-.707L4.95 4.243z"
  }));
};

BtnDelete.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 8.4854 8.4854"
};
var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    root: {
      color: theme.palette.secondary["default"],
      '&:hover': {
        color: theme.palette.primary["default"]
      }
    }
  };
});

var PickerClearButton = function PickerClearButton(_ref) {
  var show = _ref.show,
      onClick = _ref.onClick,
      asSuffix = _ref.asSuffix;
  var classes = useStyles();
  return /*#__PURE__*/_react["default"].createElement(_Fade["default"], {
    show: show,
    mountOnEnter: true,
    "data-qa": "picker-clear-button"
  }, /*#__PURE__*/_react["default"].createElement(_IconButton["default"], {
    className: classes.root,
    component: "span",
    size: "xs",
    position: asSuffix ? 'suffix' : undefined,
    onClick: onClick
  }, /*#__PURE__*/_react["default"].createElement(BtnDelete, {
    width: 12,
    height: 12
  })));
};

var _default = PickerClearButton;
exports["default"] = _default;
