// @guestyci/foundation/basePicker/PickerInputButton.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.useStyles = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _createStyles = _interopRequireDefault(require("../createStyles"));

var _Icon = _interopRequireDefault(require("../Icon"));

var _Button = _interopRequireDefault(require("../Button"));

var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    root: {
      height: '100%',
      width: 46,
      minWidth: 46,
      borderRadius: 0,
      transition: theme.transition.create(),
      fill: theme.palette.secondary["default"]
    },
    rightAligned: {
      borderLeft: "solid 1px ".concat(theme.palette.border)
    },
    leftAligned: {
      borderRight: "solid 1px ".concat(theme.palette.border)
    },
    active: {
      fill: theme.palette.primary["default"]
    }
  };
});
exports.useStyles = useStyles;

var PickerInputButton = _react["default"].forwardRef(function PickerInputButton(_ref, ref) {
  var _cn;

  var disabled = _ref.disabled,
      onClick = _ref.onClick,
      active = _ref.active,
      className = _ref.className,
      jss = _ref.jss,
      svg = _ref.svg,
      _ref$position = _ref.position,
      position = _ref$position === void 0 ? 'prefix' : _ref$position,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["disabled", "onClick", "active", "className", "jss", "svg", "position"]);
  var classes = useStyles({
    jss: jss
  });
  return /*#__PURE__*/_react["default"].createElement(_Button["default"], (0, _extends2["default"])({}, props, {
    component: "span",
    ref: ref,
    enableRipple: false,
    className: (0, _classnames["default"])(classes.root, (_cn = {}, (0, _defineProperty2["default"])(_cn, classes.rightAligned, position === 'suffix'), (0, _defineProperty2["default"])(_cn, classes.leftAligned, position === 'prefix'), _cn), className),
    disabled: disabled,
    size: "xs",
    onClick: onClick
  }), /*#__PURE__*/_react["default"].createElement(_Icon["default"], {
    svg: svg,
    className: (0, _classnames["default"])((0, _defineProperty2["default"])({}, classes.active, active), 'picker-button'),
    width: 19,
    height: 19
  }));
});

PickerInputButton.propTypes = {
  onClick: _propTypes["default"].func,
  disabled: _propTypes["default"].bool,
  className: _propTypes["default"].string,
  active: _propTypes["default"].bool,
  position: _propTypes["default"].oneOf(['prefix', 'suffix']),
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func])
};
PickerInputButton.displayName = 'PickerInputButton';
var _default = PickerInputButton;
exports["default"] = _default;
