// @guestyci/foundation/DatePicker/DateRangePicker/DateRangePicker.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.useStyles = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactMomentProptypes = _interopRequireDefault(require("react-moment-proptypes"));

var _constants = require("react-dates/lib/constants");

require("react-dates/initialize");

var _reactDates = require("react-dates");

var _moment = _interopRequireDefault(require("moment"));

var _classnames = _interopRequireDefault(require("classnames"));

var _createStyles = _interopRequireDefault(require("../../createStyles"));

var _commonUtility = require("../../utils/commonUtility");

var _Row = _interopRequireDefault(require("../../Layout/Row"));

var _PickerClearButton = _interopRequireDefault(require("../../basePicker/PickerClearButton"));

var _styles = _interopRequireDefault(require("../styles"));

var _NavButton = _interopRequireDefault(require("../NavButton"));

var _PickerInputButton = _interopRequireDefault(require("../../basePicker/PickerInputButton"));

var _constants2 = require("../constants");

var _CustomArrowIcon = _interopRequireDefault(require("../CustomArrowIcon"));

var BtnCalendar = function BtnCalendar(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M18.5 2h-4V0h-1v2h-6V0h-1v2h-4A2.503 2.503 0 0 0 0 4.5v12C0 17.879 1.122 19 2.5 19h16c1.379 0 2.5-1.121 2.5-2.5v-12C21 3.122 19.879 2 18.5 2zm-16 1h4v.667h1V3h6v.667h1V3h4c.827 0 1.5.673 1.5 1.5V7H1V4.5C1 3.673 1.673 3 2.5 3zm16 15h-16c-.827 0-1.5-.673-1.5-1.5V8h19v8.5c0 .827-.673 1.5-1.5 1.5z"
  }));
};

BtnCalendar.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 21 19"
};
var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    root: _styles["default"].root(theme),
    inputContainer: (0, _extends2["default"])({}, _styles["default"].inputContainer, {
      paddingRight: theme.spacer(3)
    }),
    focused: _styles["default"].focused(theme),
    error: _styles["default"].error(theme),
    disabled: _styles["default"].disabled(theme)
  };
}, {
  name: 'DateRangePicker'
});
exports.useStyles = useStyles;
var DateRangePicker = (0, _react.forwardRef)(function DateRangePicker(_ref, ref) {
  var value = _ref.value,
      name = _ref.name,
      onChange = _ref.onChange,
      focusedProp = _ref.focused,
      onFocusChange = _ref.onFocusChange,
      isFetching = _ref.isFetching,
      error = _ref.error,
      blockedDates = _ref.blockedDates,
      isContinuous = _ref.isContinuous,
      _ref$showClearDate = _ref.showClearDate,
      showClearDate = _ref$showClearDate === void 0 ? true : _ref$showClearDate,
      enablePastDays = _ref.enablePastDays,
      disabled = _ref.disabled,
      onMonthChange = _ref.onMonthChange,
      initialVisibleMonth = _ref.initialVisibleMonth,
      _ref$anchorDirection = _ref.anchorDirection,
      anchorDirection = _ref$anchorDirection === void 0 ? 'left' : _ref$anchorDirection,
      startDatePlaceholderText = _ref.startDatePlaceholderText,
      endDatePlaceholderText = _ref.endDatePlaceholderText,
      startDateId = _ref.startDateId,
      endDateId = _ref.endDateId,
      style = _ref.style,
      className = _ref.className,
      jss = _ref.jss,
      rest = (0, _objectWithoutProperties2["default"])(_ref, ["value", "name", "onChange", "focused", "onFocusChange", "isFetching", "error", "blockedDates", "isContinuous", "showClearDate", "enablePastDays", "disabled", "onMonthChange", "initialVisibleMonth", "anchorDirection", "startDatePlaceholderText", "endDatePlaceholderText", "startDateId", "endDateId", "style", "className", "jss"]);
  var classes = useStyles({
    jss: jss
  });

  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      focusedState = _useState2[0],
      setFocusState = _useState2[1];

  var inputFocus = focusedProp || focusedState;

  var renderMonthElement = function renderMonthElement(_ref2) {
    var month = _ref2.month;
    return month.format(_constants2.MONTH_DISPLAY_FORMAT);
  };

  var startDateInputId = (0, _react.useMemo)(function () {
    return startDateId || (0, _commonUtility.uuid)();
  }, [startDateId]);
  var endDateInputId = (0, _react.useMemo)(function () {
    return endDateId || (0, _commonUtility.uuid)();
  }, [endDateId]);

  var handleFocusChange = function handleFocusChange(focusedInput) {
    if (onFocusChange) {
      onFocusChange(focusedInput);
    }

    setFocusState(focusedInput);
  };

  var isOutsideRange = enablePastDays ? function () {
    return false;
  } : undefined;

  var handleDatesChange = function handleDatesChange(_ref3) {
    var startDate = _ref3.startDate,
        endDate = _ref3.endDate;

    if (onChange) {
      // If startDate changes - reset endDate
      if ((value === null || value === void 0 ? void 0 : value.startDate) && value.startDate !== startDate) {
        return onChange({
          startDate: startDate,
          endDate: undefined
        }, name);
      }

      onChange({
        startDate: startDate,
        endDate: endDate
      }, name);
    }
  };

  var handleClearDates = function handleClearDates() {
    handleDatesChange({
      startDate: undefined,
      endDate: undefined
    });
  };

  var upperBound = (0, _react.useMemo)(function () {
    if (isContinuous) {
      if ((value === null || value === void 0 ? void 0 : value.startDate) && blockedDates) {
        var formattedStartDate = value.startDate.format(_constants2.DATE_FORMAT);
        return Array.from(blockedDates).sort().find(function (blockedDay) {
          return blockedDay > formattedStartDate;
        });
      }
    }
  }, [value === null || value === void 0 ? void 0 : value.startDate, blockedDates, isContinuous]);

  var renderDayContents = function renderDayContents(day) {
    return isFetching ? '...' : day.date();
  };

  var blockedDatesSet = (0, _react.useMemo)(function () {
    return new Set(blockedDates || []);
  }, [blockedDates]);

  var isDayBlocked = function isDayBlocked(day) {
    var formattedDay = day.format(_constants2.DATE_FORMAT);

    if (isContinuous) {
      if (inputFocus === _constants.END_DATE && upperBound) {
        return formattedDay > upperBound || formattedDay < upperBound && !!(blockedDatesSet === null || blockedDatesSet === void 0 ? void 0 : blockedDatesSet.has(formattedDay));
      }
    }

    return isFetching || !!(blockedDatesSet === null || blockedDatesSet === void 0 ? void 0 : blockedDatesSet.has(formattedDay));
  };

  var handleCalendarButtonClick = function handleCalendarButtonClick() {
    if (!inputFocus) {
      if (onFocusChange) {
        onFocusChange(_constants.START_DATE);
      }

      setFocusState(_constants.START_DATE);
    }
  };

  var getInitialVisibleMonth = function getInitialVisibleMonth() {
    return initialVisibleMonth ? (0, _moment["default"])(initialVisibleMonth) : (0, _moment["default"])();
  };

  return /*#__PURE__*/_react["default"].createElement(_Row["default"], {
    ref: ref,
    justify: "between",
    className: (0, _classnames["default"])(classes.root, 'date-picker', className, (0, _defineProperty2["default"])({}, classes.focused, inputFocus), (0, _defineProperty2["default"])({}, classes.error, error), (0, _defineProperty2["default"])({}, classes.disabled, disabled)),
    style: style
  }, /*#__PURE__*/_react["default"].createElement(_PickerInputButton["default"], {
    disabled: disabled,
    onClick: handleCalendarButtonClick,
    svg: BtnCalendar,
    active: !!inputFocus,
    leftAligned: true
  }), /*#__PURE__*/_react["default"].createElement(_Row["default"], {
    className: classes.inputContainer
  }, /*#__PURE__*/_react["default"].createElement(_reactDates.DateRangePicker, (0, _extends2["default"])({
    renderMonthElement: renderMonthElement,
    readOnly: true,
    appendToBody: false,
    onDatesChange: handleDatesChange,
    focusedInput: inputFocus || undefined,
    onFocusChange: handleFocusChange,
    endDateId: endDateInputId,
    startDateId: startDateInputId,
    disabled: disabled,
    startDatePlaceholderText: startDatePlaceholderText,
    endDatePlaceholderText: endDatePlaceholderText,
    startDate: value === null || value === void 0 ? void 0 : value.startDate,
    endDate: value === null || value === void 0 ? void 0 : value.endDate,
    isDayBlocked: isDayBlocked,
    daySize: 40,
    noBorder: true,
    openDirection: _constants.OPEN_DOWN,
    onNextMonthClick: onMonthChange,
    onPrevMonthClick: onMonthChange,
    numberOfMonths: 1,
    hideKeyboardShortcutsPanel: true,
    verticalSpacing: 10,
    renderDayContents: renderDayContents,
    navPrev: /*#__PURE__*/_react["default"].createElement(_NavButton["default"], {
      direction: "Left"
    }),
    navNext: /*#__PURE__*/_react["default"].createElement(_NavButton["default"], {
      direction: "Right"
    }),
    customArrowIcon: /*#__PURE__*/_react["default"].createElement(_CustomArrowIcon["default"], {
      isInputEmpty: !(value === null || value === void 0 ? void 0 : value.startDate) && !(value === null || value === void 0 ? void 0 : value.endDate)
    }),
    isOutsideRange: isOutsideRange,
    initialVisibleMonth: getInitialVisibleMonth,
    anchorDirection: anchorDirection,
    displayFormat: _constants2.DATE_DISPLAY_FORMAT
  }, rest)), /*#__PURE__*/_react["default"].createElement(_PickerClearButton["default"], {
    show: showClearDate && ((value === null || value === void 0 ? void 0 : value.startDate) || (value === null || value === void 0 ? void 0 : value.endDate)),
    onClick: handleClearDates
  })));
});
DateRangePicker.displayName = 'DateRangePicker';
var _default = DateRangePicker;
exports["default"] = _default;
DateRangePicker.propTypes = {
  /** the picked date range */
  value: _propTypes["default"].shape({
    startDate: _reactMomentProptypes["default"].momentObj,
    endDate: _reactMomentProptypes["default"].momentObj
  }),

  /** date picker name to be set */
  name: _propTypes["default"].string,

  /** Callback to be called on picker value change */
  onChange: _propTypes["default"].func,

  /** Control focus state - focued => picker opens. Not required - focus state is uncontrolled by default */
  focused: _propTypes["default"].bool,

  /** Callback to be called when focus state changes. Called with a boolean  */
  onFocusChange: _propTypes["default"].func,

  /** Fetching state */
  isFetching: _propTypes["default"].bool,

  /** Indicator for form error state  */
  error: _propTypes["default"].bool,

  /** Array of blocked dates - formatted as YYYY-MM-DD */
  blockedDates: _propTypes["default"].arrayOf(_propTypes["default"].string),

  /** accept only a continuous range. Meaning, it cannot contain blocked days */
  isContinuous: _propTypes["default"].bool,

  /** When dates is selected - show a clear button */
  showClearDate: _propTypes["default"].bool,

  /** Enable past days */
  enablePastDays: _propTypes["default"].bool,

  /** Disable state in which picker interaction is not available */
  disabled: _propTypes["default"].bool,

  /** callback to be called when a switching months. Called with a momnet object of the new month */
  onMonthChange: _propTypes["default"].func,

  /** month to be show when the dialog picker opens - formatted as YYYY-MM-DD */
  initialVisibleMonth: _propTypes["default"].string,

  /** Anchor direction to open the picker - */
  anchorDirection: _propTypes["default"].oneOf(['left', 'right']),

  /** Placeholder for the start date */
  startDatePlaceholderText: _propTypes["default"].string,

  /** Placeholder for the end date */
  endDatePlaceholderText: _propTypes["default"].string,

  /** id for the start date */
  startDateId: _propTypes["default"].string,

  /** id for the end date */
  endDateId: _propTypes["default"].string,

  /** class name to be added to the root */
  className: _propTypes["default"].string,

  /** style to be added to the root */
  style: _propTypes["default"].shape(),
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func])
};
