// @guestyci/foundation/esm/legacy/Accordion/AccordionTitle.js
import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import withHover from "../../withHover";
import AccordionChevron from "./AccordionChevron";
import TextField from "../TextField/TextField";

var AccordionTitle = function AccordionTitle(_ref) {
  var title = _ref.title,
      isHovering = _ref.isHovering,
      onHoverEnter = _ref.onHoverEnter,
      onHoverLeave = _ref.onHoverLeave,
      onClick = _ref.onClick,
      isOpen = _ref.isOpen;
  return /*#__PURE__*/React.createElement("div", {
    role: "presentation",
    className: cn('w-fill clickable flex-space-between-center pt-3', "pb-".concat(isOpen ? 6 : 3)),
    onClick: onClick,
    onMouseEnter: onHoverEnter,
    onMouseLeave: onHoverLeave,
    "data-qa": "accordion-title"
  }, /*#__PURE__*/React.createElement(TextField, {
    size: "lg",
    className: "height-20 flex-start-center w-fill",
    color: "blue"
  }, title), /*#__PURE__*/React.createElement(AccordionChevron, {
    isOpen: isOpen,
    isActive: isHovering
  }));
};

AccordionTitle.propTypes = {
  /** Indicator whether or not collapse should be open */
  isOpen: PropTypes.bool.isRequired,

  /** Title body */
  title: PropTypes.node.isRequired,

  /** Callback for title click */
  onClick: PropTypes.func.isRequired
};
export default withHover(AccordionTitle);
