// @guestyci/foundation/Layout/Section/Section.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.useStyles = exports.generateGutter = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _memoize = _interopRequireDefault(require("@guestyci/memoize"));

var _commonUtility = require("../../utils/commonUtility");

var _createStyles = _interopRequireDefault(require("../../createStyles"));

var _styles = _interopRequireDefault(require("../styles"));

var _utils = require("../../utils");

var generateGutter = (0, _memoize["default"])(function (gutter, spacer) {
  if (!gutter) {
    return undefined;
  }

  if ((0, _utils.isNumber)(gutter)) {
    return spacer(gutter);
  }

  return gutter.split(' ').reduce(function (padding, item) {
    return "".concat(padding, " ").concat(spacer(item), "px");
  }, '').trim();
});
exports.generateGutter = generateGutter;
var useStyles = (0, _createStyles["default"])(function (theme) {
  return (0, _extends2["default"])({
    root: {
      backgroundColor: function backgroundColor(_ref) {
        var bgColor = _ref.bgColor;
        return bgColor;
      }
    },
    gutter: {
      padding: function padding(_ref2) {
        var gutter = _ref2.gutter;
        return generateGutter(gutter, theme.spacer);
      }
    },
    displayFlex: {
      display: 'flex',
      width: '100%'
    }
  }, _styles["default"]);
}, {
  name: 'Section'
});
exports.useStyles = useStyles;

var Section = _react["default"].forwardRef(function Section(_ref3, ref) {
  var _cn;

  var className = _ref3.className,
      style = _ref3.style,
      children = _ref3.children,
      gutter = _ref3.gutter,
      flex = _ref3.flex,
      justify = _ref3.justify,
      wrap = _ref3.wrap,
      align = _ref3.align,
      bgColor = _ref3.bgColor,
      jss = _ref3.jss,
      props = (0, _objectWithoutProperties2["default"])(_ref3, ["className", "style", "children", "gutter", "flex", "justify", "wrap", "align", "bgColor", "jss"]);
  var classes = useStyles({
    bgColor: bgColor,
    gutter: gutter,
    jss: jss
  });
  return /*#__PURE__*/_react["default"].createElement("div", (0, _extends2["default"])({}, props, {
    ref: ref,
    className: (0, _classnames["default"])(classes.root, (_cn = {}, (0, _defineProperty2["default"])(_cn, classes.gutter, gutter), (0, _defineProperty2["default"])(_cn, classes.flexWrap, wrap), (0, _defineProperty2["default"])(_cn, classes.displayFlex, flex), (0, _defineProperty2["default"])(_cn, classes["justify".concat((0, _commonUtility.firstLetterUppercase)(justify))], justify !== 'start'), (0, _defineProperty2["default"])(_cn, classes["alignItems".concat((0, _commonUtility.firstLetterUppercase)(align))], align !== 'stretch'), _cn), className),
    style: style
  }), children);
});

Section.defaultProps = {
  bgColor: 'initial',
  className: '',
  style: {},
  children: null,
  gutter: 0,
  flex: true,
  wrap: true,
  justify: 'start',
  align: 'stretch'
};
Section.propTypes = {
  /** Background color for section */
  bgColor: _propTypes["default"].string,

  /** Additional class for top level */
  className: _propTypes["default"].string,

  /** Indicator whether or not to render flex wrap */
  wrap: _propTypes["default"].bool,

  /** Additional styling for root level */
  style: _propTypes["default"].shape(),

  /** set gutter padding to component - available value - [0,1,2,3,4,5,6,7,8,9,10] will be determined by the theme spacer (default 5)
   * Also supporting string format "x y", "top right bottom left" - e.g "2 3", "2 2 0 2"
   * */
  gutter: _propTypes["default"].oneOfType([_propTypes["default"].oneOf([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]), _propTypes["default"].string]),

  /** Child components inside a section */
  children: _propTypes["default"].oneOfType([_propTypes["default"].node, _propTypes["default"].arrayOf(_propTypes["default"].node)]),

  /** Boolean to set section a flex */
  flex: _propTypes["default"].bool,

  /** Set flex alignment - available values = ['start', 'center', 'end', 'stretch', 'baseline']  */
  align: _propTypes["default"].oneOf(['start', 'center', 'end', 'stretch', 'baseline']),

  /** Set flex justify content - available values = ['start','center','end','between','around','evenly'] */
  justify: _propTypes["default"].oneOf(['start', 'center', 'end', 'between', 'around', 'evenly']),

  /** JSS overrides */
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func])
}; // Hack for withInfo plugin

Section.displayName = 'Section';
var _default = Section;
exports["default"] = _default;
