// @guestyci/foundation/RaisedButton/RaisedButton.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _noop = _interopRequireDefault(require("lodash/noop"));

var _utils = require("../theme/utils");

var _colors = require("../theme/colors");

var _createStyles = _interopRequireDefault(require("../createStyles"));

var _Button = _interopRequireDefault(require("../Button"));

var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    root: {
      minWidth: 95,
      color: _colors.white,
      boxShadow: '0 2px 3px 0 rgba(0, 0, 0, 0.14)',
      backgroundColor: theme.palette.primary["default"],
      '&:not([disabled]):hover': {
        backgroundColor: function backgroundColor(_ref) {
          var isLoading = _ref.isLoading;
          return isLoading ? theme.palette.primary["default"] : theme.palette.primary.hover;
        }
      },
      '&:not([disabled]):active': {
        border: theme.boxShadow[3]
      },
      '&:disabled': {
        backgroundColor: (0, _utils.fade)(theme.palette.primary.disabled, 0.7),
        color: (0, _utils.fade)(_colors.white, 0.7)
      }
    }
  };
}, {
  name: 'RaisedButton'
});

var RaisedButton = _react["default"].forwardRef(function RaisedButton(_ref2, ref) {
  var children = _ref2.children,
      size = _ref2.size,
      isLoading = _ref2.isLoading,
      onClick = _ref2.onClick,
      disabled = _ref2.disabled,
      className = _ref2.className,
      jss = _ref2.jss,
      props = (0, _objectWithoutProperties2["default"])(_ref2, ["children", "size", "isLoading", "onClick", "disabled", "className", "jss"]);
  var classes = useStyles({
    isLoading: isLoading,
    jss: jss
  });
  return /*#__PURE__*/_react["default"].createElement(_Button["default"], (0, _extends2["default"])({}, props, {
    disabled: disabled,
    isLoading: isLoading,
    size: size,
    onClick: onClick,
    spinnerColor: _colors.white,
    className: (0, _classnames["default"])(classes.root, className),
    rippleColor: _colors.white,
    ref: ref
  }), children);
});

RaisedButton.defaultProps = {
  size: 'md',
  onClick: _noop["default"],
  isLoading: false,
  disabled: false
};
RaisedButton.propTypes = {
  /** Child element to locate inside of button */
  children: _propTypes["default"].node.isRequired,

  /** callback to handle click event for button (evnt) => {} */
  onClick: _propTypes["default"].func,

  /** Button size - values =['md', 'lg', 'xl'] */
  size: _propTypes["default"].oneOf(['md', 'lg', 'xl']),

  /** Indicator whether or not to show loader in button */
  isLoading: _propTypes["default"].bool,

  /** Indicator whether or not to button is disabled */
  disabled: _propTypes["default"].bool,

  /** jss override object to customize the jss classes */
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func])
};
RaisedButton.displayName = 'RaisedButton';
var _default = RaisedButton;
exports["default"] = _default;
