// @guestyci/foundation/legacy/basePicker/PickerResetWrapper.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactJss = _interopRequireDefault(require("react-jss"));

var _classnames = _interopRequireDefault(require("classnames"));

var _ActionButton = _interopRequireDefault(require("../ActionButton"));

var BtnReset = function BtnReset(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M14.272 4.242a8.461 8.461 0 0 0-2.277-1.55V0l-.746.423c-1.218.69-3.92 2.506-4.033 2.583l-.63.424.638.41c.113.074 2.796 1.805 4.038 2.46l.733.387V3.819c.57.31 1.104.688 1.588 1.147 3.044 2.894 3.23 7.792.413 10.92-1.363 1.513-3.217 2.389-5.219 2.466-2.006.078-3.904-.648-5.37-2.043C.363 13.416.178 8.518 2.994 5.39l-.743-.668c-3.175 3.526-2.966 9.05.466 12.313 1.585 1.506 3.622 2.324 5.77 2.324.11 0 .219-.002.329-.007 2.275-.088 4.378-1.081 5.922-2.796 3.176-3.526 2.967-9.05-.466-12.313zm-3.277.761a91.787 91.787 0 0 1-2.586-1.591c.67-.444 1.741-1.149 2.586-1.672v3.263z"
  }));
};

BtnReset.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 16.9898434 19.3579102"
};
var styles = {
  resetBtn: {
    borderLeftWidth: '0 !important',
    height: 46
  }
};

var PickerResetWrapper = function PickerResetWrapper(_ref) {
  var resetable = _ref.resetable,
      className = _ref.className,
      style = _ref.style,
      disabled = _ref.disabled,
      onReset = _ref.onReset,
      children = _ref.children,
      classes = _ref.classes,
      id = _ref.id;
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])('picker-wrapper flex-end d-flex-fill', className),
    style: style,
    "data-qa": "picker-reset-wrapper"
  }, children, resetable && /*#__PURE__*/_react["default"].createElement(_ActionButton["default"], {
    tooltip: "Reset",
    id: id,
    svg: BtnReset,
    className: classes.resetBtn,
    disabled: disabled,
    onClick: onReset
  }));
};

PickerResetWrapper.defaultProps = {
  className: '',
  style: {},
  disabled: false,
  id: ''
};
PickerResetWrapper.propTypes = {
  children: _propTypes["default"].oneOfType([_propTypes["default"].arrayOf(_propTypes["default"].node), _propTypes["default"].node]).isRequired,
  resetable: _propTypes["default"].bool.isRequired,
  onReset: _propTypes["default"].func.isRequired,
  className: _propTypes["default"].string,
  style: _propTypes["default"].shape(),
  disabled: _propTypes["default"].bool,
  id: _propTypes["default"].string
};

var _default = (0, _reactJss["default"])(styles)(PickerResetWrapper);

exports["default"] = _default;
