// @guestyci/foundation/FormControl/FormControl.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.useStyles = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _TextField = _interopRequireDefault(require("../TextField"));

var _Row = _interopRequireDefault(require("../Layout/Row"));

var _Col = _interopRequireDefault(require("../Layout/Col"));

var _createStyles = _interopRequireDefault(require("../createStyles"));

var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    root: {},
    label: {
      alignSelf: 'flex-start',
      marginBottom: theme.spacer(2),
      color: theme.palette.label
    },
    info: {
      marginTop: '2px',
      color: theme.palette.description
    }
  };
});
exports.useStyles = useStyles;

var FormControl = _react["default"].forwardRef(function FormControl(_ref, ref) {
  var name = _ref.name,
      children = _ref.children,
      _ref$label = _ref.label,
      label = _ref$label === void 0 ? null : _ref$label,
      _ref$info = _ref.info,
      info = _ref$info === void 0 ? null : _ref$info,
      _ref$className = _ref.className,
      className = _ref$className === void 0 ? '' : _ref$className,
      _ref$error = _ref.error,
      error = _ref$error === void 0 ? '' : _ref$error,
      _ref$style = _ref.style,
      style = _ref$style === void 0 ? {} : _ref$style,
      _ref$required = _ref.required,
      required = _ref$required === void 0 ? false : _ref$required,
      _ref$jss = _ref.jss,
      jss = _ref$jss === void 0 ? {} : _ref$jss,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["name", "children", "label", "info", "className", "error", "style", "required", "jss"]);
  var classes = useStyles({
    jss: jss
  });
  var showInfo = info && !error;
  return /*#__PURE__*/_react["default"].createElement(_Col["default"], (0, _extends2["default"])({}, props, {
    ref: ref,
    tabIndex: -1,
    className: (0, _classnames["default"])(classes.root, {
      'form-control-error': !!error
    }, className),
    style: style
  }), label && /*#__PURE__*/_react["default"].createElement(_TextField["default"], {
    className: classes.label
  }, label), _react["default"].cloneElement(children, {
    error: !!error,
    name: name,
    required: required
  }), /*#__PURE__*/_react["default"].createElement(_Row["default"], {
    spacing: 3,
    justify: error ? 'start' : 'end'
  }, error && /*#__PURE__*/_react["default"].createElement(_TextField["default"], {
    color: "error"
  }, error), showInfo && /*#__PURE__*/_react["default"].createElement(_TextField["default"], {
    className: classes.info
  }, info)));
});

FormControl.propTypes = {
  /** Child component for the form control */
  children: _propTypes["default"].node.isRequired,

  /** Boolean indicator for required */
  required: _propTypes["default"].bool,

  /** Form control label field */
  label: _propTypes["default"].node,

  /* Form control info field */
  info: _propTypes["default"].node,

  /** Additional root classname */
  className: _propTypes["default"].string,

  /** additional root syle */
  style: _propTypes["default"].shape(),

  /** Jss override to create custom styles */
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func])
};
FormControl.displayName = 'FormControl';
var _default = FormControl;
exports["default"] = _default;
