// @guestyci/foundation/es/legacy/Popover/withPopover.js
import _extends from "@babel/runtime/helpers/esm/extends";
import React, { Component } from 'react';
/**
 * Create a wrapper compose to support Popover functionality
 * @param ComposedComponent
 * @returns {{Component}}
 */

const withPopover = ComposedComponent => {
  var _temp;

  return _temp = class extends Component {
    constructor(props) {
      super(props);

      this.closePopover = () => {
        this.setState({
          show: false
        });
      };

      this.openPopover = event => {
        if (event) {
          event.stopPropagation();
        }

        this.setState({
          show: true
        });
      };

      this.togglePopover = event => {
        if (event) {
          event.stopPropagation();
        }

        this.setState(prevState => ({
          show: !prevState.show
        }));
      };

      this.state = {
        show: false
      };
    }

    render() {
      const {
        show
      } = this.state;
      return /*#__PURE__*/React.createElement(ComposedComponent, _extends({}, this.props, {
        isPopoverOpen: show,
        closePopover: this.closePopover,
        openPopover: this.openPopover,
        togglePopover: this.togglePopover
      }));
    }

  }, _temp;
};

export default withPopover;
