// @guestyci/foundation/legacy/DynamicField/DynamicField.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _noop = _interopRequireDefault(require("lodash/noop"));

var _memoize = _interopRequireDefault(require("lodash/memoize"));

var _Input = _interopRequireDefault(require("../Input/Input"));

var _Checkbox = _interopRequireDefault(require("../Checkbox/Checkbox"));

var _NumberPicker = _interopRequireDefault(require("../NumberPicker/NumberPicker"));

var _TimePicker = _interopRequireDefault(require("../TimePicker/TimePicker"));

var _Dropdown = _interopRequireDefault(require("../Dropdown/Dropdown"));

var _TextArea = _interopRequireDefault(require("../TextArea/TextArea"));

var _Person = _interopRequireDefault(require("../Person/Person"));

var _InputLabel = _interopRequireDefault(require("../InputLabel/InputLabel"));

var _DatePicker = _interopRequireDefault(require("../DatePicker/DatePicker"));

var _enums = require("../../enums/enums");

var createPerson = function createPerson(data) {
  return /*#__PURE__*/_react["default"].createElement(_Person["default"], {
    name: data.fullName,
    size: "sm",
    "data-qa": "create-person"
  });
};

var getPersonOptions = function getPersonOptions() {
  var options = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : [];
  return options.map(function (option) {
    return {
      label: createPerson(option),
      value: option._id,
      name: option.fullName
    };
  });
};

var getOptions = function getOptions() {
  var options = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : [];
  return options.map(function (option) {
    return {
      label: option,
      value: option
    };
  });
};

var memoizeGetOptions = (0, _memoize["default"])(getOptions);
var memoizeGetPersonOptions = (0, _memoize["default"])(getPersonOptions);

var getLabel = function getLabel(type, label) {
  return type === 'boolean' ? 'A yes no issue' : label;
};

var getField = function getField(props) {
  var type = props.type,
      _onChange = props.onChange,
      value = props.value,
      name = props.name,
      id = props.id,
      label = props.label,
      options = props.options,
      onLoadMore = props.onLoadMore,
      pageCount = props.pageCount,
      floatingPoint = props.floatingPoint,
      isFetching = props.isFetching,
      searchable = props.searchable,
      onSearchQueryChange = props.onSearchQueryChange;

  switch (type) {
    case 'boolean':
      return /*#__PURE__*/_react["default"].createElement("div", {
        className: "row w-100 pl-3"
      }, /*#__PURE__*/_react["default"].createElement(_Checkbox["default"], {
        name: name,
        id: id,
        onChange: function onChange(val, nameVal) {
          return _onChange(!val, nameVal);
        },
        value: value,
        checked: !!value
      }, label));

    case 'number':
      return /*#__PURE__*/_react["default"].createElement("div", {
        className: "row w-100 pl-3 d-block"
      }, /*#__PURE__*/_react["default"].createElement(_NumberPicker["default"], {
        floatingPoint: floatingPoint,
        value: value,
        name: name,
        id: id,
        onChange: _onChange
      }));

    case 'time':
      return /*#__PURE__*/_react["default"].createElement("div", {
        className: "row w-100 pl-3 d-block"
      }, /*#__PURE__*/_react["default"].createElement(_TimePicker["default"], {
        onChange: _onChange,
        name: name,
        id: id,
        value: value
      }));

    case 'longtext':
      return /*#__PURE__*/_react["default"].createElement("div", {
        className: "row w-100 pl-3 d-block"
      }, /*#__PURE__*/_react["default"].createElement(_TextArea["default"], {
        onChange: _onChange,
        name: name,
        id: id,
        value: value
      }));

    case 'enum':
      {
        var dropDownOptions = memoizeGetOptions(options);
        return /*#__PURE__*/_react["default"].createElement("div", {
          className: "row w-100 pl-3 d-block"
        }, /*#__PURE__*/_react["default"].createElement(_Dropdown["default"], {
          options: dropDownOptions,
          onSelect: _onChange,
          name: name,
          id: id,
          defaultValue: value,
          onLoadMore: onLoadMore,
          pageCount: pageCount,
          isFetching: isFetching,
          searchable: searchable,
          onSearchQueryChange: onSearchQueryChange
        }));
      }

    case 'user':
    case 'contact':
      {
        var _dropDownOptions = memoizeGetPersonOptions(options);

        return /*#__PURE__*/_react["default"].createElement("div", {
          className: "row w-100 pl-3 d-block"
        }, /*#__PURE__*/_react["default"].createElement(_Dropdown["default"], {
          options: _dropDownOptions,
          onSelect: _onChange,
          name: name,
          id: id,
          defaultValue: value,
          onLoadMore: onLoadMore,
          pageCount: pageCount,
          isFetching: isFetching,
          searchable: searchable,
          onSearchQueryChange: onSearchQueryChange
        }));
      }

    case 'date':
      return /*#__PURE__*/_react["default"].createElement(_DatePicker["default"], {
        type: _enums.DatePickerType.Single,
        value: value,
        onChange: _onChange,
        id: id
      });

    case 'text':
    default:
      return /*#__PURE__*/_react["default"].createElement("div", {
        className: "row w-100 d-block pl-3"
      }, /*#__PURE__*/_react["default"].createElement(_Input["default"], {
        value: value,
        name: name,
        id: id,
        onChange: _onChange,
        className: "pr-3"
      }));
  }
};

var DynamicField = function DynamicField(props) {
  var type = props.type,
      label = props.label;
  var fieldElement = getField(props);
  var inputLabel = getLabel(type, label);
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: "form-group-body",
    "data-qa": "dynamic-field"
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: "mb-6 form-field"
  }, /*#__PURE__*/_react["default"].createElement(_InputLabel["default"], {
    className: "dynamic-field-label",
    label: inputLabel
  }), fieldElement));
};

DynamicField.defaultProps = {
  onChange: _noop["default"],
  value: undefined,
  label: undefined,
  options: undefined,
  type: 'text',
  id: undefined,
  name: '',
  pageCount: 0,
  onLoadMore: _noop["default"],
  isFetching: false,
  searchable: false,
  onSearchQueryChange: null
};
DynamicField.propTypes = {
  /** Function to be called on value change */
  onChange: _propTypes["default"].func,

  /** Element type to render */
  type: _propTypes["default"].string,

  /** A label to show on top of the input */
  label: _propTypes["default"].string,

  /** Controlled input value to be set to input */
  value: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].number, _propTypes["default"].bool, _propTypes["default"].instanceOf(Date), _propTypes["default"].shape()]),

  /** The dropdown options to show Can be array of strings or Objects Structure ({
   _id: user/contact id,
   fullName: user/contact full Name,
  }) */
  options: _propTypes["default"].arrayOf(_propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].object])),

  /** Name indicator for field */
  name: _propTypes["default"].string,

  /** id indicator for field */
  id: _propTypes["default"].string,

  /** Callback to handle pagination fetch more */
  onLoadMore: _propTypes["default"].func,

  /** total pages for pagination  */
  pageCount: _propTypes["default"].number,

  /** Indicator whether, or not we are in fetching mode in pagination */
  isFetching: _propTypes["default"].bool,

  /** Set if or not support dropdown search */
  searchable: _propTypes["default"].bool,

  /** Callback to handle search query change */
  onSearchQueryChange: _propTypes["default"].func
};
var _default = DynamicField;
exports["default"] = _default;
