// @guestyci/foundation/legacy/Dropdown/Dropdown.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.DropdownComponent = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _noop = _interopRequireDefault(require("lodash/noop"));

var _classnames = _interopRequireDefault(require("classnames"));

var _DropdownInputField = _interopRequireDefault(require("../baseDropdown/DropdownInputField"));

var _withDropdown = _interopRequireDefault(require("../baseDropdown/withDropdown"));

var _DropdownMenu = _interopRequireDefault(require("../baseDropdown/DropdownMenu"));

var _MenuItem = _interopRequireDefault(require("../Menu/MenuItem"));

var Dropdown = function Dropdown(_ref) {
  var options = _ref.options,
      searchable = _ref.searchable,
      toggleCollapse = _ref.toggleCollapse,
      isCollapsed = _ref.isCollapsed,
      selectedItem = _ref.selectedItem,
      query = _ref.query,
      name = _ref.name,
      placeholder = _ref.placeholder,
      className = _ref.className,
      style = _ref.style,
      onSearchChange = _ref.onSearchChange,
      onSelectItem = _ref.onSelectItem,
      clearSelectedItem = _ref.clearSelectedItem,
      isFetching = _ref.isFetching,
      fetchNextPage = _ref.fetchNextPage,
      pageCount = _ref.pageCount,
      currentPage = _ref.currentPage,
      searchInputValue = _ref.searchInputValue,
      clearQuery = _ref.clearQuery,
      noClear = _ref.noClear,
      closeDropdown = _ref.closeDropdown,
      disabled = _ref.disabled,
      ellipsis = _ref.ellipsis,
      emptyStateComponent = _ref.emptyStateComponent;
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])('d-flex-fill relative flex-column', className),
    style: style,
    name: name,
    "data-qa": "dropdown"
  }, /*#__PURE__*/_react["default"].createElement(_DropdownInputField["default"], {
    onQueryChange: onSearchChange,
    value: searchInputValue,
    query: query,
    name: name,
    searchable: searchable,
    onIconClick: toggleCollapse,
    isOpen: !isCollapsed,
    placeholder: placeholder,
    onClearQuery: clearQuery,
    onClearSelected: clearSelectedItem,
    noClear: noClear,
    disabled: disabled,
    ellipsis: ellipsis
  }), /*#__PURE__*/_react["default"].createElement(_DropdownMenu["default"], {
    emptyStateComponent: emptyStateComponent,
    pageCount: pageCount,
    onDismiss: closeDropdown,
    isEmptyState: !options.length,
    currentPage: currentPage,
    onLoadMore: fetchNextPage,
    isFetching: isFetching,
    isCollapsed: isCollapsed
  }, options.map(function (item) {
    return /*#__PURE__*/_react["default"].createElement(_MenuItem["default"], {
      key: item.value,
      value: item,
      isSelected: item.value === selectedItem.value,
      onClick: onSelectItem
    }, item.label);
  })));
};

Dropdown.defaultProps = {
  searchable: false,
  placeholder: 'Select',
  defaultValue: {},
  onSearchQueryChange: null,
  className: '',
  style: {},
  name: '',
  onLoadMore: _noop["default"],
  isFetching: false,
  pageCount: 0,
  value: undefined,
  noClear: false,
  onChange: _noop["default"],
  onSelect: _noop["default"],
  disabled: false,
  ellipsis: false,
  emptyStateComponent: 'No results found.'
};
Dropdown.propTypes = {
  /** The dropdown options to show - Structure ({
   label: {Node} - the label to present,
   value: {String|number} - unique value to be sent to the server,
  name: {String} - required if value to present if label is component-shown in input once selected,
  })
   */
  options: _propTypes["default"].arrayOf(_propTypes["default"].object).isRequired,

  /** Indicator whether or not the input field is searchable */
  searchable: _propTypes["default"].bool,

  /** Placeholder for input field */
  placeholder: _propTypes["default"].string,

  /** Empty state component  to display when there're no results */
  emptyStateComponent: _propTypes["default"].node,

  /** Initial value to set to dropdown */
  defaultValue: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].string]),

  /** Mandatory if searchable - callback for on input change */
  onSearchQueryChange: _propTypes["default"].func,

  /** Callback function to be called on value changed - Deprecated */
  onSelect: _propTypes["default"].func,

  /** Callback function to be called on value changed */
  onChange: _propTypes["default"].func,

  /** Callback function to be called on dropdown visibility change */
  onVisibilityChange: _propTypes["default"].func,

  /** Additional class for dropdown root */
  className: _propTypes["default"].string,

  /** Additional style for dropdown root */
  style: _propTypes["default"].shape(),

  /** form element name */
  name: _propTypes["default"].string,

  /** Callback function to fetch more items - required for infinite scroll */
  onLoadMore: _propTypes["default"].func,

  /** Total page count for pagination - required to allow infinite scroll */
  pageCount: _propTypes["default"].number,

  /** Indicator whether or not we are currently fetching value for infinite scroll */
  isFetching: _propTypes["default"].bool,

  /** Indicator whether or not we are disabling the clear button */
  noClear: _propTypes["default"].bool,

  /** Value to set for the dropdown to make controlled
   * auto sets if full dropdown item, if string -
   * will only set once the value exists in the dropdown options
   */
  value: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].string, _propTypes["default"].number]),

  /** Indicator for disabled status */
  disabled: _propTypes["default"].bool,

  /** Indicator whether or not selected value is one line & with ellipsis text */
  ellipsis: _propTypes["default"].bool
};
var DropdownComponent = Dropdown;
exports.DropdownComponent = DropdownComponent;

var _default = (0, _withDropdown["default"])(Dropdown);

exports["default"] = _default;
