// @guestyci/foundation/StayIndicator/IndicatorLine.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _helpers = require("./helpers");

var IndicatorLine = function IndicatorLine(_ref) {
  var className = _ref.className,
      vertical = _ref.vertical;
  var viewBox = "0 0 ".concat(_helpers.componentThickness, " ").concat(_helpers.componentThickness);
  var componentMiddleHeight = _helpers.componentThickness / 2;
  var d = !vertical ? "M 0 ".concat(componentMiddleHeight, " L ").concat(_helpers.componentThickness, " ").concat(componentMiddleHeight) : "M ".concat(componentMiddleHeight, " 0 L ").concat(componentMiddleHeight, " ").concat(_helpers.componentThickness);
  return /*#__PURE__*/_react["default"].createElement("svg", {
    className: className,
    viewBox: viewBox,
    preserveAspectRatio: "none",
    "data-qa": "indicator-line"
  }, /*#__PURE__*/_react["default"].createElement("path", {
    fill: "none",
    vectorEffect: "non-scaling-stroke",
    strokeWidth: _helpers.lineWidth,
    d: d
  }));
};

var _default = IndicatorLine;
exports["default"] = _default;
