// @guestyci/foundation/StayIndicator/StayIndicator.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _enums = require("../enums/enums");

var _Tooltip = _interopRequireDefault(require("../Tooltip"));

var _BaseStayIndicator = _interopRequireDefault(require("./BaseStayIndicator"));

var _tooltips;

var tooltips = (_tooltips = {}, (0, _defineProperty2["default"])(_tooltips, _enums.StayStatus.Reserved, 'Reserved'), (0, _defineProperty2["default"])(_tooltips, _enums.StayStatus.Planned, 'Upcoming stay'), (0, _defineProperty2["default"])(_tooltips, _enums.StayStatus.Staying, 'Currently staying'), (0, _defineProperty2["default"])(_tooltips, _enums.StayStatus.Complete, 'Checked out'), _tooltips);

var getWrapperStyles = function getWrapperStyles(_ref) {
  var vertical = _ref.vertical,
      fullWidth = _ref.fullWidth;
  var fullLengthStyles = vertical ? {
    height: '100%'
  } : {
    width: '100%'
  };
  return fullWidth ? fullLengthStyles : null;
};

var StayIndicator = (0, _react.forwardRef)(function (_ref2, ref) {
  var _ref2$status = _ref2.status,
      status = _ref2$status === void 0 ? _enums.StayStatus.Planned : _ref2$status,
      className = _ref2.className,
      jss = _ref2.jss,
      dark = _ref2.dark,
      id = _ref2.id,
      fullWidth = _ref2.fullWidth,
      vertical = _ref2.vertical;
  var wrapperStyles = (0, _react.useMemo)(function () {
    return getWrapperStyles({
      vertical: vertical,
      fullWidth: fullWidth
    });
  }, [vertical, fullWidth]);
  return /*#__PURE__*/_react["default"].createElement("span", {
    style: wrapperStyles
  }, /*#__PURE__*/_react["default"].createElement(_Tooltip["default"], {
    body: tooltips[status],
    placement: "top start",
    id: id
  }, /*#__PURE__*/_react["default"].createElement(_BaseStayIndicator["default"], {
    dark: dark,
    status: status,
    className: className,
    fullWidth: fullWidth,
    vertical: vertical,
    ref: ref,
    jss: jss
  })));
});
StayIndicator.displayName = 'StayIndicator';
StayIndicator.propTypes = {
  /** Id for the tooltip body container */
  id: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].number]),

  /** Stay status - Enum (StayStatus- [Reserved, Planned, Staying, Complete,]) */
  status: _propTypes["default"].oneOf([_enums.StayStatus.Planned, _enums.StayStatus.Staying, _enums.StayStatus.Complete, _enums.StayStatus.Reserved]),

  /** Additional class to add to StayIndicator */
  className: _propTypes["default"].string,

  /** jss override object to customize the jss classes */
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func]),

  /** Set orientation as vertical(default is horizontal) */
  vertical: _propTypes["default"].bool,

  /** Set theme to dark */
  dark: _propTypes["default"].bool,

  /** Set the stay indicator to fill space */
  fullWidth: _propTypes["default"].bool,

  /** reference that will be applied to Row container */
  ref: _propTypes["default"].shape()
};
var _default = StayIndicator;
exports["default"] = _default;
