// @guestyci/foundation/legacy/baseInput/InputStarDeleteActionAddon.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _noop = _interopRequireDefault(require("lodash/noop"));

var _IconButton = _interopRequireDefault(require("../IconButton"));

var _CSSTransiton = _interopRequireDefault(require("../CSSTransiton"));

var _SvgIcon = _interopRequireDefault(require("../SvgIcon/SvgIcon"));

var _Checkbox = _interopRequireDefault(require("../Checkbox/Checkbox"));

function _createSuper(Derived) { return function () { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

var BtnStarOff = function BtnStarOff(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M2.864 14l.886-5.065L0 5.347l5.183-.74L7.5 0l2.318 4.608 5.182.74-3.75 3.587.887 5.065-4.638-2.393L2.864 14zM7.5 10.482l3.305 1.706-.631-3.607 2.648-2.535-3.662-.522L7.5 2.225 5.84 5.524l-3.662.522 2.649 2.535-.63 3.606L7.5 10.482z"
  }));
};

BtnStarOff.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 15 14"
};

var BtnStarOn = function BtnStarOn(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M7.5 0l2.319 4.608L15 5.347l-3.75 3.588.887 5.065-4.638-2.393L2.864 14l.886-5.065L0 5.347l5.183-.74z"
  }));
};

BtnStarOn.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 15 14"
};

var BtnTrash = function BtnTrash(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M11.773 18h-8.5c-1.224 0-2.22-.923-2.22-2.057L1 5.503A.501.501 0 0 1 1.5 5h11.946a.5.5 0 0 1 .5.498L14 15.94c0 1.137-.999 2.06-2.227 2.06zM2.003 6l.05 9.94v.002c0 .574.558 1.058 1.22 1.058h8.5C12.44 17 13 16.516 13 15.942L12.95 6H2.001z"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    d: "M14.5 6H.5a.5.5 0 0 1-.5-.5v-3A.5.5 0 0 1 .5 2h14a.5.5 0 0 1 .5.5v3a.5.5 0 0 1-.5.5zM1 5h13V3H1v2z"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    d: "M10.5 3h-6a.5.5 0 0 1-.5-.5v-2a.5.5 0 0 1 .5-.5h6a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-.5.5zM5 2h5V1H5v1zM10.5 15a.5.5 0 0 1-.5-.5v-6a.5.5 0 1 1 1 0v6a.5.5 0 0 1-.5.5zM7.5 15a.5.5 0 0 1-.5-.5v-6a.5.5 0 1 1 1 0v6a.5.5 0 0 1-.5.5zM4.5 15a.5.5 0 0 1-.5-.5v-6a.5.5 0 1 1 1 0v6a.5.5 0 0 1-.5.5z"
  }));
};

BtnTrash.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 15 18"
};

var InputStarDeleteActionAddOn = /*#__PURE__*/function (_PureComponent) {
  (0, _inherits2["default"])(InputStarDeleteActionAddOn, _PureComponent);

  var _super = _createSuper(InputStarDeleteActionAddOn);

  function InputStarDeleteActionAddOn() {
    var _this;

    (0, _classCallCheck2["default"])(this, InputStarDeleteActionAddOn);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _super.call.apply(_super, [this].concat(args));

    _this.handleStarClick = function () {
      var _this$props = _this.props,
          onStar = _this$props.onStar,
          id = _this$props.id,
          name = _this$props.name;
      onStar(id || name);
    };

    _this.handleTrashClick = function () {
      var _this$props2 = _this.props,
          onTrash = _this$props2.onTrash,
          id = _this$props2.id,
          name = _this$props2.name;
      onTrash(id || name);
    };

    return _this;
  }

  (0, _createClass2["default"])(InputStarDeleteActionAddOn, [{
    key: "render",
    value: function render() {
      var _this$props3 = this.props,
          show = _this$props3.show,
          isPrimary = _this$props3.isPrimary,
          disabled = _this$props3.disabled;
      return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(_CSSTransiton["default"], {
        show: show || isPrimary,
        animation: "fade",
        className: "absolute px-4  right-0 h-fill flex-center bg-gray-lightest"
      }, /*#__PURE__*/_react["default"].createElement("div", {
        className: "flex-center"
      }, !isPrimary && /*#__PURE__*/_react["default"].createElement(_IconButton["default"], {
        iconHeight: 18,
        iconWidth: 15,
        type: "hint",
        disabled: disabled,
        onClick: this.handleTrashClick,
        svg: BtnTrash,
        className: " p-0 input-addon-trash-action"
      }), /*#__PURE__*/_react["default"].createElement(_Checkbox["default"], {
        className: "input-addon-star-action pl-4",
        checked: isPrimary,
        onChange: this.handleStarClick,
        disabled: disabled,
        inputClassName: "mr-0",
        icon: /*#__PURE__*/_react["default"].createElement(_SvgIcon["default"], {
          width: 15,
          height: 14,
          className: "fill-hover-blue",
          color: "muted",
          svg: BtnStarOff
        }),
        checkedIcon: /*#__PURE__*/_react["default"].createElement(_SvgIcon["default"], {
          color: "blue",
          width: 15,
          height: 14,
          svg: BtnStarOn
        })
      }))));
    }
  }]);
  return InputStarDeleteActionAddOn;
}(_react.PureComponent);

InputStarDeleteActionAddOn.defaultProps = {
  show: false,
  isPrimary: false,
  onTrash: _noop["default"],
  onStar: _noop["default"],
  disabled: false,
  id: undefined,
  name: ''
};
InputStarDeleteActionAddOn.propTypes = {
  show: _propTypes["default"].bool,
  isPrimary: _propTypes["default"].bool,
  onTrash: _propTypes["default"].func,
  onStar: _propTypes["default"].func,
  disabled: _propTypes["default"].bool,
  name: _propTypes["default"].string,
  id: _propTypes["default"].oneOfType([_propTypes["default"].number, _propTypes["default"].string])
};
var _default = InputStarDeleteActionAddOn;
exports["default"] = _default;
