// @guestyci/foundation/esm/legacy/ThreadMessage/ThreadMessageBody.js
import _extends from "@babel/runtime/helpers/esm/extends";
import React from 'react';
import withStyles from 'react-jss';
import PropTypes from 'prop-types';
import size from 'lodash/size';
import cn from 'classnames';
import ThreadAttachment from "./ThreadAttachment";
var threadMessageStyle = {
  automatedMessage: {
    backgroundImage: 'repeating-linear-gradient(-67.5deg, transparent, transparent 7px, rgba(255, 255, 255, 0.4) 7px, rgba(255, 255, 255, 0.4) 10px)'
  }
};

var ThreadMessageBody = function ThreadMessageBody(_ref) {
  var body = _ref.body,
      attachments = _ref.attachments,
      isAutoMessage = _ref.isAutoMessage,
      byUs = _ref.byUs,
      errorMessage = _ref.errorMessage,
      bgColor = _ref.bgColor,
      onAttachmentClick = _ref.onAttachmentClick,
      classes = _ref.classes;
  return /*#__PURE__*/React.createElement("div", {
    className: cn('align-items-center mb-2 px-4 py-3 font-size-md', {
      'by-guest': !byUs
    }, {
      'by-us': byUs
    }, {
      'border border-red': !!errorMessage
    }, "bg-".concat(bgColor), isAutoMessage ? "autoMsg ".concat(classes.automatedMessage) : null),
    "data-qa": "thread-message-body"
  }, /*#__PURE__*/React.createElement("span", {
    dangerouslySetInnerHTML: {
      __html: body ? body.replace(/\n/g, '<br/>') : null
    }
  }), !!size(attachments) && /*#__PURE__*/React.createElement("div", {
    className: "mt-4"
  }, attachments.map(function (attachment) {
    return /*#__PURE__*/React.createElement(ThreadAttachment, _extends({
      key: attachment.url,
      attachment: attachment,
      onAttachmentClick: onAttachmentClick
    }, attachment));
  })));
};

ThreadMessageBody.defaultProps = {
  isAutoMessage: false,
  byUs: true,
  bgColor: null,
  errorMessage: false
};
ThreadMessageBody.propTypes = {
  body: PropTypes.node.isRequired,
  attachments: PropTypes.arrayOf(PropTypes.shape()).isRequired,
  isAutoMessage: PropTypes.bool,
  byUs: PropTypes.bool,
  bgColor: PropTypes.string,
  onAttachmentClick: PropTypes.func,
  errorMessage: PropTypes.bool
};
export default withStyles(threadMessageStyle)(ThreadMessageBody);
