// @guestyci/foundation/esm/legacy/ThreadMessage/ThreadMessageLoadingState.js
import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';

var InfoLoadingState = function InfoLoadingState(_ref) {
  var byUs = _ref.byUs;
  return /*#__PURE__*/React.createElement("div", {
    className: cn('d-flex align-items-center bg-gray-lightest text-muted', {
      'justify-content-end': byUs,
      'pr-4': byUs,
      'pl-4': !byUs
    }),
    style: {
      minHeight: '10px',
      width: '40%'
    },
    "data-qa": "info-loading-state"
  });
};

var MessageBodyLoadingState = function MessageBodyLoadingState(_ref2) {
  var byUs = _ref2.byUs;
  return /*#__PURE__*/React.createElement("div", {
    className: cn('align-items-center bg-gray-lightest mb-2 p-4', {
      'by-guest': !byUs
    }, {
      'by-us': byUs
    }),
    "data-qa": "message-body-loading-state"
  });
};

var ThreadMessageIconLoadingState = function ThreadMessageIconLoadingState(_ref3) {
  var byUs = _ref3.byUs;
  return /*#__PURE__*/React.createElement("div", {
    className: cn('bg-gray-lightest', {
      'mr-3': !byUs,
      'ml-3': byUs
    }),
    style: {
      height: '40px',
      width: '40px',
      borderRadius: '50%'
    },
    "data-qa": "thread-message-icon-loading-state"
  });
};

var ThreadMessageLoadingState = function ThreadMessageLoadingState(_ref4) {
  var byUs = _ref4.byUs,
      className = _ref4.className;
  return /*#__PURE__*/React.createElement("div", {
    className: cn('threadMessageBubble d-flex', {
      'flex-row-reverse by-us': byUs
    }, className),
    "data-qa": "thread-message-loading-state"
  }, /*#__PURE__*/React.createElement(ThreadMessageIconLoadingState, {
    byUs: byUs
  }), /*#__PURE__*/React.createElement("div", {
    className: "d-flex flex-column flex-1"
  }, /*#__PURE__*/React.createElement(MessageBodyLoadingState, {
    byUs: byUs
  }), /*#__PURE__*/React.createElement(InfoLoadingState, {
    byUs: byUs
  })));
};

ThreadMessageLoadingState.defaultProps = {
  byUs: false,
  className: 'null'
};
ThreadMessageLoadingState.propTypes = {
  /**
   * Bool, decide the render side of the message,
   */
  byUs: PropTypes.bool,

  /**
   * className. extra classes
   */
  className: PropTypes.string
};
export default ThreadMessageLoadingState;
