// @guestyci/foundation/esm/legacy/TextDivider/TextDivider.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import Divider from "../Divider/Divider";

var TextDivider = function TextDivider(_ref) {
  var children = _ref.children,
      style = _ref.style,
      textStyle = _ref.textStyle,
      color = _ref.color,
      textColor = _ref.textColor,
      width = _ref.width,
      className = _ref.className,
      textClassName = _ref.textClassName;
  return /*#__PURE__*/React.createElement("div", {
    className: cn('flex-center w-100', className),
    "data-qa": "text-divider"
  }, /*#__PURE__*/React.createElement(Divider, {
    className: "flex-1",
    width: width,
    style: style,
    color: color
  }), /*#__PURE__*/React.createElement("span", {
    className: cn('nowrap m-2', _defineProperty({}, "text-".concat(textColor), textColor), textClassName),
    style: textStyle
  }, children), /*#__PURE__*/React.createElement(Divider, {
    className: "flex-1",
    width: width,
    style: style,
    color: color
  }));
};

TextDivider.defaultProps = {
  color: 'gray-light',
  textColor: 'muted',
  className: '',
  textStyle: {},
  style: {}
};
TextDivider.propTypes = {
  /** Set divider line color */
  color: PropTypes.string,

  /** Set divider text color */
  textColor: PropTypes.string,

  /** Set additional class names to divider */
  className: PropTypes.string,

  /** Set style to divider text */
  textStyle: PropTypes.shape(),

  /** Set style to divider line */
  style: PropTypes.shape(),

  /** The divider message body */
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired
};
export default TextDivider;
