// @guestyci/foundation/es/AddButton/AddButton.js
import React from 'react';
import PropTypes from 'prop-types';

var IcoPlusCircle = function IcoPlusCircle(props) {
  return /*#__PURE__*/React.createElement("svg", props, /*#__PURE__*/React.createElement("g", {
    fill: "#206CFF",
    fillRule: "evenodd"
  }, /*#__PURE__*/React.createElement("ellipse", {
    cx: "6.491",
    cy: "6.539",
    opacity: ".114",
    rx: "6.491",
    ry: "6.539"
  }), /*#__PURE__*/React.createElement("path", {
    d: "M6.978 7.036l2.33.005c.275 0 .496-.22.495-.495a.497.497 0 0 0-.497-.497L6.98 6.043l.004-2.329a.492.492 0 0 0-.495-.495.497.497 0 0 0-.497.497l-.006 2.325-2.333-.004a.494.494 0 0 0-.495.495c0 .274.222.496.497.497l2.329.005-.003 2.336a.492.492 0 0 0 .495.495.5.5 0 0 0 .498-.498l.002-2.33z"
  })));
};

IcoPlusCircle.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  width: "13",
  height: "14",
  viewBox: "0 0 13 14"
};
import cn from 'classnames';
import createStyles from "../createStyles";
import Button from "../Button";
import Row from "../Layout/Row";
import Icon from "../Icon";
import TextField from "../TextField";
const useStyles = createStyles(theme => ({
  root: {
    color: theme.palette.text.link,
    padding: theme.spacer(1),
    transition: theme.transition.create('color'),
    '&:hover': {
      color: theme.palette.primary.hover
    }
  }
}));
const AddButton = React.forwardRef(function AddButton({
  onClick,
  text,
  className,
  jss
}, ref) {
  const classes = useStyles({
    jss
  });
  return /*#__PURE__*/React.createElement(Button, {
    ref: ref,
    className: cn(classes.root, className),
    onClick: onClick
  }, /*#__PURE__*/React.createElement(Row, {
    spacing: 1,
    align: "center"
  }, /*#__PURE__*/React.createElement(Icon, {
    svg: IcoPlusCircle,
    width: 14,
    height: 14
  }), /*#__PURE__*/React.createElement(TextField, null, text)));
});
AddButton.displayName = 'AddButton';
AddButton.propTypes = {
  onClick: PropTypes.func,
  text: PropTypes.string,
  className: PropTypes.string,
  jss: PropTypes.oneOfType([PropTypes.shape(), PropTypes.func])
};
export default AddButton;
