// @guestyci/foundation/legacy/RadioGroup/RadioGroup.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.RadioGroupComponent = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _withSelectionGroup = _interopRequireDefault(require("../baseSelectionControl/withSelectionGroup"));

function _createSuper(Derived) { return function () { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

var RadioGroup = /*#__PURE__*/function (_Component) {
  (0, _inherits2["default"])(RadioGroup, _Component);

  var _super = _createSuper(RadioGroup);

  function RadioGroup() {
    var _this;

    (0, _classCallCheck2["default"])(this, RadioGroup);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _super.call.apply(_super, [this].concat(args));

    _this.changeRadio = function (value, childName, id, generatedChildId) {
      var _this$props = _this.props,
          onOptionClicked = _this$props.onOptionClicked,
          onChange = _this$props.onChange;
      onOptionClicked(generatedChildId);
      onChange(value, childName || id);
    };

    return _this;
  }

  (0, _createClass2["default"])(RadioGroup, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props2 = this.props,
          options = _this$props2.options,
          center = _this$props2.center,
          className = _this$props2.className,
          fill = _this$props2.fill,
          style = _this$props2.style,
          col = _this$props2.col;
      return /*#__PURE__*/_react["default"].createElement("div", {
        className: (0, _classnames["default"])('d-flex-fill', 'flex-column', className),
        style: style,
        "data-qa": "radio-group"
      }, /*#__PURE__*/_react["default"].createElement("div", {
        className: (0, _classnames["default"])('d-flex h-fill')
      }, /*#__PURE__*/_react["default"].createElement("div", {
        className: (0, _classnames["default"])('relative', {
          'w-100': fill
        })
      }, /*#__PURE__*/_react["default"].createElement("div", {
        className: (0, _classnames["default"])('radioGroup-wrapper d-flex text-gray-dark', {
          'flex-column': col
        }, {
          'flex-center': center
        }, {
          'flex-space-between-center': fill
        })
      }, Array.from(options.values(), function (obj) {
        return _react["default"].cloneElement(obj.child, {
          key: obj.uid,
          onChange: _this2.changeRadio,
          checked: obj.checked,
          id: obj.uid,
          generatedChildId: obj.uid,
          name: obj.child.props.name,
          value: obj.child.props.value,
          className: (0, _classnames["default"])(obj.child.props.className, {
            'height-45': col,
            'px-4': col,
            'py-3': col
          })
        });
      })))));
    }
  }]);
  return RadioGroup;
}(_react.Component);

RadioGroup.defaultProps = {
  center: false,
  fill: false,
  className: '',
  style: {},
  col: false,
  name: undefined,
  id: undefined
};
RadioGroup.propTypes = {
  /** The radioGroup children to render */
  children: _propTypes["default"].node.isRequired,

  /** Callback for on radio change */
  onChange: _propTypes["default"].func.isRequired,

  /** Indicator whether or not to align radioGroup to center or space between */
  center: _propTypes["default"].bool,

  /** Indicator whether or not to make radioGroup space between */
  fill: _propTypes["default"].bool,

  /** Additional class to radioGroup root */
  className: _propTypes["default"].string,

  /** Additional style to radioGroup root */
  style: _propTypes["default"].shape(),

  /** Select view mode: column view if true  */
  col: _propTypes["default"].bool,

  /** value will be return when click on this item */
  name: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].number]),

  /** Identifier will be return when click on this element */
  id: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].number])
};
var RadioGroupComponent = RadioGroup;
exports.RadioGroupComponent = RadioGroupComponent;

var _default = (0, _withSelectionGroup["default"])(RadioGroup);

exports["default"] = _default;
