// @guestyci/foundation/esm/legacy/Calendar/CalendarTooltipReserved.js
import React from 'react';
import PropTypes from 'prop-types';
import CalendarTooltipReservedItem from "./CalendarTooltipReservedItem";
import { getCurrencySymbol, colorFromStatus } from "./calendarHelper";

var CalendarTooltipReserved = function CalendarTooltipReserved(_ref) {
  var data = _ref.data;
  return /*#__PURE__*/React.createElement("div", {
    className: "flex-start d-flex-fill text-white flex-column",
    "data-qa": "calendar-tooltip-reserved"
  }, /*#__PURE__*/React.createElement("div", {
    className: "text-left font-weight-bold bb-2 w-100 border-black-dark pb-2 mb-1"
  }, data.title), /*#__PURE__*/React.createElement(CalendarTooltipReservedItem, {
    title: "Status",
    value: data.status,
    color: colorFromStatus(data.status)
  }), /*#__PURE__*/React.createElement(CalendarTooltipReservedItem, {
    title: "ETA",
    value: data.plannedArrival,
    alert: data.isValidCheckIn
  }), /*#__PURE__*/React.createElement(CalendarTooltipReservedItem, {
    title: "ETD",
    value: data.plannedDeparture,
    alert: data.isValidCheckOut
  }), /*#__PURE__*/React.createElement(CalendarTooltipReservedItem, {
    title: "Guests",
    value: data.guestsCount
  }), data.money && /*#__PURE__*/React.createElement(CalendarTooltipReservedItem, {
    title: "Payout",
    value: getCurrencySymbol(data.money.currency, data.money.hostPayout)
  }));
};

CalendarTooltipReserved.propTypes = {
  data: PropTypes.shape({
    title: PropTypes.string.isRequired,
    status: PropTypes.string,
    plannedArrival: PropTypes.string.isRequired,
    plannedDeparture: PropTypes.string.isRequired,
    isValidCheckIn: PropTypes.bool,
    isValidCheckOut: PropTypes.bool,
    guestsCount: PropTypes.number,
    money: PropTypes.shape()
  }).isRequired
};
export default CalendarTooltipReserved;
