// @guestyci/foundation/esm/legacy/Calendar/CalendarTooltip.js
import React from 'react';
import PropTypes from 'prop-types';
import ListItem from "../List/ListItem";
import CalendarTooltipNote from "./CalendarTooltipNote";
import CalendarTooltipPrice from "./CalendarTooltipPrice";
import List from "../List/List";
import CalendarTooltipBlock from "./CalendarTooltipBlock";
import CalendarTooltipReserved from "./CalendarTooltipReserved";

var CalendarTooltip = function CalendarTooltip(_ref) {
  var price = _ref.price,
      note = _ref.note,
      blocks = _ref.blocks,
      isReserved = _ref.isReserved,
      data = _ref.data;
  return /*#__PURE__*/React.createElement(List, {
    className: "calendar-tooltip-list",
    "data-qa": "calendar-tooltip"
  }, !isReserved && note && /*#__PURE__*/React.createElement(ListItem, {
    dense: true,
    className: "calendar-tooltip-item"
  }, /*#__PURE__*/React.createElement(CalendarTooltipNote, {
    note: note
  })), isReserved && /*#__PURE__*/React.createElement(CalendarTooltipReserved, {
    data: data
  }), !isReserved && blocks.map(function (block) {
    return /*#__PURE__*/React.createElement(ListItem, {
      key: block,
      dense: true,
      className: "calendar-tooltip-item"
    }, /*#__PURE__*/React.createElement(CalendarTooltipBlock, {
      block: block
    }));
  }), !isReserved && price && /*#__PURE__*/React.createElement(ListItem, {
    dense: true,
    className: "calendar-tooltip-item"
  }, /*#__PURE__*/React.createElement(CalendarTooltipPrice, {
    price: price
  })));
};

CalendarTooltip.defaultProps = {
  price: undefined,
  note: undefined,
  blocks: [],
  isReserved: false,
  data: {}
};
CalendarTooltip.propTypes = {
  price: PropTypes.string,
  note: PropTypes.string,
  blocks: PropTypes.arrayOf(PropTypes.string),
  isReserved: PropTypes.bool,
  data: PropTypes.shape()
};
export default CalendarTooltip;
