// @guestyci/foundation/esm/legacy/Calendar/CalendarDayLGBody.js
import React from 'react';
import Bullet from "../Bullet/Bullet";
import Person from "../Person/Person";

var CalendarDayLGBody = function CalendarDayLGBody(_ref) {
  var day = _ref.day,
      isHovering = _ref.isHovering,
      hasNote = _ref.hasNote,
      guest = _ref.guest,
      startOfBlock = _ref.startOfBlock,
      opacity = _ref.personOpacity,
      price = _ref.price,
      showAvatar = _ref.showAvatar;
  return /*#__PURE__*/React.createElement("div", {
    className: "d-flex-fill flex-space-between-center h-fill flex-column pt-1",
    "data-qa": "calendar-day-lg-body"
  }, /*#__PURE__*/React.createElement("div", {
    className: "w-100 flex-end-base h-fill pr-3"
  }, hasNote && /*#__PURE__*/React.createElement(Bullet, {
    className: "calendar-day-bullet",
    color: isHovering ? 'blue' : 'muted',
    style: {
      height: 6,
      width: 6
    }
  }), /*#__PURE__*/React.createElement("div", {
    className: "day pl-1"
  }, day)), showAvatar && /*#__PURE__*/React.createElement("div", {
    className: "w-100 pl-7 pb-1 flex-start-center"
  }, startOfBlock && guest && /*#__PURE__*/React.createElement(Person, {
    style: {
      opacity: opacity
    },
    isUser: true,
    name: guest.name,
    img: guest.avatar,
    size: "sm"
  })), /*#__PURE__*/React.createElement("div", {
    className: "w-100 flex-end pr-3"
  }, price && /*#__PURE__*/React.createElement("div", null, price)));
};

export default CalendarDayLGBody;
