// @guestyci/foundation/esm/legacy/Calendar/CalendarWeekDayBar.js
import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';

var CalendarWeekDayBar = function CalendarWeekDayBar(_ref) {
  var weekday = _ref.weekday,
      locale = _ref.locale,
      localeUtils = _ref.localeUtils,
      className = _ref.className,
      style = _ref.style,
      dayCharCount = _ref.dayCharCount;
  var weekdayName = localeUtils.formatWeekdayLong(weekday, locale);
  return /*#__PURE__*/React.createElement("div", {
    className: cn('text-muted font-size-md text-capitalize pt-0 pb-2', className),
    style: style,
    "data-qa": "calendar-week-day-bar"
  }, weekdayName.slice(0, dayCharCount));
};

CalendarWeekDayBar.defaultProps = {
  className: '',
  style: {},
  weekday: 0,
  localeUtils: {},
  locale: '',
  dayCharCount: 3
};
CalendarWeekDayBar.propTypes = {
  className: PropTypes.string,
  style: PropTypes.shape(),
  weekday: PropTypes.number,
  localeUtils: PropTypes.shape(),
  locale: PropTypes.string,
  dayCharCount: PropTypes.number
};
export default CalendarWeekDayBar;
