// @guestyci/foundation/PhonePicker/CountryFlagButton.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.useStyles = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _createStyles = _interopRequireDefault(require("../createStyles"));

var _Button = _interopRequireDefault(require("../Button"));

var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    root: {
      height: '100%',
      width: 46,
      minWidth: 46,
      borderRadius: 0,
      transition: theme.transition.create(),
      fill: theme.palette.secondary["default"],
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      borderRight: "solid 1px ".concat(theme.palette.border)
    },
    pickerButton: {
      width: '25px',
      height: '25px'
    }
  };
});
exports.useStyles = useStyles;

var CountryFlagButton = _react["default"].forwardRef(function CountryFlagButton(_ref, ref) {
  var CountryIcon = _ref.CountryIcon,
      disabled = _ref.disabled,
      style = _ref.style,
      onClick = _ref.onClick,
      className = _ref.className,
      jss = _ref.jss,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["CountryIcon", "disabled", "style", "onClick", "className", "jss"]);
  var classes = useStyles({
    jss: jss
  });
  return /*#__PURE__*/_react["default"].createElement(_Button["default"], (0, _extends2["default"])({}, props, {
    style: style,
    component: "span",
    ref: ref,
    enableRipple: false,
    className: (0, _classnames["default"])(classes.root, className),
    disabled: disabled,
    size: "xs",
    onClick: onClick
  }), /*#__PURE__*/_react["default"].createElement(CountryIcon, {
    className: classes.pickerButton
  }));
});

CountryFlagButton.propTypes = {
  /** The country Icon SVG */
  CountryIcon: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func]),

  /** Cb for onClick */
  onClick: _propTypes["default"].func,

  /** Additional class to the top component level */
  className: _propTypes["default"].string,

  /** Additional style to the top component level */
  style: _propTypes["default"].shape(),

  /** Boolean disable/enable component */
  disabled: _propTypes["default"].bool,

  /** jss override object to customize the jss classes */
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func])
};
var _default = CountryFlagButton;
exports["default"] = _default;
