// @guestyci/foundation/PhonePicker/PhonePickerDropdownItem.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.useStyles = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _createStyles = _interopRequireDefault(require("../createStyles"));

var _TextField = _interopRequireDefault(require("../TextField"));

var _Row = _interopRequireDefault(require("../Layout/Row"));

var flags = _interopRequireWildcard(require("./flags"));

var useStyles = (0, _createStyles["default"])(function () {
  return {
    root: {
      height: '25px',
      width: '25px'
    }
  };
});
exports.useStyles = useStyles;

var PhonePickerDropdownItem = function PhonePickerDropdownItem(_ref) {
  var jss = _ref.jss,
      country = _ref.country;
  var classes = useStyles({
    jss: jss
  });
  var name = country.name;
  var CountryIcon = flags[country.iso2.toUpperCase()];
  return /*#__PURE__*/_react["default"].createElement(_Row["default"], {
    spacing: 2,
    align: "center",
    "data-qa": "phone-picker-dropdown-item"
  }, /*#__PURE__*/_react["default"].createElement(CountryIcon, {
    className: classes.root
  }), /*#__PURE__*/_react["default"].createElement(_TextField["default"], null, name));
};

PhonePickerDropdownItem.propTypes = {
  country: _propTypes["default"].PropTypes.shape()
};
var _default = PhonePickerDropdownItem;
exports["default"] = _default;
