// @guestyci/foundation/es/legacy/Fade/Fade.js
import _objectWithoutPropertiesLoose from "@babel/runtime/helpers/esm/objectWithoutPropertiesLoose";
import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import CSSTransition from "../CSSTransiton";

const Fade = (_ref) => {
  let {
    className,
    timeout,
    children,
    tag: Tag,
    show,
    speed,
    mountOnEnter
  } = _ref,
      rest = _objectWithoutPropertiesLoose(_ref, ["className", "timeout", "children", "tag", "show", "speed", "mountOnEnter"]);

  return /*#__PURE__*/React.createElement(CSSTransition, {
    show: !!show,
    animation: `css-transition-fade-${speed}`,
    className: cn(!show ? 'opacity-0' : '', 'css-transition-fade', className),
    timeout: timeout,
    unmountOnExit: mountOnEnter,
    mountOnEnter: mountOnEnter,
    "data-qa": "fade"
  }, /*#__PURE__*/React.createElement(Tag, rest, children));
};

Fade.defaultProps = {
  className: '',
  tag: 'span',
  speed: 'reg',
  children: null,
  timeout: 225,
  mountOnEnter: false
};
Fade.propTypes = {
  /** Bool whether or not to show (fade in) children */
  show: PropTypes.bool,

  /** Additional class for root level */
  className: PropTypes.string,

  /** children to render for the fade component */
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.arrayOf(PropTypes.node)]),

  /** Timeout for fade animation */
  timeout: PropTypes.oneOfType([PropTypes.shape({
    enter: PropTypes.number.isRequired,
    exit: PropTypes.number.isRequired
  }), PropTypes.number]),

  /** Tag to render for fade */
  tag: PropTypes.string,

  /** Custom fade speed */
  speed: PropTypes.oneOf(['quick', 'reg', 'slow']),

  /** Mount and unmount on show/hide (true) or keep mounted -default */
  mountOnEnter: PropTypes.bool
};
export default Fade;
