// @guestyci/foundation/Dialog/DialogHeader.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _TextField = _interopRequireDefault(require("../TextField"));

var _Section = _interopRequireDefault(require("../Layout/Section"));

var _createStyles = _interopRequireDefault(require("../createStyles"));

var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    text: {
      color: theme.palette.title
    }
  };
}, {
  name: 'DialogHeader'
});

var DialogHeader = function DialogHeader(_ref) {
  var children = _ref.children,
      _ref$jss = _ref.jss,
      jss = _ref$jss === void 0 ? {} : _ref$jss,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["children", "jss"]);
  var classes = useStyles({
    jss: jss
  });
  return /*#__PURE__*/_react["default"].createElement(_Section["default"], (0, _extends2["default"])({
    gutter: "6 8 0 6"
  }, props, {
    "data-qa": "dialog-header"
  }), /*#__PURE__*/_react["default"].createElement(_TextField["default"], {
    variant: "h2",
    className: classes.text
  }, children));
};

DialogHeader.propTypes = {
  /** Child elements to render in header */
  children: _propTypes["default"].node.isRequired,

  /** JSS override object  */
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func])
};
var _default = DialogHeader;
exports["default"] = _default;
