// @guestyci/foundation/Dialog/DialogContent.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Section = _interopRequireDefault(require("../Layout/Section"));

var _TextField = _interopRequireDefault(require("../TextField"));

var _createStyles = _interopRequireDefault(require("../createStyles"));

var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    text: {
      color: theme.palette.text.secondary
    }
  };
}, {
  name: 'DialogContent'
});

var DialogContent = function DialogContent(_ref) {
  var children = _ref.children,
      _ref$jss = _ref.jss,
      jss = _ref$jss === void 0 ? {} : _ref$jss,
      className = _ref.className,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["children", "jss", "className"]);
  var classes = useStyles({
    jss: jss
  });
  return /*#__PURE__*/_react["default"].createElement(_Section["default"], (0, _extends2["default"])({
    gutter: "6 8 6 6",
    className: className
  }, props, {
    "data-qa": "dialog-content"
  }), /*#__PURE__*/_react["default"].createElement(_TextField["default"], {
    className: classes.text
  }, children));
};

DialogContent.propTypes = {
  /** Child elements to render in header */
  children: _propTypes["default"].node.isRequired,

  /** JSS override object */
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func]),

  /** Additional classname to root component */
  className: _propTypes["default"].string
};
var _default = DialogContent;
exports["default"] = _default;
