// @guestyci/foundation/Dialog/Dialog.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.useStyles = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _noop = _interopRequireDefault(require("lodash/noop"));

var _classnames = _interopRequireDefault(require("classnames"));

var _useLockBody = _interopRequireDefault(require("../useLockBody"));

var _Paper = _interopRequireDefault(require("../Paper"));

var _Portal = _interopRequireDefault(require("../legacy/Portal"));

var _CSSTransition = _interopRequireDefault(require("../CSSTransition"));

var _Backdrop = _interopRequireDefault(require("../Backdrop"));

var _createStyles = _interopRequireDefault(require("../createStyles"));

var _colors = require("../theme/colors");

var _Fade = _interopRequireDefault(require("../Fade"));

var _IconButton = _interopRequireDefault(require("../IconButton"));

var BtnDelete = function BtnDelete(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M8.485.707L7.778 0 4.243 3.536.707 0 0 .707l3.536 3.536L0 7.778l.707.707L4.243 4.95l3.535 3.535.707-.707L4.95 4.243z"
  }));
};

BtnDelete.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 8.4854 8.4854"
};
var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    root: {
      position: 'fixed',
      top: 0,
      bottom: 0,
      left: 0,
      right: 0,
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      height: '100%',
      overflowY: 'auto',
      zIndex: theme.zIndex.dialog
    },
    dialog: {
      margin: 'auto'
    },
    content: {
      position: 'relative',
      border: "1px solid ".concat(theme.palette.border),
      minWidth: 500,
      backgroundColor: _colors.white,
      borderRadius: 2
    },
    closeButton: {
      position: 'absolute',
      top: 15,
      right: 25,
      color: theme.palette.text.secondary,
      '&:hover': {
        color: theme.palette.text["default"]
      }
    },
    enter: {
      opacity: 0.01,
      transform: 'scale(0)'
    },
    enterActive: {
      transform: 'scale(1)',
      opacity: 1,
      transition: theme.transition.create('transform', {
        duration: 200
      })
    },
    enterDone: {
      transform: 'scale(1)',
      opacity: 1
    },
    exit: {
      opacity: 1,
      transform: 'scale(1)'
    },
    exitDone: {
      opacity: 0,
      transform: 'scale(0)'
    },
    exitActive: {
      transform: 'scale(0)',
      opacity: 0.01,
      transition: theme.transition.create('transform', {
        duration: 200
      })
    }
  };
}, {
  name: 'Dialog'
});
exports.useStyles = useStyles;

var Dialog = _react["default"].forwardRef(function Dialog(_ref, ref) {
  var _ref$open = _ref.open,
      open = _ref$open === void 0 ? false : _ref$open,
      _ref$className = _ref.className,
      className = _ref$className === void 0 ? '' : _ref$className,
      _ref$children = _ref.children,
      children = _ref$children === void 0 ? null : _ref$children,
      _ref$style = _ref.style,
      style = _ref$style === void 0 ? {} : _ref$style,
      _ref$jss = _ref.jss,
      jss = _ref$jss === void 0 ? {} : _ref$jss,
      _ref$onClose = _ref.onClose,
      onClose = _ref$onClose === void 0 ? _noop["default"] : _ref$onClose,
      transition = _ref.transition,
      _ref$noClear = _ref.noClear,
      noClear = _ref$noClear === void 0 ? false : _ref$noClear,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["open", "className", "children", "style", "jss", "onClose", "transition", "noClear"]);
  var classes = useStyles({
    jss: jss
  });

  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      show = _useState2[0],
      setShow = _useState2[1];

  var dialogTransition = transition || {
    enter: classes.enter,
    enterActive: classes.enterActive,
    enterDone: classes.enterDone,
    exit: classes.exit,
    exitActive: classes.exitActive,
    exitDone: classes.exitDone
  };
  (0, _useLockBody["default"])(open);
  (0, _react.useEffect)(function () {
    setTimeout(function () {
      return setShow(open);
    }, 50);
  }, [open]);
  return /*#__PURE__*/_react["default"].createElement(_Portal["default"], null, /*#__PURE__*/_react["default"].createElement(_Fade["default"], {
    duration: 150,
    show: open,
    unmountOnExit: true
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: classes.root
  }, /*#__PURE__*/_react["default"].createElement(_Backdrop["default"], {
    onClick: onClose
  }), /*#__PURE__*/_react["default"].createElement(_CSSTransition["default"], {
    animation: dialogTransition,
    show: show,
    unmountOnExit: true
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: classes.dialog
  }, /*#__PURE__*/_react["default"].createElement(_Paper["default"], (0, _extends2["default"])({
    ref: ref,
    elevation: 2,
    className: (0, _classnames["default"])(classes.content, className),
    style: style
  }, props), /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, !noClear && /*#__PURE__*/_react["default"].createElement(_IconButton["default"], {
    className: classes.closeButton,
    size: "xs",
    variant: "secondary",
    position: "suffix",
    onClick: onClose
  }, /*#__PURE__*/_react["default"].createElement(BtnDelete, {
    width: 12,
    height: 12
  })), children)))))));
});

Dialog.propTypes = {
  /** Child elements to render inside of the Dialog */
  children: _propTypes["default"].node.isRequired,

  /** Boolean flag whether or not to show dialog */
  open: _propTypes["default"].bool,

  /** Custom transition to pass to dialog */
  transition: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].shape()]),

  /** Custom jss overrides */
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func]),

  /** Callback function for closing dialog */
  onClose: _propTypes["default"].func,

  /** Additional classes for the paper level */
  className: _propTypes["default"].string,

  /** Boolean flat to show/hide x button */
  noClear: _propTypes["default"].bool
};
Dialog.displayName = 'Dialog';
var _default = Dialog;
exports["default"] = _default;
