// @guestyci/foundation/Dialog/DialogFooter.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Section = _interopRequireDefault(require("../Layout/Section"));

var _Row = _interopRequireDefault(require("../Layout/Row"));

var DialogFooter = function DialogFooter(_ref) {
  var children = _ref.children,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["children"]);
  return /*#__PURE__*/_react["default"].createElement(_Section["default"], (0, _extends2["default"])({
    gutter: "0 8 6 6"
  }, props, {
    "data-qa": "dialog-footer"
  }), /*#__PURE__*/_react["default"].createElement(_Row["default"], {
    spacing: 3,
    align: "center",
    justify: "end",
    fullWidth: true,
    wrap: false
  }, children));
};

DialogFooter.propTypes = {
  /** Child elements to render in header */
  children: _propTypes["default"].node.isRequired
};
var _default = DialogFooter;
exports["default"] = _default;
