// @guestyci/foundation/legacy/TelInput/TelInput.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.TelComponent = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _isObject = _interopRequireDefault(require("lodash/isObject"));

var _classnames = _interopRequireDefault(require("classnames"));

var _InputStarDeleteActionAddon = _interopRequireDefault(require("../baseInput/InputStarDeleteActionAddon"));

var _withHover = _interopRequireDefault(require("../../withHover"));

var _IntlTelInputWrapperLazy = _interopRequireDefault(require("./IntlTelInputWrapperLazy"));

function _createSuper(Derived) { return function () { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

var TelInput = /*#__PURE__*/function (_Component) {
  (0, _inherits2["default"])(TelInput, _Component);

  var _super = _createSuper(TelInput);

  function TelInput(_props) {
    var _this;

    (0, _classCallCheck2["default"])(this, TelInput);
    _this = _super.call(this, _props);

    _this.onChangePhoneNumber = function (isPhoneNumberValid, newValue, object, formatted) {
      if (_this.isComponentMounted) {
        var _this$props = _this.props,
            id = _this$props.id,
            name = _this$props.name,
            onChange = _this$props.onChange;

        _this.setState({
          value: newValue,
          formatted: formatted
        });

        onChange({
          value: formatted,
          isTelInput: true,
          isValid: isPhoneNumberValid
        }, id || name);
      }
    };

    _this.onSelectFlag = function (value, object, formatted, isPhoneNumberValid) {
      var _this$props2 = _this.props,
          id = _this$props2.id,
          name = _this$props2.name,
          onChange = _this$props2.onChange;
      onChange({
        value: formatted,
        isTelInput: true,
        isValid: isPhoneNumberValid
      }, id || name);
    };

    _this.getInitialValue = function (props) {
      if (props.defaultValue) {
        return (0, _isObject["default"])(props.defaultValue) ? props.defaultValue.value : props.defaultValue;
      }

      return (0, _isObject["default"])(props.value) ? props.value.value : props.value;
    };

    _this.customPlaceholder = function (placeholder) {
      return "e.g. ".concat(placeholder);
    };

    var value = _this.getInitialValue(_props);

    _this.state = {
      value: value,
      formatted: value
    };
    _this.isComponentMounted = false; // control onChange callback before component init

    _this.onIntlTelInputMount = _this.onIntlTelInputMount.bind((0, _assertThisInitialized2["default"])(_this));
    return _this;
  }

  (0, _createClass2["default"])(TelInput, [{
    key: "componentDidUpdate",
    value: function componentDidUpdate(prevProps) {
      var value = this.props.value;
      var _this$state = this.state,
          formatted = _this$state.formatted,
          stateValue = _this$state.value;
      var prevVal = !(0, _isObject["default"])(prevProps.value) ? prevProps.value : prevProps.value.value;
      var val = !(0, _isObject["default"])(value) ? value : value.value;

      if (val !== prevVal && val !== formatted && val !== stateValue) {
        this.setPhoneNumber(val);
      }
    }
  }, {
    key: "onIntlTelInputMount",
    value: function onIntlTelInputMount() {
      if (!this.isComponentMounted) {
        this.isComponentMounted = true;
      }
    }
  }, {
    key: "setPhoneNumber",
    value: function setPhoneNumber(val) {
      this.setState({
        formatted: val,
        value: val
      });
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props3 = this.props,
          className = _this$props3.className,
          style = _this$props3.style,
          withActions = _this$props3.withActions,
          onTrashClick = _this$props3.onTrashClick,
          onStarClick = _this$props3.onStarClick,
          isPrimary = _this$props3.isPrimary,
          inputClassName = _this$props3.inputClassName,
          onHoverEnter = _this$props3.onHoverEnter,
          onHoverLeave = _this$props3.onHoverLeave,
          isHovering = _this$props3.isHovering,
          disabled = _this$props3.disabled,
          name = _this$props3.name,
          id = _this$props3.id;
      var value = this.state.value;
      return /*#__PURE__*/_react["default"].createElement("div", {
        className: (0, _classnames["default"])('d-flex', 'action-input height-46', className),
        onMouseEnter: onHoverEnter,
        onMouseLeave: onHoverLeave,
        style: style,
        "data-qa": "tel-input"
      }, /*#__PURE__*/_react["default"].createElement(_IntlTelInputWrapperLazy["default"], {
        onIntlTelInputMount: this.onIntlTelInputMount,
        fieldId: id,
        fieldName: name,
        disabled: disabled,
        value: value,
        onSelectFlag: this.onSelectFlag,
        containerClassName: "intl-tel-input w-fill",
        inputClassName: (0, _classnames["default"])('flex-1 h-100', inputClassName, {
          'action-input': withActions
        }, 'form-control', 'w-fill'),
        onPhoneNumberChange: this.onChangePhoneNumber,
        formatOnInit: false,
        customPlaceholder: this.customPlaceholder,
        format: true
      }), /*#__PURE__*/_react["default"].createElement(_InputStarDeleteActionAddon["default"], {
        isPrimary: isPrimary,
        disabled: disabled,
        show: withActions && isHovering,
        onStar: onStarClick,
        onTrash: onTrashClick,
        name: name,
        id: id
      }));
    }
  }]);
  return TelInput;
}(_react.Component);

TelInput.defaultProps = {
  onChange: function onChange() {},
  onTrashClick: function onTrashClick() {},
  onStarClick: function onStarClick() {},
  onBlur: function onBlur() {},
  withActions: false,
  isPrimary: false,
  className: '',
  inputClassName: '',
  style: {},
  disabled: false,
  value: '',
  defaultValue: ''
};
TelInput.propTypes = {
  /** Cb for input change - params: {value: String, isValid: Boolean}, id|name {String | number} */
  onChange: _propTypes["default"].func,

  /**  Cb for clicking on the trash icon */
  onTrashClick: _propTypes["default"].func,

  /** Cb for clicking on the star (set primary) icon */
  onStarClick: _propTypes["default"].func,

  /** Indicator whether to show the star/trash actions */
  withActions: _propTypes["default"].bool,

  /** Boolean setting the email as primary */
  isPrimary: _propTypes["default"].bool,

  /** Additional class to the top component level */
  className: _propTypes["default"].string,

  /** Input level addition class */
  inputClassName: _propTypes["default"].string,

  /** Additional style to the top component level */
  style: _propTypes["default"].shape(),

  /** Boolean disable/enable component */
  disabled: _propTypes["default"].bool,

  /** on Blur call back function */
  onBlur: _propTypes["default"].func,

  /** value for controlled component */
  value: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].shape()]),

  /** default value for component */
  defaultValue: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].shape()])
};
/** Work around to fix storybook HOC propTypes bug  - DO NOT IMPORT THIS */

var TelComponent = TelInput;
exports.TelComponent = TelComponent;

var _default = (0, _withHover["default"])(TelInput);

exports["default"] = _default;
