// @guestyci/foundation/esm/legacy/Paper/Paper.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
var Paper = React.forwardRef(function (_ref, ref) {
  var className = _ref.className,
      bgColor = _ref.bgColor,
      style = _ref.style,
      elevation = _ref.elevation,
      borderRadius = _ref.borderRadius,
      children = _ref.children,
      props = _objectWithoutProperties(_ref, ["className", "bgColor", "style", "elevation", "borderRadius", "children"]);

  return /*#__PURE__*/React.createElement("div", _extends({
    ref: ref,
    className: cn(className, "box-shadow-".concat(elevation), "bg-".concat(bgColor), "border-radius-".concat(borderRadius)),
    style: style
  }, props), children);
});
Paper.defaultProps = {
  bgColor: 'white',
  elevation: 1,
  borderRadius: 0,
  className: '',
  style: {}
};
Paper.propTypes = {
  /** Paper inner content */
  children: PropTypes.node.isRequired,

  /** Paper background color */
  bgColor: PropTypes.string,

  /** Paper box shadow elevation 1-10 */
  elevation: PropTypes.number,

  /** Paper roundness  0 - 25 leaps of 5  */
  borderRadius: PropTypes.number,

  /** Paper additional classes */
  className: PropTypes.string,

  /** Paper additional styles */
  style: PropTypes.shape()
};
export default Paper;
