// @guestyci/foundation/es/legacy/Person/PersonReturningGuestPill.js
import React from 'react';
import SvgIcon from "../SvgIcon/SvgIcon";

var BtnStarOn = function BtnStarOn(props) {
  return /*#__PURE__*/React.createElement("svg", props, /*#__PURE__*/React.createElement("path", {
    d: "M7.5 0l2.319 4.608L15 5.347l-3.75 3.588.887 5.065-4.638-2.393L2.864 14l.886-5.065L0 5.347l5.183-.74z"
  }));
};

BtnStarOn.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 15 14"
};
import Pill from "../Pill/Pill";
import Row from "../Row/Row";
import TextField from "../TextField/TextField";

const PersonReturningGuestPill = ({
  className
}) => /*#__PURE__*/React.createElement(Pill, {
  color: "white",
  className: className,
  bgColor: "black",
  "data-qa": "person-returning-guest-pill"
}, /*#__PURE__*/React.createElement(Row, {
  spacing: "sm",
  align: "center"
}, /*#__PURE__*/React.createElement(SvgIcon, {
  svg: BtnStarOn,
  color: "white",
  width: 12,
  height: 12,
  style: {
    marginBottom: 2
  }
}), /*#__PURE__*/React.createElement(TextField, {
  size: "md"
}, "Returning guest")));

export default PersonReturningGuestPill;
