// @guestyci/history/NavMenu/NavMenuItem.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _noop = _interopRequireDefault(require("lodash/noop"));

var _classnames = _interopRequireDefault(require("classnames"));

var _createStyles = _interopRequireDefault(require("@guestyci/foundation/createStyles"));

var _Icon = _interopRequireDefault(require("@guestyci/foundation/Icon"));

var _List = require("@guestyci/foundation/List");

var _Layout = require("@guestyci/foundation/Layout");

var _Divider = _interopRequireDefault(require("@guestyci/foundation/Divider"));

var _TextField = _interopRequireDefault(require("@guestyci/foundation/TextField"));

var _colors = require("@guestyci/foundation/theme/colors");

var _Link = _interopRequireDefault(require("../Link"));

var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    root: {
      letterSpacing: '0.4px',
      '&:hover $label': {
        color: theme.palette.link,
        fontWeight: theme.typography.fontWeight.bold,
        letterSpacing: '0.1px !important'
      }
    },
    divider: {
      borderColor: theme.palette.brand.primary,
      height: 'auto'
    },
    content: {},
    label: {},
    iconContainer: {
      minWidth: '24px',
      height: '24px',
      width: '24px',
      marginRight: theme.spacer(2)
    },
    icon: {
      fill: _colors.gray400
    },
    overflowHidden: {
      overflow: 'hidden'
    }
  };
});

var listItemJssOverrides = function listItemJssOverrides(theme) {
  return {
    selected: {
      color: theme.palette.brand.primary,
      backgroundColor: _colors.gray200,
      letterSpacing: '0.1px !important',
      borderRadius: 0
    }
  };
};

var MenuItem = function MenuItem(_ref) {
  var children = _ref.children,
      _ref$to = _ref.to,
      to = _ref$to === void 0 ? '' : _ref$to,
      _ref$onClick = _ref.onClick,
      onClick = _ref$onClick === void 0 ? _noop["default"] : _ref$onClick,
      _ref$selected = _ref.selected,
      selected = _ref$selected === void 0 ? false : _ref$selected,
      icon = _ref.icon,
      className = _ref.className,
      _ref$jss = _ref.jss,
      jss = _ref$jss === void 0 ? {} : _ref$jss;
  var classes = useStyles({
    jss: jss
  });
  return /*#__PURE__*/_react["default"].createElement(_Link["default"], {
    to: to
  }, /*#__PURE__*/_react["default"].createElement(_Layout.Row, {
    className: (0, _classnames["default"])(classes.root, className),
    justify: "between"
  }, /*#__PURE__*/_react["default"].createElement(_List.ListItem, {
    variant: "button",
    value: to,
    selected: selected,
    onClick: onClick,
    className: (0, _classnames["default"])(classes.overflowHidden, classes.content),
    jss: listItemJssOverrides
  }, icon && /*#__PURE__*/_react["default"].createElement(_Layout.Section, {
    className: classes.iconContainer,
    justify: "center",
    align: "center"
  }, /*#__PURE__*/_react["default"].createElement(_Icon["default"], {
    svg: icon,
    className: classes.icon
  })), /*#__PURE__*/_react["default"].createElement(_TextField["default"], {
    overflowEllipsis: true,
    className: classes.label,
    bold: selected
  }, children)), /*#__PURE__*/_react["default"].createElement(_Divider["default"], {
    thickness: 3,
    orientation: "vertical",
    className: selected ? classes.divider : ''
  })));
};

MenuItem.propTypes = {
  /** children passed to the ListItem - string or Node */
  children: _propTypes["default"].oneOfType([_propTypes["default"].node, _propTypes["default"].string]).isRequired,

  /** A route value to be passed when clicked */
  to: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].shape()]),

  /** Function to be called when clicked */
  onClick: _propTypes["default"].func,

  /** Flag that marks the menu item as selected */
  selected: _propTypes["default"].bool,

  /** Icon to be shown to the left of the title */
  icon: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func]),

  /** Jss override to create custom styles */
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func]),

  /* Additional class name to be added to the MenuItem root */
  className: _propTypes["default"].string
};
var _default = MenuItem;
exports["default"] = _default;
