// @guestyci/history/NavMenu/NavMenuAccordion.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Accordion = _interopRequireDefault(require("@guestyci/foundation/Accordion"));

var _colors = require("@guestyci/foundation/theme/colors");

var accordionJssOverrides = function accordionJssOverrides(theme) {
  return {
    header: {
      paddingLeft: theme.spacer(4),
      paddingRight: theme.spacer(4),
      height: '45px',
      '.accordion-opened &': {
        color: theme.palette.title,
        fontWeight: theme.typography.fontWeight.bold
      },
      '.accordion-closed &': {
        color: _colors.gray800,
        fontWeight: theme.typography.fontWeight.regular
      },
      '&:hover $title': {
        color: theme.palette.link,
        fontWeight: theme.typography.fontWeight.bold,
        letterSpacing: '0.1px !important'
      }
    },
    icon: {
      fill: _colors.gray400
    },
    content: {
      padding: 0,
      backgroundColor: theme.palette.action.selected
    },
    title: {
      color: 'inherit',
      fontWeight: 'inherit',
      letterSpacing: '0.4px',
      '.accordion-opened &': {
        letterSpacing: '0.1px !important'
      }
    },
    iconContainer: {
      minWidth: '24px',
      height: '24px',
      width: '24px'
    },
    root: {
      padding: 0,
      display: 'flex',
      flexDirection: 'column'
    }
  };
};

var NavMenuAccordion = function NavMenuAccordion(_ref) {
  var children = _ref.children,
      title = _ref.title,
      icon = _ref.icon,
      onChange = _ref.onChange,
      open = _ref.open,
      className = _ref.className;
  return /*#__PURE__*/_react["default"].createElement(_Accordion["default"], {
    title: title,
    icon: icon,
    open: open,
    onChange: onChange,
    jss: accordionJssOverrides,
    className: className
  }, children);
};

NavMenuAccordion.propTypes = {
  /** children passed to the Accordion - string or Node */
  children: _propTypes["default"].oneOfType([_propTypes["default"].node, _propTypes["default"].string]).isRequired,

  /** title passed to the Accordion - string or Node */
  title: _propTypes["default"].oneOfType([_propTypes["default"].node, _propTypes["default"].string]).isRequired,

  /** Icon to be passed to Accordion */
  icon: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func]),

  /** Function to be called when clicked */
  onChange: _propTypes["default"].func,

  /** Flag that marks if the Accordion should be open */
  open: _propTypes["default"].bool,

  /* Additional class name to be added to the NavMenuAccordion */
  className: _propTypes["default"].string
};
var _default = NavMenuAccordion;
exports["default"] = _default;
