// @guestyci/foundation/legacy/TextDivider/TextDivider.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Divider = _interopRequireDefault(require("../Divider/Divider"));

var TextDivider = function TextDivider(_ref) {
  var children = _ref.children,
      style = _ref.style,
      textStyle = _ref.textStyle,
      color = _ref.color,
      textColor = _ref.textColor,
      width = _ref.width,
      className = _ref.className,
      textClassName = _ref.textClassName;
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])('flex-center w-100', className),
    "data-qa": "text-divider"
  }, /*#__PURE__*/_react["default"].createElement(_Divider["default"], {
    className: "flex-1",
    width: width,
    style: style,
    color: color
  }), /*#__PURE__*/_react["default"].createElement("span", {
    className: (0, _classnames["default"])('nowrap m-2', (0, _defineProperty2["default"])({}, "text-".concat(textColor), textColor), textClassName),
    style: textStyle
  }, children), /*#__PURE__*/_react["default"].createElement(_Divider["default"], {
    className: "flex-1",
    width: width,
    style: style,
    color: color
  }));
};

TextDivider.defaultProps = {
  color: 'gray-light',
  textColor: 'muted',
  className: '',
  textStyle: {},
  style: {}
};
TextDivider.propTypes = {
  /** Set divider line color */
  color: _propTypes["default"].string,

  /** Set divider text color */
  textColor: _propTypes["default"].string,

  /** Set additional class names to divider */
  className: _propTypes["default"].string,

  /** Set style to divider text */
  textStyle: _propTypes["default"].shape(),

  /** Set style to divider line */
  style: _propTypes["default"].shape(),

  /** The divider message body */
  children: _propTypes["default"].oneOfType([_propTypes["default"].arrayOf(_propTypes["default"].node), _propTypes["default"].node]).isRequired
};
var _default = TextDivider;
exports["default"] = _default;
