// @guestyci/foundation/legacy/Fade/Fade.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _CSSTransiton = _interopRequireDefault(require("../CSSTransiton"));

var Fade = function Fade(_ref) {
  var className = _ref.className,
      timeout = _ref.timeout,
      children = _ref.children,
      Tag = _ref.tag,
      show = _ref.show,
      speed = _ref.speed,
      mountOnEnter = _ref.mountOnEnter,
      rest = (0, _objectWithoutProperties2["default"])(_ref, ["className", "timeout", "children", "tag", "show", "speed", "mountOnEnter"]);
  return /*#__PURE__*/_react["default"].createElement(_CSSTransiton["default"], {
    show: !!show,
    animation: "css-transition-fade-".concat(speed),
    className: (0, _classnames["default"])(!show ? 'opacity-0' : '', 'css-transition-fade', className),
    timeout: timeout,
    unmountOnExit: mountOnEnter,
    mountOnEnter: mountOnEnter,
    "data-qa": "fade"
  }, /*#__PURE__*/_react["default"].createElement(Tag, rest, children));
};

Fade.defaultProps = {
  className: '',
  tag: 'span',
  speed: 'reg',
  children: null,
  timeout: 225,
  mountOnEnter: false
};
Fade.propTypes = {
  /** Bool whether or not to show (fade in) children */
  show: _propTypes["default"].bool,

  /** Additional class for root level */
  className: _propTypes["default"].string,

  /** children to render for the fade component */
  children: _propTypes["default"].oneOfType([_propTypes["default"].node, _propTypes["default"].arrayOf(_propTypes["default"].node)]),

  /** Timeout for fade animation */
  timeout: _propTypes["default"].oneOfType([_propTypes["default"].shape({
    enter: _propTypes["default"].number.isRequired,
    exit: _propTypes["default"].number.isRequired
  }), _propTypes["default"].number]),

  /** Tag to render for fade */
  tag: _propTypes["default"].string,

  /** Custom fade speed */
  speed: _propTypes["default"].oneOf(['quick', 'reg', 'slow']),

  /** Mount and unmount on show/hide (true) or keep mounted -default */
  mountOnEnter: _propTypes["default"].bool
};
var _default = Fade;
exports["default"] = _default;
