// @guestyci/history/esm/NavTabs/Tab.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import React, { useCallback, useMemo } from 'react';
import PropTypes from 'prop-types';
import isObject from 'lodash/isObject';
import cn from 'classnames';
import createStyles from '@guestyci/foundation/createStyles';
import Link from "../Link";
var useStyles = createStyles(function (theme) {
  return {
    root: {
      textTransform: 'uppercase',
      padding: 0,
      minHeight: 50,
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      '&:hover': {
        color: theme.palette.brand.primary
      }
    },
    active: {
      color: theme.palette.brand.primary,
      fontWeight: 'bold'
    }
  };
}, {
  name: 'Tab'
});

var Tab = function Tab(_ref) {
  var onClick = _ref.onClick,
      tabIndex = _ref.tabIndex,
      to = _ref.to,
      value = _ref.value,
      className = _ref.className,
      disabled = _ref.disabled,
      isActive = _ref.isActive,
      children = _ref.children;
  var classes = useStyles();
  var pathname = useMemo(function () {
    return isObject(to) ? to.pathname : to;
  }, [to]);
  var onTabClick = useCallback(function () {
    onClick({
      tabIndex: tabIndex,
      to: to,
      value: value
    });
  }, [onClick, tabIndex, to, value]);
  return /*#__PURE__*/React.createElement(Link, {
    "data-cy": isActive ? 'active' : '',
    href: pathname,
    className: cn('tab-item', classes.root, _defineProperty({}, classes.active, isActive), className),
    onClick: onTabClick,
    disabled: disabled
  }, children);
};

Tab.defaultProps = {
  children: null,
  disabled: false,
  to: undefined
};
Tab.propTypes = {
  /** Tab name */
  children: PropTypes.node,

  /** Indicator whether or not the tab is disabled */
  disabled: PropTypes.bool,

  /** The tab value to be passed when clicked */
  to: PropTypes.oneOfType([PropTypes.string, PropTypes.shape()]),

  /** value to pass on click  */
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.shape()])
};
export default Tab;
