// @guestyci/foundation/Person/ReturningGuestBadge.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _createStyles = _interopRequireDefault(require("../createStyles"));

var BtnStarOn = function BtnStarOn(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M7.5 0l2.319 4.608L15 5.347l-3.75 3.588.887 5.065-4.638-2.393L2.864 14l.886-5.065L0 5.347l5.183-.74z"
  }));
};

BtnStarOn.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 15 14"
};
var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    root: {
      height: 20,
      width: 20,
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      fill: 'white',
      backgroundColor: theme.palette.secondary["default"],
      borderRadius: '50%'
    }
  };
}, {
  name: 'ReturningGuestBadge'
});

var ReturningGuestBadge = function ReturningGuestBadge(_ref) {
  var _ref$className = _ref.className,
      className = _ref$className === void 0 ? '' : _ref$className,
      _ref$style = _ref.style,
      style = _ref$style === void 0 ? {} : _ref$style,
      _ref$jss = _ref.jss,
      jss = _ref$jss === void 0 ? {} : _ref$jss,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["className", "style", "jss"]);
  var classes = useStyles({
    jss: jss
  });
  return /*#__PURE__*/_react["default"].createElement("div", (0, _extends2["default"])({}, props, {
    className: (0, _classnames["default"])(classes.root, className),
    style: style,
    "data-qa": "returning-guest-badge"
  }), /*#__PURE__*/_react["default"].createElement(BtnStarOn, {
    height: 13,
    width: 13
  }));
};

ReturningGuestBadge.propTypes = {
  /** Additional class for badge */
  className: _propTypes["default"].string,

  /** Additional style for badge */
  style: _propTypes["default"].shape(),

  /** jss override object to customize the jss classes */
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func])
};
var _default = ReturningGuestBadge;
exports["default"] = _default;
