// @guestyci/foundation/withFocus/withFocus.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _react = _interopRequireWildcard(require("react"));

function _createSuper(Derived) { return function () { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

/**
 * Create a wrapper compose to support focus functionality
 * @param ComposedComponent
 * @returns {{Component}}
 */
function withFocus(ComposedComponent) {
  var FocusWrapper = /*#__PURE__*/function (_Component) {
    (0, _inherits2["default"])(FocusWrapper, _Component);

    var _super = _createSuper(FocusWrapper);

    function FocusWrapper(props) {
      var _this;

      (0, _classCallCheck2["default"])(this, FocusWrapper);
      _this = _super.call(this, props);

      _this.onBlur = function () {
        _this.setState({
          isFocus: false
        });
      };

      _this.onFocus = function () {
        _this.setState({
          isFocus: true
        });
      };

      _this.state = {
        isFocus: false
      };
      return _this;
    }

    (0, _createClass2["default"])(FocusWrapper, [{
      key: "render",
      value: function render() {
        var isFocus = this.state.isFocus;
        return /*#__PURE__*/_react["default"].createElement(ComposedComponent, (0, _extends2["default"])({}, this.props, {
          isFocused: isFocus,
          onBlur: this.onBlur,
          onFocus: this.onFocus,
          "data-qa": "focus-wrapper"
        }));
      }
    }]);
    return FocusWrapper;
  }(_react.Component);

  return FocusWrapper;
}

var _default = withFocus;
exports["default"] = _default;
