// @guestyci/foundation/legacy/Placeholder/Placeholder.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _reactJss = require("react-jss");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _colors = require("../../theme/colors");

var useStyles = (0, _reactJss.createUseStyles)({
  '@keyframes shimmer': {
    from: {
      backgroundPosition: -1000
    },
    to: {
      backgroundPosition: 1000
    }
  },
  placeholder: {
    backgroundColor: _colors.grayLighter,
    display: 'inline-block'
  },
  animate: {
    animation: '$shimmer 2s infinite linear',
    background: "linear-gradient(to right, ".concat(_colors.grayLighter, " 8%, ").concat(_colors.grayLight, " 18%, ").concat(_colors.grayLighter, " 33%);"),
    backgroundSize: 1000
  }
});

var Placeholder = function Placeholder(_ref) {
  var className = _ref.className,
      style = _ref.style,
      animate = _ref.animate;
  var classes = useStyles();
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])(className, classes.placeholder, (0, _defineProperty2["default"])({}, classes.animate, animate)),
    style: style,
    "data-qa": "placeholder"
  });
};

Placeholder.defaultProps = {
  className: '',
  style: {},
  animate: true
};
Placeholder.propTypes = {
  /** Additional classes to style your placeholder */
  className: _propTypes["default"].string,

  /** additional styles to pass to placeholder */
  style: _propTypes["default"].shape(),

  /** if placeholder should be animated */
  animate: _propTypes["default"].bool
};
var _default = Placeholder;
exports["default"] = _default;
