// @guestyci/foundation/legacy/Image/Image.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _cloudinaryReact = require("cloudinary-react");

var _constants = require("../../constants/constants");

var cloudinaryCropToObjectFitDict = {
  scale: 'fill',
  fit: 'contain',
  fill: 'cover',
  limit: 'scale-down',
  crop: 'none'
};

var isCloudinary = function isCloudinary(id, isImg) {
  return id || !isImg;
};

var ImageView = function ImageView(_ref) {
  var id = _ref.id,
      _ref$src = _ref.src,
      src = _ref$src === void 0 ? '' : _ref$src,
      children = _ref.children,
      crop = _ref.crop,
      width = _ref.width,
      height = _ref.height,
      radius = _ref.radius,
      responsive = _ref.responsive;
  var isNotCloudinary = (0, _react.useMemo)(function () {
    return src.startsWith('//') || src.startsWith('http');
  }, [src]);
  var source = id || src;
  return isCloudinary(id, isNotCloudinary) ? /*#__PURE__*/_react["default"].createElement(_cloudinaryReact.Image, {
    responsive: responsive,
    publicId: source,
    cloudName: _constants.CLOUD_NAME
  }, /*#__PURE__*/_react["default"].createElement(_cloudinaryReact.Transformation, {
    crop: crop,
    width: width,
    height: height,
    radius: radius
  }), children) : /*#__PURE__*/_react["default"].createElement("img", {
    alt: "",
    src: source,
    width: width,
    height: height,
    className: (0, _classnames["default"])({
      'w-fill': width === 'auto'
    }),
    style: {
      objectFit: cloudinaryCropToObjectFitDict[crop]
    }
  });
};

ImageView.defaultProps = {
  children: null,
  crop: 'scale',
  width: 300,
  height: 300,
  radius: 0,
  responsive: false
};

var requiredOnlyOnePropCheck = function requiredOnlyOnePropCheck(props, propName, componentName) {
  if (!props.src === !props.id) {
    return new Error("One and only one of 'id' or 'src' is required by '".concat(componentName, "' component."));
  }
};

ImageView.propTypes = {
  /** Image Id as stored in coudinary  - Deprecated use src */
  id: requiredOnlyOnePropCheck,

  /** Image source url */
  src: requiredOnlyOnePropCheck,

  /** Inner children for image component - usually used with Transfrom component via cloudinary */
  children: _propTypes["default"].oneOfType([_propTypes["default"].node, _propTypes["default"].arrayOf(_propTypes["default"].node)]),

  /** Crop type of image("scale"=objectFit.fill, "fit"=objectFit.contain, "fill":objectFit.cover, "limit":objectFit.scaleDown, "crop":objectFit.none) */
  crop: _propTypes["default"].string,

  /** Image width */
  width: _propTypes["default"].oneOfType([_propTypes["default"].number, _propTypes["default"].oneOf(['auto'])]),

  /** Image height */
  height: _propTypes["default"].number,

  /** Image border radius */
  radius: _propTypes["default"].number,

  /** Define if image's width will be responsive */
  responsive: _propTypes["default"].bool
};
var _default = ImageView;
exports["default"] = _default;
