// @guestyci/foundation/esm/legacy/Avatar/ImageAvatar.js
import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import { createUseStyles } from 'react-jss';
export var getClass = function getClass(size) {
  switch (size) {
    case 'xl':
      return '80px';

    case 'md':
      return '40px';

    case 'lg':
      return '60px';

    case 'sm':
      return '25px';

    default:
      return '40px';
  }
};
var useStyles = createUseStyles({
  icon: {
    height: function height(_ref) {
      var size = _ref.size;
      return getClass(size);
    },
    width: function width(_ref2) {
      var size = _ref2.size;
      return getClass(size);
    }
  }
});

var ImageAvatar = function ImageAvatar(_ref3) {
  var text = _ref3.text,
      src = _ref3.src,
      _ref3$size = _ref3.size,
      size = _ref3$size === void 0 ? 'md' : _ref3$size,
      bordered = _ref3.bordered,
      disabled = _ref3.disabled;
  var classes = useStyles({
    size: size
  });
  return /*#__PURE__*/React.createElement("img", {
    className: cn("img-avatar avatar", 'd-inline-block', {
      'avatar-border': bordered
    }, {
      disabled: disabled
    }, classes.icon),
    src: src,
    alt: text ? text.trim() : '',
    "data-qa": "image-avatar"
  });
};

ImageAvatar.propTypes = {
  src: PropTypes.string.isRequired,
  size: PropTypes.oneOf(['sm', 'md', 'lg', 'xl']),
  disabled: PropTypes.bool,
  text: PropTypes.string,
  bordered: PropTypes.bool
};
export default ImageAvatar;
