// @guestyci/foundation/legacy/ImageGallery/ImageGallery.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Carousel = _interopRequireDefault(require("../Carousel"));

var _Image = _interopRequireDefault(require("../Image"));

var _enums = require("../../enums/enums");

var ImageGallery = function ImageGallery(_ref) {
  var images = _ref.images,
      arrows = _ref.arrows,
      arrowTheme = _ref.arrowTheme,
      showArrowsOnHover = _ref.showArrowsOnHover,
      infinite = _ref.infinite,
      height = _ref.height,
      className = _ref.className,
      style = _ref.style,
      crop = _ref.crop;
  var imagesElements = (0, _react.useMemo)(function () {
    return images.map(function (image) {
      return /*#__PURE__*/_react["default"].createElement(_Image["default"], {
        responsive: true,
        src: image,
        key: image,
        crop: crop,
        width: "auto",
        height: height
      });
    });
  }, [images]);
  var settings = {
    arrows: arrows,
    infinite: infinite,
    showArrowsOnHover: showArrowsOnHover,
    className: className,
    style: style,
    arrowsFade: true,
    arrowTheme: arrowTheme
  };
  return /*#__PURE__*/_react["default"].createElement(_Carousel["default"], (0, _extends2["default"])({}, settings, {
    "data-qa": "image-gallery"
  }), imagesElements);
};

ImageGallery.defaultProps = {
  arrows: true,
  showArrowsOnHover: true,
  infinite: true,
  className: '',
  style: {},
  arrowTheme: _enums.Theme.Light,
  height: 300,
  crop: 'fill'
};
ImageGallery.propTypes = {
  /** list of images to show in the gallery */
  images: _propTypes["default"].arrayOf(_propTypes["default"].string).isRequired,

  /** indicates if to display arrows */
  arrows: _propTypes["default"].bool,

  /** indicate if to show the arrows when not hovering the carousel */
  showArrowsOnHover: _propTypes["default"].bool,

  /** Infinitely wrap around contents */
  infinite: _propTypes["default"].bool,

  /** Additional style */
  style: _propTypes["default"].shape(),

  /** Additional class name */
  className: _propTypes["default"].string,

  /** arrow's theme */
  arrowTheme: _propTypes["default"].oneOf([_enums.Theme.Light, _enums.Theme.Dark]),

  /** height of the gallery */
  height: _propTypes["default"].number,

  /** Crop type */
  crop: _propTypes["default"].string
};
var _default = ImageGallery;
exports["default"] = _default;
