// @guestyci/foundation/esm/withActive/withActive.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _possibleConstructorReturn from "@babel/runtime/helpers/esm/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/esm/getPrototypeOf";

function _createSuper(Derived) { return function () { var Super = _getPrototypeOf(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

import React, { Component } from 'react';
/**
 * Create a wrapper compose to support active functionality
 * @param ComposedComponent
 * @returns {{Component}}
 */

function withActive(ComposedComponent) {
  var ActiveWrapper = /*#__PURE__*/function (_Component) {
    _inherits(ActiveWrapper, _Component);

    var _super = _createSuper(ActiveWrapper);

    function ActiveWrapper(props) {
      var _this;

      _classCallCheck(this, ActiveWrapper);

      _this = _super.call(this, props);

      _this.activeOn = function () {
        _this.setState({
          isActive: true
        });
      };

      _this.activeOff = function () {
        _this.setState({
          isActive: false
        });
      };

      _this.toggleActive = function () {
        _this.setState(function (prevState) {
          return {
            isActive: !prevState.isActive
          };
        });
      };

      _this.state = {
        isActive: false
      };
      return _this;
    }

    _createClass(ActiveWrapper, [{
      key: "render",
      value: function render() {
        var isActive = this.state.isActive;
        return /*#__PURE__*/React.createElement(ComposedComponent, _extends({}, this.props, {
          isActive: isActive,
          toggleActive: this.toggleActive,
          activeOn: this.activeOn,
          activeOff: this.activeOff,
          "data-qa": "active-wrapper"
        }));
      }
    }]);

    return ActiveWrapper;
  }(Component);

  return ActiveWrapper;
}

export default withActive;
