// @guestyci/foundation/esm/legacy/NumberPicker/NumberPickerActionButtons.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { createUseStyles } from 'react-jss';
import ActionButton from "../ActionButton";

var BtnArrowUp = function BtnArrowUp(props) {
  return /*#__PURE__*/React.createElement("svg", props, /*#__PURE__*/React.createElement("path", {
    d: "M4.529 0a.474.474 0 0 0-.357.137L.137 4.172a.488.488 0 0 0 .69.689L4.53 1.156l3.705 3.705a.488.488 0 0 0 .69-.69L4.89.138A.477.477 0 0 0 4.53 0z"
  }));
};

BtnArrowUp.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 9.0618038 4.9977479"
};

var BtnArrowDown = function BtnArrowDown(props) {
  return /*#__PURE__*/React.createElement("svg", props, /*#__PURE__*/React.createElement("path", {
    d: "M4.533 4.997c.13.006.26-.039.357-.136L8.925.826a.488.488 0 0 0-.69-.69L4.532 3.843.826.137a.488.488 0 0 0-.69.69L4.172 4.86c.097.097.23.143.362.136z"
  }));
};

BtnArrowDown.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 9.0618038 4.9977479"
};
var useStyles = createUseStyles({
  actionButton: {
    width: 45,
    height: 23
  },
  btnTop: {
    borderRadius: '0px !important',
    borderTopRightRadius: '2px !important',
    borderWidth: '1px 1px 1px 0px !important'
  },
  btnBottom: {
    borderRadius: '0px !important',
    borderBottomRightRadius: '2px !important',
    borderWidth: '0px 1px 1px 0px !important'
  },
  btnAsLabelWrapper: {
    border: '0px'
  },
  btnAsLabel: {
    border: '0px',
    padding: '0px 5px',
    height: '15px'
  }
});

var NumberPickerActionButtons = function NumberPickerActionButtons(_ref) {
  var asLabel = _ref.asLabel,
      onIncrement = _ref.onIncrement,
      onDecrement = _ref.onDecrement,
      disableMax = _ref.disableMax,
      disableMin = _ref.disableMin;
  var classes = useStyles();
  return /*#__PURE__*/React.createElement("div", {
    className: cn('flex-center bl-1 border-gray-light flex-column', _defineProperty({}, classes.btnAsLabelWrapper, asLabel)),
    "data-qa": "number-picker-action-buttons"
  }, /*#__PURE__*/React.createElement(ActionButton, {
    onClick: onIncrement,
    iconWidth: 10,
    iconHeight: 10,
    disabled: disableMax,
    svg: BtnArrowUp,
    className: cn("".concat(classes.actionButton, " ").concat(classes.btnTop), _defineProperty({}, classes.btnAsLabel, asLabel))
  }), /*#__PURE__*/React.createElement(ActionButton, {
    onClick: onDecrement,
    iconWidth: 10,
    iconHeight: 10,
    disabled: disableMin,
    svg: BtnArrowDown,
    className: cn("".concat(classes.actionButton, " ").concat(classes.btnBottom), _defineProperty({}, classes.btnAsLabel, asLabel))
  }));
};

NumberPickerActionButtons.defaultProps = {
  disableMin: false,
  disableMax: false
};
NumberPickerActionButtons.propTypes = {
  onIncrement: PropTypes.func.isRequired,
  onDecrement: PropTypes.func.isRequired,
  disableMax: PropTypes.bool,
  disableMin: PropTypes.bool
};
export default NumberPickerActionButtons;
