// @guestyci/foundation/legacy/Table/Column/Column.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var Column = function Column() {
  return null;
};

Column.defaultProps = {
  width: 200,
  fixed: undefined,
  align: 'left',
  dataKey: '',
  sortable: false,
  resizable: false
};
Column.propTypes = {
  /** Column width */
  width: _propTypes["default"].number,

  /** Indicator whether or not column should be fixed (left | right)  */
  fixed: _propTypes["default"].oneOf(['left', 'right']),

  /** Set content alignment  */
  align: _propTypes["default"].oneOf(['left', 'center', 'right']),

  /** Data source to retrieve data from   */
  dataKey: _propTypes["default"].string,

  /** Indicator whether or not column is sortable  */
  sortable: _propTypes["default"].bool,

  /** Indicator whether or not column can be resized */
  resizable: _propTypes["default"].bool
};
var _default = Column;
exports["default"] = _default;
