// @guestyci/foundation/esm/legacy/Dialog/DialogFooter.js
import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import RaisedButton from "../RaisedButton/RaisedButton";
import FlatButton from "../FlatButton/FlatButton";
import Row from "../Row/Row";

var DialogFooter = function DialogFooter(_ref) {
  var onConfirm = _ref.onConfirm,
      onCancel = _ref.onCancel,
      confirmTitle = _ref.confirmTitle,
      cancelTitle = _ref.cancelTitle,
      leftComponent = _ref.leftComponent,
      className = _ref.className,
      style = _ref.style,
      disabled = _ref.disabled,
      showCancel = _ref.showCancel,
      showConfirm = _ref.showConfirm,
      isSubmitting = _ref.isSubmitting;
  return /*#__PURE__*/React.createElement("div", {
    className: cn('flex-space-between-center w-fill py-6', className),
    style: style,
    "data-qa": "dialog-footer"
  }, leftComponent && leftComponent, /*#__PURE__*/React.createElement(Row, {
    justify: "end",
    align: "center",
    className: "flex-1"
  }, showCancel && /*#__PURE__*/React.createElement(FlatButton, {
    className: "dialog-footer-cancel",
    onClick: onCancel
  }, cancelTitle), showConfirm && /*#__PURE__*/React.createElement(RaisedButton, {
    className: "dialog-footer-confirm",
    disabled: disabled,
    onClick: onConfirm,
    isLoading: isSubmitting
  }, confirmTitle)));
};

DialogFooter.defaultProps = {
  cancelTitle: 'Cancel',
  confirmTitle: 'Save',
  onConfirm: undefined,
  onCancel: undefined,
  leftComponent: null,
  className: '',
  style: {},
  disabled: false,
  showCancel: true,
  showConfirm: true,
  isSubmitting: false
};
DialogFooter.propTypes = {
  cancelTitle: PropTypes.string,
  confirmTitle: PropTypes.string,
  onConfirm: PropTypes.func,
  onCancel: PropTypes.func,
  leftComponent: PropTypes.node,
  className: PropTypes.string,
  style: PropTypes.shape(),
  disabled: PropTypes.bool,
  showCancel: PropTypes.bool,
  showConfirm: PropTypes.bool,
  isSubmitting: PropTypes.bool
};
export default DialogFooter;
