// @guestyci/foundation/esm/legacy/Dialog/DialogXHeader.js
import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

var BtnDelete = function BtnDelete(props) {
  return /*#__PURE__*/React.createElement("svg", props, /*#__PURE__*/React.createElement("path", {
    d: "M8.485.707L7.778 0 4.243 3.536.707 0 0 .707l3.536 3.536L0 7.778l.707.707L4.243 4.95l3.535 3.535.707-.707L4.95 4.243z"
  }));
};

BtnDelete.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 8.4854 8.4854"
};
import IconButton from "../IconButton";

var DialogXHeader = function DialogXHeader(_ref) {
  var onClick = _ref.onClick,
      className = _ref.className,
      style = _ref.style,
      setAbsoluteRight = _ref.setAbsoluteRight;
  return /*#__PURE__*/React.createElement("div", {
    className: cn(setAbsoluteRight ? 'absolute top-1 right-2' : 'dialog-x-header flex-end-center height-10 w-fill', className),
    style: style,
    "data-qa": "dialog-x-header"
  }, /*#__PURE__*/React.createElement(IconButton, {
    onClick: onClick,
    svg: BtnDelete,
    iconHeight: 10,
    iconWidth: 10
  }));
};

DialogXHeader.defaultProps = {
  className: '',
  style: {},
  setAbsoluteRight: false
};
DialogXHeader.propTypes = {
  onClick: PropTypes.func.isRequired,
  className: PropTypes.string,
  style: PropTypes.shape(),
  setAbsoluteRight: PropTypes.bool
};
export default DialogXHeader;
