// @guestyci/foundation/esm/legacy/Dialog/withDialog.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _possibleConstructorReturn from "@babel/runtime/helpers/esm/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/esm/getPrototypeOf";

function _createSuper(Derived) { return function () { var Super = _getPrototypeOf(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

import React, { Component } from 'react';
/**
 * Create a wrapper compose to support Dialog functionality
 * @param ComposedComponent
 * @returns {{Component}}
 */

function withDialog(ComposedComponent) {
  var DialogWrapper = /*#__PURE__*/function (_Component) {
    _inherits(DialogWrapper, _Component);

    var _super = _createSuper(DialogWrapper);

    function DialogWrapper(props) {
      var _this;

      _classCallCheck(this, DialogWrapper);

      _this = _super.call(this, props);

      _this.closeDialog = function () {
        _this.setState({
          isOpen: false
        });
      };

      _this.openDialog = function () {
        _this.setState({
          isOpen: true
        });
      };

      _this.toggleDialog = function () {
        _this.setState(function (prevState) {
          return {
            isOpen: !prevState.isOpen
          };
        });
      };

      _this.state = {
        isOpen: false
      };
      return _this;
    }

    _createClass(DialogWrapper, [{
      key: "render",
      value: function render() {
        var isOpen = this.state.isOpen;
        return /*#__PURE__*/React.createElement(ComposedComponent, _extends({}, this.props, {
          isDialogOpen: isOpen,
          closeDialog: this.closeDialog,
          openDialog: this.openDialog,
          toggleDialog: this.toggleDialog,
          "data-qa": "dialog-wrapper"
        }));
      }
    }]);

    return DialogWrapper;
  }(Component);

  return DialogWrapper;
}

export default withDialog;
