// @guestyci/foundation/RadioGroup/RadioGroup.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _noop = _interopRequireDefault(require("lodash/noop"));

var _Layout = require("../Layout");

var _RadioGroupContext = _interopRequireDefault(require("./RadioGroupContext"));

var RadioGroup = _react["default"].forwardRef(function RadioGroup(_ref, ref) {
  var children = _ref.children,
      name = _ref.name,
      className = _ref.className,
      inline = _ref.inline,
      valueProp = _ref.value,
      style = _ref.style,
      onChange = _ref.onChange,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["children", "name", "className", "inline", "value", "style", "onChange"]);

  var _useState = (0, _react.useState)(valueProp),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      activeItem = _useState2[0],
      setActiveItem = _useState2[1];

  var handleOnChange = (0, _react.useCallback)(function (e) {
    var value = e.target.value;
    setActiveItem(value);

    if (onChange) {
      onChange(e);
    }
  }, [onChange]);
  return /*#__PURE__*/_react["default"].createElement(_RadioGroupContext["default"].Provider, {
    value: {
      name: name,
      onChange: handleOnChange,
      value: activeItem
    }
  }, inline ? /*#__PURE__*/_react["default"].createElement(_Layout.Row, (0, _extends2["default"])({}, props, {
    ref: ref
  }), children) : /*#__PURE__*/_react["default"].createElement(_Layout.Col, (0, _extends2["default"])({}, props, {
    ref: ref
  }), children));
});

RadioGroup.defaultProps = {
  name: '',
  className: '',
  value: '',
  onChange: _noop["default"],
  style: {},
  inline: false
};
RadioGroup.propTypes = {
  /** The radioGroup children to render */
  children: _propTypes["default"].node.isRequired,

  /** Callback for on radio change */
  onChange: _propTypes["default"].func,

  /** Additional class to radioGroup root */
  className: _propTypes["default"].string,

  /** Additional style to radioGroup root */
  style: _propTypes["default"].shape(),

  /** value will be return when click on this item */
  name: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].number]),

  /** Identifier will be return when click on this element */
  id: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].number]),

  /** The value of the selected radio  */
  value: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].number, _propTypes["default"].shape()]),

  /** Render group inline or not */
  inline: _propTypes["default"].bool
};
RadioGroup.displayName = 'RadioGroup';
var _default = RadioGroup;
exports["default"] = _default;
