// @guestyci/foundation/Placeholder/Placeholder.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _createStyles = _interopRequireDefault(require("../createStyles"));

var _colors = require("../theme/colors");

var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    root: {
      width: '100%',
      backgroundColor: theme.palette.background.system,
      display: 'inline-block'
    },
    animate: {
      animation: '$shimmer 2s infinite linear',
      background: "linear-gradient(to right, ".concat(theme.palette.background.system, " 8%, ").concat(_colors.gray200, " 18%, ").concat(theme.palette.background.system, " 28%);"),
      backgroundSize: 1000
    },
    '@keyframes shimmer': {
      from: {
        backgroundPosition: -1000
      },
      to: {
        backgroundPosition: 1000
      }
    }
  };
}, {
  name: 'Placeholder'
});

var Placeholder = _react["default"].forwardRef(function Placeholder(_ref, ref) {
  var _ref$className = _ref.className,
      className = _ref$className === void 0 ? '' : _ref$className,
      _ref$style = _ref.style,
      style = _ref$style === void 0 ? {} : _ref$style,
      _ref$animate = _ref.animate,
      animate = _ref$animate === void 0 ? true : _ref$animate,
      _ref$jss = _ref.jss,
      jss = _ref$jss === void 0 ? {} : _ref$jss,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["className", "style", "animate", "jss"]);
  var classes = useStyles({
    jss: jss
  });
  return /*#__PURE__*/_react["default"].createElement("div", (0, _extends2["default"])({}, props, {
    ref: ref,
    className: (0, _classnames["default"])(className, classes.root, (0, _defineProperty2["default"])({}, classes.animate, animate)),
    style: style
  }));
});

Placeholder.displayName = 'Placeholder';
Placeholder.propTypes = {
  /** Additional classes to style your placeholder */
  className: _propTypes["default"].string,

  /** additional styles to pass to placeholder */
  style: _propTypes["default"].shape(),

  /** if placeholder should be animated */
  animate: _propTypes["default"].bool,

  /** jss override object to customize the jss classes */
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func])
};
var _default = Placeholder;
exports["default"] = _default;
