// @guestyci/foundation/esm/legacy/ProxyEmailInput/ProxyEmailInput.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _possibleConstructorReturn from "@babel/runtime/helpers/esm/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/esm/getPrototypeOf";

function _createSuper(Derived) { return function () { var Super = _getPrototypeOf(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

import React, { Component } from 'react';
import trim from 'lodash/trim';
import debounce from 'lodash/debounce';
import cn from 'classnames';
import withStyle from 'react-jss';
import PropTypes from 'prop-types';
import Input from "../Input/Input";
import IndicationIcon from "./IndicationIcon";
var style = {
  fadeOut: {
    opacity: 0,
    width: 0,
    transition: 'width 0.45s 0.45s, height 0.45s 0.45s, opacity 0.5s',
    borderLeft: 0
  },
  fadeIn: {
    opacity: 1,
    width: '46px',
    transition: 'width 0.45s, height 0.45s, opacity 0.5s'
  }
};
var DEBOUNCE_DELAY = 500;

var ProxyEmailInput = /*#__PURE__*/function (_Component) {
  _inherits(ProxyEmailInput, _Component);

  var _super = _createSuper(ProxyEmailInput);

  function ProxyEmailInput(props) {
    var _this;

    _classCallCheck(this, ProxyEmailInput);

    _this = _super.call(this, props);

    _this.onInputValueChange = function (value) {
      var inputField = _this.state.inputField;
      var trimmedValue = trim(value);

      if (inputField === trimmedValue) {
        return;
      }

      _this.setState({
        inputField: trimmedValue
      });

      _this.handleValueChange();
    };

    _this.state = {
      inputField: props.defaultInputValue
    };
    _this.handleValueChange = debounce(function () {
      var onChange = _this.props.onChange;
      var inputField = _this.state.inputField;
      onChange(inputField);
    }, DEBOUNCE_DELAY);
    return _this;
  }

  _createClass(ProxyEmailInput, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          placeholder = _this$props.placeholder,
          _this$props$classes = _this$props.classes,
          fadeIn = _this$props$classes.fadeIn,
          fadeOut = _this$props$classes.fadeOut,
          required = _this$props.required,
          maxLength = _this$props.maxLength,
          minLength = _this$props.minLength,
          domain = _this$props.domain,
          isValid = _this$props.isValid,
          isValidating = _this$props.isValidating,
          name = _this$props.name;
      var inputField = this.state.inputField;
      return /*#__PURE__*/React.createElement("div", {
        className: cn('d-flex-fill', 'border', 'border-gray-lighter', 'bg-gray-lightest'),
        "data-qa": "proxy-email-input"
      }, /*#__PURE__*/React.createElement(Input, {
        onChange: this.onInputValueChange,
        name: name,
        value: inputField,
        containerClassName: "border-0",
        placeholder: placeholder,
        minLength: minLength,
        maxLength: maxLength,
        required: required
      }), /*#__PURE__*/React.createElement("div", {
        className: cn('label', 'flex-center', 'bg-gray-lightest', 'px-4', 'font-size-md', 'letter-spacing-sm', 'bl-1', 'border-gray-light')
      }, "@".concat(domain)), /*#__PURE__*/React.createElement("div", {
        className: cn(inputField.length > 0 ? fadeIn : fadeOut, 'd-flex-fill', 'bl-1', 'border-gray-light')
      }, /*#__PURE__*/React.createElement(IndicationIcon, {
        isValidating: isValidating,
        isValid: isValid
      })));
    }
  }], [{
    key: "getDerivedStateFromProps",
    value: function getDerivedStateFromProps(props, state) {
      var newState = null;

      if (props.value && props.value !== state.inputField) {
        newState = {
          inputField: props.value
        };
      }

      return newState;
    }
  }]);

  return ProxyEmailInput;
}(Component);

ProxyEmailInput.defaultProps = {
  defaultInputValue: '',
  value: '',
  minLength: undefined,
  maxLength: undefined,
  placeholder: '5-30 characters, with no spaces',
  name: ''
};
ProxyEmailInput.propTypes = {
  /** An indicator for the icon indicator */
  isValid: PropTypes.bool.isRequired,

  /** A default input value */
  defaultInputValue: PropTypes.string,

  /** A value for controlled input field */
  value: PropTypes.string,

  /** An indicator if the req is validating */
  isValidating: PropTypes.bool.isRequired,

  /** Function to toggle the input changes */
  onChange: PropTypes.func.isRequired,

  /** A placeholder */
  placeholder: PropTypes.string,

  /** A domain placeholder */
  domain: PropTypes.string.isRequired,

  /** Min input length */
  minLength: PropTypes.number,

  /** Max input length */
  maxLength: PropTypes.number,

  /** A unique name */
  name: PropTypes.string
};
export var ProxyEmailInputComponent = ProxyEmailInput;
export default withStyle(style)(ProxyEmailInput);
