// @guestyci/foundation/legacy/Link/Link.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

function _createSuper(Derived) { return function () { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

var Link = /*#__PURE__*/function (_PureComponent) {
  (0, _inherits2["default"])(Link, _PureComponent);

  var _super = _createSuper(Link);

  function Link() {
    var _this;

    (0, _classCallCheck2["default"])(this, Link);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _super.call.apply(_super, [this].concat(args));

    _this.handleClick = function () {
      var href = _this.props.href;

      if (href) {
        window.open(href, '_blank');
      }
    };

    return _this;
  }

  (0, _createClass2["default"])(Link, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          className = _this$props.className,
          style = _this$props.style,
          children = _this$props.children,
          color = _this$props.color,
          props = (0, _objectWithoutProperties2["default"])(_this$props, ["className", "style", "children", "color"]);
      return /*#__PURE__*/_react["default"].createElement("div", (0, _extends2["default"])({
        role: "presentation",
        className: (0, _classnames["default"])('text-hover-blue', 'd-flex', 'align-items-center', 'fill-hover-blue', 'transition-all', 'clickable', (0, _defineProperty2["default"])({}, "text-".concat(color, " fill-").concat(color), color), className),
        onClick: this.handleClick,
        style: style
      }, props, {
        "data-qa": "link"
      }), children);
    }
  }]);
  return Link;
}(_react.PureComponent);

Link.defaultProps = {
  className: '',
  style: {},
  href: '',
  color: 'white'
};
Link.propTypes = {
  /** Additional class to add to the link component */
  className: _propTypes["default"].string,

  /** Additional styles to add to the link component  */
  style: _propTypes["default"].shape(),

  /** Url to redirect on click */
  href: _propTypes["default"].string,

  /** The item to render inside the link */
  children: _propTypes["default"].oneOfType([_propTypes["default"].arrayOf(_propTypes["default"].node), _propTypes["default"].node]).isRequired,

  /** Color for link text */
  color: _propTypes["default"].string
};
var _default = Link;
exports["default"] = _default;
