// @guestyci/foundation/es/legacy/baseInput/InputStarDeleteActionAddon.js
import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import noop from 'lodash/noop';
import IconButton from "../IconButton";
import CSSTransition from "../CSSTransiton";
import SvgIcon from "../SvgIcon/SvgIcon";

var BtnStarOff = function BtnStarOff(props) {
  return /*#__PURE__*/React.createElement("svg", props, /*#__PURE__*/React.createElement("path", {
    d: "M2.864 14l.886-5.065L0 5.347l5.183-.74L7.5 0l2.318 4.608 5.182.74-3.75 3.587.887 5.065-4.638-2.393L2.864 14zM7.5 10.482l3.305 1.706-.631-3.607 2.648-2.535-3.662-.522L7.5 2.225 5.84 5.524l-3.662.522 2.649 2.535-.63 3.606L7.5 10.482z"
  }));
};

BtnStarOff.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 15 14"
};

var BtnStarOn = function BtnStarOn(props) {
  return /*#__PURE__*/React.createElement("svg", props, /*#__PURE__*/React.createElement("path", {
    d: "M7.5 0l2.319 4.608L15 5.347l-3.75 3.588.887 5.065-4.638-2.393L2.864 14l.886-5.065L0 5.347l5.183-.74z"
  }));
};

BtnStarOn.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 15 14"
};

var BtnTrash = function BtnTrash(props) {
  return /*#__PURE__*/React.createElement("svg", props, /*#__PURE__*/React.createElement("path", {
    d: "M11.773 18h-8.5c-1.224 0-2.22-.923-2.22-2.057L1 5.503A.501.501 0 0 1 1.5 5h11.946a.5.5 0 0 1 .5.498L14 15.94c0 1.137-.999 2.06-2.227 2.06zM2.003 6l.05 9.94v.002c0 .574.558 1.058 1.22 1.058h8.5C12.44 17 13 16.516 13 15.942L12.95 6H2.001z"
  }), /*#__PURE__*/React.createElement("path", {
    d: "M14.5 6H.5a.5.5 0 0 1-.5-.5v-3A.5.5 0 0 1 .5 2h14a.5.5 0 0 1 .5.5v3a.5.5 0 0 1-.5.5zM1 5h13V3H1v2z"
  }), /*#__PURE__*/React.createElement("path", {
    d: "M10.5 3h-6a.5.5 0 0 1-.5-.5v-2a.5.5 0 0 1 .5-.5h6a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-.5.5zM5 2h5V1H5v1zM10.5 15a.5.5 0 0 1-.5-.5v-6a.5.5 0 1 1 1 0v6a.5.5 0 0 1-.5.5zM7.5 15a.5.5 0 0 1-.5-.5v-6a.5.5 0 1 1 1 0v6a.5.5 0 0 1-.5.5zM4.5 15a.5.5 0 0 1-.5-.5v-6a.5.5 0 1 1 1 0v6a.5.5 0 0 1-.5.5z"
  }));
};

BtnTrash.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 15 18"
};
import Checkbox from "../Checkbox/Checkbox";

class InputStarDeleteActionAddOn extends PureComponent {
  constructor(...args) {
    super(...args);

    this.handleStarClick = () => {
      const {
        onStar,
        id,
        name
      } = this.props;
      onStar(id || name);
    };

    this.handleTrashClick = () => {
      const {
        onTrash,
        id,
        name
      } = this.props;
      onTrash(id || name);
    };
  }

  render() {
    const {
      show,
      isPrimary,
      disabled
    } = this.props;
    return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(CSSTransition, {
      show: show || isPrimary,
      animation: "fade",
      className: "absolute px-4  right-0 h-fill flex-center bg-gray-lightest"
    }, /*#__PURE__*/React.createElement("div", {
      className: "flex-center"
    }, !isPrimary && /*#__PURE__*/React.createElement(IconButton, {
      iconHeight: 18,
      iconWidth: 15,
      type: "hint",
      disabled: disabled,
      onClick: this.handleTrashClick,
      svg: BtnTrash,
      className: " p-0 input-addon-trash-action"
    }), /*#__PURE__*/React.createElement(Checkbox, {
      className: "input-addon-star-action pl-4",
      checked: isPrimary,
      onChange: this.handleStarClick,
      disabled: disabled,
      inputClassName: "mr-0",
      icon: /*#__PURE__*/React.createElement(SvgIcon, {
        width: 15,
        height: 14,
        className: "fill-hover-blue",
        color: "muted",
        svg: BtnStarOff
      }),
      checkedIcon: /*#__PURE__*/React.createElement(SvgIcon, {
        color: "blue",
        width: 15,
        height: 14,
        svg: BtnStarOn
      })
    }))));
  }

}

InputStarDeleteActionAddOn.defaultProps = {
  show: false,
  isPrimary: false,
  onTrash: noop,
  onStar: noop,
  disabled: false,
  id: undefined,
  name: ''
};
InputStarDeleteActionAddOn.propTypes = {
  show: PropTypes.bool,
  isPrimary: PropTypes.bool,
  onTrash: PropTypes.func,
  onStar: PropTypes.func,
  disabled: PropTypes.bool,
  name: PropTypes.string,
  id: PropTypes.oneOfType([PropTypes.number, PropTypes.string])
};
export default InputStarDeleteActionAddOn;
