// @guestyci/foundation/legacy/GenericError/GenericError.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var Oops = function Oops(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("defs", null, /*#__PURE__*/_react["default"].createElement("path", {
    id: "a",
    d: "M.236.574H26.78v11.53H.236z"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    id: "c",
    d: "M.104.477h13.02v26.388H.103z"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    id: "e",
    d: "M0 .502h58.252v121.262H0z"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    id: "g",
    d: "M.213.382h17.358v18.666H.213z"
  })), /*#__PURE__*/_react["default"].createElement("g", {
    fill: "none",
    fillRule: "evenodd"
  }, /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#D4ECF7",
    d: "M0 323.616s32.537-18.774 32.166-33.728c-.372-14.955-6.172-21.962-13.109-32.384-14.87-22.344-12.914-55.771 6.383-74.218 12.67-12.11 30.718-16.866 48.037-16.624 17.32.241 34.243 4.981 50.945 9.688-6.244-25.34-10.644-53.538 1.507-76.514 12.099-22.878 39.768-35.037 64.78-30.593 25.011 4.444 46.321 23.984 55.777 48.136 5.772 14.742 7.483 30.832 12.5 45.862 5.018 15.03 14.626 29.916 29.456 34.52 13.19 4.094 29.69.003 39.337 10.092 8.133 8.505 6.416 22.803 1.076 33.38-5.34 10.578-13.616 19.519-17.896 30.592-4.278 11.073-4.144 29.958 7.341 31.748 9.36 1.459 14.272 6.45 12.691 12.69-1.11 4.387-5.113 7.353-9.535 7.353H0z"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#212121",
    d: "M235.606 211.402c4.75-8.536 10.022-16.748 15.333-24.952 2.623-4.05 5.255-8.097 7.95-12.102 2.909-4.323 5.635-8.982 9.056-12.93.528-.61 1.712-.05 1.319.739-4.292 8.607-10.38 16.614-15.506 24.765-5.374 8.543-10.565 17.192-16.163 25.596-.7 1.05-2.609-.002-1.99-1.116m-18.379-12.228c4.75-8.537 10.023-16.748 15.335-24.952 2.622-4.05 5.254-8.098 7.949-12.102 2.91-4.323 5.635-8.982 9.056-12.93.529-.61 1.712-.05 1.318.738-4.291 8.608-10.378 16.615-15.505 24.765-5.374 8.544-10.565 17.193-16.163 25.597-.699 1.05-2.61-.003-1.99-1.116"
  }), /*#__PURE__*/_react["default"].createElement("g", {
    transform: "scale(-1 1) rotate(-19 315.311 898.36)"
  }, /*#__PURE__*/_react["default"].createElement("mask", {
    id: "b",
    fill: "#fff"
  }, /*#__PURE__*/_react["default"].createElement("use", {
    xlinkHref: "#a"
  })), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#616161",
    d: "M9.733 1.88a100.71 100.71 0 0 1 7.352-.985c1.927-.193 4.922-.777 6.647.367 1.491.989 2.522 2.551 2.977 4.249.323 1.208-.49 2.431-1.797 2.653l-.164.028a2.369 2.369 0 0 1-2.452-1.142c-.51-.888-1.179-1.537-2.133-1.487-2.48.13-4.936.319-7.398.62a49.02 49.02 0 0 0-7.433 1.483c-.289.082-.18.776.078 1.634.362 1.206-.454 2.447-1.765 2.683-.149.028-.297.055-.446.081-1.222.22-2.425-.515-2.667-1.66C.322 9.41.245 8.387.236 7.479.192 2.876 6.14 2.508 9.733 1.88",
    mask: "url(#b)"
  })), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#757575",
    d: "M265.114 151.644c2.45 1.06 5.953 2.957 7.051 5.495.518 1.194.456 3.431-.599 4.425-.69.651-1.854.095-1.78-.848.08-.999.382-1.958-.068-2.999-.398-.92-1.544-1.681-2.294-2.312-.821-.69-1.614-1.407-2.433-2.102-.68-.579-.763-2.042.123-1.659"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#212121",
    d: "M145.335 285.951c.993.315 1.653.951 2.009 1.74-2.79-.845-5.252 1.269-6.179 3.94a3.328 3.328 0 0 1-.34-1.302c-.142-2.453 2-5.172 4.51-4.378"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#424242",
    d: "M147.369 287.833c.59 1.35.313 3.14-.703 4.55-.713.988-1.76 1.349-2.815 1.094-1.127.115-2.199-.484-2.725-1.568.952-2.74 3.443-4.92 6.243-4.076"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#FFB300",
    d: "M146.499 289.553c-.57-1.644-2.201-2.373-3.823-2.145-1.952.274-2.824 2.9-2.582 4.565.299 2.05 2.361 3.007 4.192 2.842a3.13 3.13 0 0 0 1.457-.492c1.156-1.167 1.25-3.342.756-4.77m-8.562-3.845c2.25-2.964 6.031-3.469 8.924-1.187 1.946 1.534 2.567 3.505 2.616 5.968.047 2.373-1.042 4.496-2.894 5.935-2.87 2.108-6.568 1.463-8.827-.844-2.89-2.953-2.222-6.705.18-9.872"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#FFD54F",
    d: "M145.712 282.525c1.08.368.72 2.08-.418 2.027-1.356-.064-2.584.127-3.853.694-1.36.606-2.101 1.625-2.938 2.824-.193.276-.708.154-.643-.22.623-3.602 4.314-6.534 7.852-5.325"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#FF8F00",
    d: "M142.884 292.922c.764.01 1.623-.28 2.211-.997.77-.94.968-2.25.816-3.407.549 1.213.284 3.37-.568 4.367-.805.944-2.377.882-3.353-.032-.024-.023.758.065.836.068l.058.001"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#424242",
    d: "M147.844 207.981c2.581-.412 3.965 1.348 6.013 2.363-.288.463-.581.926-.864 1.39-1.697-.66-4.45-2.883-5.352-1.784-1.563 1.904-3.06 3.832-4.465 5.845-1.42 2.036-2.712 4.242-3.744 6.486-.227.492 3.127 3.202 4.85 3.826l-.71 1.172c-2.195-1.262-4.252-2.65-5.962-4.357 2.722-5.61 5.888-10.487 10.234-14.94"
  }), /*#__PURE__*/_react["default"].createElement("g", {
    transform: "scale(-1 1) rotate(-19 539.088 566.937)"
  }, /*#__PURE__*/_react["default"].createElement("mask", {
    id: "d",
    fill: "#fff"
  }, /*#__PURE__*/_react["default"].createElement("use", {
    xlinkHref: "#c"
  })), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#616161",
    d: "M12.138 14.77a96.216 96.216 0 0 0-1.751-7.176c-.523-1.854-1.06-4.85-2.793-5.984C5.429.192 2.462.142.104 1.255c.166.784.323 1.566.483 2.348 2.079-.3 3.807-1.532 5.921-.289 2.433 5.7 3.716 11.417 4.395 17.692-1.995 1.075-4.194 1.73-6.476 2.217l.707 3.634c1.715.081 3.61-.477 5.109-1.085 4.44-1.801 2.61-7.436 1.895-11.002",
    mask: "url(#d)"
  })), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#FFD54F",
    d: "M125.918 253.328c-1.455 4.543-1.87 9.574-.22 14.053 2.243 6.114 7.846 10.273 13.2 14a768.3 768.3 0 0 0 38.251 24.96c-3.959-2.927-7.256-6.452-8.619-11.092-1.449-4.936-.365-10.247 1.25-15.131 2.595-7.837 6.531-15.162 10.68-22.307 10.22-17.63 21.915-34.59 36.546-48.815 3.89-3.78 8.328-7.532 13.73-8.211 4.964-.628 9.663 1.437 14 4.154a66.906 66.906 0 0 1 3.334 2.244c.49.34.976.694 1.458 1.04 1.325.958 2.698 1.968 3.492 3.398.075.141.136.277.195.42-.332-1.51-5.251-8.304-11.476-11.684-.325-.175-.638-.355-.958-.537a457.735 457.735 0 0 0-6.092-3.333c-13.844-7.431-28.23-13.984-43.604-17.076-2.489-.501-5.034-.907-7.567-.675-2.782.262-5.44 1.292-7.91 2.597-10.242 5.386-17.205 15.018-23.29 24.98a456.396 456.396 0 0 0-2.1 3.497c-.45.758-.894 1.52-1.336 2.275l-7.303 12.458-2.592 4.42-6.767 11.535c-2.41 4.123-4.847 8.286-6.302 12.83m61.255 59.487c-.195-.115-.388-.232-.582-.347.218.096.44.18.662.252l-.08.095"
  }), /*#__PURE__*/_react["default"].createElement("g", {
    transform: "scale(-1 1) rotate(-19 490.095 872.154)"
  }, /*#__PURE__*/_react["default"].createElement("mask", {
    id: "f",
    fill: "#fff"
  }, /*#__PURE__*/_react["default"].createElement("use", {
    xlinkHref: "#e"
  })), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#FFB300",
    d: "M58.246 96.941c-.068 5.147-.777 10.526-3.745 14.727-2.79 3.949-7.039 6.216-11.715 7.7a48.808 48.808 0 0 1-4.085 1.096c-2.014.464-4.05.824-6.038 1.158a8.188 8.188 0 0 1-1.369.142c-.374-.006-.742-.06-1.082-.217-.953-.436-1.43-1.485-1.831-2.458A430.779 430.779 0 0 1 .313 15.186C-.163 11.81-.395 7.867 1.899 5.43c.102-.116.205-.225.32-.334C3.431 4.001 5.051 3.49 6.61 3.013c.653-.198 1.314-.402 1.981-.593a45.867 45.867 0 0 1 1.573-.442c5.589-1.519 11.464-2.41 16.542.096C31.567 4.47 34.521 9.46 36.951 14.3c9.136 18.213 14.623 38.059 18.504 58.062 1.58 8.109 2.9 16.32 2.791 24.578",
    mask: "url(#f)"
  })), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#FF8F00",
    d: "M179.598 307.966c.449.297.908.583 1.367.869 12.526-15.66 24.06-32.239 35.267-48.844 11.38-16.854 22.07-34.18 31.34-52.279a44.764 44.764 0 0 0-1.334-.922c-9.575 17.93-20.355 35.182-31.78 51.984-11.307 16.622-23.626 32.532-34.86 49.192"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#FFB300",
    d: "M234.656 196.893c3.591 1.947 5.07 3.225 6.965 4.83 1.217 1.033 3.739 3.572 4.551 4.507 1.109.694.485.313 1.544 1.067-1.746-2.728-4.607-5.007-7.025-7.078-.033-.026-.058-.055-.092-.08a450.254 450.254 0 0 0-5.943-3.246"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#424242",
    d: "M179.507 305.3c1.84-2.75 5.23-4.22 8.451-4.04 2.196.124 4.293.949 6.095 2.219 1.599 1.127 1.863 3.465.602 5.024-1.572 1.943-3.687 3.492-5.257 5.377-.428.514-.851 1.055-1.418 1.4-1.423.864-3.798-.933-5.11-1.593-2.201-1.11-44.198-29.069-47.017-32.43-1.063-1.203-1.37-3.409.143-3.25.428.046 32.998 21.821 33.357 22.058l5.475 3.619c1.287.85 3.148 1.868 4.68 1.615"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#212121",
    d: "M188.972 311.582c1.39.443 2.323 1.31 2.834 2.373-3.903-1.19-7.288 1.593-8.525 5.155a4.41 4.41 0 0 1-.501-1.752c-.247-3.292 2.68-6.892 6.192-5.776"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#424242",
    d: "M191.54 313.647c.85 1.82.502 4.215-.884 6.083-.973 1.311-2.424 1.774-3.9 1.41-1.567.131-3.072-.694-3.827-2.158 1.269-3.654 4.693-6.526 8.611-5.335"
  }), /*#__PURE__*/_react["default"].createElement("g", {
    transform: "scale(-1 1) rotate(-19 824.284 750.333)"
  }, /*#__PURE__*/_react["default"].createElement("mask", {
    id: "h",
    fill: "#fff"
  }, /*#__PURE__*/_react["default"].createElement("use", {
    xlinkHref: "#g"
  })), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#FFB300",
    d: "M4.795 7.13c1.381-1.755 3.67-1.958 5.545-1.005 2.26 1.147 2.22 4.692 1.228 6.608-1.221 2.36-4.128 2.659-6.286 1.695a4.034 4.034 0 0 1-1.565-1.204c-.917-1.896-.12-4.572 1.078-6.095m12.018-1.086C15.32 1.513 10.933-.677 6.461.88 3.453 1.928 1.873 4.057.783 7.021c-1.05 2.856-.615 5.883 1.033 8.397 2.607 3.753 7.373 4.514 11.086 2.66 4.75-2.37 5.508-7.196 3.911-12.035",
    mask: "url(#h)"
  })), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#FFD54F",
    d: "M189.488 305.832c1.618.553 1.12 3.004-.58 2.903-2.026-.122-3.853.125-5.736.911-2.015.842-3.1 2.29-4.322 3.992-.282.392-1.052.207-.963-.33.848-5.16 6.294-9.288 11.6-7.476"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#FF8F00",
    d: "M186.294 320.557c1.269.031 2.689-.417 3.65-1.558 1.26-1.496 1.56-3.596 1.284-5.46.936 1.96.541 5.424-.853 7.01-1.317 1.498-3.926 1.367-5.564-.122-.042-.037 1.26.119 1.387.126l.096.004"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#FFB300",
    d: "M175.776 229.255c2.71-4.298 5.628-8.455 8.598-12.574 2.907-4.031 5.796-8.389 9.25-11.976 2.677-2.78 6-4.904 9.987-4.18 2.778.505 4.23 3.2 3.985 5.804-.003.027-.043.03-.047.003-.898-5.71-7.926-3.491-10.684-1.292-2.636 2.102-4.663 5.26-6.721 7.902a236.996 236.996 0 0 0-11.082 15.488 181.053 181.053 0 0 0-9.573 16.41c-1.695 3.283-3.217 6.652-4.659 10.052-.63 1.486-1.201 2.894-1.406 4.51 1.143 3.19 3.017 4.166 5.62 2.929a8.977 8.977 0 0 0 1.758-.865c.504-.283.914.437.53.806-1.868 1.79-4.725 2.97-7.213 1.603-3.114-1.71-3.337-4.547-2.31-7.678 3.1-9.446 8.696-18.58 13.967-26.942m-21.57-13.323c2.709-4.298 5.628-8.454 8.598-12.574 2.906-4.03 5.796-8.387 9.25-11.975 2.677-2.78 6-4.904 9.987-4.18 2.778.504 4.228 3.2 3.985 5.804-.003.026-.043.03-.048.003-.898-5.71-7.925-3.492-10.683-1.292-2.636 2.102-4.663 5.26-6.721 7.901a236.773 236.773 0 0 0-11.082 15.489 181.099 181.099 0 0 0-9.574 16.41c-1.694 3.283-3.215 6.652-4.658 10.052-.63 1.486-1.2 2.894-1.406 4.508 1.144 3.192 3.017 4.169 5.62 2.93a8.977 8.977 0 0 0 1.759-.865c.503-.283.914.438.529.806-1.868 1.79-4.725 2.97-7.213 1.603-3.114-1.71-3.337-4.547-2.31-7.677 3.1-9.446 8.696-18.58 13.967-26.943"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#FFEE58",
    d: "M189.998 232.46c5.453-8.518 11.165-17.013 18.047-24.5.07-.076.186.038.135.118-5.479 8.503-11.63 16.593-17.371 24.928-6.002 8.715-11.324 17.83-16.46 27.055-.236.425-.871-.01-.64-.43 5.073-9.249 10.599-18.28 16.29-27.17m-22.776-12.844c5.384-8.302 10.7-16.773 16.695-24.63.093-.122.261-.003.187.13-9.829 17.275-22.345 32.854-31.494 50.564-.079.154-.287.004-.21-.147 4.486-8.918 9.398-17.553 14.822-25.917m30.274-37.111c3.315-.565 7.3 2.395 9.434 4.881.107.124-.065.337-.182.314-1.803-.343-3.56-.896-5.317-1.394-1.595-.452-3.365-.467-4.743-1.463-.833-.602-.067-2.188.808-2.338m-3.634-.516c.294-.056.544-.123.8.01.248.13.42.332.605.568.294.371-.044 1.348-.505 1.461-.291.072-.552.127-.827.076-.284-.052-.44-.258-.638-.482-.438-.499-.089-1.509.565-1.634m-58.72 62.665c.088-.154.306-.042.233.125-3.308 7.55-5.578 14.08-4.716 22.108.077.707-.772 1.053-1.03.46-3.113-7.143 1.831-16.25 5.513-22.693m-4.15 25.751c.224.34.423.65.524 1.022.06.22-.294.31-.423.1-.213-.348-.297-.684-.384-1.056-.034-.147.19-.204.283-.066"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#F57100",
    d: "M252.055 24.645l3.08-9.356 2.47 1.246-3.886 9.333-1.664-1.223m-2.429-.899c0-1.31 1.078-15.926 1.248-18.217a.239.239 0 0 1 .314-.208l2.148.689a.24.24 0 0 1 .167.25L251.9 23.746h-2.274m-2.05-.673l-2.802-7.744L241 13.645v-2.188l5.33 1.515 2.457 10.101h-1.211m12.67 3.124l6.493-13.562 3.174 1.42s-7.018 7.176-8.016 13.353l-1.651-1.211"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#FB9300",
    d: "M259.4 24.645l3.08-9.356 2.469 1.246-3.886 9.333-1.664-1.223m-8.687-.899l2.47-12.29h1.965l-2.218 12.964-2.217-.674m-3.136-.673V10.084c0-.172.14-.311.313-.311h1.796c.173 0 .313.14.313.311v13.663l-2.422-.674"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#7DB1A1",
    d: "M259.088 22.4l3.887 1.515s-2.028 10.378-5.914 12.007c0 0-2.197-5.104 2.027-13.522"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#B3E0D0",
    d: "M246.33 22.4c.761-.168 12.758 0 12.758 0s-2.534 7.01-2.027 13.522c0 0-5.999-.335-12.04-2.215l1.31-11.307"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#FB9300",
    d: "M258.168 5.227l2.304 1.01s.63 7.24-.62 10.523h-1.684V5.227zM242.73 1.845l2.395 6.207c.096.25.35.354.559.23l.994-.595c.18-.108.266-.353.203-.577L244.95.29c-.076-.268-.35-.375-.548-.215l-1.548 1.248a.468.468 0 0 0-.124.522m20.582 7.928s1.86-4.967 2.366-8.081h3.042l-3.64 8.67-1.768-.59"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#CC395D",
    d: "M289.646 100.67l-2.358 3.97-36.794 21.108-43.026-11.142V104.32l82.178-3.65"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#E73A5E",
    d: "M207.046 111.565l42.689 11.656 39.91-22.552-27.533-6.334-65.17 11.909 10.104 5.321"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#3F2C48",
    d: "M198.457 37.377c-.475-.627-1.061-1.188-1.778-1.51-.716-.32-1.575-.375-2.277-.025-1.032.515-1.507 1.805-1.356 2.952.15 1.146.81 2.163 1.556 3.043a14.468 14.468 0 0 0 3.434 2.954c.292.18.614.357.957.333a.72.72 0 0 0 .351-.127c1.073-2.454.729-5.482-.887-7.62m.815 7.792l.082-.178-.009.006c-.025.057-.047.115-.073.172m.082-.178c.18-.126.298-.328.252-.539l-.252.54"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#F9B6A8",
    d: "M199.88 105.707s41.014 10.419 53.644 4.59c12.63-5.827 28.039-19.764 28.039-19.764s-29.316 5.322-61.262-6.08c0 0-2.898 10.388-26.642 19.004l6.22 2.25"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#CC395D",
    d: "M234.585 111.746s8.078-.688 13.382-6.262l5.557 4.814s-5.473 3.463-18.939 1.448"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#F9B6A8",
    d: "M195.175 55.819c0-.53.883-.975 2.079-1.102.127-.54.194-.897.194-.897l1.431-.873s-1.094-1.858.59-6.335c1.684-4.476-2.358-6.503-2.358-6.503l-8.42-1.774-15.1 23.65 14.848 3.294 3.031-5.828s.926.169 2.863.422c.992.13 1.785-1.462 2.322-3.046-.88-.191-1.48-.57-1.48-1.008"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#0080A3",
    d: "M82.012 152.867s-7.731 4.139 16.09 29.816c23.82 25.676 60.951 28.464 60.951 28.464-29.714-24.969-66.721-59.954-77.041-58.28"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#E73A5E",
    d: "M66.014 146.533H55l7.646 5.068c.372 6.358 1.027 13.248 1.93 20.175 2.554-1.234 5.294-1.947 8.115-2.567-3.969-12.018-6.677-22.676-6.677-22.676"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#F9B6A8",
    d: "M97.797 174.998l-13.933 20.861c-4.036-6.599-8.005-17.057-11.173-26.65-2.82.62-5.561 1.333-8.115 2.567 3.177 24.358 9.438 49.193 17.436 53.138 7.22 3.563 27.004-17.748 38.12-30.723-5.191-7.406-13.378-13.37-22.335-19.193"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#CC395D",
    d: "M106.43 162.074l-8.633 12.924c6.79 7.69 13.967 13.699 22.336 19.193.386-.45 7.851-9.481 7.851-9.481l-21.555-22.636"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#E73A5E",
    d: "M87.569 314.445l9.767 9.46-3.368-9.797 14.666-20.004c-2.752-1.99-5.55-3.915-8.32-5.881l-12.745 26.222"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#F9B6A8",
    d: "M120.574 246.537l-20.26 41.686c2.77 1.966 5.568 3.891 8.32 5.88l22.063-30.094a119.563 119.563 0 0 0 10.486-17.166l17.693-35.342c-9.838-3.703-19.73-8.188-29.982-11.246l-8.32 46.282"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#CC395D",
    d: "M131.689 184.71l-2.795 15.545c9.58 4.9 19.575 9.103 29.982 11.246l.177-.354-27.364-26.437"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#00ADFF",
    d: "M82.012 152.867s23.239-15.456 33.848-31.927c10.608-16.47 35.666-56.542 50.014-58.026 14.347-1.484 33.342.507 40.667 22.045 7.326 21.538 11.62 46.37 11.62 46.37l26.27-9.122 2.02 6.082s-34.605 27.619-42.688 6.587c-8.083-21.03-11.872-29.393-11.872-29.393s.758 10.896-7.83 12.923c0 0-13.64 65.881-25.008 92.74-43.951-17.23-64.916-53.21-77.041-58.279"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#523B5D",
    d: "M194.96 34.062c-4.6-2.483-10.803-1.478-14.39 2.33-2.909 3.09-4.234 7.745-7.934 9.814-2.94 1.643-6.544 1.167-9.883 1.587-5.76.725-11.18 4.593-13.172 10.063-1.992 5.47-.067 12.292 4.826 15.426 4.893 3.135 12.307 1.702 15.239-3.324 1.066-1.827 1.573-4.013 3.022-5.552 2.516-2.672 6.764-2.405 10.18-3.73 4.303-1.67 7.347-5.503 7.099-10.124-.824-.618-1.68-1.276-2.109-2.213-.43-.938-.261-2.238.637-2.74 1.332-.929 3.037 2.302 3.037 2.302s1.64-3.178 2.941-4.199c1.53-1.2 3.413-2.318 3.84-4.218.502-2.228-1.328-4.34-3.332-5.422"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#F9B6A8",
    d: "M244.43 122.207s5.053-7.095 5.558-6.715c.505.38-1.137 5.575-1.137 5.575l8.967-3.42s.253 3.04-1.894 4.56c-2.147 1.52-9.472 6.081-9.472 6.081l-2.021-6.081"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#CC395D",
    d: "M83.864 195.859l2.793 5.858.912-11.405-3.705 5.547"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#3A3B39",
    d: "M196.943 48.09a.885.885 0 1 0-1.768 0 .885.885 0 1 0 1.768 0m.537-1.314a.356.356 0 0 1-.354-.347c-.01-.398-.2-.687-.508-.774-.332-.093-.699.08-.931.443a.354.354 0 0 1-.491.106.357.357 0 0 1-.107-.492c.406-.631 1.082-.923 1.72-.743.615.172 1.008.725 1.027 1.441a.356.356 0 0 1-.346.366h-.01"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#CC395D",
    d: "M190.813 49.542a.237.237 0 0 1-.212-.132 5.887 5.887 0 0 0-1.942-2.217.238.238 0 0 1-.061-.33.236.236 0 0 1 .329-.061 6.352 6.352 0 0 1 2.098 2.396.238.238 0 0 1-.212.344m.657 9.909l-2.677-.464 2.02 1.727.657-1.263"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#FFFFFE",
    d: "M190.813 51.195a1.011 1.011 0 0 0-2.02 0 1.011 1.011 0 1 0 2.02 0"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#0080A3",
    d: "M218.16 131.329l.758 4.392 1.383-5.135-2.14.743"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#B3E0D0",
    d: "M213.123 26l.773.654-5.657 6.718-.772-.655L213.123 26m6.878 6.895l.445.914-9.644 4.721-.445-.914L220 32.895m-.58 11.04l.036 1.016-7.072.256-.037-1.016 7.072-.256"
  }), /*#__PURE__*/_react["default"].createElement("g", null, /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#B3E0D0",
    d: "M250.228 59.997v21.946L239.59 83.85l-5.64-18.946 16.278-4.908"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#3A3B39",
    d: "M234.776 65.718c5.595.134 10.758-1.716 15.452-5.721 0 0-9.58-1.44-15.452 5.72"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#B3E0D0",
    d: "M233.27 65.523L232 61.11l9.382-8.11 3.438 2.027-11.55 10.496"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#7DB1A1",
    d: "M232 68.505l15.18-13.528L244.82 53 232 65.718v2.787"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#523B5D",
    d: "M235.279 70.174c5.253-.04 10.43-1.67 15.182-4.409l.154 12.661-12.417 2.244-2.92-10.496"
  })), /*#__PURE__*/_react["default"].createElement("g", null, /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#00A650",
    d: "M289.663 57.689c-.26-.09-.554-.121-.765-.296-.514-.426-.144-1.226-.1-1.885.086-1.32-1.298-2.257-2.577-2.674a11.691 11.691 0 0 0-7.558.132c-.667.243-1.333.566-1.78 1.109-.447.543-.62 1.354-.247 1.949.252.402.785.6 1.242.523.371.383.413 1.065.372 1.636-.05.699-.027 1.563.607 1.886 1.19.605 2.377-1.587 3.636-1.135.533.191.776.775 1.056 1.26.28.484.819.969 1.347.766.582-.225.63-1.091 1.178-1.386.395-.212.895-.037 1.257.227.36.264.66.615 1.062.813.79.389 1.875-.001 2.222-.8.346-.797-.119-1.835-.952-2.125"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#585755",
    d: "M274.615 51.322c3.646-.935 7.084.115 7.678 2.347.594 2.231-1.88 4.798-5.526 5.733-3.647.935-7.084-.115-7.679-2.347-.594-2.231 1.88-4.798 5.527-5.733"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#3A3B39",
    d: "M276.396 58.012c-3.258.836-6.343.083-7.388-1.679-.004.243.017.485.08.722.595 2.232 4.032 3.282 7.679 2.347 3.646-.935 6.12-3.502 5.526-5.733a2.645 2.645 0 0 0-.29-.669c-.033 2.038-2.348 4.177-5.607 5.012"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#FB9300",
    d: "M269.185 61.074s-2.114 7.512 7.967 8.68c10.08 1.167 11.056-5.506 11.056-5.506l-19.023-3.174"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#F57100",
    d: "M269.185 61.074s6.666 4.509 19.023 3.174c0 0-12.357-3.678-19.023-3.174"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#FB9300",
    d: "M271.38 51.674s3.74 5.568 18.698 4.566c2.207-8.702-16.492-13.268-18.698-4.566"
  })), /*#__PURE__*/_react["default"].createElement("path", {
    stroke: "#616161",
    d: "M184 166.583c13.474-1.333 25.474 1.562 36 8.684m-24-18.057c8.983-.888 16.983 1.042 24 5.79"
  })));
};

Oops.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  xmlnsXlink: "http://www.w3.org/1999/xlink",
  width: "334",
  height: "325",
  viewBox: "0 0 334 325"
};

var GenericError = function GenericError(_ref) {
  var className = _ref.className,
      text = _ref.text,
      title = _ref.title,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["className", "text", "title"]);
  return /*#__PURE__*/_react["default"].createElement("div", (0, _extends2["default"])({
    className: (0, _classnames["default"])('text-center w-100 d-flex flex-column align-items-center', className)
  }, props, {
    "data-qa": "generic-error"
  }), /*#__PURE__*/_react["default"].createElement(Oops, {
    className: "mb-6"
  }), /*#__PURE__*/_react["default"].createElement("h1", {
    className: "mb-4"
  }, title), /*#__PURE__*/_react["default"].createElement("h3", null, text));
};

GenericError.defaultProps = {
  className: '',
  text: 'Please refresh the page',
  title: 'Oops, something went wrong'
};
GenericError.propTypes = {
  /** Component additional classes */
  className: _propTypes["default"].string,

  /** Additional text to be displayed to the user */
  text: _propTypes["default"].string,

  /** Text to be displayed to the user */
  title: _propTypes["default"].string
};
var _default = GenericError;
exports["default"] = _default;
