// @guestyci/foundation/FormGroup/FormGroup.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Col = _interopRequireDefault(require("../Layout/Col"));

var _Row = _interopRequireDefault(require("../Layout/Row"));

var FormGroup = function FormGroup(_ref) {
  var children = _ref.children,
      className = _ref.className,
      style = _ref.style,
      horizontal = _ref.horizontal;
  return horizontal ? /*#__PURE__*/_react["default"].createElement(_Row["default"], {
    className: className,
    style: style,
    wrap: false,
    spacing: 4
  }, children) : /*#__PURE__*/_react["default"].createElement(_Col["default"], {
    className: className,
    style: style,
    spacing: 2
  }, children);
};

FormGroup.defaultProps = {
  className: '',
  style: {},
  horizontal: false
};
FormGroup.propTypes = {
  /** Form Group inner children */
  children: _propTypes["default"].node.isRequired,

  /** Class  to be added to FormGroup */
  className: _propTypes["default"].string,

  /** Style to add to FormGroup */
  style: _propTypes["default"].shape(),

  /** Boolean to indicate if Form group should be horizontal or not */
  horizontal: _propTypes["default"].bool
};
var _default = FormGroup;
exports["default"] = _default;
