// @guestyci/foundation/legacy/Table/CheckboxCell/CheckboxCell.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Checkbox = _interopRequireDefault(require("../../../Checkbox"));

var _createStyles = _interopRequireDefault(require("../../../createStyles"));

var useStyles = (0, _createStyles["default"])({
  root: {
    display: 'table-cell',
    background: 'inherit',
    width: 36,
    verticalAlign: 'middle'
  },
  fixed: {
    position: 'sticky',
    left: 0
  }
}, {
  name: 'CheckboxCell'
});

var CheckboxCell = _react["default"].memo(function (_ref) {
  var className = _ref.className,
      style = _ref.style,
      onChange = _ref.onChange,
      checked = _ref.checked,
      indeterminate = _ref.indeterminate,
      value = _ref.value,
      fixed = _ref.fixed;
  var classes = useStyles({
    fixed: fixed
  });

  var handleCheck = function handleCheck() {
    onChange(value);
  };

  var handleClick = function handleClick(e) {
    e.stopPropagation();
  };

  var inputProps = {
    onClick: handleClick
  };
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])('checkbox-cell', classes.root, (0, _defineProperty2["default"])({}, classes.fixed, fixed), className),
    style: style
  }, /*#__PURE__*/_react["default"].createElement(_Checkbox["default"], {
    checked: checked,
    inputProps: inputProps,
    indeterminate: indeterminate,
    onChange: handleCheck,
    value: value
  }));
});

CheckboxCell.defaultProps = {
  className: '',
  style: {},
  checked: false,
  value: '',
  fixed: false
};
CheckboxCell.propTypes = {
  onChange: _propTypes["default"].func.isRequired,
  className: _propTypes["default"].string,
  style: _propTypes["default"].shape(),
  checked: _propTypes["default"].bool,
  indeterminate: _propTypes["default"].bool,
  fixed: _propTypes["default"].bool,
  value: _propTypes["default"].string
};
CheckboxCell.displayName = 'CheckboxCell';
var _default = CheckboxCell;
exports["default"] = _default;
