// @guestyci/foundation/Dropdown/DropdownSearch.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _Input = _interopRequireDefault(require("../Input"));

var _DropdownClearButton = _interopRequireDefault(require("./DropdownClearButton"));

var _createStyles = _interopRequireDefault(require("../createStyles"));

var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    root: {
      margin: "".concat(theme.spacer(2), "px 0")
    }
  };
}, {
  name: 'DropdownSearch'
});

var DropdownSearch = _react["default"].forwardRef(function DropdownSearchInput(_ref, ref) {
  var value = _ref.value,
      onChange = _ref.onChange,
      _ref$placeholder = _ref.placeholder,
      placeholder = _ref$placeholder === void 0 ? 'Search...' : _ref$placeholder,
      className = _ref.className,
      onClear = _ref.onClear;
  var classes = useStyles();
  return /*#__PURE__*/_react["default"].createElement(_Input["default"], {
    ref: ref,
    value: value,
    onChange: onChange,
    placeholder: placeholder,
    className: (0, _classnames["default"])(classes.root, className),
    suffix: /*#__PURE__*/_react["default"].createElement(_DropdownClearButton["default"], {
      show: !!value,
      asSuffix: true,
      onClick: onClear
    })
  });
});

DropdownSearch.displayName = 'DropdownSearchInput';
var _default = DropdownSearch;
exports["default"] = _default;
