// @guestyci/foundation/Dropdown/DropdownArrowButton.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.useStyles = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _createStyles = _interopRequireDefault(require("../createStyles"));

var _Icon = _interopRequireDefault(require("../legacy/Icon"));

var _Button = _interopRequireDefault(require("../Button"));

var BtnArrowUp = function BtnArrowUp(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M4.529 0a.474.474 0 0 0-.357.137L.137 4.172a.488.488 0 0 0 .69.689L4.53 1.156l3.705 3.705a.488.488 0 0 0 .69-.69L4.89.138A.477.477 0 0 0 4.53 0z"
  }));
};

BtnArrowUp.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 9.0618038 4.9977479"
};
var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    root: {
      height: '100%',
      width: 46,
      borderRadius: 0,
      borderLeft: "solid 1px ".concat(theme.palette.border)
    },
    arrow: {
      fill: theme.palette.secondary["default"],
      transition: theme.transition.create()
    },
    active: {
      fill: theme.palette.primary["default"]
    },
    rotate: {
      transform: 'rotate(180deg)'
    }
  };
}, {
  name: 'DropdownArrowButton'
});
exports.useStyles = useStyles;

var DropdownArrowButton = _react["default"].forwardRef(function DropdownArrowButton(_ref, ref) {
  var _cn;

  var disabled = _ref.disabled,
      onClick = _ref.onClick,
      active = _ref.active,
      className = _ref.className,
      jss = _ref.jss,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["disabled", "onClick", "active", "className", "jss"]);
  var classes = useStyles({
    jss: jss
  });
  return /*#__PURE__*/_react["default"].createElement(_Button["default"], (0, _extends2["default"])({}, props, {
    component: "span",
    ref: ref,
    enableRipple: false,
    className: (0, _classnames["default"])(classes.root, classes.arrow, className),
    disabled: disabled,
    size: "xs"
  }), /*#__PURE__*/_react["default"].createElement(_Icon["default"], {
    svg: BtnArrowUp,
    className: (0, _classnames["default"])(classes.arrow, (_cn = {}, (0, _defineProperty2["default"])(_cn, classes.rotate, !active), (0, _defineProperty2["default"])(_cn, classes.active, active), _cn), 'dropdown-arrow-button'),
    width: 10,
    height: 10
  }));
});

DropdownArrowButton.propTypes = {
  onClick: _propTypes["default"].func,
  disabled: _propTypes["default"].bool,
  className: _propTypes["default"].string,
  active: _propTypes["default"].bool,
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func])
};
DropdownArrowButton.displayName = 'DropdownArrowButton';
var _default = DropdownArrowButton;
exports["default"] = _default;
