// @guestyci/foundation/Dropdown/Option.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.useStyles = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _MenuItem = _interopRequireDefault(require("../Menu/MenuItem"));

var _TextField = _interopRequireDefault(require("../TextField"));

var _styles = _interopRequireDefault(require("../Checkbox/styles"));

var _CheckedIcon = _interopRequireDefault(require("../Checkbox/CheckedIcon"));

var _createStyles = _interopRequireDefault(require("../createStyles"));

var _Row = _interopRequireDefault(require("../Layout/Row"));

var CheckboxIconOff = function CheckboxIconOff(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M12.5 15h-10A2.503 2.503 0 0 1 0 12.5v-10C0 1.122 1.122 0 2.5 0h10C13.879 0 15 1.122 15 2.5v10c0 1.379-1.121 2.5-2.5 2.5zM2.5 1C1.673 1 1 1.673 1 2.5v10c0 .827.673 1.5 1.5 1.5h10c.827 0 1.5-.673 1.5-1.5v-10c0-.827-.673-1.5-1.5-1.5h-10z"
  }));
};

CheckboxIconOff.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 15 15"
};
var useStyles = (0, _createStyles["default"])(function (theme) {
  return (0, _extends2["default"])({}, (0, _styles["default"])(theme));
}, {
  name: 'Option'
});
exports.useStyles = useStyles;

var jssOverrides = function jssOverrides(theme) {
  return {
    selected: {
      color: theme.palette.primary["default"]
    }
  };
};

var Option = _react["default"].forwardRef(function Option(_ref, ref) {
  var _cn;

  var value = _ref.value,
      _ref$multiselect = _ref.multiselect,
      multiselect = _ref$multiselect === void 0 ? false : _ref$multiselect,
      children = _ref.children,
      _ref$selected = _ref.selected,
      selected = _ref$selected === void 0 ? false : _ref$selected,
      onClick = _ref.onClick,
      _ref$overflowEllipsis = _ref.overflowEllipsis,
      overflowEllipsis = _ref$overflowEllipsis === void 0 ? true : _ref$overflowEllipsis,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["value", "multiselect", "children", "selected", "onClick", "overflowEllipsis"]);
  var classes = useStyles();
  var handleOptionClick = (0, _react.useCallback)(function (e, selectedItem) {
    onClick(e, selectedItem);
  }, [onClick]);
  var CheckboxIcon = (0, _react.useMemo)(function () {
    return selected ? _CheckedIcon["default"] : CheckboxIconOff;
  }, [selected]);
  var selectedStyles = !multiselect ? jssOverrides : undefined;
  return /*#__PURE__*/_react["default"].createElement(_MenuItem["default"], (0, _extends2["default"])({
    selected: selected,
    jss: selectedStyles,
    value: value,
    onClick: handleOptionClick
  }, props, {
    ref: ref
  }), /*#__PURE__*/_react["default"].createElement(_Row["default"], {
    spacing: 2,
    fullWidth: true,
    align: "center"
  }, multiselect && /*#__PURE__*/_react["default"].createElement(CheckboxIcon, {
    height: 16,
    width: 16,
    className: (0, _classnames["default"])(classes.icon, (_cn = {}, (0, _defineProperty2["default"])(_cn, classes.uncheckedDefaultBg, !selected), (0, _defineProperty2["default"])(_cn, classes.checked, selected), _cn))
  }), /*#__PURE__*/_react["default"].createElement(_TextField["default"], {
    overflowEllipsis: overflowEllipsis
  }, children)));
});

Option.propTypes = {
  value: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].number, _propTypes["default"].shape()]).isRequired,
  children: _propTypes["default"].node.isRequired,
  overflowEllipsis: _propTypes["default"].bool
};
Option.displayName = 'Option';
var _default = Option;
exports["default"] = _default;
