// @guestyci/foundation/es/legacy/Listing/Listing.js
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import withStyles from 'react-jss';
import isFunction from 'lodash/isFunction';
import Avatar from "../Avatar/Avatar";
import TextField from "../TextField/TextField";
import Row from "../Row/Row";
import Col from "../Col/Col";
const styles = {
  text: {
    width: ({
      size
    }) => `calc(100% - ${size === 'sm' ? 30 : 45}px)`
  }
};

class Listing extends Component {
  constructor(...args) {
    super(...args);

    this.handleListingClick = () => {
      const {
        disabled,
        onClick,
        name
      } = this.props;

      if (isFunction(onClick) && !disabled) {
        onClick(name);
      }
    };
  }

  render() {
    const {
      img,
      title = '',
      subTitle,
      size,
      className,
      style,
      onClick,
      disabled,
      isMultiUnit,
      classes
    } = this.props;
    return /*#__PURE__*/React.createElement("div", {
      role: "presentation",
      className: cn('flex-start-center w-inherit transition-color', className, {
        'text-hover-blue clickable': !disabled && isFunction(onClick)
      }, {
        'disabled-light': disabled
      }),
      style: style,
      onClick: this.handleListingClick,
      "data-qa": "listing"
    }, /*#__PURE__*/React.createElement(Row, {
      align: "center",
      fullWidth: true,
      justify: "start",
      spacing: "sm"
    }, /*#__PURE__*/React.createElement(Avatar, {
      listing: true,
      bordered: isMultiUnit,
      src: img,
      size: size
    }), /*#__PURE__*/React.createElement(Col, {
      dense: true,
      className: classes.text
    }, /*#__PURE__*/React.createElement(TextField, {
      size: size === 'md' ? 'lg' : 'md',
      className: "overflow-ellipsis"
    }, title), subTitle && /*#__PURE__*/React.createElement(TextField, {
      size: size,
      className: "listing-subtitle  overflow-ellipsis"
    }, subTitle))));
  }

}

Listing.defaultProps = {
  img: '',
  subTitle: '',
  size: 'md',
  className: '',
  style: {},
  onClick: null,
  disabled: false,
  isMultiUnit: false
};
Listing.propTypes = {
  /** Image url for listing */
  img: PropTypes.string,

  /** Listing full name */
  title: PropTypes.string.isRequired,

  /** Listing address */
  subTitle: PropTypes.string,

  /** avatar icon size */
  size: PropTypes.oneOf(['sm', 'md']),

  /** Additional class name for listing */
  className: PropTypes.string,

  /** Additional style for listing */
  style: PropTypes.shape(),

  /** CB function for listing click */
  onClick: PropTypes.func,

  /** Indicator whether or not the listing component is disabled */
  disabled: PropTypes.bool,

  /** Indicator whether or not we are in multi unit */
  isMultiUnit: PropTypes.bool
};
export const ListingComponent = Listing;
export default withStyles(styles)(Listing);
