// @guestyci/localize/index.js
/** @license Guesty v1.5.1
 *
 * This source code is licensed under the ISC license found in the
 * LICENSE file in the root directory of this source tree.
 */
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "FormattedMessage", {
  enumerable: true,
  get: function get() {
    return _FormattedMessage["default"];
  }
});
Object.defineProperty(exports, "FormattedDate", {
  enumerable: true,
  get: function get() {
    return _FormattedDate["default"];
  }
});
Object.defineProperty(exports, "FormattedTime", {
  enumerable: true,
  get: function get() {
    return _FormattedTime["default"];
  }
});
Object.defineProperty(exports, "Moment", {
  enumerable: true,
  get: function get() {
    return _Moment["default"];
  }
});
Object.defineProperty(exports, "FormattedCurrency", {
  enumerable: true,
  get: function get() {
    return _FormattedCurrency["default"];
  }
});
Object.defineProperty(exports, "LocalizationProvider", {
  enumerable: true,
  get: function get() {
    return _LocalizationProvider["default"];
  }
});
Object.defineProperty(exports, "withLocalization", {
  enumerable: true,
  get: function get() {
    return _withLocalization["default"];
  }
});
Object.defineProperty(exports, "MomentFormatTypes", {
  enumerable: true,
  get: function get() {
    return _utils.MomentFormatTypes;
  }
});
Object.defineProperty(exports, "warningHandler", {
  enumerable: true,
  get: function get() {
    return _utils.warningHandler;
  }
});
Object.defineProperty(exports, "getLanguage", {
  enumerable: true,
  get: function get() {
    return _utils.getLanguage;
  }
});
Object.defineProperty(exports, "intl", {
  enumerable: true,
  get: function get() {
    return _utils.intl;
  }
});
Object.defineProperty(exports, "init", {
  enumerable: true,
  get: function get() {
    return _utils.init;
  }
});
Object.defineProperty(exports, "locales", {
  enumerable: true,
  get: function get() {
    return _utils.locales;
  }
});
exports["default"] = void 0;

var _FormattedMessage = _interopRequireDefault(require("./FormattedMessage"));

var _FormattedDate = _interopRequireDefault(require("./FormattedDate"));

var _FormattedTime = _interopRequireDefault(require("./FormattedTime"));

var _Moment = _interopRequireDefault(require("./Moment"));

var _FormattedCurrency = _interopRequireDefault(require("./FormattedCurrency"));

var _LocalizationProvider = _interopRequireDefault(require("./LocalizationProvider"));

var _withLocalization = _interopRequireDefault(require("./withLocalization"));

var _utils = require("./utils");

var locale = {
  intl: _utils.intl,
  init: _utils.init,
  locales: _utils.locales,
  warningHandler: _utils.warningHandler,
  getLanguage: _utils.getLanguage,
  FormattedMessage: _FormattedMessage["default"],
  FormattedDate: _FormattedDate["default"],
  FormattedTime: _FormattedTime["default"],
  FormattedCurrency: _FormattedCurrency["default"],
  Moment: _Moment["default"],
  LocalizationProvider: _LocalizationProvider["default"],
  withLocalization: _withLocalization["default"],
  MomentFormatTypes: _utils.MomentFormatTypes
};
console.log('please');
var _default = locale;
exports["default"] = _default;
