// @guestyci/foundation/esm/legacy/Col/Col.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import _toConsumableArray from "@babel/runtime/helpers/esm/toConsumableArray";
import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { createUseStyles } from 'react-jss';

var spacer = _toConsumableArray(Array(7).keys()).map(function (i) {
  return i * 5;
});

var useStyles = createUseStyles({
  col: {
    '& > * ': {
      marginBottom: function marginBottom(_ref) {
        var spacing = _ref.spacing;
        return spacer[spacing];
      }
    },
    '& > *:last-child ': {
      marginBottom: 0
    }
  }
});
var Col = React.forwardRef(function Col(_ref2, ref) {
  var className = _ref2.className,
      style = _ref2.style,
      children = _ref2.children,
      dense = _ref2.dense,
      fullWidth = _ref2.fullWidth,
      align = _ref2.align,
      justify = _ref2.justify,
      spacing = _ref2.spacing,
      onClick = _ref2.onClick,
      props = _objectWithoutProperties(_ref2, ["className", "style", "children", "dense", "fullWidth", "align", "justify", "spacing", "onClick"]);

  var classes = useStyles({
    spacing: spacing
  });
  return /*#__PURE__*/React.createElement("div", _extends({}, props, {
    ref: ref,
    role: "presentation",
    onClick: onClick,
    className: cn({
      clickable: onClick
    }, 'd-flex flex-column', _defineProperty({}, "col-wrapper ".concat(classes.col), !dense), {
      'w-fill': fullWidth
    }, _defineProperty({}, "align-items-".concat(align), align), _defineProperty({}, "justify-content-".concat(justify), justify), className),
    style: style
  }), children);
});
Col.defaultProps = {
  className: '',
  style: {},
  children: null,
  dense: false,
  fullWidth: false,
  align: 'inherit',
  justify: 'inherit',
  spacing: 6,
  onClick: null
};
Col.propTypes = {
  /** Additional classes for row */
  className: PropTypes.string,

  /** Additional classes for row */
  align: PropTypes.string,

  /** Additional classes for row */
  justify: PropTypes.string,

  /** Additional styles for row */
  style: PropTypes.shape(),

  /** Children for row wrapper */
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.arrayOf(PropTypes.node)]),

  /** Boolean whether or not we are in dense (no margins) mode */
  dense: PropTypes.bool,

  /** Boolean whether or not we are in full width mode */
  fullWidth: PropTypes.bool,

  /** Spacing between components [0, 1, 2, 3, 4, 5, 6] -
   *  multiplies of 5 (1 => 5, 2=> 10). - defaults to 6
   */
  spacing: PropTypes.oneOf([0, 1, 2, 3, 4, 5, 6]),

  /** handle click event for Col */
  onClick: PropTypes.func
}; // Hack for withInfo plugin

Col.displayName = 'Col';
export default Col;
