// @guestyci/foundation/OverlayDrawer/OverlayDrawer.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.useStyles = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _Paper = _interopRequireDefault(require("../Paper"));

var _useLockBody = _interopRequireDefault(require("../useLockBody"));

var _Portal = _interopRequireDefault(require("../legacy/Portal"));

var _Fade = _interopRequireDefault(require("../Fade"));

var _Backdrop = _interopRequireDefault(require("../Backdrop"));

var _createStyles = _interopRequireDefault(require("../createStyles"));

var _colors = require("../theme/colors");

var _CSSTransition = _interopRequireDefault(require("../CSSTransition"));

var _utils = require("../utils");

var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    root: {
      position: 'fixed',
      inset: 0,
      zIndex: theme.zIndex.dialog
    },
    holder: {
      position: 'absolute',
      height: '100%'
    },
    holdLeft: {
      left: 0
    },
    holdRight: {
      right: 0
    },
    content: {
      position: 'relative',
      backgroundColor: _colors.white,
      height: '100%',
      overflow: 'auto'
    },
    enterRight: {
      transform: 'translateX(100%)'
    },
    enterLeft: {
      transform: 'translateX(-100%)'
    },
    enterActive: {
      transform: 'translateX(0)',
      transition: "transform ".concat(theme.transition.duration.complex, "ms ").concat(theme.transition.easing.easeInOut)
    },
    enterDone: {
      transform: 'translateX(0)'
    },
    exit: {
      transform: 'translateX(0)'
    },
    exitRightActive: {
      transform: 'translateX(100%)',
      transition: "transform ".concat(theme.transition.duration.complex, "ms ").concat(theme.transition.easing.easeInOut)
    },
    exitLeftActive: {
      transform: 'translateX(-100%)',
      transition: "transform ".concat(theme.transition.duration.complex, "ms ").concat(theme.transition.easing.easeInOut)
    },
    exitLeftDone: {
      transform: 'translateX(-100%)'
    },
    exitRightDone: {
      transform: 'translateX(100%)'
    },
    hideSurface: {
      visibility: 'hidden'
    },
    showSurface: {
      visibility: 'visible'
    }
  };
}, {
  name: 'OverlayDrawer'
});
exports.useStyles = useStyles;

var OverlayDrawer = _react["default"].forwardRef(function OverlayDrawer(_ref, ref) {
  var _ref$open = _ref.open,
      open = _ref$open === void 0 ? false : _ref$open,
      _ref$wrapperClassName = _ref.wrapperClassName,
      wrapperClassName = _ref$wrapperClassName === void 0 ? '' : _ref$wrapperClassName,
      _ref$className = _ref.className,
      className = _ref$className === void 0 ? '' : _ref$className,
      _ref$children = _ref.children,
      children = _ref$children === void 0 ? null : _ref$children,
      style = _ref.style,
      _ref$jss = _ref.jss,
      jss = _ref$jss === void 0 ? {} : _ref$jss,
      onClose = _ref.onClose,
      _onEnter = _ref.onEnter,
      _onExited = _ref.onExited,
      _ref$direction = _ref.direction,
      direction = _ref$direction === void 0 ? 'right' : _ref$direction,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["open", "wrapperClassName", "className", "children", "style", "jss", "onClose", "onEnter", "onExited", "direction"]);

  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      showSurface = _useState2[0],
      setShowSurface = _useState2[1];

  var classes = useStyles({
    jss: jss
  });
  var directionKey = (0, _utils.firstLetterUppercase)(direction);
  var slideTransition = {
    enter: classes["enter".concat(directionKey)],
    enterActive: classes.enterActive,
    enterDone: classes.enterDone,
    exit: classes.exit,
    exitActive: classes["exit".concat(directionKey, "Active")],
    exitDone: classes["exit".concat(directionKey, "Done")]
  };
  (0, _useLockBody["default"])(open);
  return /*#__PURE__*/_react["default"].createElement(_Portal["default"], null, /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])(classes.root, (0, _defineProperty2["default"])({}, classes.showSurface, showSurface), (0, _defineProperty2["default"])({}, classes.hideSurface, !showSurface))
  }, /*#__PURE__*/_react["default"].createElement(_Fade["default"], {
    show: open,
    duration: 300,
    timeout: 300,
    unmountOnExit: true
  }, /*#__PURE__*/_react["default"].createElement(_Backdrop["default"], {
    onClick: onClose
  })), /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])(classes.holder, classes["hold".concat(directionKey)], wrapperClassName)
  }, /*#__PURE__*/_react["default"].createElement(_CSSTransition["default"], {
    timeout: 375,
    animation: slideTransition,
    show: open,
    unmountOnExit: true,
    onEnter: function onEnter() {
      setShowSurface(true);
      if (_onEnter) _onEnter();
    },
    onExited: function onExited() {
      setShowSurface(false);
      if (_onExited) _onExited();
    }
  }, /*#__PURE__*/_react["default"].createElement(_Paper["default"], (0, _extends2["default"])({
    ref: ref,
    elevation: 2,
    className: (0, _classnames["default"])(classes.content, className),
    style: style,
    square: true
  }, props), children)))));
});

OverlayDrawer.propTypes = {
  /** Child elements to render inside of the drawer */
  children: _propTypes["default"].node.isRequired,

  /** Boolean flag whether or not to show drawer */
  open: _propTypes["default"].bool,

  /** Custom jss overrides */
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func]),

  /** Callback function for closing drawer */
  onClose: _propTypes["default"].func,

  /** Callback function when drawer animation enters */
  onEnter: _propTypes["default"].func,

  /** Callback function when drawer animation exited */
  onExited: _propTypes["default"].func,

  /** Additional classes for the paper div parent */
  wrapperClassName: _propTypes["default"].string,

  /** Additional classes for the paper level */
  className: _propTypes["default"].string,

  /** Enum for the drawer anchor [ 'right', 'left' ] */
  direction: _propTypes["default"].oneOf(['right', 'left'])
};
OverlayDrawer.displayName = 'OverlayDrawer';
var _default = OverlayDrawer;
exports["default"] = _default;
