// @guestyci/foundation/Toast/StackedToast.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireWildcard(require("react"));

var _reactJss = require("react-jss");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Toast = _interopRequireWildcard(require("./Toast"));

var slideAndPop = {
  entering: {
    transform: 'translate3d(120%, 0, 0)'
  },
  entered: {
    transform: 'translate3d(0,0,0)'
  },
  exiting: {
    transform: 'scale(0.66)',
    opacity: 0
  },
  exited: {
    transform: 'scale(0.66)',
    opacity: 0
  }
};
var collapse = {
  entering: 'auto',
  entered: 120,
  exiting: 0,
  exited: 'auto'
};
var useStyles = (0, _reactJss.createUseStyles)({
  toast: function toast(_ref) {
    var transition = _ref.transition,
        duration = _ref.duration;
    return (0, _extends2["default"])({
      transition: "transform ".concat(duration, "ms cubic-bezier(0.2, 0, 0, 1), opacity ").concat(duration, "ms")
    }, slideAndPop[transition]);
  },
  container: {
    transition: function transition(_ref2) {
      var duration = _ref2.duration;
      return "max-height ".concat(duration - 100, "ms 100ms");
    },
    maxHeight: function maxHeight(_ref3) {
      var transition = _ref3.transition;
      return collapse[transition];
    }
  }
});

var StackedToast = function StackedToast(_ref4) {
  var variant = _ref4.variant,
      onDismiss = _ref4.onDismiss,
      _ref4$autoHideDelay = _ref4.autoHideDelay,
      autoHideDelay = _ref4$autoHideDelay === void 0 ? 5000 : _ref4$autoHideDelay,
      _ref4$autoHide = _ref4.autoHide,
      autoHide = _ref4$autoHide === void 0 ? true : _ref4$autoHide,
      transition = _ref4.transition,
      duration = _ref4.duration,
      rest = (0, _objectWithoutProperties2["default"])(_ref4, ["variant", "onDismiss", "autoHideDelay", "autoHide", "transition", "duration"]);
  var timeout = (0, _react.useRef)();
  var loadingVariant = variant === 'loading';

  var _useStyles = useStyles({
    autoHideDelay: autoHideDelay,
    transition: transition,
    duration: duration
  }),
      container = _useStyles.container,
      toast = _useStyles.toast;

  (0, _react.useEffect)(function () {
    if (!loadingVariant && autoHide) {
      timeout.current = setTimeout(onDismiss, autoHideDelay);
      return function () {
        return timeout.current && clearTimeout(timeout.current);
      };
    }
  }, [loadingVariant, autoHide, onDismiss]);
  var dismiss = (0, _react.useCallback)(function () {
    if (timeout.current) {
      clearTimeout(timeout.current);
    }

    onDismiss();
  }, [onDismiss]);
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: container,
    "data-qa": "stacked-toast"
  }, /*#__PURE__*/_react["default"].createElement(_Toast["default"], (0, _extends2["default"])({}, rest, {
    onDismiss: dismiss,
    variant: variant,
    className: toast
  })));
};

StackedToast.propTypes = {
  variant: _propTypes["default"].oneOf(_Toast.VARIANTS_NAMES).isRequired,
  onDismiss: _propTypes["default"].func.isRequired,
  autoHideDelay: _propTypes["default"].number,
  autoHide: _propTypes["default"].bool,
  duration: _propTypes["default"].number,
  transition: _propTypes["default"].oneOf(['exited', 'exiting', 'entered', 'entering'])
};
var _default = StackedToast;
exports["default"] = _default;
