// @guestyci/foundation/legacy/Drawer/withDrawer.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _react = _interopRequireWildcard(require("react"));

function _createSuper(Derived) { return function () { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

/**
 * Create a wrapper compose to support Drawer functionality
 * @param ComposedComponent
 * @returns {{Component}}
 */
var withDrawer = function withDrawer(ComposedComponent) {
  var _temp;

  return _temp = /*#__PURE__*/function (_Component) {
    (0, _inherits2["default"])(_temp, _Component);

    var _super = _createSuper(_temp);

    function _temp(props) {
      var _this;

      (0, _classCallCheck2["default"])(this, _temp);
      _this = _super.call(this, props);

      _this.defaultState = function (isOpen) {
        _this.setState({
          show: isOpen
        });
      };

      _this.closeDrawer = function () {
        _this.setState({
          show: false
        });
      };

      _this.openDrawer = function (event) {
        if (event) {
          event.stopPropagation();
        }

        _this.setState({
          show: true
        });
      };

      _this.toggleDrawer = function (event) {
        if (event) {
          event.stopPropagation();
        }

        _this.setState(function (prevState) {
          return {
            show: !prevState.show
          };
        });
      };

      _this.state = {
        show: false
      };
      return _this;
    }

    (0, _createClass2["default"])(_temp, [{
      key: "render",
      value: function render() {
        var show = this.state.show;
        return /*#__PURE__*/_react["default"].createElement(ComposedComponent, (0, _extends2["default"])({}, this.props, {
          isDrawerOpen: show,
          closeDrawer: this.closeDrawer,
          openDrawer: this.openDrawer,
          toggleDrawer: this.toggleDrawer,
          drawerDefaultState: this.defaultState
        }));
      }
    }]);
    return _temp;
  }(_react.Component), _temp;
};

var _default = withDrawer;
exports["default"] = _default;
