// @guestyci/foundation/es/legacy/DatePicker/SinglePicker.js
import React from 'react';
import { SingleDatePicker } from 'react-dates';
import { DateDisplayFormat } from "../../enums/enums";

const SinglePicker = ({
  id,
  style,
  prevMonthButton,
  nextMonthButton,
  renderMonthElement,
  verticalSpacing,
  disabled,
  date,
  readOnly,
  daySize,
  numberOfMonths,
  isOutsideRange,
  isDayBlocked,
  initialVisibleMonth,
  onDateChange,
  isFocused,
  onFocusChange,
  renderDayContents,
  anchorDirection,
  openDirection,
  weekDayFormat,
  appendToBody
}) => /*#__PURE__*/React.createElement("div", {
  style: style,
  "data-qa": "single-picker"
}, /*#__PURE__*/React.createElement(SingleDatePicker, {
  date: date,
  placeholder: "Select",
  id: id,
  onDateChange: onDateChange,
  noBorder: true,
  daySize: daySize,
  numberOfMonths: numberOfMonths,
  verticalSpacing: verticalSpacing,
  navPrev: prevMonthButton,
  navNext: nextMonthButton,
  renderMonthElement: renderMonthElement,
  focused: isFocused,
  onFocusChange: onFocusChange,
  disabled: disabled,
  hideKeyboardShortcutsPanel: true,
  readOnly: readOnly,
  renderDayContents: renderDayContents,
  isOutsideRange: isOutsideRange,
  isDayBlocked: isDayBlocked,
  initialVisibleMonth: initialVisibleMonth,
  anchorDirection: anchorDirection,
  openDirection: openDirection,
  weekDayFormat: weekDayFormat,
  displayFormat: DateDisplayFormat,
  appendToBody: appendToBody
}));

export default SinglePicker;
