// @guestyci/foundation/esm/legacy/FormField/FormFieldComponent.js
import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import isUndefined from 'lodash/isUndefined';
import cn from 'classnames';
import withPopover from "../Popover/withPopover";
import { getPlaceholder } from "../../utils/formUtility";
import Popover from "../Popover";

var getChildValue = function getChildValue(props) {
  return props.value || props.defaultValue;
};

var FormFieldComponent = function FormFieldComponent(_ref) {
  var input = _ref.input,
      children = _ref.children,
      isPopoverOpen = _ref.isPopoverOpen,
      className = _ref.className,
      meta = _ref.meta,
      _ref$input = _ref.input,
      name = _ref$input.name,
      inputValue = _ref$input.value,
      style = _ref.style,
      popoverId = _ref.popoverId,
      required = _ref.required,
      openPopover = _ref.openPopover,
      closePopover = _ref.closePopover;
  useEffect(function () {
    var propsValue = getChildValue(children.props);

    if (!input.value && propsValue) {
      input.onChange(propsValue);
    }
  }, []);

  var onChangeValue = function onChangeValue() {
    var _children$props = children.props,
        onChange = _children$props.onChange,
        onSelect = _children$props.onSelect;

    if (onChange) {
      onChange.apply(void 0, arguments);
    }

    if (onSelect) {
      onSelect.apply(void 0, arguments);
    }

    input.onChange.apply(input, arguments);
  };

  var getFieldValue = function getFieldValue(propsValue, inptVal, isVisited) {
    if (isUndefined(propsValue)) {
      return inptVal;
    }

    if (isVisited) {
      return inptVal;
    }

    return propsValue;
  };

  var handleFocus = function handleFocus() {
    input.onFocus();

    if (!meta.visited) {
      input.onChange(input.value);
    }

    if (meta.error) {
      openPopover();
    }
  };

  var handleBlur = function handleBlur() {
    input.onBlur();
    closePopover();
  };

  var _children$props2 = children.props,
      value = _children$props2.value,
      placeholder = _children$props2.placeholder,
      disabled = _children$props2.disabled;
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Popover, {
    isOpen: isPopoverOpen,
    body: meta.error,
    id: popoverId,
    theme: "error",
    containerClassName: "d-flex-fill",
    className: "font-weight-bold"
  }, /*#__PURE__*/React.createElement("div", {
    className: "w-fill"
  })), /*#__PURE__*/React.createElement("div", {
    className: cn(className, 'form-field-component-wrapper', 'transition-borderColor', 'w-fill', 'b-1', "border-".concat(meta.touched && meta.error ? 'red' : 'transparent'), "".concat(meta.error ? 'invalid-field' : null)),
    onBlur: handleBlur,
    onFocus: handleFocus,
    tabIndex: -1,
    style: style
  }, React.cloneElement(children, {
    name: name,
    onChange: onChangeValue,
    required: required,
    disabled: disabled,
    placeholder: getPlaceholder(placeholder, required, disabled),
    value: getFieldValue(value, inputValue, meta.visited)
  })));
};

FormFieldComponent.defaultProps = {
  className: '',
  style: {},
  popoverId: null,
  required: false
};
FormFieldComponent.propTypes = {
  children: PropTypes.node.isRequired,
  required: PropTypes.bool,
  className: PropTypes.string,
  style: PropTypes.shape(),
  input: PropTypes.shape().isRequired,
  meta: PropTypes.shape().isRequired,
  popoverId: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
};
export default withPopover(FormFieldComponent);
