// @guestyci/foundation/FormField/FormField.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.FieldComponent = exports.getFormFieldValidation = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _memoize = _interopRequireDefault(require("@guestyci/memoize"));

var _FormFieldComponent = _interopRequireDefault(require("./FormFieldComponent"));

var _validators = require("../validators");

var _enums = require("../enums");

var _FormContext = _interopRequireDefault(require("../Form/FormContext"));

var buildValidators = (0, _memoize["default"])(function (validations, required) {
  return required ? [_validators.required].concat((0, _toConsumableArray2["default"])(validations)) : validations;
});
var composeValidators = (0, _memoize["default"])(function (validators) {
  return function (value) {
    return validators.reduce(function (error, validator) {
      return error || validator(value);
    }, undefined);
  };
});

var getFormFieldValidation = function getFormFieldValidation(validations, required, provider) {
  return provider === _enums.FormProvider.Redux ? buildValidators(validations, required) : composeValidators(required ? [_validators.required].concat((0, _toConsumableArray2["default"])(validations)) : validations);
};

exports.getFormFieldValidation = getFormFieldValidation;

var FormField = function FormField(_ref) {
  var name = _ref.name,
      label = _ref.label,
      info = _ref.info,
      children = _ref.children,
      className = _ref.className,
      style = _ref.style,
      validate = _ref.validate,
      field = _ref.field,
      required = _ref.required,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["name", "label", "info", "children", "className", "style", "validate", "field", "required"]);

  var _useContext = (0, _react.useContext)(_FormContext["default"]),
      provider = _useContext.provider,
      fieldInstance = _useContext.fieldInstance;

  var Field = field || fieldInstance;
  var validations = (0, _react.useMemo)(function () {
    return getFormFieldValidation(validate, required, provider);
  }, [provider, validate, required]);
  return /*#__PURE__*/_react["default"].createElement(Field, (0, _extends2["default"])({
    name: name,
    style: style,
    className: className,
    validate: validations,
    component: _FormFieldComponent["default"],
    required: required,
    label: label,
    info: info
  }, props, {
    "data-qa": "form-field"
  }), children);
};

FormField.defaultProps = {
  label: null,
  info: null,
  className: '',
  field: undefined,
  style: {},
  validate: [],
  required: false
};
FormField.propTypes = {
  /** form field name to validate accordingly */
  name: _propTypes["default"].string.isRequired,

  /** the redux form field props - required due to redux-form limitations */
  field: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func]),

  /** Form field level validations to run - to use required, use the boolean required prop - Must be an Array */
  validate: _propTypes["default"].arrayOf(_propTypes["default"].func),

  /** Form element to render */
  children: _propTypes["default"].node.isRequired,

  /** form field label */
  label: _propTypes["default"].string,

  /** form field Info description */
  info: _propTypes["default"].string,

  /** Form field root level additional class */
  className: _propTypes["default"].string,

  /** Form field root level additional class */
  style: _propTypes["default"].shape(),

  /** Form field component level additional class */
  required: _propTypes["default"].bool
};
var FieldComponent = FormField;
exports.FieldComponent = FieldComponent;
var _default = FormField;
exports["default"] = _default;
