// @guestyci/foundation/legacy/baseDropdown/withDropdown.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _react = _interopRequireWildcard(require("react"));

var _isEmpty = _interopRequireDefault(require("lodash/isEmpty"));

var _isUndefined = _interopRequireDefault(require("lodash/isUndefined"));

var _isObject = _interopRequireDefault(require("lodash/isObject"));

var _debounce = _interopRequireDefault(require("lodash/debounce"));

var _get = _interopRequireDefault(require("lodash/get"));

var _classnames = _interopRequireDefault(require("classnames"));

var _withCollapse = _interopRequireDefault(require("../Collapse/withCollapse"));

function _createSuper(Derived) { return function () { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

var isEqualValue = function isEqualValue(prevValue, value) {
  if ((0, _isObject["default"])(prevValue) && (0, _isObject["default"])(value)) {
    return prevValue.value === value.value;
  }

  return prevValue === value;
};
/**
 * Create a wrapper compose to support dropdown functionality
 * @param ComposedComponent
 * @returns {{Component}}
 */


var withDropdown = function withDropdown(ComposedComponent) {
  var DropdownWrapper = /*#__PURE__*/function (_Component) {
    (0, _inherits2["default"])(DropdownWrapper, _Component);

    var _super = _createSuper(DropdownWrapper);

    function DropdownWrapper(props) {
      var _this;

      (0, _classCallCheck2["default"])(this, DropdownWrapper);
      _this = _super.call(this, props);
      _this.handleSearchQueryChange = (0, _debounce["default"])(function (val, identifier) {
        var onSearchQueryChange = _this.props.onSearchQueryChange;
        onSearchQueryChange(val, identifier);
      }, 500);

      _this.setSelectedItemControlledValue = function (defaultValue) {
        var options = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : [];
        var value = {};

        if (!(0, _isUndefined["default"])(defaultValue)) {
          if (!(0, _isObject["default"])(defaultValue) && !(0, _isEmpty["default"])(options)) {
            value = options.find(function (option) {
              return option.value === defaultValue;
            }) || {
              label: undefined,
              value: defaultValue
            };
          } else {
            value = defaultValue;
          }
        }

        _this.setSelectedItem(value);
      };

      _this.handleClickOutside = function () {
        var _this$props = _this.props,
            isCollapsed = _this$props.isCollapsed,
            closeCollapse = _this$props.closeCollapse;
        var query = _this.state.query;

        if (!isCollapsed) {
          closeCollapse();

          if (!(0, _isEmpty["default"])(query)) {
            _this.searchQueryChange('');
          }
        }
      };

      _this.handleLoadMore = function (page) {
        var onLoadMore = _this.props.onLoadMore;
        var query = _this.state.query;

        _this.setState({
          currentPage: page
        });

        onLoadMore(page, query);
      };

      _this.handleSelectedItemChange = function (value) {
        var _this$props2 = _this.props,
            id = _this$props2.id,
            name = _this$props2.name,
            onSelect = _this$props2.onSelect,
            onChange = _this$props2.onChange;

        if (onSelect) {
          onSelect(value, id || name);
        } else if (onChange) {
          onChange(value, id || name);
        }
      };

      _this.handleSelectItem = function () {
        var item = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
        var closeCollapse = _this.props.closeCollapse;
        closeCollapse();

        _this.setSelectedItem(item);

        _this.handleSelectedItemChange(item.value);
      };

      _this.clearSelectedItem = function () {
        _this.handleSelectedItemChange('');

        _this.setSelectedItem({});
      };

      _this.clearQuery = function () {
        _this.searchQueryChange('');
      };

      _this.handleSearchChange = function (e) {
        var query = e.currentTarget.value;
        var _this$props3 = _this.props,
            isCollapsed = _this$props3.isCollapsed,
            openCollapse = _this$props3.openCollapse;

        if (isCollapsed) {
          openCollapse();
        }

        _this.searchQueryChange(query);
      };

      _this.dropdownRef = _react["default"].createRef();
      _this.state = {
        selectedItem: {},
        query: '',
        currentPage: props.intialPage || 0,
        searchInputValue: ''
      };
      return _this;
    }

    (0, _createClass2["default"])(DropdownWrapper, [{
      key: "componentDidMount",
      value: function componentDidMount() {
        var _this$props4 = this.props,
            value = _this$props4.value,
            defaultValue = _this$props4.defaultValue,
            options = _this$props4.options;
        this.setSelectedItemControlledValue(value || defaultValue, options);
      }
    }, {
      key: "componentDidUpdate",
      value: function componentDidUpdate(prevProps) {
        var _this$props5 = this.props,
            defaultValue = _this$props5.defaultValue,
            options = _this$props5.options,
            value = _this$props5.value;
        var selectedItem = this.state.selectedItem;

        if (!isEqualValue(prevProps.defaultValue, defaultValue) && (0, _isEmpty["default"])(selectedItem)) {
          this.setSelectedItemControlledValue(defaultValue, options);
        }

        if (!isEqualValue(prevProps.value, value) && !isEqualValue(value, selectedItem)) {
          this.setSelectedItemControlledValue(value, options);
        }

        if (options.length !== prevProps.options.length && !selectedItem.label) {
          this.setSelectedItemControlledValue(selectedItem.value, options);
        }
      }
    }, {
      key: "setSelectedItem",
      value: function setSelectedItem() {
        var item = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
        this.setState({
          selectedItem: item || {},
          searchInputValue: (0, _isObject["default"])((0, _get["default"])(item, 'label')) ? item.name : (0, _get["default"])(item, 'label')
        });
      }
    }, {
      key: "searchQueryChange",
      value: function searchQueryChange(val) {
        var _this$props6 = this.props,
            id = _this$props6.id,
            name = _this$props6.name,
            onSearchQueryChange = _this$props6.onSearchQueryChange;
        this.setState({
          query: val,
          currentPage: 0
        });

        if (onSearchQueryChange) {
          this.handleSearchQueryChange(val, id || name);
        }
      }
    }, {
      key: "render",
      value: function render() {
        var _this$props7 = this.props,
            containerClassName = _this$props7.containerClassName,
            onChange = _this$props7.onChange,
            props = (0, _objectWithoutProperties2["default"])(_this$props7, ["containerClassName", "onChange"]);
        return /*#__PURE__*/_react["default"].createElement("div", {
          ref: this.dropdownRef,
          className: (0, _classnames["default"])('d-flex w-inherit dropdown-wrapper', containerClassName),
          "data-qa": "dropdown-wrapper"
        }, /*#__PURE__*/_react["default"].createElement(ComposedComponent, (0, _extends2["default"])({}, props, this.state, {
          onSearchChange: this.handleSearchChange,
          clearSelectedItem: this.clearSelectedItem,
          onSelectItem: this.handleSelectItem,
          onChange: this.handleSelectedItemChange,
          fetchNextPage: this.handleLoadMore,
          clearQuery: this.clearQuery,
          closeDropdown: this.handleClickOutside
        })));
      }
    }]);
    return DropdownWrapper;
  }(_react.Component);

  return (0, _withCollapse["default"])(DropdownWrapper);
};

var _default = withDropdown;
exports["default"] = _default;
