// @guestyci/foundation/es/legacy/Table/HeaderCell/SortIndicator.js
import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import Icon from "../../Icon";

var ArrowIcon = function ArrowIcon(props) {
  return /*#__PURE__*/React.createElement("svg", props, /*#__PURE__*/React.createElement("path", {
    d: "M.178 9.293l3.342 3.515a.597.597 0 0 0 .491.19c.17.01.342-.06.466-.19L7.82 9.293a.648.648 0 0 0 .083-.79l.009-.008-.085-.09a.004.004 0 0 1-.002-.002l-.01.009a.579.579 0 0 0-.33-.102.597.597 0 0 0-.434.19l-2.512 2.642V.539a.54.54 0 0 0-1.082 0v10.603L.945 8.5a.596.596 0 0 0-.434-.192.573.573 0 0 0-.328.104l-.01-.01-.001.001-.001.001-.086.091.01.01a.657.657 0 0 0 .083.788z"
  }));
};

ArrowIcon.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 8 13"
};
const SortIndicator = React.memo(({
  direction,
  isHint
}) => /*#__PURE__*/React.createElement(Icon, {
  svg: ArrowIcon,
  height: 10,
  width: 10,
  fillColor: isHint ? 'gray-dark' : 'black',
  className: cn('transition-transform', {
    'transform-180': direction === 'asc'
  })
}));
SortIndicator.defaultProps = {
  direction: 'desc',
  isHint: false
};
SortIndicator.prototype = {
  /** Boolean indicator for the direction */
  order: PropTypes.oneOf[('asc', 'desc')],

  /** Boolean indicator for showing the arrow as a hint */
  isHint: PropTypes.string
};
SortIndicator.displayName = 'SortIndicator';
export default SortIndicator;
