// @guestyci/foundation/es/legacy/Drawer/Drawer.js
import React, { useEffect, useRef } from 'react';
import cn from 'classnames';
import { createUseStyles } from 'react-jss';
import PropTypes from 'prop-types';
import Paper from "../Paper";

const calculateDrawerWrapperHeight = (isOpen, overlay) => {
  if (overlay) {
    return '100%';
  }

  return isOpen ? '100%' : 0;
};

const useStyles = createUseStyles({
  wrapper: {
    transition: 'width 225ms cubic-bezier(0, 0, 0.2, 1) 0ms, height 220ms cubic-bezier(0, 0, 0.2, 1) 0ms',
    width: ({
      isOpen,
      width
    }) => isOpen ? width : 0,
    height: ({
      isOpen,
      overlay
    }) => calculateDrawerWrapperHeight(isOpen, overlay),
    overflow: 'hidden'
  },
  drawer: {
    transition: 'transform 225ms cubic-bezier(0, 0, 0.2, 1) 0ms',
    transform: `translateX(${({
      isOpen
    }) => isOpen ? 0 : '-100%'})`,
    width: ({
      width
    }) => width
  }
});

const Drawer = ({
  overlay,
  className,
  style,
  isOpen,
  children,
  width,
  elevation,
  right,
  bgColor,
  containerClassName,
  fixed,
  onDrawerTransitionEnd
}) => {
  const drawerRef = useRef();
  const classes = useStyles({
    isOpen,
    width,
    overlay
  });
  useEffect(() => {
    if (drawerRef.current && overlay) {
      const el = drawerRef.current;
      const parent = el.parentElement;
      parent.classList.toggle('relative');
    }
  }, [overlay]);
  return /*#__PURE__*/React.createElement(Paper, {
    elevation: elevation,
    ref: drawerRef,
    bgColor: bgColor,
    className: cn(classes.wrapper, 'zindex-5 flex-shrink-0', {
      'right-0': overlay && right
    }, {
      absolute: overlay
    }, {
      fixed
    }, containerClassName),
    style: style,
    onTransitionEnd: onDrawerTransitionEnd,
    "data-qa": "drawer"
  }, /*#__PURE__*/React.createElement("div", {
    className: cn(classes.drawer, 'd-flex-fill h-fill flex-column overyflow-y-auto', className)
  }, children));
};

Drawer.defaultProps = {
  className: '',
  style: {},
  width: 250,
  overlay: false,
  right: false,
  elevation: 1,
  bgColor: '',
  containerClassName: '',
  fixed: false
};
Drawer.propTypes = {
  /** Indicator whether or not to show drawer */
  isOpen: PropTypes.bool.isRequired,

  /** Child elements inside drawer */
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired,

  /** Additional classes to drawer level */
  className: PropTypes.string,

  /** Additional style to root level */
  style: PropTypes.shape(),

  /** drawer width */
  width: PropTypes.number,

  /** indicator whether or not drawer should overlay over content (position absolute) or static
   * *Note - overlay auto toggles parent component with position relative
   * */
  overlay: PropTypes.bool,

  /** Only applicable for overlay = true -
   *  Indicator whether or not we want to open drawer on right - default left
   *  */
  right: PropTypes.bool,

  /** The drawer paper elevation level * */
  elevation: PropTypes.number,

  /** Background color for drawer */
  bgColor: PropTypes.string,

  /** Add class to root level */
  containerClassName: PropTypes.string,

  /** Set drawer as fixed */
  fixed: PropTypes.bool,

  /** Callback hook for the Drawer slide transition end */
  onDrawerTransitionEnd: PropTypes.func
};
export default Drawer;
