// @guestyci/foundation/esm/legacy/MultiSelectDropdown/MultiSelectDropdown.js
import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import noop from 'lodash/noop';
import DropdownInputField from "../baseDropdown/DropdownInputField";
import DropdownMenu from "../baseDropdown/DropdownMenu";
import withDropdown from "../baseDropdown/withDropdown";
import withMultiSelect from "../baseDropdown/withMultiSelect";
import MultiSelectDropdownSelectedMenu from "./MultiSelectDropdownSelectedMenu";
import MultiSelectDropdownMenuItem from "./MultiSelectDropdownMenuItem";

var MultiSelectDropdown = function MultiSelectDropdown(_ref) {
  var toggleCollapse = _ref.toggleCollapse,
      isCollapsed = _ref.isCollapsed,
      name = _ref.name,
      className = _ref.className,
      onSearchChange = _ref.onSearchChange,
      clearQuery = _ref.clearQuery,
      closeDropdown = _ref.closeDropdown,
      style = _ref.style,
      isFetching = _ref.isFetching,
      fetchNextPage = _ref.fetchNextPage,
      pageCount = _ref.pageCount,
      currentPage = _ref.currentPage,
      query = _ref.query,
      placeholder = _ref.placeholder,
      searchInputValue = _ref.searchInputValue,
      clearSelected = _ref.clearSelected,
      unselectedOptions = _ref.unselectedOptions,
      selectedOptions = _ref.selectedOptions,
      onChange = _ref.onChange,
      disabled = _ref.disabled,
      emptyStateComponent = _ref.emptyStateComponent;
  return /*#__PURE__*/React.createElement("div", {
    className: cn('d-flex-fill relative flex-column', className),
    style: style,
    name: name,
    "data-qa": "multi-select-dropdown"
  }, /*#__PURE__*/React.createElement(DropdownInputField, {
    query: query,
    name: name,
    value: searchInputValue,
    onQueryChange: onSearchChange,
    onIconClick: toggleCollapse,
    isOpen: !isCollapsed,
    searchable: true,
    onClearQuery: clearQuery,
    onClearSelected: clearSelected,
    placeholder: placeholder,
    disabled: disabled
  }), /*#__PURE__*/React.createElement(DropdownMenu, {
    pageCount: pageCount,
    onDismiss: closeDropdown,
    isEmptyState: !unselectedOptions.length,
    emptyStateComponent: emptyStateComponent,
    currentPage: currentPage,
    onLoadMore: fetchNextPage,
    isFetching: isFetching,
    isCollapsed: isCollapsed
  }, /*#__PURE__*/React.createElement(MultiSelectDropdownSelectedMenu, {
    options: selectedOptions,
    onChange: onChange
  }), unselectedOptions.map(function (option) {
    return /*#__PURE__*/React.createElement(MultiSelectDropdownMenuItem, {
      key: option.value,
      checked: false,
      onChange: onChange,
      value: option
    }, option.label);
  })));
};

MultiSelectDropdown.defaultProps = {
  values: [],
  value: [],
  options: [],
  single: 'more',
  plural: 'more',
  className: '',
  id: '',
  name: '',
  onLoadMore: noop,
  isFetching: false,
  pageCount: 0,
  placeholder: 'Select or type',
  onChange: noop,
  onSelect: noop,
  disabled: false,
  emptyStateComponent: 'No results found.'
};
MultiSelectDropdown.propTypes = {
  /** Callback function to be called on input value changed - Deprecated */
  onSelect: PropTypes.func,

  /** Callback function to be called on input value changed */
  onChange: PropTypes.func,

  /** Array to populate the dropdown selected list - DEPRECATED USE VALUE  */
  values: PropTypes.arrayOf(PropTypes.string),

  /** Array to populate the dropdown selected list  */
  value: PropTypes.arrayOf(PropTypes.string),

  /** Array, current selected values - structure :
   *  { label: 'component' ? name prop is mandatory / 'String'.
   *    value: the value return to the parent */
  options: PropTypes.arrayOf(PropTypes.object),

  /** Callback function to be called on input value changed */
  onSearchQueryChange: PropTypes.func.isRequired,

  /** Callback function to be called on dropdown visibility change */
  onVisibilityChange: PropTypes.func,

  /** String, to display placeholder with 2 values */
  single: PropTypes.string,

  /** String, to display placeholder with more than 2 values */
  plural: PropTypes.string,

  /** String, classes to pass */
  className: PropTypes.string,

  /** String, identifier */
  name: PropTypes.string,

  /** String, identifier */
  id: PropTypes.string,

  /** Callback function to fetch more items - required for infinite scroll */
  onLoadMore: PropTypes.func,

  /** Total page count for pagination - required to allow infinite scroll */
  pageCount: PropTypes.number,

  /** Indicator whether or not we are currently fetching value for infinite scroll */
  isFetching: PropTypes.bool,

  /** Placeholder for multi select dropdown */
  placeholder: PropTypes.string,

  /** Empty state component  to display when there're no results */
  emptyStateComponent: PropTypes.node,

  /** Indicator for disabled status */
  disabled: PropTypes.bool
};
export var MultiSelectDropdownComponent = MultiSelectDropdown;
export default withDropdown(withMultiSelect(MultiSelectDropdown));
