// @guestyci/redux-tools/actions/api.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.apiError = exports.apiSuccess = exports.apiRequest = exports.API_ERROR = exports.API_SUCCESS = exports.API_REQUEST = void 0;
// action types
var API_REQUEST = 'API_REQUEST';
exports.API_REQUEST = API_REQUEST;
var API_SUCCESS = 'API_SUCCESS';
exports.API_SUCCESS = API_SUCCESS;
var API_ERROR = 'API_ERROR'; // action creators

exports.API_ERROR = API_ERROR;

var apiRequest = function apiRequest(_ref) {
  var body = _ref.body,
      method = _ref.method,
      url = _ref.url,
      feature = _ref.feature;
  return {
    type: "".concat(feature, " ").concat(API_REQUEST),
    payload: body,
    meta: {
      method: method,
      url: url,
      feature: feature
    }
  };
};

exports.apiRequest = apiRequest;

var apiSuccess = function apiSuccess(_ref2) {
  var response = _ref2.response,
      feature = _ref2.feature;
  return {
    type: "".concat(feature, " ").concat(API_SUCCESS),
    payload: response,
    meta: {
      feature: feature
    }
  };
};

exports.apiSuccess = apiSuccess;

var apiError = function apiError(_ref3) {
  var error = _ref3.error,
      response = _ref3.response,
      feature = _ref3.feature;
  return {
    type: "".concat(feature, " ").concat(API_ERROR),
    payload: {
      error: error,
      response: response
    },
    meta: {
      feature: feature
    }
  };
};

exports.apiError = apiError;
