// @guestyci/foundation/withHover/withHover.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _react = _interopRequireWildcard(require("react"));

function _createSuper(Derived) { return function () { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

/**
 * Create a wrapper compose to support Hover functionality
 * @param ComposedComponent
 * @returns {{Component}}
 */
function withHover(ComposedComponent) {
  var HoverWrapper = /*#__PURE__*/function (_Component) {
    (0, _inherits2["default"])(HoverWrapper, _Component);

    var _super = _createSuper(HoverWrapper);

    function HoverWrapper(props) {
      var _this;

      (0, _classCallCheck2["default"])(this, HoverWrapper);
      _this = _super.call(this, props);

      _this.onHoverLeave = function () {
        _this.setState({
          isHovering: false
        });
      };

      _this.onHoverEnter = function () {
        _this.setState({
          isHovering: true
        });
      };

      _this.state = {
        isHovering: false
      };
      _this.hoverRef = _react["default"].createRef();
      return _this;
    }

    (0, _createClass2["default"])(HoverWrapper, [{
      key: "componentDidMount",
      value: function componentDidMount() {
        var node = this.hoverRef.current;

        if (node) {
          node.addEventListener('mouseover', this.onHoverEnter);
          node.addEventListener('mouseleave', this.onHoverLeave);
          node.addEventListener('blur', this.onHoverLeave);
        }
      }
    }, {
      key: "componentWillUnmount",
      value: function componentWillUnmount() {
        var node = this.hoverRef.current;

        if (node) {
          node.removeEventListener('mouseover', this.onHoverEnter);
          node.removeEventListener('mouseleave', this.onHoverLeave);
          node.removeEventListener('blur', this.onHoverLeave);
        }
      }
    }, {
      key: "render",
      value: function render() {
        var isHovering = this.state.isHovering;
        return /*#__PURE__*/_react["default"].createElement(ComposedComponent, (0, _extends2["default"])({}, this.props, {
          isHovering: isHovering,
          hoverRef: this.hoverRef,
          onHoverLeave: this.onHoverLeave,
          onHoverEnter: this.onHoverEnter,
          "data-qa": "hover-wrapper"
        }));
      }
    }]);
    return HoverWrapper;
  }(_react.Component);

  return HoverWrapper;
}

var _default = withHover;
exports["default"] = _default;
