// @guestyci/foundation/esm/Layout/styles.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import memoize from '@guestyci/memoize';
import { capitalize } from "../utils/commonUtility";
var COL_SPANS = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
export var spacing = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
export var generateSpacing = memoize(function (spacer) {
  var spacings = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : spacing;
  var merging = arguments.length > 2 ? arguments[2] : undefined;
  return spacings.reduce(function (acc, space) {
    var spacingObj = _defineProperty({}, "spacing".concat(space), {
      '& > * ': _defineProperty({}, "margin".concat(capitalize(merging)), spacer(space))
    });

    return _extends({}, acc, {}, spacingObj);
  }, {});
});
export var generateMediaSupport = memoize(function () {
  var breakpoints = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {
    keys: []
  };
  return breakpoints.keys.reduce(function (acc, breakpoint) {
    var styles = {};
    var control = {};
    COL_SPANS.forEach(function (span) {
      var key = "col-".concat(breakpoint, "-").concat(span); // Keep 7 significant numbers.

      var width = "".concat(Math.round(span / 12 * 10e7) / 10e5, "%");
      styles[key] = {
        flexBasis: width,
        flexGrow: 0,
        maxWidth: width
      };
      control[key] = {
        display: 'flex'
      };
    });

    if (breakpoint === 'xs') {
      styles["col-".concat(breakpoint, "-auto")] = {
        flexGrow: 1
      };
      Object.assign(acc, styles);
    } else {
      acc[breakpoints.create(breakpoint)] = styles;
      Object.assign(acc, control);
    }

    return acc;
  }, {});
});
export default {
  flexWrap: {
    flexWrap: 'wrap'
  },
  // justify styling
  justifyBetween: {
    justifyContent: 'space-between'
  },
  justifyAround: {
    justifyContent: 'space-around'
  },
  justifyEvenly: {
    justifyContent: 'space-evenly'
  },
  justifyCenter: {
    justifyContent: 'center'
  },
  justifyStart: {
    justifyContent: 'flex-start'
  },
  justifyEnd: {
    justifyContent: 'flex-end'
  },
  // align styling
  alignItemsBaseline: {
    alignItems: 'baseline'
  },
  alignItemsCenter: {
    alignItems: 'center'
  },
  alignItemsStart: {
    alignItems: 'flex-start'
  },
  alignItemsEnd: {
    alignItems: 'flex-end'
  },
  // support full width
  fullWidth: {
    width: '100%'
  }
};
