// @guestyci/foundation/legacy/Person/PersonBadge.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _Badge = _interopRequireDefault(require("../Badge/Badge"));

var BtnStarOn = function BtnStarOn(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M7.5 0l2.319 4.608L15 5.347l-3.75 3.588.887 5.065-4.638-2.393L2.864 14l.886-5.065L0 5.347l5.183-.74z"
  }));
};

BtnStarOn.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 15 14"
};

var PersonBadge = function PersonBadge() {
  return /*#__PURE__*/_react["default"].createElement(_Badge["default"], {
    bgColor: "black",
    svg: BtnStarOn,
    iconWidth: 11,
    iconHeight: 11,
    color: "white",
    "data-qa": "person-badge"
  });
};

var _default = PersonBadge;
exports["default"] = _default;
