// @guestyci/foundation/legacy/Person/PersonReturningGuestPill.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _SvgIcon = _interopRequireDefault(require("../SvgIcon/SvgIcon"));

var _Pill = _interopRequireDefault(require("../Pill/Pill"));

var _Row = _interopRequireDefault(require("../Row/Row"));

var _TextField = _interopRequireDefault(require("../TextField/TextField"));

var BtnStarOn = function BtnStarOn(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M7.5 0l2.319 4.608L15 5.347l-3.75 3.588.887 5.065-4.638-2.393L2.864 14l.886-5.065L0 5.347l5.183-.74z"
  }));
};

BtnStarOn.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 15 14"
};

var PersonReturningGuestPill = function PersonReturningGuestPill(_ref) {
  var className = _ref.className;
  return /*#__PURE__*/_react["default"].createElement(_Pill["default"], {
    color: "white",
    className: className,
    bgColor: "black",
    "data-qa": "person-returning-guest-pill"
  }, /*#__PURE__*/_react["default"].createElement(_Row["default"], {
    spacing: "sm",
    align: "center"
  }, /*#__PURE__*/_react["default"].createElement(_SvgIcon["default"], {
    svg: BtnStarOn,
    color: "white",
    width: 12,
    height: 12,
    style: {
      marginBottom: 2
    }
  }), /*#__PURE__*/_react["default"].createElement(_TextField["default"], {
    size: "md"
  }, "Returning guest")));
};

var _default = PersonReturningGuestPill;
exports["default"] = _default;
