// @guestyci/foundation/es/legacy/Pill/Pill.js
import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import { createUseStyles } from 'react-jss';
const useStyles = createUseStyles({
  pill: {
    height: ({
      height
    }) => height,
    width: ({
      width
    }) => width,
    borderRadius: ({
      borderRadius
    }) => borderRadius
  }
});

const Pill = ({
  className,
  color,
  width,
  height,
  bgColor,
  borderColor,
  textHover,
  bgHover,
  borderHover,
  filter,
  children,
  borderRadius
}) => {
  const classes = useStyles({
    width,
    height,
    borderRadius
  });
  return /*#__PURE__*/React.createElement("div", {
    className: cn(className, classes.pill, {
      [`text-${color}`]: color,
      [`border-${borderColor}`]: borderColor,
      [`bg-${bgColor}`]: bgColor,
      [`text-hover-${textHover}`]: textHover,
      [`border-hover-${borderHover}`]: borderHover,
      [`bg-hover-${bgHover}`]: bgHover,
      'b-1': borderColor,
      'bg-hover-blue box-shadow-hover-heavy box-shadow-active-0': filter
    }, 'px-2  pill-element overflow-ellipsis w-fit-content flex-center'),
    "data-qa": "pill"
  }, children);
};

Pill.defaultProps = {
  className: '',
  color: 'white',
  bgColor: 'black',
  borderColor: '',
  textHover: '',
  bgHover: '',
  borderHover: '',
  filter: false,
  children: null,
  width: undefined,
  height: 21,
  borderRadius: 10
};
Pill.propTypes = {
  /** ClassName url for pill */
  className: PropTypes.string,

  /** Text for pill */
  color: PropTypes.string,

  /** Background Color for pill */
  bgColor: PropTypes.string,

  /** borderColor for pill */
  borderColor: PropTypes.string,

  /** textHover for pill */
  textHover: PropTypes.string,

  /** Background on hover for pill */
  bgHover: PropTypes.string,

  /** borderHover for pill */
  borderHover: PropTypes.string,

  /** Is it a filter */
  filter: PropTypes.bool,

  /** Array or nodes , or nodes */
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),

  /** Set the pill width */
  width: PropTypes.number,

  /** Set the pill height */
  height: PropTypes.number,

  /** Set the pill border radius */
  borderRadius: PropTypes.number
};
export default Pill;
