// @guestyci/foundation/utils/fileUploadUtility.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.openUploadWidget = exports.createUploadWidget = exports.setUploadFolder = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));

var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));

var _agni = _interopRequireDefault(require("@guestyci/agni"));

var _commonUtility = require("./commonUtility");

var _constants = require("../constants/constants");

var getFromTokenIssuer = function getFromTokenIssuer() {
  var issuer = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : '';
  return issuer.split('.')[0];
};

var _Resource$create = _agni["default"].create({
  domain: '/user-generated-content/api'
}),
    api = _Resource$create.api;

var generateSignature = /*#__PURE__*/function () {
  var _ref = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee(callback, params) {
    var _yield$api$post, data;

    return _regenerator["default"].wrap(function _callee$(_context) {
      while (1) {
        switch (_context.prev = _context.next) {
          case 0:
            _context.prev = 0;
            _context.next = 3;
            return api.post('/media-provider/generate-signature', params);

          case 3:
            _yield$api$post = _context.sent;
            data = _yield$api$post.data;
            callback(data.signature);
            _context.next = 11;
            break;

          case 8:
            _context.prev = 8;
            _context.t0 = _context["catch"](0);
            console.log('error', _context.t0);

          case 11:
          case "end":
            return _context.stop();
        }
      }
    }, _callee, null, [[0, 8]]);
  }));

  return function generateSignature(_x, _x2) {
    return _ref.apply(this, arguments);
  };
}();

var setUploadFolder = function setUploadFolder() {
  var _ref2 = (0, _commonUtility.getUserData)() || {},
      accountId = _ref2.accountId,
      iss = _ref2.iss;

  if (!accountId || !iss) {
    return 'general';
  }

  return "".concat(getFromTokenIssuer(iss), "/").concat(accountId);
};
/**
 * Create a cloudinary upload widget
 * @param onSuccess  {Function} - CB function to run on success
 * @param onFailure {Function} - CB function to run on failure
 * @param options = { Object} - options to pass to the widget
 * @return {Object} Cloudinary widget object
 */


exports.setUploadFolder = setUploadFolder;

var createUploadWidget = function createUploadWidget(_ref3) {
  var onSuccess = _ref3.onSuccess,
      onFailure = _ref3.onFailure,
      options = (0, _objectWithoutProperties2["default"])(_ref3, ["onSuccess", "onFailure"]);

  if (window.cloudinary) {
    var _window = window,
        cloudinary = _window.cloudinary;
    return cloudinary.createUploadWidget((0, _extends2["default"])({
      cloudName: _constants.CLOUD_NAME,
      apiKey: _constants.CLOUDINARY_API_KEY,
      uploadSignature: generateSignature,
      sources: ['local'],
      showPoweredBy: false,
      folder: setUploadFolder()
    }, options), function (uploadError, result) {
      if (uploadError && onFailure) {
        onFailure(uploadError);
      }

      if (result && result.event === 'success') {
        onSuccess(result.info);
      }
    });
  }
};
/**
 * Open a cloudinary upload widget
 * @param onSuccess  {Function} - CB function to run on success
 * @param onFailure {Function} - CB function to run on failure
 * @param options = { Object} - options to pass to the widget
 */


exports.createUploadWidget = createUploadWidget;

var openUploadWidget = function openUploadWidget(_ref4) {
  var onSuccess = _ref4.onSuccess,
      onFailure = _ref4.onFailure,
      options = (0, _objectWithoutProperties2["default"])(_ref4, ["onSuccess", "onFailure"]);

  if (window.cloudinary) {
    window.cloudinary.openUploadWidget((0, _extends2["default"])({
      cloudName: _constants.CLOUD_NAME,
      api_key: _constants.CLOUDINARY_API_KEY,
      sources: ['local'],
      showPoweredBy: false,
      uploadSignature: generateSignature,
      folder: setUploadFolder()
    }, options), function (uploadError, result) {
      if (uploadError && onFailure) {
        onFailure(uploadError);
      }

      if (result && result.event === 'success') {
        onSuccess(result.info);
      }
    });
  }
};

exports.openUploadWidget = openUploadWidget;
