// @guestyci/foundation/utils/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "uuid", {
  enumerable: true,
  get: function get() {
    return _commonUtility.uuid;
  }
});
Object.defineProperty(exports, "firstLetterUppercase", {
  enumerable: true,
  get: function get() {
    return _commonUtility.firstLetterUppercase;
  }
});
Object.defineProperty(exports, "getUserData", {
  enumerable: true,
  get: function get() {
    return _commonUtility.getUserData;
  }
});
Object.defineProperty(exports, "isEmpty", {
  enumerable: true,
  get: function get() {
    return _commonUtility.isEmpty;
  }
});
Object.defineProperty(exports, "isNumber", {
  enumerable: true,
  get: function get() {
    return _commonUtility.isNumber;
  }
});
Object.defineProperty(exports, "link", {
  enumerable: true,
  get: function get() {
    return _commonUtility.link;
  }
});
Object.defineProperty(exports, "toMap", {
  enumerable: true,
  get: function get() {
    return _commonUtility.toMap;
  }
});
Object.defineProperty(exports, "nameToInitials", {
  enumerable: true,
  get: function get() {
    return _commonUtility.nameToInitials;
  }
});

var _commonUtility = require("./commonUtility");
