// @guestyci/foundation/legacy/Calendar/CalendarDay.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _isEmpty = _interopRequireDefault(require("lodash/isEmpty"));

var _calendarHelper = require("./calendarHelper");

var _withHover = _interopRequireDefault(require("../../withHover/withHover"));

var _Tooltip = _interopRequireDefault(require("../Tooltip/Tooltip"));

var _enums = require("../../enums/enums");

var _commonUtility = require("../../utils/commonUtility");

var _CalendarTooltip = _interopRequireDefault(require("./CalendarTooltip"));

var _CalendarDaySMBody = _interopRequireDefault(require("./CalendarDaySMBody"));

var _CalendarDayLGBody = _interopRequireDefault(require("./CalendarDayLGBody"));

function _createSuper(Derived) { return function () { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

var CalendarDay = /*#__PURE__*/function (_PureComponent) {
  (0, _inherits2["default"])(CalendarDay, _PureComponent);

  var _super = _createSuper(CalendarDay);

  function CalendarDay(props) {
    var _this;

    (0, _classCallCheck2["default"])(this, CalendarDay);
    _this = _super.call(this, props);

    _this.goToLink = function () {
      var calendarDayData = _this.state.calendarDayData;

      if ((calendarDayData.dayStatus === _enums.CalendarDayStatus.Booked || calendarDayData.dayStatus === _enums.CalendarDayStatus.Reserved || calendarDayData.blocks && (calendarDayData.blocks.sr || calendarDayData.blocks.o)) && calendarDayData.link) {
        (0, _commonUtility.link)(calendarDayData.link);
      }
    };

    _this.serializeReservationData = function (reservation, status, isReserved, isOwner, isSR) {
      var data = (0, _extends2["default"])({}, reservation);

      if (!(0, _isEmpty["default"])(data)) {
        data.plannedArrival = (0, _calendarHelper.getStayDateTime)(data.checkIn, data.plannedArrival);

        if (isReserved) {
          data.title = (0, _calendarHelper.getReservationTitle)(data.confirmationCode, data.source, isOwner, isSR);
        }

        data.plannedDeparture = (0, _calendarHelper.getStayDateTime)(data.checkOut, data.plannedDeparture);
        data.isValidCheckIn = (0, _calendarHelper.isInvalidPlannedDate)(data.plannedArrival, data.checkIn);
        data.isValidCheckOut = (0, _calendarHelper.isInvalidPlannedDate)(data.checkOut, data.plannedDeparture);
      }

      return data;
    };

    _this.showTooltip = function (sm, cost, availableNotes, unavailableBlocks, isReserved) {
      return sm && !(0, _isEmpty["default"])(cost) || !(0, _isEmpty["default"])(availableNotes) || !(0, _isEmpty["default"])(unavailableBlocks) || isReserved;
    };

    _this.renderCalendarTooltip = function () {
      var calendarDayData = _this.state.calendarDayData;
      var sm = _this.props.sm;
      var cost = calendarDayData.cost,
          availableNotes = calendarDayData.availableNotes,
          dayStatus = calendarDayData.dayStatus,
          isReserved = calendarDayData.isReserved,
          unavailableBlocks = calendarDayData.unavailableBlocks,
          isOwner = calendarDayData.isOwner,
          isSR = calendarDayData.isSR,
          reservation = calendarDayData.reservation;

      var data = _this.serializeReservationData(reservation, dayStatus, isReserved, isOwner, isSR);

      if (_this.showTooltip(sm, cost, availableNotes, unavailableBlocks, isReserved)) {
        return /*#__PURE__*/_react["default"].createElement(_CalendarTooltip["default"], {
          data: data,
          blocks: unavailableBlocks,
          isReserved: isReserved,
          price: sm ? cost : undefined,
          note: availableNotes
        });
      }

      return null;
    };

    _this.state = {
      d: (0, _calendarHelper.formatCalendarDay)(props.day, true),
      m: (0, _calendarHelper.formatCalendarMonth)(props.day, true),
      calendarDayData: {}
    };
    return _this;
  }

  (0, _createClass2["default"])(CalendarDay, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      var _this$props = this.props,
          calendar = _this$props.calendar,
          day = _this$props.day,
          activeDates = _this$props.activeDates;
      this.initCalendarDay(calendar, day, activeDates);
    }
  }, {
    key: "componentDidUpdate",
    value: function componentDidUpdate(prevProps) {
      var _this$props2 = this.props,
          calendar = _this$props2.calendar,
          day = _this$props2.day,
          activeDates = _this$props2.activeDates;

      if ((0, _isEmpty["default"])(prevProps.calendar) && !(0, _isEmpty["default"])(calendar) || prevProps.day !== day) {
        this.initCalendarDay(calendar, day, activeDates);
      }
    }
  }, {
    key: "initCalendarDay",
    value: function initCalendarDay(calendar, day, activeDates) {
      var calendarDayData = (0, _calendarHelper.checkReservationDetails)(calendar, day, activeDates);
      this.setState({
        calendarDayData: calendarDayData
      });
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props3 = this.props,
          onHoverEnter = _this$props3.onHoverEnter,
          onHoverLeave = _this$props3.onHoverLeave,
          isHovering = _this$props3.isHovering,
          sm = _this$props3.sm,
          isFetching = _this$props3.isFetching,
          tooltipBoundariesElement = _this$props3.tooltipBoundariesElement;
      var _this$state = this.state,
          d = _this$state.d,
          m = _this$state.m,
          calendarDayData = _this$state.calendarDayData;
      var availableNotes = calendarDayData.availableNotes,
          startStyles = calendarDayData.startStyles,
          endStyles = calendarDayData.endStyles,
          endOpacity = calendarDayData.endOpacity,
          startOpacity = calendarDayData.startOpacity,
          startOfBlock = calendarDayData.startOfBlock,
          color = calendarDayData.color,
          link = calendarDayData.link,
          guest = calendarDayData.guest,
          showPrice = calendarDayData.showPrice,
          cost = calendarDayData.cost,
          isAvailable = calendarDayData.isAvailable,
          showAvatar = calendarDayData.showAvatar;
      return /*#__PURE__*/_react["default"].createElement("div", {
        className: (0, _classnames["default"])('DayPicker-Day--wrap', {
          lg: !sm
        }),
        "data-qa": "calendar-day"
      }, !isFetching && startStyles && /*#__PURE__*/_react["default"].createElement("div", {
        className: (0, _classnames["default"])('DayPicker-Day--range', startStyles),
        style: {
          opacity: startOpacity
        }
      }), !isFetching && endStyles && /*#__PURE__*/_react["default"].createElement("div", {
        className: (0, _classnames["default"])('DayPicker-Day--range', endStyles),
        style: {
          opacity: endOpacity
        }
      }), /*#__PURE__*/_react["default"].createElement(_Tooltip["default"], {
        boundariesElement: tooltipBoundariesElement,
        id: "cell-".concat(sm ? 'sm' : 'lg', "-").concat(d, "-").concat(m),
        containerClassName: "w-fill h-fill",
        body: this.renderCalendarTooltip(),
        delay: {
          show: 300,
          hide: 250
        }
      }, /*#__PURE__*/_react["default"].createElement("div", {
        role: "presentation",
        onClick: this.goToLink,
        onMouseEnter: onHoverEnter,
        onMouseLeave: onHoverLeave,
        className: (0, _classnames["default"])('flex-center flex-column font-size-xs calendar-day-content w-fill h-fill', {
          'text-hover-blue': isAvailable
        }, "text-".concat(color), {
          clickable: link
        })
      }, sm ? /*#__PURE__*/_react["default"].createElement(_CalendarDaySMBody["default"], {
        isHovering: isAvailable && isHovering,
        hasNote: availableNotes,
        day: isFetching ? '-' : d
      }) : /*#__PURE__*/_react["default"].createElement(_CalendarDayLGBody["default"], {
        isHovering: isAvailable && isHovering,
        hasNote: availableNotes,
        day: isFetching ? '-' : d,
        price: showPrice ? cost : undefined,
        personOpacity: startOpacity,
        startOfBlock: startOfBlock,
        guest: guest,
        showAvatar: showAvatar
      }))));
    }
  }]);
  return CalendarDay;
}(_react.PureComponent);

CalendarDay.defaultProps = {
  isFetching: false,
  sm: false,
  tooltipBoundariesElement: undefined
};
CalendarDay.propTypes = {
  day: _propTypes["default"].instanceOf(Date).isRequired,
  calendar: _propTypes["default"].shape().isRequired,
  activeDates: _propTypes["default"].shape().isRequired,
  sm: _propTypes["default"].bool,
  isFetching: _propTypes["default"].bool,
  tooltipBoundariesElement: _propTypes["default"].string
};

var _default = (0, _withHover["default"])(CalendarDay);

exports["default"] = _default;
