// @guestyci/foundation/legacy/Calendar/CalendarTooltip.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _ListItem = _interopRequireDefault(require("../List/ListItem"));

var _CalendarTooltipNote = _interopRequireDefault(require("./CalendarTooltipNote"));

var _CalendarTooltipPrice = _interopRequireDefault(require("./CalendarTooltipPrice"));

var _List = _interopRequireDefault(require("../List/List"));

var _CalendarTooltipBlock = _interopRequireDefault(require("./CalendarTooltipBlock"));

var _CalendarTooltipReserved = _interopRequireDefault(require("./CalendarTooltipReserved"));

var CalendarTooltip = function CalendarTooltip(_ref) {
  var price = _ref.price,
      note = _ref.note,
      blocks = _ref.blocks,
      isReserved = _ref.isReserved,
      data = _ref.data;
  return /*#__PURE__*/_react["default"].createElement(_List["default"], {
    className: "calendar-tooltip-list",
    "data-qa": "calendar-tooltip"
  }, !isReserved && note && /*#__PURE__*/_react["default"].createElement(_ListItem["default"], {
    dense: true,
    className: "calendar-tooltip-item"
  }, /*#__PURE__*/_react["default"].createElement(_CalendarTooltipNote["default"], {
    note: note
  })), isReserved && /*#__PURE__*/_react["default"].createElement(_CalendarTooltipReserved["default"], {
    data: data
  }), !isReserved && blocks.map(function (block) {
    return /*#__PURE__*/_react["default"].createElement(_ListItem["default"], {
      key: block,
      dense: true,
      className: "calendar-tooltip-item"
    }, /*#__PURE__*/_react["default"].createElement(_CalendarTooltipBlock["default"], {
      block: block
    }));
  }), !isReserved && price && /*#__PURE__*/_react["default"].createElement(_ListItem["default"], {
    dense: true,
    className: "calendar-tooltip-item"
  }, /*#__PURE__*/_react["default"].createElement(_CalendarTooltipPrice["default"], {
    price: price
  })));
};

CalendarTooltip.defaultProps = {
  price: undefined,
  note: undefined,
  blocks: [],
  isReserved: false,
  data: {}
};
CalendarTooltip.propTypes = {
  price: _propTypes["default"].string,
  note: _propTypes["default"].string,
  blocks: _propTypes["default"].arrayOf(_propTypes["default"].string),
  isReserved: _propTypes["default"].bool,
  data: _propTypes["default"].shape()
};
var _default = CalendarTooltip;
exports["default"] = _default;
