// @guestyci/foundation/legacy/Calendar/CalendarTooltipReserved.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _CalendarTooltipReservedItem = _interopRequireDefault(require("./CalendarTooltipReservedItem"));

var _calendarHelper = require("./calendarHelper");

var CalendarTooltipReserved = function CalendarTooltipReserved(_ref) {
  var data = _ref.data;
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: "flex-start d-flex-fill text-white flex-column",
    "data-qa": "calendar-tooltip-reserved"
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: "text-left font-weight-bold bb-2 w-100 border-black-dark pb-2 mb-1"
  }, data.title), /*#__PURE__*/_react["default"].createElement(_CalendarTooltipReservedItem["default"], {
    title: "Status",
    value: data.status,
    color: (0, _calendarHelper.colorFromStatus)(data.status)
  }), /*#__PURE__*/_react["default"].createElement(_CalendarTooltipReservedItem["default"], {
    title: "ETA",
    value: data.plannedArrival,
    alert: data.isValidCheckIn
  }), /*#__PURE__*/_react["default"].createElement(_CalendarTooltipReservedItem["default"], {
    title: "ETD",
    value: data.plannedDeparture,
    alert: data.isValidCheckOut
  }), /*#__PURE__*/_react["default"].createElement(_CalendarTooltipReservedItem["default"], {
    title: "Guests",
    value: data.guestsCount
  }), data.money && /*#__PURE__*/_react["default"].createElement(_CalendarTooltipReservedItem["default"], {
    title: "Payout",
    value: (0, _calendarHelper.getCurrencySymbol)(data.money.currency, data.money.hostPayout)
  }));
};

CalendarTooltipReserved.propTypes = {
  data: _propTypes["default"].shape({
    title: _propTypes["default"].string.isRequired,
    status: _propTypes["default"].string,
    plannedArrival: _propTypes["default"].string.isRequired,
    plannedDeparture: _propTypes["default"].string.isRequired,
    isValidCheckIn: _propTypes["default"].bool,
    isValidCheckOut: _propTypes["default"].bool,
    guestsCount: _propTypes["default"].number,
    money: _propTypes["default"].shape()
  }).isRequired
};
var _default = CalendarTooltipReserved;
exports["default"] = _default;
