// @guestyci/foundation/legacy/Calendar/CalendarTooltipBlock.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var mapMessageToBlock = function mapMessageToBlock(block) {
  var blockMap = {
    m: 'Blocked manually',
    bd: 'Blocked by availability settings',
    abl: 'Blocked by annual booking limit',
    a: 'Blocked by multi unit',
    bw: 'Blocked by availability settings',
    an: 'Blocked by availability settings',
    pt: 'Blocked by availability settings'
  };
  return blockMap[block];
};

var CalendarTooltipBlock = function CalendarTooltipBlock(_ref) {
  var block = _ref.block;
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: "text-left font-weight-bold font-size-md text-white",
    "data-qa": "calendar-tooltip-block"
  }, mapMessageToBlock(block));
};

CalendarTooltipBlock.propTypes = {
  block: _propTypes["default"].string.isRequired
};
var _default = CalendarTooltipBlock;
exports["default"] = _default;
