// @guestyci/foundation/legacy/Calendar/CalendarWeekDayBar.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var CalendarWeekDayBar = function CalendarWeekDayBar(_ref) {
  var weekday = _ref.weekday,
      locale = _ref.locale,
      localeUtils = _ref.localeUtils,
      className = _ref.className,
      style = _ref.style,
      dayCharCount = _ref.dayCharCount;
  var weekdayName = localeUtils.formatWeekdayLong(weekday, locale);
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])('text-muted font-size-md text-capitalize pt-0 pb-2', className),
    style: style,
    "data-qa": "calendar-week-day-bar"
  }, weekdayName.slice(0, dayCharCount));
};

CalendarWeekDayBar.defaultProps = {
  className: '',
  style: {},
  weekday: 0,
  localeUtils: {},
  locale: '',
  dayCharCount: 3
};
CalendarWeekDayBar.propTypes = {
  className: _propTypes["default"].string,
  style: _propTypes["default"].shape(),
  weekday: _propTypes["default"].number,
  localeUtils: _propTypes["default"].shape(),
  locale: _propTypes["default"].string,
  dayCharCount: _propTypes["default"].number
};
var _default = CalendarWeekDayBar;
exports["default"] = _default;
