// @guestyci/foundation/legacy/Calendar/CalendarTooltipNote.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _SvgIcon = _interopRequireDefault(require("../SvgIcon/SvgIcon"));

var IcoNotice = function IcoNotice(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M10.5 0C4.701 0 0 4.701 0 10.5S4.701 21 10.5 21 21 16.299 21 10.5 16.299 0 10.5 0zm-.923 5.53h1.885v3.717c0 .4-.024.795-.071 1.187-.048.392-.11.792-.189 1.199H9.837c-.078-.407-.14-.807-.189-1.2a9.873 9.873 0 0 1-.07-1.186V5.53zm1.98 8.888c-.059.137-.14.255-.245.354a1.17 1.17 0 0 1-.812.319c-.165 0-.316-.028-.455-.084a1.088 1.088 0 0 1-.601-.589c-.059-.136-.088-.282-.088-.439a1.076 1.076 0 0 1 .329-.786 1.145 1.145 0 0 1 .815-.325 1.175 1.175 0 0 1 .813.325 1.082 1.082 0 0 1 .331.786c0 .157-.03.303-.087.439z"
  }));
};

IcoNotice.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 21 21"
};

var CalendarTooltipNote = function CalendarTooltipNote(_ref) {
  var note = _ref.note;
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: "flex-start-center text-left",
    "data-qa": "calendar-tooltip-note"
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: "flex-1"
  }, /*#__PURE__*/_react["default"].createElement(_SvgIcon["default"], {
    svg: IcoNotice,
    color: "gray-lighter"
  })), /*#__PURE__*/_react["default"].createElement("div", {
    className: "text-white font-size-md font-weight-bold ml-2"
  }, note));
};

CalendarTooltipNote.propTypes = {
  note: _propTypes["default"].string.isRequired
};
var _default = CalendarTooltipNote;
exports["default"] = _default;
