// @guestyci/foundation/legacy/Calendar/CalendarDayLGBody.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _Bullet = _interopRequireDefault(require("../Bullet/Bullet"));

var _Person = _interopRequireDefault(require("../Person/Person"));

var CalendarDayLGBody = function CalendarDayLGBody(_ref) {
  var day = _ref.day,
      isHovering = _ref.isHovering,
      hasNote = _ref.hasNote,
      guest = _ref.guest,
      startOfBlock = _ref.startOfBlock,
      opacity = _ref.personOpacity,
      price = _ref.price,
      showAvatar = _ref.showAvatar;
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: "d-flex-fill flex-space-between-center h-fill flex-column pt-1",
    "data-qa": "calendar-day-lg-body"
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: "w-100 flex-end-base h-fill pr-3"
  }, hasNote && /*#__PURE__*/_react["default"].createElement(_Bullet["default"], {
    className: "calendar-day-bullet",
    color: isHovering ? 'blue' : 'muted',
    style: {
      height: 6,
      width: 6
    }
  }), /*#__PURE__*/_react["default"].createElement("div", {
    className: "day pl-1"
  }, day)), showAvatar && /*#__PURE__*/_react["default"].createElement("div", {
    className: "w-100 pl-7 pb-1 flex-start-center"
  }, startOfBlock && guest && /*#__PURE__*/_react["default"].createElement(_Person["default"], {
    style: {
      opacity: opacity
    },
    isUser: true,
    name: guest.name,
    img: guest.avatar,
    size: "sm"
  })), /*#__PURE__*/_react["default"].createElement("div", {
    className: "w-100 flex-end pr-3"
  }, price && /*#__PURE__*/_react["default"].createElement("div", null, price)));
};

var _default = CalendarDayLGBody;
exports["default"] = _default;
