// @guestyci/foundation/legacy/Calendar/CalendarNavBar.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _moment = _interopRequireDefault(require("moment"));

var _IconButton = _interopRequireDefault(require("../IconButton/IconButton"));

function _createSuper(Derived) { return function () { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

var BtnRight = function BtnRight(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M.849.146L5.854 5.15a.497.497 0 0 1 0 .703L.85 10.86a.497.497 0 1 1-.703-.703l4.653-4.654L.146.849A.497.497 0 1 1 .849.146z"
  }));
};

BtnRight.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 6 11.0054388"
};

var BtnLeft = function BtnLeft(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M5.854.146a.497.497 0 0 1 0 .703L1.201 5.503l4.653 4.654a.497.497 0 1 1-.703.703L.146 5.854a.497.497 0 0 1 0-.703L5.15.146a.497.497 0 0 1 .703 0z"
  }));
};

BtnLeft.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 6 11.0054388"
};

var CalendarNavBar = /*#__PURE__*/function (_PureComponent) {
  (0, _inherits2["default"])(CalendarNavBar, _PureComponent);

  var _super = _createSuper(CalendarNavBar);

  function CalendarNavBar() {
    var _this;

    (0, _classCallCheck2["default"])(this, CalendarNavBar);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _super.call.apply(_super, [this].concat(args));

    _this.handlePrevClick = function () {
      var onPreviousClick = _this.props.onPreviousClick;

      if (onPreviousClick) {
        onPreviousClick();
      }
    };

    _this.handleNextClick = function () {
      var onNextClick = _this.props.onNextClick;

      if (onNextClick) {
        onNextClick();
      }
    };

    return _this;
  }

  (0, _createClass2["default"])(CalendarNavBar, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          className = _this$props.className,
          style = _this$props.style,
          previousMonth = _this$props.previousMonth;
      return /*#__PURE__*/_react["default"].createElement("div", {
        className: (0, _classnames["default"])('flex-center d-flex-fill', className),
        style: style,
        "data-qa": "calendar-nav-bar"
      }, /*#__PURE__*/_react["default"].createElement(_IconButton["default"], {
        onClick: this.handlePrevClick,
        className: "p-2",
        svg: BtnLeft,
        iconWidth: 6,
        iconHeight: 11
      }), /*#__PURE__*/_react["default"].createElement("div", {
        className: "font-size-lg mt-1 text-center min-width-115 text-muted mx-2"
      }, (0, _moment["default"])(previousMonth).add(1, 'M').format('MMMM YYYY')), /*#__PURE__*/_react["default"].createElement(_IconButton["default"], {
        onClick: this.handleNextClick,
        className: "p-2",
        svg: BtnRight,
        iconWidth: 6,
        iconHeight: 11
      }));
    }
  }]);
  return CalendarNavBar;
}(_react.PureComponent);

CalendarNavBar.defaultProps = {
  className: '',
  style: {},
  onPreviousClick: null,
  previousMonth: undefined,
  onNextClick: null
};
CalendarNavBar.propTypes = {
  onPreviousClick: _propTypes["default"].func,
  onNextClick: _propTypes["default"].func,
  className: _propTypes["default"].string,
  style: _propTypes["default"].shape(),
  previousMonth: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].instanceOf(Date), _propTypes["default"].number])
};
var _default = CalendarNavBar;
exports["default"] = _default;
