// @guestyci/foundation/es/legacy/SearchBar/withSearch.js
import _extends from "@babel/runtime/helpers/esm/extends";
import React, { Component } from 'react';
/**
 * Create a wrapper compose to support searchBar functionality
 * @param ComposedComponent
 * @returns {{Component}}
 */

function withSearch(ComposedComponent) {
  class SearchWrapper extends Component {
    constructor(props) {
      super(props);

      this.onSearchClose = () => {
        this.setState({
          isSearchOpen: false
        });
      };

      this.onSearchOpen = () => {
        this.setState({
          isSearchOpen: true
        });
      };

      this.onSearchToggle = val => {
        this.setState({
          isSearchOpen: val
        });
      };

      this.state = {
        isSearchOpen: false
      };
    }

    render() {
      const {
        isSearchOpen
      } = this.state;
      return /*#__PURE__*/React.createElement(ComposedComponent, _extends({}, this.props, {
        isSearchOpen: isSearchOpen,
        onSearchClose: this.onSearchClose,
        onSearchOpen: this.onSearchOpen,
        onSearchToggle: this.onSearchToggle,
        "data-qa": "search-wrapper"
      }));
    }

  }

  return SearchWrapper;
}

export default withSearch;
