// @guestyci/foundation/legacy/Table/Table.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.useStyles = void 0;

var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _noop = _interopRequireDefault(require("lodash/noop"));

var _useTranslations5 = _interopRequireDefault(require("@guestyci/localize/useTranslations"));

var _utils = require("../../utils");

var _EmptyState = _interopRequireDefault(require("../EmptyState"));

var _useHorizontalScrollState = _interopRequireDefault(require("./useHorizontalScrollState"));

var _tableUtility = require("./tableUtility");

var _HeaderRow = _interopRequireDefault(require("./HeaderRow"));

var _CellRow = _interopRequireDefault(require("./CellRow"));

var _useInfiniteScroll = _interopRequireDefault(require("../../useInfiniteScroll"));

var _LoadingIndicator = _interopRequireDefault(require("./LoadingIndicator"));

var _createStyles = _interopRequireDefault(require("../../createStyles"));

var Rug = function Rug(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("g", {
    fill: "none",
    fillRule: "evenodd"
  }, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M.062 89.255h6M0 81.87h6m-6-7.386h6m-6-7.387h6m-6-7.386h6m-6-7.386h6M0 44.94h6m-6-7.386h6m-6-7.386h6m-6-7.386h6m-6-7.385h6M0 8.01h6M0 .625h6M128 44.94h6m-6-7.386h6m-6-7.386h6m-6-7.386h6m-6-7.385h6m-6-7.387h6m-6-7.385h6",
    stroke: "#AEB6BA"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#FF5F5F",
    d: "M5.197.063v89.775h67.272l.57-1.209 54.11-41.435h1.489V.064z"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#30ABE5",
    d: "M117.192 9.265v44.888L84.705 78.84H16.194V9.265z"
  }), /*#__PURE__*/_react["default"].createElement("g", {
    fill: "#48BDF3"
  }, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M40.391 65.912l7.07 7.07 7.072-7.07-7.072-7.07zm14.106-.142l7.071 7.071 7.072-7.07-7.072-7.07zM27.462 78.841h11.54l-5.77-5.77zm13.965 0H53.25l-5.91-5.91zM68.764 37.54l7.07 7.07 7.073-7.07-7.072-7.072zm14.106-.141l7.072 7.07 7.07-7.07-7.07-7.071z"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    d: "M54.533 51.77l7.071 7.07 7.072-7.07-7.072-7.071zm14.107-.14l7.07 7.07 7.072-7.07-7.071-7.072zM55.393 78.841h12.07l-6.035-6.035zm13.966 0h12.35l-6.174-6.176zM82.73 51.505l7.07 7.07 7.072-7.07-7.072-7.07z"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    d: "M68.5 65.735l7.07 7.07 7.071-7.07-7.072-7.07zm28.735-56.47l6.973 6.973 6.973-6.973zM82.906 23.396l7.07 7.072 7.073-7.072-7.073-7.07zm14.106-.14l7.071 7.071 7.072-7.07-7.072-7.072zm14.09-.124l6.09 6.09v-12.18zm-14.23 14.231l7.07 7.07 7.073-7.07-7.072-7.071zm14.106-.141l6.214 6.214V31.007zM16.194 20.109v7.358l3.392 3.393 7.071-7.072-7.071-7.071z"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    d: "M16.194 9.265v4.166l3.428 3.428 7.071-7.071-.522-.523zm10.845 0l-.382.382 7.072 7.07 7.07-7.07-.381-.382zm-10.845 24.81v7.64l3.251 3.25 7.071-7.07-7.071-7.072zm10.286 3.679l7.071 7.07 7.072-7.07-7.072-7.07zM16.194 48.04v7.888l3.128 3.127 7.07-7.071-7.07-7.07zm24.81-38.775l-.257.257 7.07 7.07 7.071-7.07-.257-.257zm13.965 0l-.116.116 7.07 7.07 7.073-7.07-.117-.116z"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    d: "M26.516 23.752l7.071 7.071 7.072-7.071-7.072-7.07zm14.107-.14l7.07 7.07 7.073-7.07-7.072-7.071zM26.428 51.948l7.07 7.071 7.073-7.07-7.072-7.072zm14.106-.14l7.07 7.07 7.073-7.07-7.072-7.071zm-24.34 10.374v7.993l3.074 3.075 7.072-7.07-7.072-7.073zm10.11 3.856l7.07 7.072 7.072-7.072-7.071-7.07zm-10.11 10.11v2.693h8.305L19 73.342zM54.8 23.576l7.071 7.07 7.072-7.07-7.072-7.07zm14.106-.142l7.071 7.071 7.072-7.07-7.072-7.07z"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    d: "M40.57 37.805l7.07 7.071 7.072-7.07-7.072-7.07zm14.106-.14l7.071 7.071 7.072-7.07-7.072-7.072zm14.436-28.4l-.169.169 7.071 7.07 7.071-7.07-.169-.17zm13.966 0l-.03.029 7.072 7.07 7.07-7.07-.028-.03z"
  })), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#0073AD",
    fillOpacity: ".5",
    d: "M73.324 87.421L127.5 47.793 85 32.633z"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    fill: "#E3E8EB",
    d: "M72.584 89.838l56.11-42.644-40.797-11.24z"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    d: "M90.079 30.328l-1.604 5.782m8.737-3.869l-1.603 5.782m8.721-3.808l-1.604 5.782m8.722-3.807l-1.604 5.78m8.721-3.807l-1.604 5.782m8.721-3.808l-1.604 5.781",
    stroke: "#E3E8EB"
  })));
};

Rug.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  width: "134",
  height: "90",
  viewBox: "0 0 134 90"
};
var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    root: {
      position: 'relative',
      overflowY: 'hidden'
    },
    table: {
      display: 'table',
      tableLayout: 'fixed',
      width: '100%',
      borderCollapse: 'separate',
      '& .cell-group-left': {
        '&.cell-group-item-last': {
          boxShadow: "2px 0 3px 0 ".concat(theme.palette.border)
        },
        '&.cell-group-item-no-padding': {
          '& .cell-content, .header-cell-content': {
            paddingLeft: '0px !important'
          }
        }
      },
      '& .cell-group-right.cell-group-item-first': {
        boxShadow: "-2px 0 3px 0 ".concat(theme.palette.border)
      },
      '& .fixed-cell-group': {
        position: 'sticky'
      }
    },
    wrapper: {
      overflow: 'auto',
      height: function height(_ref) {
        var _height = _ref.height;
        return _height || '100%';
      },
      '&.start-shadow:before': {
        content: '""',
        position: 'absolute',
        top: 50,
        bottom: 0,
        left: 0,
        transform: 'scale(1)',
        width: '10px',
        zIndex: 1,
        background: 'radial-gradient(ellipse at left, rgba(0, 0, 0, 0.2) 0%, rgba(0, 0, 0, 0) 75%) 0 center'
      },
      '&.end-shadow:after': {
        content: '""',
        position: 'absolute',
        top: 50,
        transform: 'scale(1)',
        bottom: 0,
        right: 0,
        width: '10px',
        zIndex: 1,
        background: 'radial-gradient(ellipse at right, rgba(0, 0, 0, 0.2) 0%, rgba(0, 0, 0, 0) 75%) 100% center'
      }
    },
    emptyState: {
      margin: 'auto',
      height: 'calc(100% - 60px)'
    },
    errorBanner: {
      bottom: 0,
      width: '100%',
      zIndex: 5,
      position: 'absolute'
    }
  };
}, {
  name: 'Table'
});
exports.useStyles = useStyles;

var areCellsEqual = function areCellsEqual() {
  var columns = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : [];
  var cells = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : [];
  var headerCellsMeta = cells[1];

  if (columns.length !== headerCellsMeta.length) {
    return false;
  }

  for (var i = 0; i < columns.length; i += 1) {
    var _columns$i, _columns$i$props, _headerCellsMeta$i, _headerCellsMeta$i$pr;

    var columnDataKey = (_columns$i = columns[i]) === null || _columns$i === void 0 ? void 0 : (_columns$i$props = _columns$i.props) === null || _columns$i$props === void 0 ? void 0 : _columns$i$props.dataKey;
    var cellDataKey = (_headerCellsMeta$i = headerCellsMeta[i]) === null || _headerCellsMeta$i === void 0 ? void 0 : (_headerCellsMeta$i$pr = _headerCellsMeta$i.props) === null || _headerCellsMeta$i$pr === void 0 ? void 0 : _headerCellsMeta$i$pr.dataKey;

    if (columnDataKey !== cellDataKey) {
      return false;
    }
  }

  return true;
};

var Table = _react["default"].forwardRef(function Table(_ref2, ref) {
  var height = _ref2.height,
      data = _ref2.data,
      children = _ref2.children,
      className = _ref2.className,
      style = _ref2.style,
      onRowClick = _ref2.onRowClick,
      rowClassName = _ref2.rowClassName,
      rowIdDataKey = _ref2.rowIdDataKey,
      onColumnOrderChange = _ref2.onColumnOrderChange,
      draggableColumns = _ref2.draggableColumns,
      sortBy = _ref2.sortBy,
      onSort = _ref2.onSort,
      multiselect = _ref2.multiselect,
      rowSelection = _ref2.rowSelection,
      onCheckedRowsChange = _ref2.onCheckedRowsChange,
      onColumnResize = _ref2.onColumnResize,
      rowComponent = _ref2.rowComponent,
      infiniteScrollOptions = _ref2.infiniteScrollOptions,
      isLoading = _ref2.isLoading,
      errorBanner = _ref2.errorBanner,
      pageSize = _ref2.pageSize,
      jss = _ref2.jss,
      hasScrollShadow = _ref2.hasScrollShadow;
  var classes = useStyles({
    height: height,
    jss: jss
  });
  var cachedCells = (0, _react.useRef)(null);

  var _useTranslations = (0, _useTranslations5["default"])(['smartviews_v3:table.table.empty_state_title', 'We searched everywhere']),
      _useTranslations2 = (0, _slicedToArray2["default"])(_useTranslations, 1),
      emptyStateTitle = _useTranslations2[0];

  var _useTranslations3 = (0, _useTranslations5["default"])(['smartviews_v3:table.table.empty_state_description', 'And couldn’t find anything. Try again?']),
      _useTranslations4 = (0, _slicedToArray2["default"])(_useTranslations3, 1),
      emptyStateDescription = _useTranslations4[0];

  var scrollableRef = (0, _useInfiniteScroll["default"])((0, _extends2["default"])({}, infiniteScrollOptions, {
    isLoading: isLoading,
    pageSize: pageSize
  }));
  var isMasterChecked = rowSelection.allSelected,
      selectedRows = rowSelection.items,
      unselectedRows = rowSelection.exceptItems;
  var handleSelectRowChange = (0, _react.useCallback)(function (rowId) {
    onCheckedRowsChange({
      rowId: rowId
    });
  }, [onCheckedRowsChange]);
  var handleMasterCheckboxChange = (0, _react.useCallback)(function () {
    onCheckedRowsChange({
      selectAll: true
    });
  }, [onCheckedRowsChange]);
  var handleColumnResize = (0, _react.useCallback)(function () {
    cachedCells.current = null;
    onColumnResize.apply(void 0, arguments);
  }, [onColumnResize]);
  var handleColumnReorderChange = (0, _react.useCallback)(function (_ref3) {
    var sourceData = _ref3.sourceData,
        targetData = _ref3.targetData;
    cachedCells.current = null;
    var columns = (0, _toConsumableArray2["default"])(children);

    if (sourceData !== targetData) {
      var orderedColumns = (0, _tableUtility.getOrderedColumnsById)(columns, sourceData, targetData);
      onColumnOrderChange(orderedColumns);
    }
  }, [onColumnOrderChange, children]);
  var handleHeaderSort = (0, _react.useCallback)(function (args) {
    if (scrollableRef.current) {
      scrollableRef.current.scrollTop = 0;
    }

    onSort(args);
  }, [onSort]);

  var _useMemo = (0, _react.useMemo)(function () {
    var current = cachedCells.current;

    if (current && areCellsEqual(children, current)) {
      return current;
    }

    cachedCells.current = (0, _tableUtility.getTableCells)(children, multiselect);
    return cachedCells.current;
  }, [children, multiselect]),
      _useMemo2 = (0, _slicedToArray2["default"])(_useMemo, 2),
      headerCells = _useMemo2[0],
      bodyCells = _useMemo2[1];

  var CellRowComponent = rowComponent || _CellRow["default"];
  var showEmptyState = (data === null || data === void 0 ? void 0 : data.length) === 0 && !isLoading;
  var totalCount = (infiniteScrollOptions === null || infiniteScrollOptions === void 0 ? void 0 : infiniteScrollOptions.totalCount) || (data === null || data === void 0 ? void 0 : data.length) || 0;
  var isMasterIndeterminate = Boolean(isMasterChecked ? unselectedRows.size && unselectedRows.size !== totalCount : selectedRows.size && selectedRows.size !== totalCount);
  var hasStickyColumns = (headerCells === null || headerCells === void 0 ? void 0 : headerCells.some(function (headerCell) {
    return headerCell.props.fixed;
  })) || false;
  var canRenderScrollShadows = !hasStickyColumns && hasScrollShadow;

  var _useHorizontalScrollS = (0, _useHorizontalScrollState["default"])([canRenderScrollShadows, scrollableRef]),
      _useHorizontalScrollS2 = (0, _slicedToArray2["default"])(_useHorizontalScrollS, 3),
      showStart = _useHorizontalScrollS2[0],
      showEnd = _useHorizontalScrollS2[1],
      onScrollUpdateShadow = _useHorizontalScrollS2[2];

  return /*#__PURE__*/_react["default"].createElement("div", {
    className: classes.root
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])(classes.wrapper, {
      'start-shadow': showStart,
      'end-shadow': showEnd
    }),
    ref: scrollableRef,
    onScroll: onScrollUpdateShadow
  }, /*#__PURE__*/_react["default"].createElement("div", {
    ref: ref,
    className: (0, _classnames["default"])(classes.table, className),
    style: style
  }, /*#__PURE__*/_react["default"].createElement(_HeaderRow["default"], {
    sortBy: sortBy,
    onSort: handleHeaderSort,
    draggable: draggableColumns,
    onDragDrop: handleColumnReorderChange,
    multiSelect: multiselect,
    masterCheckbox: {
      checked: isMasterChecked,
      indeterminate: isMasterIndeterminate,
      onChange: handleMasterCheckboxChange
    },
    onResize: handleColumnResize
  }, headerCells), /*#__PURE__*/_react["default"].createElement(_react.Suspense, {
    fallback: null
  }, data === null || data === void 0 ? void 0 : data.map(function (row) {
    var id = row[rowIdDataKey];
    return /*#__PURE__*/_react["default"].createElement(CellRowComponent, {
      multiSelect: multiselect,
      onCheckboxChange: handleSelectRowChange,
      checked: selectedRows.has(id),
      onClick: onRowClick,
      className: rowClassName,
      data: row,
      rowId: id.toString(),
      key: id
    }, bodyCells);
  }), /*#__PURE__*/_react["default"].createElement(_LoadingIndicator["default"], {
    show: isLoading,
    isTableEmpty: (0, _utils.isEmpty)(data),
    pageSize: pageSize,
    multiSelect: multiselect,
    checked: isMasterChecked,
    className: rowClassName,
    cellComponents: bodyCells
  }))), showEmptyState && /*#__PURE__*/_react["default"].createElement(_EmptyState["default"], {
    className: classes.emptyState,
    svg: Rug,
    iconHeight: 90,
    iconWidth: 134,
    title: emptyStateTitle,
    description: emptyStateDescription
  })), errorBanner && /*#__PURE__*/_react["default"].createElement("div", {
    className: classes.errorBanner
  }, errorBanner));
});

Table.defaultProps = {
  className: '',
  style: {},
  data: undefined,
  onRowClick: _noop["default"],
  rowClassName: '',
  height: undefined,
  multiselect: false,
  onCheckedRowsChange: _noop["default"],
  rowSelection: {
    allSelected: false,
    items: new Set()
  },
  rowIdDataKey: '_id',
  draggableColumns: false,
  onColumnOrderChange: _noop["default"],
  onColumnResize: _noop["default"],
  sortBy: undefined,
  onSort: undefined,
  rowComponent: null,
  isLoading: false,
  pageSize: 25,
  errorBanner: null,
  infiniteScrollOptions: {
    skip: 0,
    onLoadMore: _noop["default"],
    scrollOffset: 200
  },
  hasScrollShadow: true
};
Table.propTypes = {
  /** Data array to pass to the table - expects an array of object. Notice, undefined data is not considered in an empty state */
  data: _propTypes["default"].arrayOf(_propTypes["default"].object),

  /** Column data to pass to the table - expects an array of Columns */
  children: _propTypes["default"].arrayOf(_propTypes["default"].node).isRequired,

  /** Optional class name to the table level */
  className: _propTypes["default"].string,

  /** Optional styles to table level */
  style: _propTypes["default"].shape(),

  /** Row click Callback  => ({data: {Object} row data + rowId}) */
  onRowClick: _propTypes["default"].func,

  /** Optional className to pass to cell rows */
  rowClassName: _propTypes["default"].string,

  /** Height for the entire table */
  height: _propTypes["default"].number,

  /** Row id data source to set from data object as unique identifier for each row */
  rowIdDataKey: _propTypes["default"].string,

  /** Boolean indicator for allowing bulk actions */
  multiselect: _propTypes["default"].bool,

  /** Returns selection "action" with either { rowId } or { selectAll } */
  onCheckedRowsChange: _propTypes["default"].func,

  /** Config that describes selection in the table */
  rowSelection: _propTypes["default"].shape({
    /** Marks table as everything should be selected, even not yet loaded data */
    allSelected: _propTypes["default"].bool,

    /** List of selected items  */
    items: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].array])
  }),

  /** Boolean indicator for reordering columns */
  draggableColumns: _propTypes["default"].bool,

  /** CB function for draggable drop event => ({ columns: {Array} Array of columns id by order }) */
  onColumnOrderChange: _propTypes["default"].func,

  /** String indicator for the sorted column. '-' prefix defines descending order */
  sortBy: _propTypes["default"].string,

  /** Callback function to handle sorting by a column */
  onSort: _propTypes["default"].func,

  /** Custom Cell row component to be rendered */
  rowComponent: _propTypes["default"].oneOfType([_propTypes["default"].node, _propTypes["default"].func]),

  /** callback function for resize column width => ({ payload: {object} returns {[columnId: resizedWidth]} }) */
  onColumnResize: _propTypes["default"].func,

  /** indicator for data fetching state */
  isLoading: _propTypes["default"].bool,

  /** number of rows per page */
  pageSize: _propTypes["default"].number,

  /** Object to define infinite scrolling behaviour */
  infiniteScrollOptions: _propTypes["default"].shape({
    /** starting index of the chunk */
    skip: _propTypes["default"].number,

    /** cb to notify about required loads */
    onLoadMore: _propTypes["default"].func,

    /** total number of entries */
    totalCount: _propTypes["default"].number,

    /** offset from the bottom, from which loading the next chunk should be triggered */
    scrollOffset: _propTypes["default"].number
  }),

  /** Node error component */
  errorBanner: _propTypes["default"].node,

  /** jss override object to customize the jss classes */
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func]),

  /** Adds shadow indicators at the sides of the table when it is overflowed horizontally, disabled by default with fixed columns */
  hasScrollShadow: _propTypes["default"].bool
};
Table.displayName = 'Table';
var _default = Table;
exports["default"] = _default;
