// @guestyci/foundation/legacy/DatePicker/DatePickerNavButton.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _reactJss = _interopRequireDefault(require("react-jss"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _noop = _interopRequireDefault(require("lodash/noop"));

var _index = _interopRequireDefault(require("../ActionButton/index"));

var _enums = require("../../enums/enums");

var styles = {
  button: function button(_ref) {
    var sm = _ref.sm,
        type = _ref.type;
    return {
      position: 'absolute',
      top: '20px',
      width: type === _enums.DatePickerType.Single || sm ? 40 : undefined,
      height: type === _enums.DatePickerType.Single || sm ? 40 : undefined
    };
  },
  nextButton: {
    right: '22px'
  },
  prevButton: {
    left: '22px'
  }
};

var DatePickerNavButton = function DatePickerNavButton(_ref2) {
  var _cn;

  var direction = _ref2.direction,
      classes = _ref2.classes;
  return /*#__PURE__*/_react["default"].createElement(_index["default"], {
    onClick: _noop["default"],
    icon: "Btn".concat(direction, "Small"),
    iconHeight: 11,
    iconWidth: 6,
    className: (0, _classnames["default"])(classes.button, (_cn = {}, (0, _defineProperty2["default"])(_cn, classes.nextButton, direction === 'Right'), (0, _defineProperty2["default"])(_cn, classes.prevButton, direction === 'Left'), _cn)),
    "data-qa": "date-picker-nav-button"
  });
};

DatePickerNavButton.defaultProps = {
  sm: false
};
DatePickerNavButton.propTypes = {
  direction: _propTypes["default"].oneOf(['Right', 'Left']).isRequired,
  sm: _propTypes["default"].bool,
  type: _propTypes["default"].oneOf([_enums.DatePickerType.Single, _enums.DatePickerType.Range]).isRequired
};

var _default = (0, _reactJss["default"])(styles)(DatePickerNavButton);

exports["default"] = _default;
