// @guestyci/foundation/es/Dialog/DialogContent.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _objectWithoutPropertiesLoose from "@babel/runtime/helpers/esm/objectWithoutPropertiesLoose";
import React from 'react';
import PropTypes from 'prop-types';
import Section from "../Layout/Section";
import TextField from "../TextField";
import createStyles from "../createStyles";
const useStyles = createStyles(theme => ({
  text: {
    color: theme.palette.text.secondary
  }
}), {
  name: 'DialogContent'
});

const DialogContent = (_ref) => {
  let {
    children,
    jss = {},
    className
  } = _ref,
      props = _objectWithoutPropertiesLoose(_ref, ["children", "jss", "className"]);

  const classes = useStyles({
    jss
  });
  return /*#__PURE__*/React.createElement(Section, _extends({
    gutter: "6 8 6 6",
    className: className
  }, props, {
    "data-qa": "dialog-content"
  }), /*#__PURE__*/React.createElement(TextField, {
    className: classes.text
  }, children));
};

DialogContent.propTypes = {
  /** Child elements to render in header */
  children: PropTypes.node.isRequired,

  /** JSS override object */
  jss: PropTypes.oneOfType([PropTypes.shape(), PropTypes.func]),

  /** Additional classname to root component */
  className: PropTypes.string
};
export default DialogContent;
