// @guestyci/foundation/esm/legacy/VirtualTable/HeaderRenderer.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import _extends from "@babel/runtime/helpers/esm/extends";
import React from 'react';
import SortDirection from 'react-virtualized/dist/commonjs/Table/SortDirection';
import cn from 'classnames';
import { createUseStyles } from 'react-jss';
import SvgIcon from "../SvgIcon";

var BtnSort = function BtnSort(props) {
  return /*#__PURE__*/React.createElement("svg", props, /*#__PURE__*/React.createElement("path", {
    d: "M0 0h8L4 4z"
  }));
};

BtnSort.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 8 4"
};
var useStyles = createUseStyles({
  sort: {
    opacity: 0
  },
  hover: {
    display: 'flex',
    alignItems: 'center',
    userSelect: 'none',
    '&:hover $sort': {
      opacity: 1,
      transition: 'opacity 0.2s ease-in'
    }
  }
});

var defaultHeaderRenderer = function defaultHeaderRenderer(props) {
  return /*#__PURE__*/React.createElement(HeaderRenderer, _extends({}, props, {
    "data-qa": "default-header-renderer"
  }));
};

var HeaderRenderer = function HeaderRenderer(_ref) {
  var dataKey = _ref.dataKey,
      disableSort = _ref.disableSort,
      label = _ref.label,
      sortBy = _ref.sortBy,
      sortDirection = _ref.sortDirection;
  var classes = useStyles();
  var showSortIndicator = sortBy === dataKey;
  return /*#__PURE__*/React.createElement("div", {
    className: classes.hover,
    "data-qa": "header-renderer"
  }, /*#__PURE__*/React.createElement("span", {
    className: cn('ReactVirtualized__Table__headerTruncatedText', {
      'text-black': showSortIndicator,
      'font-weight-bold': showSortIndicator
    }),
    title: label
  }, label), !disableSort && /*#__PURE__*/React.createElement(SvgIcon, {
    svg: BtnSort,
    className: cn('relative', 'ml-1', _defineProperty({
      'transform-180': showSortIndicator && sortDirection === SortDirection.ASC
    }, classes.sort, !showSortIndicator)),
    height: 9,
    width: 9,
    color: "black"
  }));
};

export default defaultHeaderRenderer;
