// @guestyci/foundation/es/legacy/Table/CheckboxCell/CheckboxCell.js
import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import Checkbox from "../../../Checkbox";
import createStyles from "../../../createStyles";
const useStyles = createStyles({
  root: {
    display: 'table-cell',
    background: 'inherit',
    width: 36,
    verticalAlign: 'middle'
  },
  fixed: {
    position: 'sticky',
    left: 0
  }
}, {
  name: 'CheckboxCell'
});
const CheckboxCell = React.memo(({
  className,
  style,
  onChange,
  checked,
  indeterminate,
  value,
  fixed
}) => {
  const classes = useStyles({
    fixed
  });

  const handleCheck = () => {
    onChange(value);
  };

  const handleClick = e => {
    e.stopPropagation();
  };

  const inputProps = {
    onClick: handleClick
  };
  return /*#__PURE__*/React.createElement("div", {
    className: cn('checkbox-cell', classes.root, {
      [classes.fixed]: fixed
    }, className),
    style: style
  }, /*#__PURE__*/React.createElement(Checkbox, {
    checked: checked,
    inputProps: inputProps,
    indeterminate: indeterminate,
    onChange: handleCheck,
    value: value
  }));
});
CheckboxCell.defaultProps = {
  className: '',
  style: {},
  checked: false,
  value: '',
  fixed: false
};
CheckboxCell.propTypes = {
  onChange: PropTypes.func.isRequired,
  className: PropTypes.string,
  style: PropTypes.shape(),
  checked: PropTypes.bool,
  indeterminate: PropTypes.bool,
  fixed: PropTypes.bool,
  value: PropTypes.string
};
CheckboxCell.displayName = 'CheckboxCell';
export default CheckboxCell;
