// @guestyci/foundation/StatusPill/StatusPill.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.useStyles = void 0;

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _createStyles = _interopRequireDefault(require("../createStyles"));

var _TextField = _interopRequireDefault(require("../TextField"));

var _Row = _interopRequireDefault(require("../Layout/Row"));

var useStyles = (0, _createStyles["default"])(function (theme) {
  return {
    statusWrapper: {
      height: 21,
      borderRadius: 3,
      padding: theme.spacer(0, 2)
    },
    status: function status(_ref) {
      var _theme$palette$varian, _theme$palette$varian2;

      var variant = _ref.variant;
      return {
        backgroundColor: ((_theme$palette$varian = theme.palette[variant]) === null || _theme$palette$varian === void 0 ? void 0 : _theme$palette$varian.background) || theme.palette.secondary.background,
        color: ((_theme$palette$varian2 = theme.palette[variant]) === null || _theme$palette$varian2 === void 0 ? void 0 : _theme$palette$varian2.text) || theme.palette.text["default"]
      };
    },
    dot: function dot(_ref2) {
      var _theme$palette$varian3;

      var variant = _ref2.variant;
      return {
        borderRadius: '50%',
        height: '7px',
        width: '7px',
        backgroundColor: ((_theme$palette$varian3 = theme.palette[variant]) === null || _theme$palette$varian3 === void 0 ? void 0 : _theme$palette$varian3.text) || theme.palette.text["default"]
      };
    }
  };
});
exports.useStyles = useStyles;
var StatusPill = (0, _react.forwardRef)(function StatusPill(_ref3, ref) {
  var _ref3$variant = _ref3.variant,
      variant = _ref3$variant === void 0 ? 'info' : _ref3$variant,
      _ref3$dotIcon = _ref3.dotIcon,
      dotIcon = _ref3$dotIcon === void 0 ? false : _ref3$dotIcon,
      className = _ref3.className,
      children = _ref3.children,
      _ref3$jss = _ref3.jss,
      jss = _ref3$jss === void 0 ? {} : _ref3$jss;
  var classes = useStyles({
    jss: jss,
    variant: variant
  });
  return /*#__PURE__*/_react["default"].createElement(_Row["default"], {
    align: "center",
    justify: "start",
    spacing: 1,
    className: (0, _classnames["default"])(classes.statusWrapper, classes.status, className),
    ref: ref
  }, dotIcon && /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])(classes.dot)
  }), /*#__PURE__*/_react["default"].createElement(_TextField["default"], {
    variant: "h7",
    overflowEllipsis: true
  }, children));
});
StatusPill.displayName = 'StatusPill';
StatusPill.propTypes = {
  variant: _propTypes["default"].oneOf(['success', 'alert', 'info', 'primary', 'secondary', 'error']),

  /** root level class name to add */
  className: _propTypes["default"].string,

  /** text node element */
  children: _propTypes["default"].node.isRequired,

  /** jss override object to customize the jss classes [statusWrapper] */
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func]),

  /** Boolean to determine if to show dot icon to left of the children */
  dotIcon: _propTypes["default"].bool
};
var _default = StatusPill;
exports["default"] = _default;
