// @guestyci/foundation/legacy/Note/Note.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var Note = function Note(_ref) {
  var message = _ref.message,
      className = _ref.className,
      px = _ref.px,
      children = _ref.children;
  var content = children || message;
  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, content ? /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])('note overflow-y-auto b-1 w-fill border-gray-light font-size-md py-3 bg-yellow-light min-height-50 max-height-200', "px-".concat(px), className)
  }, content) : null);
};

Note.defaultProps = {
  children: null,
  message: '',
  className: '',
  px: 6
};
Note.propTypes = {
  /** Note content children */
  children: _propTypes["default"].node,

  /** Additional class name for Note */
  className: _propTypes["default"].string,

  /** Note message, if none provided will not render anything - DEPRECATED - use children */
  message: _propTypes["default"].string,

  /** padding x for note - ( 6===30 9===45) */
  px: _propTypes["default"].oneOf([6, 9])
};
var _default = Note;
exports["default"] = _default;
