// @guestyci/foundation/legacy/Popover/withOverflowPopoverState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _react = _interopRequireDefault(require("react"));

function _createSuper(Derived) { return function () { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

var withOverflowPopoverState = function withOverflowPopoverState(WrappedComponent) {
  var WithOverflowState = /*#__PURE__*/function (_React$Component) {
    (0, _inherits2["default"])(WithOverflowState, _React$Component);

    var _super = _createSuper(WithOverflowState);

    function WithOverflowState(props) {
      var _this;

      (0, _classCallCheck2["default"])(this, WithOverflowState);
      _this = _super.call(this, props);

      _this.onMouseEnter = function () {
        var _this$ref$current = _this.ref.current,
            clientWidth = _this$ref$current.clientWidth,
            scrollWidth = _this$ref$current.scrollWidth;

        if (scrollWidth > clientWidth) {
          _this.setState({
            showPopover: true
          });
        }
      };

      _this.onMouseLeave = function () {
        _this.setState({
          showPopover: false
        });
      };

      _this.state = {
        showPopover: false
      };
      _this.ref = _react["default"].createRef();
      return _this;
    }

    (0, _createClass2["default"])(WithOverflowState, [{
      key: "render",
      value: function render() {
        var showPopover = this.state.showPopover;
        return /*#__PURE__*/_react["default"].createElement("div", {
          className: "w-fill",
          ref: this.ref,
          onMouseEnter: this.onMouseEnter,
          onMouseLeave: this.onMouseLeave,
          "data-qa": "with-overflow-state"
        }, /*#__PURE__*/_react["default"].createElement(WrappedComponent, (0, _extends2["default"])({
          showPopover: showPopover
        }, this.props)));
      }
    }]);
    return WithOverflowState;
  }(_react["default"].Component);

  WithOverflowState.displayName = "WithOverflowState(".concat(WrappedComponent.displayName || WrappedComponent.name || 'Component', ")");
  return WithOverflowState;
};

var _default = withOverflowPopoverState;
exports["default"] = _default;
