// @guestyci/foundation/Collapse/Collapse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _Transition = _interopRequireDefault(require("react-transition-group/Transition"));

var _createStyles = _interopRequireDefault(require("../createStyles"));

/* eslint no-param-reassign: 0 */
var useStyles = (0, _createStyles["default"])(function () {
  return {
    content: {
      width: '100%',
      overflow: 'hidden',
      height: '0px'
    }
  };
}, {
  name: 'Collapse'
});

var Collapse = _react["default"].forwardRef(function Collapse(_ref, ref) {
  var children = _ref.children,
      show = _ref.show,
      _ref$className = _ref.className,
      className = _ref$className === void 0 ? '' : _ref$className,
      _ref$duration = _ref.duration,
      duration = _ref$duration === void 0 ? 200 : _ref$duration,
      onEntered = _ref.onEntered,
      onExited = _ref.onExited,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["children", "show", "className", "duration", "onEntered", "onExited"]);
  var classes = useStyles();

  var handleEnter = function handleEnter(node, isAppearing) {
    if (isAppearing) {
      node.style.height = 'auto';
    } else {
      node.style.height = 0;
    }
  };

  var handleEntering = function handleEntering(node) {
    var height = node.scrollHeight;
    node.style.transitionDuration = "".concat(duration, "ms");
    node.style.height = "".concat(height, "px");
  };

  var handleEntered = function handleEntered(node, isAppearing) {
    node.style.height = 'auto';

    if (onEntered) {
      onEntered(node, isAppearing);
    }
  };

  var handleExit = function handleExit(node) {
    var height = node.scrollHeight;
    node.style.height = "".concat(height, "px");
  };

  var handleExiting = function handleExiting(node) {
    setTimeout(function () {
      node.style.transitionDuration = "".concat(duration, "ms");
      node.style.height = 0;
    }, 0);
  };

  return /*#__PURE__*/_react["default"].createElement(_Transition["default"], (0, _extends2["default"])({
    "in": show,
    ref: ref,
    appear: true,
    timeout: duration,
    onEnter: handleEnter,
    onEntering: handleEntering,
    onExit: handleExit,
    onExiting: handleExiting,
    onEntered: handleEntered,
    onExited: onExited
  }, props), /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])(classes.content, className)
  }, children));
});

Collapse.propTypes = {
  /** Indicator whether or not collapse should be open */
  show: _propTypes["default"].bool.isRequired,

  /** Children to render when collapse is open */
  children: _propTypes["default"].oneOfType([_propTypes["default"].arrayOf(_propTypes["default"].node), _propTypes["default"].node]).isRequired,

  /** Duration of the animation */
  duration: _propTypes["default"].number,

  /** Function to be executed when the collapse is opened
   * (node: HtmlElement, isAppearing: bool) -> void
   */
  onEntered: _propTypes["default"].func,

  /** Function to be executed when the collapse is closed
   * (node: HtmlElement, isAppearing: bool) -> void
   */
  onExited: _propTypes["default"].func,

  /** Additional class for root level */
  className: _propTypes["default"].string
};
Collapse.displayName = 'Collapse';
var _default = Collapse;
exports["default"] = _default;
