// @guestyci/foundation/legacy/FormGroup/FormGroup.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var FormGroup = function FormGroup(_ref) {
  var title = _ref.title,
      children = _ref.children,
      inline = _ref.inline,
      className = _ref.className,
      style = _ref.style;
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])('form-group-wrapper d-flex-fill flex-column', className),
    style: style,
    "data-qa": "form-group"
  }, title && /*#__PURE__*/_react["default"].createElement("div", {
    className: "text-muted font-size-md form-group-title mb-4"
  }, title), /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])('form-group-body', 'd-flex-fill', {
      'flex-column': !inline
    })
  }, children));
};

FormGroup.defaultProps = {
  inline: false,
  className: '',
  style: {}
};
FormGroup.propTypes = {
  /** Form Group inner children */
  children: _propTypes["default"].node.isRequired,

  /** Form Group Title */
  title: _propTypes["default"].string.isRequired,

  /** Indicator whether form should be column or row */
  inline: _propTypes["default"].bool,

  /** Class  to be added to FormGroup */
  className: _propTypes["default"].string,

  /** Style to add to FormGroup */
  style: _propTypes["default"].shape()
};
var _default = FormGroup;
exports["default"] = _default;
