// @guestyci/foundation/legacy/Badge/Badge.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.BadgeComponent = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _reactJss = _interopRequireDefault(require("react-jss"));

var _Icon = _interopRequireDefault(require("../Icon"));

var _TextField = _interopRequireDefault(require("../TextField/TextField"));

var styles = {
  badge: {
    minWidth: function minWidth(_ref) {
      var sm = _ref.sm;
      return sm ? 13 : 21;
    },
    height: function height(_ref2) {
      var sm = _ref2.sm;
      return sm ? 13 : 21;
    },
    borderRadius: 10
  },
  value: {
    padding: '3px 7px'
  }
};

var getIconSize = function getIconSize(width, height, sm) {
  return {
    height: height || (sm ? 13 : 21),
    width: width || (sm ? 13 : 21)
  };
};

var Badge = function Badge(_ref3) {
  var _cn;

  var classes = _ref3.classes,
      value = _ref3.value,
      bgColor = _ref3.bgColor,
      color = _ref3.color,
      className = _ref3.className,
      style = _ref3.style,
      borderColor = _ref3.borderColor,
      sm = _ref3.sm,
      icon = _ref3.icon,
      svg = _ref3.svg,
      iconHeight = _ref3.iconHeight,
      iconWidth = _ref3.iconWidth;

  var _getIconSize = getIconSize(iconWidth, iconHeight, sm),
      height = _getIconSize.height,
      width = _getIconSize.width;

  return /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])('badge-wrapper flex-center', classes.badge, (_cn = {}, (0, _defineProperty2["default"])(_cn, "bg-".concat(bgColor), bgColor), (0, _defineProperty2["default"])(_cn, "text-".concat(color, " fill-").concat(color), color), (0, _defineProperty2["default"])(_cn, "border-".concat(borderColor), borderColor), (0, _defineProperty2["default"])(_cn, 'b-1', borderColor), _cn), className),
    style: style,
    "data-qa": "badge"
  }, icon || svg ? /*#__PURE__*/_react["default"].createElement(_Icon["default"], {
    svg: svg,
    icon: icon,
    width: width,
    height: height,
    noFill: true
  }) : /*#__PURE__*/_react["default"].createElement(_TextField["default"], {
    className: classes.value,
    color: color,
    size: sm ? 'xs' : 'md'
  }, value));
};

Badge.defaultProps = {
  bgColor: 'blue',
  className: '',
  borderColor: '',
  style: {},
  color: 'white',
  sm: false,
  iconHeight: null,
  iconWidth: null,
  icon: null,
  value: null
};
Badge.propTypes = {
  /** Background color for badge root */
  bgColor: _propTypes["default"].string,

  /** Border color for badge root */
  borderColor: _propTypes["default"].string,

  /** Color for badge child */
  color: _propTypes["default"].string,

  /** Additional class for badge */
  className: _propTypes["default"].string,

  /** Additional style for badge */
  style: _propTypes["default"].shape(),

  /** Set badge as small */
  sm: _propTypes["default"].bool,

  /** Icon to locate in badge */
  icon: _propTypes["default"].string,

  /** The svg icon to render */
  svg: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func]),

  /* Width for icon */
  iconWidth: _propTypes["default"].number,

  /* Height for icon */
  iconHeight: _propTypes["default"].number,

  /* Value to put in badge if counter type */
  value: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].number])
};
/** Work around to fix storybook HOC propTypes bug  - DO NOT IMPORT THIS */

var BadgeComponent = Badge;
exports.BadgeComponent = BadgeComponent;

var _default = (0, _reactJss["default"])(styles)(Badge);

exports["default"] = _default;
