// @guestyci/testify/utils.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.shallow = exports.mount = exports.asyncFlush = exports.clearAllMocks = exports.clearMocks = exports.mockFn = exports.createContainer = void 0;

var _enzyme = require("enzyme");

/**
 * Create  a new container and append to document body
 * @param id  {String}  Container id
 * @return {HTMLElement}
 */
var createContainer = function createContainer(id) {
  var container = document.createElement('div');
  container.setAttribute('id', id);
  document.body.appendChild(container);
  return container;
};
/**
 * Generate mock function for unit tests
 * @returns {*}
 */


exports.createContainer = createContainer;
var mockFn = jest.fn;
/**
 * Clear all mocks
 * @returns {*}
 */

exports.mockFn = mockFn;
var clearMocks = jest.clearAllMocks;
/**
 * Clear all mocks
 * @returns {*}
 */

exports.clearMocks = clearMocks;
var _jest = jest,
    clearAllMocks = _jest.clearAllMocks;
/**
 * Helper function to create context switch to await async operations
 * @returns {Promise<any>}
 */

exports.clearAllMocks = clearAllMocks;

var asyncFlush = function asyncFlush() {
  var count = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 0;
  return new Promise(function (resolve) {
    return setTimeout(resolve, count);
  });
};

exports.asyncFlush = asyncFlush;
var mount = _enzyme.mount;
exports.mount = mount;
var shallow = _enzyme.shallow;
exports.shallow = shallow;
