// @guestyci/foundation/legacy/Bullet/Bullet.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _reactJss = require("react-jss");

var useStyles = (0, _reactJss.createUseStyles)({
  bullet: {
    width: function width(_ref) {
      var _width = _ref.width;
      return _width;
    },
    height: function height(_ref2) {
      var _height = _ref2.height;
      return _height;
    },
    borderRadius: '50%'
  }
});

var Bullet = function Bullet(_ref3) {
  var height = _ref3.height,
      width = _ref3.width,
      color = _ref3.color,
      className = _ref3.className,
      style = _ref3.style;
  var classes = useStyles({
    width: width,
    height: height
  });
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])(classes.bullet, (0, _defineProperty2["default"])({}, "bg-".concat(color), color), className),
    style: style,
    "data-qa": "bullet"
  });
};

Bullet.defaultProps = {
  color: 'muted',
  width: 4,
  height: 4,
  className: '',
  style: {}
};
Bullet.propTypes = {
  /** Bullet color */
  color: _propTypes["default"].string,

  /** Bullet height */
  height: _propTypes["default"].number,

  /** Bullet width */
  width: _propTypes["default"].number,

  /** Additional class for bullet root level */
  className: _propTypes["default"].string,

  /** Additional style to bullet root level */
  style: _propTypes["default"].shape()
};
var _default = Bullet;
exports["default"] = _default;
