// @guestyci/foundation/es/DatePicker/SingleDatePicker/SingleDatePicker.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _objectWithoutPropertiesLoose from "@babel/runtime/helpers/esm/objectWithoutPropertiesLoose";
import React, { useState, useMemo, forwardRef } from 'react';
import PropTypes from 'prop-types';
import { OPEN_DOWN } from 'react-dates/lib/constants';
import moment from 'moment';
import 'react-dates/initialize';
import { SingleDatePicker as ReactDatesPicker } from 'react-dates';
import cn from 'classnames';
import createStyles from "../../createStyles";
import DatePickerStyles from "../styles";
import { uuid } from "../../utils/commonUtility";
import NavButton from "../NavButton";
import PickerClearButton from "../../basePicker/PickerClearButton";
import PickerInputButton from "../../basePicker/PickerInputButton";
import Row from "../../Layout/Row";

var BtnCalendar = function BtnCalendar(props) {
  return /*#__PURE__*/React.createElement("svg", props, /*#__PURE__*/React.createElement("path", {
    d: "M18.5 2h-4V0h-1v2h-6V0h-1v2h-4A2.503 2.503 0 0 0 0 4.5v12C0 17.879 1.122 19 2.5 19h16c1.379 0 2.5-1.121 2.5-2.5v-12C21 3.122 19.879 2 18.5 2zm-16 1h4v.667h1V3h6v.667h1V3h4c.827 0 1.5.673 1.5 1.5V7H1V4.5C1 3.673 1.673 3 2.5 3zm16 15h-16c-.827 0-1.5-.673-1.5-1.5V8h19v8.5c0 .827-.673 1.5-1.5 1.5z"
  }));
};

BtnCalendar.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 21 19"
};
import { DATE_FORMAT, DATE_DISPLAY_FORMAT, MONTH_DISPLAY_FORMAT } from "../constants";
export const useStyles = createStyles(theme => ({
  root: DatePickerStyles.root(theme),
  inputContainer: DatePickerStyles.inputContainer,
  focused: DatePickerStyles.focused(theme),
  error: DatePickerStyles.error(theme),
  disabled: DatePickerStyles.disabled(theme)
}), {
  name: 'SingleDatePicker'
});

const renderMonthElement = ({
  month
}) => month.format(MONTH_DISPLAY_FORMAT);

const SingleDatePicker = forwardRef(function SingleDatePicker(_ref, ref) {
  let {
    id,
    name,
    blockedDates,
    disabled,
    initialVisibleMonth,
    value,
    onMonthChange,
    onChange,
    enablePastDays,
    focused: focusedProp,
    onFocusChange,
    isFetching = false,
    anchorDirection = 'left',
    showClearDate = true,
    error,
    className,
    style,
    jss
  } = _ref,
      rest = _objectWithoutPropertiesLoose(_ref, ["id", "name", "blockedDates", "disabled", "initialVisibleMonth", "value", "onMonthChange", "onChange", "enablePastDays", "focused", "onFocusChange", "isFetching", "anchorDirection", "showClearDate", "error", "className", "style", "jss"]);

  const classes = useStyles({
    jss
  });
  const [focusedState, setFocusState] = useState(false);
  const isFocused = focusedProp || focusedState;

  const handleFocusChange = ({
    focused
  }) => {
    if (onFocusChange) {
      onFocusChange(focused);
    }

    setFocusState(focused);
  };

  const handleCalendarButtonClick = () => {
    if (!isFocused) {
      if (onFocusChange) {
        onFocusChange(true);
      }

      setFocusState(true);
    }
  };

  const renderDayContents = day => {
    if (isFetching) {
      return '...';
    }

    return day.date();
  };

  const inputId = useMemo(() => id || uuid(), [id]);
  const blockedDatesSet = useMemo(() => new Set(blockedDates || []), [blockedDates]);

  const isDayBlocked = day => {
    const formattedDay = day.format(DATE_FORMAT);
    return isFetching || !!blockedDatesSet.has(formattedDay);
  };

  const getInitialVisibleMonth = () => initialVisibleMonth ? moment(initialVisibleMonth) : moment();

  const isOutsideRange = enablePastDays ? () => false : undefined;

  const handleDateChange = newDate => {
    if (onChange) {
      onChange(newDate, name);
    }
  };

  const handleClearDates = () => {
    handleDateChange(undefined);
  };

  return /*#__PURE__*/React.createElement(Row, {
    justify: "between",
    className: cn('date-picker', classes.root, {
      [classes.focused]: isFocused
    }, {
      [classes.error]: error
    }, {
      [classes.disabled]: disabled
    }, className),
    style: style,
    ref: ref
  }, /*#__PURE__*/React.createElement(Row, {
    className: classes.inputContainer
  }, /*#__PURE__*/React.createElement(PickerInputButton, {
    disabled: disabled,
    onClick: handleCalendarButtonClick,
    svg: BtnCalendar,
    active: !!isFocused
  }), /*#__PURE__*/React.createElement(ReactDatesPicker, _extends({
    id: inputId,
    date: value ? moment(value) : value,
    onDateChange: handleDateChange,
    placeholder: "Select",
    focused: isFocused,
    disabled: disabled,
    initialVisibleMonth: getInitialVisibleMonth,
    onFocusChange: handleFocusChange,
    isDayBlocked: isDayBlocked,
    isOutsideRange: isOutsideRange,
    hideKeyboardShortcutsPanel: true,
    noBorder: true,
    openDirection: OPEN_DOWN,
    readOnly: true,
    daySize: 40,
    verticalSpacing: 4,
    numberOfMonths: 1,
    displayFormat: DATE_DISPLAY_FORMAT,
    renderMonthElement: renderMonthElement,
    renderDayContents: renderDayContents,
    appendToBody: false,
    anchorDirection: anchorDirection,
    onNextMonthClick: onMonthChange,
    onPrevMonthClick: onMonthChange,
    navPrev: /*#__PURE__*/React.createElement(NavButton, {
      direction: "Left"
    }),
    navNext: /*#__PURE__*/React.createElement(NavButton, {
      direction: "Right"
    })
  }, rest)), /*#__PURE__*/React.createElement(PickerClearButton, {
    show: value && showClearDate,
    onClick: handleClearDates
  })));
});
SingleDatePicker.displayName = 'SingleDatePicker';
export default SingleDatePicker;
SingleDatePicker.propTypes = {
  /** Id for date picker */
  id: PropTypes.string,

  /** date picker name to be set */
  name: PropTypes.string,

  /** Array of blocked dates - formatted as YYYY-MM-DD */
  blockedDates: PropTypes.arrayOf(PropTypes.string),

  /** Disable state in which picker interaction is not available */
  disabled: PropTypes.bool,

  /** month to be show when the dialog picker opens - formatted as YYYY-MM-DD */
  initialVisibleMonth: PropTypes.string,

  /** the picked date - formatted as YYYY-MM-DD */
  value: PropTypes.string,

  /** callback to be called when a switching months. Called with a momnet object of the new month */
  onMonthChange: PropTypes.func,

  /** Callback to be called on picker value change */
  onChange: PropTypes.func,

  /** Enable past days */
  enablePastDays: PropTypes.bool,

  /** Control focus state - focued => picker opens. Not required - focus state is uncontrolled by default */
  focused: PropTypes.bool,

  /** Callback to be called when focus state changes. Called with a boolean  */
  onFocusChange: PropTypes.func,

  /** Fetching state */
  isFetching: PropTypes.bool,

  /** Anchor direction to open the picker - */
  anchorDirection: PropTypes.oneOf(['left', 'right']),

  /** When dates is selected - show a clear button */
  showClearDate: PropTypes.bool,

  /** Indicator for form error state  */
  error: PropTypes.bool,

  /** class name to be added to the root */
  className: PropTypes.string,

  /** style to be added to the root */
  style: PropTypes.shape(),
  jss: PropTypes.oneOfType([PropTypes.shape(), PropTypes.func])
};
