// @guestyci/foundation/es/theme/colors.js
// New color palette
// blue
export const blue800 = '#2a57ec';
export const blue700 = '#206cff';
export const blue500 = '#008dff';
export const blue400 = '#558FFF';
export const blue300 = '#4fb1ff';
export const blue200 = '#89c8ff';
export const blue10 = '#e6f3ff'; // dark blue

export const darkBlue900 = '#10275B';
export const darkBlue800 = '#1b3772';
export const darkBlue700 = '#0B1A3B';
export const darkBlue400 = '#344773';
export const darkBlue300 = '#57678C';
export const darkBlue100 = '#BDBECD';
export const darkBlue10 = '#D4D8E1'; // teal

export const teal500 = '#14C4CA';
export const teal200 = '#89E1E4';
export const teal100 = '#B8EDEF';
export const teal50 = '#E7F9F9'; // red

export const red700 = '#d40033';
export const red400 = '#ef3f54';
export const red200 = '#ee939b';
export const red10 = '#fff8f9'; // green

export const green800 = '#00782e';
export const green400 = '#2eb769';
export const green300 = '#5dc483';
export const green200 = '#8fd4a7';
export const green10 = '#eef6f4'; // yellow

export const yellow800 = '#ff8a00';
export const yellow400 = '#fec802';
export const yellow300 = '#ffdf6b';
export const yellow200 = '#fede7b';
export const yellow10 = '#fef9e5'; // purple

export const purple600 = '#8d24aa';
export const purple400 = '#ab47bc';
export const purple300 = '#ba68c8';
export const purple200 = '#ce93d8';
export const purple10 = '#eee4f3'; // gray

export const gray900 = '#212121';
export const gray800 = '#424242';
export const gray700 = '#616161';
export const gray600 = '#757575';
export const gray500 = '#9e9e9e';
export const gray400 = '#bdbdbd';
export const gray300 = '#e0e0e0';
export const gray200 = '#eeeeee';
export const gray100 = '#f5f5f5';
export const gray50 = '#fafafa'; // blueGray

export const blueGray700 = '#2B3A47';
export const blueGray500 = '#3e4e5d';
export const blueGray400 = '#77838D';
export const blueGray300 = '#949FA8';
export const blueGray100 = '#B7BECD';
export const blueGray50 = '#F3F5F7'; // white

export const white = '#ffffff'; // Legacy color palette
// blue

export const blue = blue700;
export const blueBorder = '#239ed8';
export const blueDark = blue700;
export const blueHover = '#085174';
export const blueContact = '#a8d5eb';
export const blueLight = '#d4ecf7';
export const blueLightBorder = '#48bdf3';
export const blueDarkTheme = '#4bbdf3'; // gray scale

export const black = gray900;
export const blackDark = '#000000';
export const grayDarkest = '#515c61';
export const grayDarker = '#556268';
export const grayDark = blueGray400;
export const muted = blueGray500;
export const gray = '#d1d8db';
export const grayLight = '#e3e8eb';
export const grayLighter = '#eef1f3';
export const grayLightest = '#f4f6f7'; // basic

export const green = green400;
export const greenDark = '#4aa666';
export const yellow = yellow400;
export const yellowLight = '#ffe9ab';
export const yellowDark = '#dfa600';
export const red = red400;
export const redDark = '#e02a2a';
export const purple = purple400;
