// @guestyci/foundation/TimePicker/styles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.timePickerInputStyles = exports.timePickerStyles = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _styles = _interopRequireDefault(require("../Input/styles"));

var timePickerStyles = function timePickerStyles(theme) {
  return {
    root: {
      display: 'flex',
      width: '100%',
      position: 'relative',
      flexFlow: 'column',
      '&:not([disabled]):hover': {
        '& .picker-input-button': {
          fill: theme.palette.primary["default"]
        }
      }
    },
    menu: {
      maxHeight: 270
    }
  };
};

exports.timePickerStyles = timePickerStyles;

var timePickerInputStyles = function timePickerInputStyles(theme) {
  return {
    root: (0, _extends2["default"])({}, theme.typography.primary, {}, _styles["default"].shared(theme), {
      paddingRight: 0,
      paddingLeft: 0,
      '&:hover': (0, _extends2["default"])({}, _styles["default"].hover(theme)),
      height: '46px',
      '&::placeholder': _styles["default"].placeholder(theme)
    }),
    clearIcon: {
      color: theme.palette.secondary["default"],
      '&:hover': {
        color: theme.palette.primary["default"]
      }
    },
    row: {
      height: '100%',
      paddingLeft: theme.spacer(3)
    },
    input: {
      width: 25,
      padding: 0,
      border: '0px solid',
      boxShadow: 'none',
      '&:hover': {
        border: '0px solid'
      }
    },
    separator: {
      paddingRight: theme.spacer(1)
    },
    disabled: _styles["default"].disabled(theme),
    focus: _styles["default"].focused(theme),
    error: _styles["default"].error(theme)
  };
};

exports.timePickerInputStyles = timePickerInputStyles;
