// @guestyci/foundation/TimePicker/TimePicker.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.useStyles = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Menu = require("../Menu");

var _TextField = _interopRequireDefault(require("../TextField"));

var _MenuItem = _interopRequireDefault(require("../Menu/MenuItem"));

var _composeEvent = _interopRequireDefault(require("../utils/composeEvent"));

var _createStyles = _interopRequireDefault(require("../createStyles"));

var _TimePickerInput = _interopRequireDefault(require("./TimePickerInput"));

var _styles = require("./styles");

var useStyles = (0, _createStyles["default"])(function (theme) {
  return (0, _extends2["default"])({}, (0, _styles.timePickerStyles)(theme));
}, {
  name: 'TimePicker'
});
exports.useStyles = useStyles;

var generateOptions = function generateOptions(interval) {
  var itemCount = 24 * 60 / interval;

  var padTime = function padTime(t) {
    var padCb = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : function () {
      return "0".concat(t);
    };
    return t < 10 ? padCb(t) : t;
  };

  var minutes = 0;
  return (0, _toConsumableArray2["default"])(Array(Math.round(itemCount))).map(function () {
    var hours = Math.floor(minutes / 60);
    var isAM = hours < 12;
    var result = "".concat(padTime(hours % 12, function (t) {
      return t === 0 ? '12' : "0".concat(t);
    }), " : ").concat(padTime(minutes % 60), " ").concat(isAM ? 'AM' : 'PM');
    minutes += interval;
    return result;
  });
};

var menuItemjssOverrides = function menuItemjssOverrides(theme) {
  return {
    selected: {
      color: theme.palette.brand.primary
    }
  };
};

var TimePicker = _react["default"].forwardRef(function TimePicker(_ref, ref) {
  var value = _ref.value,
      onChange = _ref.onChange,
      className = _ref.className,
      style = _ref.style,
      name = _ref.name,
      defaultValue = _ref.defaultValue,
      resetable = _ref.resetable,
      disabled = _ref.disabled,
      jss = _ref.jss,
      interval = _ref.interval,
      error = _ref.error,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["value", "onChange", "className", "style", "name", "defaultValue", "resetable", "disabled", "jss", "interval", "error"]);
  var classes = useStyles({
    jss: jss
  });
  var anchorRef = (0, _react.useRef)();
  var selectedItemRef = (0, _react.useRef)();

  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      open = _useState2[0],
      setOpen = _useState2[1];

  (0, _react.useEffect)(function () {
    var selectedItem = selectedItemRef.current;

    if (open && selectedItem) {
      var parent = selectedItem.parentElement;

      if (parent) {
        parent.scroll(0, selectedItem.offsetTop - parent.clientHeight / 2);
      }
    }
  }, [open]);

  var handleInputClick = function handleInputClick() {
    setOpen(!open);
  };

  var handleTimePickerClose = function handleTimePickerClose() {
    setOpen(false);
  };

  var handleInputChange = function handleInputChange(e, inputValue) {
    // work around to solve - see https://github.com/guestyorg/foundation-ui/issues/284
    Object.assign(e.target, {
      name: name,
      value: inputValue
    });
    onChange(e);
    handleTimePickerClose();
  };

  var handleItemClick = function handleItemClick(e, selectedValue) {
    (0, _composeEvent["default"])(e, {
      name: name,
      value: selectedValue
    });
    onChange(e);
    handleTimePickerClose();
  };

  var handleClearClick = function handleClearClick(e) {
    e.stopPropagation();
    (0, _composeEvent["default"])(e, {
      name: name
    });
    onChange(e);
  };

  var options = (0, _react.useMemo)(function () {
    return generateOptions(interval);
  }, [interval]);
  return /*#__PURE__*/_react["default"].createElement("div", (0, _extends2["default"])({
    ref: ref,
    className: (0, _classnames["default"])(classes.root, className),
    style: style
  }, props), /*#__PURE__*/_react["default"].createElement(_TimePickerInput["default"], {
    ref: anchorRef,
    onClick: handleInputClick,
    onChange: handleInputChange,
    onClear: handleClearClick,
    open: open,
    disabled: disabled,
    defaultValue: defaultValue,
    resetable: resetable,
    value: value,
    error: error
  }), /*#__PURE__*/_react["default"].createElement(_Menu.Menu, {
    anchorDirection: "start",
    open: open,
    className: classes.menu,
    onClose: handleTimePickerClose,
    variant: "menu",
    fitAnchor: true,
    anchor: anchorRef
  }, options.map(function (option) {
    return /*#__PURE__*/_react["default"].createElement(_MenuItem["default"], {
      key: option,
      value: option,
      onClick: handleItemClick,
      selected: value === option,
      ref: value === option ? selectedItemRef : undefined,
      jss: menuItemjssOverrides
    }, /*#__PURE__*/_react["default"].createElement(_TextField["default"], null, option));
  })));
});

TimePicker.defaultProps = {
  interval: 15,
  className: '',
  style: {},
  name: '',
  value: undefined,
  jss: {},
  disabled: false,
  resetable: false,
  defaultValue: undefined
};
TimePicker.propTypes = {
  /** Callback function to be called on value changed - arguments - (event) */
  onChange: _propTypes["default"].func.isRequired,

  /** Additional class for timepicker root */
  className: _propTypes["default"].string,

  /** Additional style for timepicker root */
  style: _propTypes["default"].shape(),

  /** form element name */
  name: _propTypes["default"].string,

  /** Controlled value to set to the time picker */
  value: _propTypes["default"].string,

  /** Indicator for disabled status */
  disabled: _propTypes["default"].bool,

  /** The interval leaps for the picker items in minutes */
  interval: _propTypes["default"].number,

  /** JSS overrides for TimePicker */
  jss: _propTypes["default"].oneOfType([_propTypes["default"].shape(), _propTypes["default"].func]),

  /** default value to reset to */
  defaultValue: _propTypes["default"].string,

  /** indicator whether or not to allow reset */
  resetable: _propTypes["default"].bool
};
TimePicker.displayName = 'TimePicker';
var _default = TimePicker;
exports["default"] = _default;
