// @guestyci/foundation/legacy/Pill/Pill.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactJss = require("react-jss");

var useStyles = (0, _reactJss.createUseStyles)({
  pill: {
    height: function height(_ref) {
      var _height = _ref.height;
      return _height;
    },
    width: function width(_ref2) {
      var _width = _ref2.width;
      return _width;
    },
    borderRadius: function borderRadius(_ref3) {
      var _borderRadius = _ref3.borderRadius;
      return _borderRadius;
    }
  }
});

var Pill = function Pill(_ref4) {
  var _cn;

  var className = _ref4.className,
      color = _ref4.color,
      width = _ref4.width,
      height = _ref4.height,
      bgColor = _ref4.bgColor,
      borderColor = _ref4.borderColor,
      textHover = _ref4.textHover,
      bgHover = _ref4.bgHover,
      borderHover = _ref4.borderHover,
      filter = _ref4.filter,
      children = _ref4.children,
      borderRadius = _ref4.borderRadius;
  var classes = useStyles({
    width: width,
    height: height,
    borderRadius: borderRadius
  });
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])(className, classes.pill, (_cn = {}, (0, _defineProperty2["default"])(_cn, "text-".concat(color), color), (0, _defineProperty2["default"])(_cn, "border-".concat(borderColor), borderColor), (0, _defineProperty2["default"])(_cn, "bg-".concat(bgColor), bgColor), (0, _defineProperty2["default"])(_cn, "text-hover-".concat(textHover), textHover), (0, _defineProperty2["default"])(_cn, "border-hover-".concat(borderHover), borderHover), (0, _defineProperty2["default"])(_cn, "bg-hover-".concat(bgHover), bgHover), (0, _defineProperty2["default"])(_cn, 'b-1', borderColor), (0, _defineProperty2["default"])(_cn, 'bg-hover-blue box-shadow-hover-heavy box-shadow-active-0', filter), _cn), 'px-2  pill-element overflow-ellipsis w-fit-content flex-center'),
    "data-qa": "pill"
  }, children);
};

Pill.defaultProps = {
  className: '',
  color: 'white',
  bgColor: 'black',
  borderColor: '',
  textHover: '',
  bgHover: '',
  borderHover: '',
  filter: false,
  children: null,
  width: undefined,
  height: 21,
  borderRadius: 10
};
Pill.propTypes = {
  /** ClassName url for pill */
  className: _propTypes["default"].string,

  /** Text for pill */
  color: _propTypes["default"].string,

  /** Background Color for pill */
  bgColor: _propTypes["default"].string,

  /** borderColor for pill */
  borderColor: _propTypes["default"].string,

  /** textHover for pill */
  textHover: _propTypes["default"].string,

  /** Background on hover for pill */
  bgHover: _propTypes["default"].string,

  /** borderHover for pill */
  borderHover: _propTypes["default"].string,

  /** Is it a filter */
  filter: _propTypes["default"].bool,

  /** Array or nodes , or nodes */
  children: _propTypes["default"].oneOfType([_propTypes["default"].arrayOf(_propTypes["default"].node), _propTypes["default"].node]),

  /** Set the pill width */
  width: _propTypes["default"].number,

  /** Set the pill height */
  height: _propTypes["default"].number,

  /** Set the pill border radius */
  borderRadius: _propTypes["default"].number
};
var _default = Pill;
exports["default"] = _default;
