// @guestyci/foundation/legacy/Table/useColumnAdapter/useColumnAdapter.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _interopRequireWildcard2 = _interopRequireDefault(require("@babel/runtime/helpers/interopRequireWildcard"));

var _react = require("react");

var _memoize = _interopRequireDefault(require("lodash/memoize"));

var useColumnAdapter = function useColumnAdapter(_ref) {
  var configColumns = _ref.configColumns,
      viewColumns = _ref.viewColumns;
  var overridingAdapters = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : [];
  var getCellComponent = (0, _react.useCallback)((0, _memoize["default"])(function (type) {
    return (0, _react.lazy)(function () {
      return new Promise(function (resolve) {
        Promise.resolve().then(function () {
          return (0, _interopRequireWildcard2["default"])(require("../".concat(type, "Cell")));
        }).then(resolve)["catch"](function () {
          return resolve(Promise.resolve().then(function () {
            return (0, _interopRequireWildcard2["default"])(require("../Cell"));
          }));
        });
      });
    });
  }), []);

  var defaultAdapter = function defaultAdapter(_ref2) {
    var _ref2$configColumn = _ref2.configColumn,
        configColumn = _ref2$configColumn === void 0 ? {} : _ref2$configColumn,
        _ref2$viewColumn = _ref2.viewColumn,
        viewColumn = _ref2$viewColumn === void 0 ? {} : _ref2$viewColumn;
    var dataKey = configColumn.id,
        configHeaderProps = configColumn.headerProps,
        configCellProps = configColumn.cellProps,
        cell = configColumn.cell,
        title = configColumn.title,
        configColumnProps = (0, _objectWithoutProperties2["default"])(configColumn, ["id", "headerProps", "cellProps", "cell", "title"]);
    var _ = viewColumn.id,
        viewHeaderProps = viewColumn.headerProps,
        viewCellProps = viewColumn.cellProps,
        viewColumnProps = (0, _objectWithoutProperties2["default"])(viewColumn, ["id", "headerProps", "cellProps"]);
    return {
      columnProps: (0, _extends2["default"])({}, configColumnProps, {}, viewColumnProps, {
        dataKey: dataKey
      }),
      headerProps: (0, _extends2["default"])({}, configHeaderProps, {}, viewHeaderProps, {
        children: title
      }),
      cellProps: (0, _extends2["default"])({}, configCellProps, {}, viewCellProps),
      CellComponent: getCellComponent(cell)
    };
  };

  var tableStructure = (0, _react.useMemo)(function () {
    if (!configColumns || !viewColumns) return [];
    var adapters = [defaultAdapter].concat((0, _toConsumableArray2["default"])(overridingAdapters));
    return viewColumns.map(function (viewColumn) {
      return adapters.reduce(function (acc, currentAdapter) {
        var currentColumnMetaData = currentAdapter({
          configColumn: configColumns[viewColumn.id],
          viewColumn: viewColumn
        });
        return (0, _extends2["default"])({}, acc, {}, currentColumnMetaData);
      }, {});
    });
  }, [configColumns, viewColumns].concat((0, _toConsumableArray2["default"])(overridingAdapters)));
  return tableStructure;
};

var _default = useColumnAdapter;
exports["default"] = _default;
