// @guestyci/foundation/esm/legacy/Menu/MenuAutoPosition.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _possibleConstructorReturn from "@babel/runtime/helpers/esm/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/esm/getPrototypeOf";

function _createSuper(Derived) { return function () { var Super = _getPrototypeOf(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import withStyles from 'react-jss';
import memoize from '@guestyci/memoize';
import { AnchorDirectionType, MenuSlideAnchorDirectionType } from "../../enums/enums";
import Portal from "../Portal/Portal";
import CSSTransition from "../CSSTransiton";
var styles = {
  blocker: {
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    zIndex: 1300,
    position: 'fixed',
    touchAction: 'none',
    backgroundColor: 'transparent'
  },
  menu: {
    zIndex: 1301
  }
};

var MenuAutoPosition = /*#__PURE__*/function (_PureComponent) {
  _inherits(MenuAutoPosition, _PureComponent);

  var _super = _createSuper(MenuAutoPosition);

  function MenuAutoPosition(props) {
    var _this;

    _classCallCheck(this, MenuAutoPosition);

    _this = _super.call(this, props);
    _this.getAnchorDirection = memoize(function (anchor, parentWidth) {
      var childWidth = _this.childRef.current.offsetWidth;

      switch (anchor) {
        case AnchorDirectionType.Left:
          return -childWidth;

        case AnchorDirectionType.Right:
          return parentWidth;

        case AnchorDirectionType.EndCenter:
          return -(childWidth - parentWidth / 2);

        case AnchorDirectionType.StartCenter:
          return parentWidth / 2;

        case AnchorDirectionType.Center:
          return -(childWidth - parentWidth) / 2;

        case AnchorDirectionType.End:
          return -(childWidth - parentWidth);

        default:
          return 0;
      }
    });

    _this.isInViewPort = function (offsetHeight) {
      return offsetHeight <= window.scrollY + (window.innerHeight || document.documentElement.clientHeight);
    };

    _this.calculateComponentPosition = memoize(function (left, parentWidth, top, parentHeight) {
      var _this$childRef$curren = _this.childRef.current,
          offsetHeight = _this$childRef$curren.offsetHeight,
          offsetWidth = _this$childRef$curren.offsetWidth;
      var direction = MenuSlideAnchorDirectionType.Top;
      var offsetTop = top + window.scrollY;
      var width = offsetWidth >= parentWidth ? offsetWidth : parentWidth;
      var position = {
        left: left,
        top: offsetTop,
        width: width
      };

      if (!_this.isInViewPort(offsetTop + offsetHeight + parentHeight)) {
        position.top -= offsetHeight;
        direction = MenuSlideAnchorDirectionType.Bottom;
      } else {
        position.top += parentHeight;
      }

      return {
        position: position,
        direction: direction
      };
    });
    _this.autoPositionRef = React.createRef();
    _this.childRef = React.createRef();
    _this.state = {
      position: {},
      direction: MenuSlideAnchorDirectionType.Top
    };
    return _this;
  }

  _createClass(MenuAutoPosition, [{
    key: "componentDidUpdate",
    value: function componentDidUpdate(prevProps) {
      var trigger = this.props.trigger;

      if (trigger && prevProps.trigger !== trigger) {
        this.calculatePosition();
      }
    }
  }, {
    key: "calculatePosition",
    value: function calculatePosition() {
      var anchorDirection = this.props.anchorDirection;
      var el = this.autoPositionRef.current;

      if (el) {
        var parent = el.parentElement;
        var offsetHeight = parent.offsetHeight;

        var _parent$getBoundingCl = parent.getBoundingClientRect(),
            left = _parent$getBoundingCl.left,
            width = _parent$getBoundingCl.width,
            top = _parent$getBoundingCl.top;

        var relativeAnchorDirection = this.getAnchorDirection(anchorDirection, width);

        var _this$calculateCompon = this.calculateComponentPosition(left + relativeAnchorDirection, width, top, offsetHeight),
            position = _this$calculateCompon.position,
            direction = _this$calculateCompon.direction;

        this.setState({
          position: position,
          direction: direction
        });
      }
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props = this.props,
          children = _this$props.children,
          className = _this$props.className,
          style = _this$props.style,
          trigger = _this$props.trigger,
          classes = _this$props.classes;
      var _this$state = this.state,
          position = _this$state.position,
          direction = _this$state.direction;
      return /*#__PURE__*/React.createElement("div", {
        ref: this.autoPositionRef,
        "data-qa": "menu-auto-position"
      }, /*#__PURE__*/React.createElement(Portal, {
        withState: true,
        isOpen: trigger
      }, trigger && /*#__PURE__*/React.createElement("div", {
        className: classes.blocker
      }), /*#__PURE__*/React.createElement(CSSTransition, {
        animation: "scale",
        show: trigger,
        className: cn("menu-auto-position absolute transform-center-".concat(direction), classes.menu, className),
        style: _extends({}, style, {}, position)
      }, /*#__PURE__*/React.createElement("div", {
        ref: this.childRef
      }, children))));
    }
  }]);

  return MenuAutoPosition;
}(PureComponent);

MenuAutoPosition.defaultProps = {
  className: '',
  style: {},
  anchorDirection: AnchorDirectionType.Start
};
MenuAutoPosition.propTypes = {
  /** Child node for the auto position to reference and wrap */
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.arrayOf(PropTypes.node)]).isRequired,

  /** Indicator when to re calculate the position */
  trigger: PropTypes.bool.isRequired,

  /** Additional class to root level */
  className: PropTypes.string,

  /** Additional style to root level */
  style: PropTypes.shape(),

  /** Direction to open the menu (center, left, right, end ,start, start center, end center) */
  anchorDirection: PropTypes.oneOf([AnchorDirectionType.Center, AnchorDirectionType.Left, AnchorDirectionType.Right, AnchorDirectionType.StartCenter, AnchorDirectionType.EndCenter, AnchorDirectionType.Start, AnchorDirectionType.End])
};
export default withStyles(styles)(MenuAutoPosition);
