// @guestyci/foundation/es/legacy/Bullet/Bullet.js
import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { createUseStyles } from 'react-jss';
const useStyles = createUseStyles({
  bullet: {
    width: ({
      width
    }) => width,
    height: ({
      height
    }) => height,
    borderRadius: '50%'
  }
});

const Bullet = ({
  height,
  width,
  color,
  className,
  style
}) => {
  const classes = useStyles({
    width,
    height
  });
  return /*#__PURE__*/React.createElement("div", {
    className: cn(classes.bullet, {
      [`bg-${color}`]: color
    }, className),
    style: style,
    "data-qa": "bullet"
  });
};

Bullet.defaultProps = {
  color: 'muted',
  width: 4,
  height: 4,
  className: '',
  style: {}
};
Bullet.propTypes = {
  /** Bullet color */
  color: PropTypes.string,

  /** Bullet height */
  height: PropTypes.number,

  /** Bullet width */
  width: PropTypes.number,

  /** Additional class for bullet root level */
  className: PropTypes.string,

  /** Additional style to bullet root level */
  style: PropTypes.shape()
};
export default Bullet;
