// @guestyci/ebisu/util/slack.js
const { IncomingWebhook } = require('@slack/webhook');

const sendInventoryUpdateNotification = async ({
  platformWebhookURL,
  teamWebhookURL,
  ...rest
}) => {
  const platformWebhook = new IncomingWebhook(platformWebhookURL);
  const platformUpdate = createPlatformMessage(rest);
  await platformWebhook.send(platformUpdate);

  const teamWebhook = new IncomingWebhook(teamWebhookURL);
  const teamUpdate = createTeamMessage(rest);
  await teamWebhook.send(teamUpdate);
};

const createPlatformMessage = ({ env, project, branch, updater, prURL }) => {
  return {
    attachments: [
      {
        fallback: 'its mandatory for some reason',
        actions: [
          {
            type: 'button',
            text: 'View PR',
            url: `${prURL}/files`,
            style: 'primary',
          },
          {
            type: 'button',
            text: 'View Project Branch',
            url: `https://www.github.com/guestyorg/${project}/commits/${branch}`,
          },
        ],
      },
    ],
    blocks: [
      {
        type: 'header',
        text: {
          type: 'plain_text',
          text: 'Inventory PR Awaiting Your Approval',
        },
      },
      {
        type: 'divider',
      },
      {
        type: 'context',
        elements: [
          {
            type: 'mrkdwn',
            text: `Submitted by *${updater}*`,
          },
        ],
      },
      {
        type: 'section',
        text: {
          type: 'mrkdwn',
          text: `For *${env.toLowerCase()}*\nProject Updated: *${project}*`,
        },
        accessory: {
          type: 'image',
          image_url: `${randFromArray(images)}`,
          alt_text: 'image of the day',
        },
      },
    ],
  };
};

const createTeamMessage = ({ env, project, branch, updater, prURL }) => {
  return {
    attachments: [
      {
        fallback: 'its mandatory for some reason',
        actions: [
          {
            type: 'button',
            text: 'View PR',
            url: `${prURL}/files`,
            style: 'primary',
          },
          {
            type: 'button',
            text: 'View Project Branch',
            url: `https://www.github.com/guestyorg/${project}/commits/${branch}`,
          },
        ],
      },
    ],
    blocks: [
      {
        type: 'header',
        text: {
          type: 'plain_text',
          text: 'Inventory PR Opened and Reviewed By Platform!',
        },
      },
      {
        type: 'divider',
      },
      {
        type: 'section',
        text: {
          type: 'mrkdwn',
          text: `You have changed the manifest of *${project}* in *${env.toLowerCase()}*. A Pull request has been opened and is now being reviewed by the platform team.`,
        },
      },
      {
        type: 'section',
        text: {
          type: 'mrkdwn',
          text: `The team will get back to you as soon as its handled :D\nPlease have patience :pray:`,
        },
      },
    ],
  };
};

const images = [
  'https://upload.wikimedia.org/wikipedia/en/4/45/Berserk_vol01.jpg',
  'https://upload.wikimedia.org/wikipedia/commons/9/97/Joseph_Mallord_William_Turner_-_Flint_Castle.jpg',
  'https://upload.wikimedia.org/wikipedia/commons/a/ab/Apollo_and_Daphne_%28Bernini%29_%28cropped%29.jpg',
  'https://upload.wikimedia.org/wikipedia/commons/9/99/Joseph_Mallord_William_Turner_%28British_-_Modern_Rome-Campo_Vaccino_-_Google_Art_Project.jpg',
  'https://upload.wikimedia.org/wikipedia/commons/0/0d/Joseph_Mallord_William_Turner_-_Fishermen_at_Sea_-_Google_Art_Project.jpg',
  'https://upload.wikimedia.org/wikipedia/commons/3/30/The_Fighting_Temeraire%2C_JMW_Turner%2C_National_Gallery.jpg',
  'https://upload.wikimedia.org/wikipedia/commons/6/66/J._R._R._Tolkien%2C_1940s.jpg',
  'https://upload.wikimedia.org/wikipedia/en/7/72/Evangelion_retouched.png',
  'https://upload.wikimedia.org/wikipedia/en/f/f4/Berserk_%281997_anime%29%2C_DVD_cover_1.jpg',
  'https://upload.wikimedia.org/wikipedia/commons/c/c9/Kinkaku3402CBcropped.jpg',
  'https://upload.wikimedia.org/wikipedia/commons/5/54/Ninnaji_Kyoto07n4500.jpg',
  'https://upload.wikimedia.org/wikipedia/commons/d/dc/Shoyu_Ramen.jpg',
  'https://upload.wikimedia.org/wikipedia/commons/8/83/Tokyo_from_the_top_of_the_SkyTree_%28cropped%29.JPG',
  'https://upload.wikimedia.org/wikipedia/commons/5/57/Shibuya_crossing.jpg',
  'https://upload.wikimedia.org/wikipedia/commons/c/ca/Tokyo_Tower_at_night_8.JPG',
  'https://upload.wikimedia.org/wikipedia/commons/f/f4/Osaka_Dotonbori_Ebisu_Bridge.jpg',
  'https://upload.wikimedia.org/wikipedia/commons/1/19/Namba_20150531.JPG',
  'https://upload.wikimedia.org/wikipedia/commons/0/03/Dotonbori_19.jpg',
  'https://upload.wikimedia.org/wikipedia/en/d/db/Spirited_Away_Japanese_poster.png',
  'https://upload.wikimedia.org/wikipedia/commons/a/a6/Ogata_Gekko_-_Ryu_sho_ten_edit.jpg',
  'https://upload.wikimedia.org/wikipedia/en/1/14/Andrzej_Sapkowski_-_The_Last_Wish.jpg',
  'https://upload.wikimedia.org/wikipedia/en/0/00/WoT01_TheEyeOfTheWorld.jpg',
  'https://upload.wikimedia.org/wikipedia/en/1/18/Baynes-Map_of_Middle-earth.jpg',
  'https://upload.wikimedia.org/wikipedia/en/9/91/WoW_Box_Art1.jpg',
  'https://upload.wikimedia.org/wikipedia/en/e/e7/Illidan_Stormrage.png',
  'https://en.wikipedia.org/wiki/File:Arthas_Menethil.png',
];

const randFromArray = arr => {
  return arr[Math.floor(Math.random() * arr.length)];
};
module.exports = {
  sendInventoryUpdateNotification,
};
