// @guestyci/foundation/es/InfiniteScroll/InfiniteScroll.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _objectWithoutPropertiesLoose from "@babel/runtime/helpers/esm/objectWithoutPropertiesLoose";
import React from 'react';
import PropTypes from 'prop-types';
import noop from 'lodash/noop';
import cn from 'classnames';
import useInfiniteScroll from "../useInfiniteScroll";
import useComposeRefs from "../useComposeRefs";
import createStyles from "../createStyles";
import Spinner from "../Spinner";
import Row from "../Layout/Row";
const useStyles = createStyles({
  root: {
    display: 'block',
    height: 'inherit',
    maxHeight: 'inherit',
    overflow: 'auto'
  },
  loader: {}
}, {
  name: 'InfiniteScroll'
});
const InfiniteScroll = React.forwardRef(function InfiniteScroll(_ref, ref) {
  let {
    children,
    className,
    loader = /*#__PURE__*/React.createElement(Spinner, null),
    component = 'span',
    isLoading,
    totalCount,
    onLoadMore = noop,
    skip,
    scrollOffset = 200,
    pageSize,
    jss
  } = _ref,
      props = _objectWithoutPropertiesLoose(_ref, ["children", "className", "loader", "component", "isLoading", "totalCount", "onLoadMore", "skip", "scrollOffset", "pageSize", "jss"]);

  const scrollableRef = useInfiniteScroll({
    onLoadMore,
    skip,
    pageSize,
    scrollOffset,
    isLoading,
    totalCount
  });
  const infiniteScrollRef = useComposeRefs(scrollableRef, ref);
  const classes = useStyles({
    jss
  });
  const Tag = component || 'span';
  return totalCount ? /*#__PURE__*/React.createElement(Tag, _extends({
    className: cn(classes.root, className),
    ref: infiniteScrollRef
  }, props), children, isLoading && /*#__PURE__*/React.createElement(Row, {
    className: classes.loader,
    justify: "center",
    align: "center"
  }, loader)) : children;
});
InfiniteScroll.propTypes = {
  /** child elements to render into the infinite scroll */
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.arrayOf(PropTypes.node)]).isRequired,

  /** Custom loader component */
  loader: PropTypes.node,

  /** Custom root component to render */
  component: PropTypes.node,

  /** loading state of the data */
  isLoading: PropTypes.bool,

  /** total count of the data entries */
  totalCount: PropTypes.number,

  /** Scroll offset, from the bottom, from which loading the next chunk should be triggered */
  scrollOffset: PropTypes.number,

  /** index of the first item in the current chunk */
  skip: PropTypes.number,

  /** cb function handle loading more data */
  onLoadMore: PropTypes.func,

  /** number of rows per page */
  pageSize: PropTypes.number,

  /** JSS overrides */
  jss: PropTypes.oneOfType([PropTypes.shape(), PropTypes.func])
};
InfiniteScroll.displayName = 'InfiniteScroll';
export default InfiniteScroll;
