// @guestyci/foundation/esm/Layout/Row/Row.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import _extends from "@babel/runtime/helpers/esm/extends";
import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { firstLetterUppercase as capitalize } from "../../utils";
import createStyles from "../../createStyles";
import styles, { generateMediaSupport, generateSpacing, spacing as spacingArray } from "../styles";
export var useStyles = createStyles(function (theme) {
  return _extends({
    root: {
      display: 'flex'
    },
    flexDirectionReverse: {
      flexDirection: 'row-reverse'
    },
    lastChildSpacing: {
      '& > *:last-child': {
        marginRight: 0
      }
    }
  }, styles, {}, generateMediaSupport(theme.breakpoints), {}, generateSpacing(theme.spacer, spacingArray, 'right'));
}, {
  name: 'Row'
});
var Row = React.forwardRef(function Row(_ref, ref) {
  var _cn;

  var className = _ref.className,
      style = _ref.style,
      children = _ref.children,
      spacing = _ref.spacing,
      align = _ref.align,
      justify = _ref.justify,
      wrap = _ref.wrap,
      fullWidth = _ref.fullWidth,
      reverse = _ref.reverse,
      breakpoints = _ref.breakpoints,
      span = _ref.span,
      _ref$jss = _ref.jss,
      jss = _ref$jss === void 0 ? {} : _ref$jss,
      props = _objectWithoutProperties(_ref, ["className", "style", "children", "spacing", "align", "justify", "wrap", "fullWidth", "reverse", "breakpoints", "span", "jss"]);

  var classes = useStyles({
    jss: jss
  });

  var _ref2 = breakpoints || {},
      xs = _ref2.xs,
      sm = _ref2.sm,
      md = _ref2.md,
      lg = _ref2.lg,
      xl = _ref2.xl,
      xlg = _ref2.xlg;

  var xsSize = span || xs;
  return /*#__PURE__*/React.createElement("div", _extends({}, props, {
    ref: ref,
    className: cn(classes.root, (_cn = {}, _defineProperty(_cn, classes.flexWrap, wrap), _defineProperty(_cn, classes["spacing".concat(spacing)], spacing), _defineProperty(_cn, classes.lastChildSpacing, spacing), _defineProperty(_cn, classes.fullWidth, fullWidth), _defineProperty(_cn, classes.flexDirectionReverse, reverse), _defineProperty(_cn, classes["justify".concat(capitalize(justify))], justify !== 'start'), _defineProperty(_cn, classes["alignItems".concat(capitalize(align))], align !== 'stretch'), _defineProperty(_cn, classes["col-xs-".concat(xsSize)], xsSize), _defineProperty(_cn, classes["col-sm-".concat(sm)], sm), _defineProperty(_cn, classes["col-md-".concat(md)], md), _defineProperty(_cn, classes["col-lg-".concat(lg)], lg), _defineProperty(_cn, classes["col-xl-".concat(xl)], xl), _defineProperty(_cn, classes["col-xlg-".concat(xlg)], xlg), _cn), 'row-wrapper', className),
    style: style
  }), children);
});
Row.defaultProps = {
  className: '',
  spacing: 0,
  style: {},
  children: null,
  justify: 'start',
  align: 'stretch',
  reverse: false,
  wrap: false,
  fullWidth: false,
  span: 0,
  breakpoints: null
};
Row.propTypes = {
  /** Additional class for top level */
  className: PropTypes.string,

  /** Indicator whether or not to render flex wrap */
  wrap: PropTypes.bool,

  /** Indicator whether or not to render flex direction reverse */
  reverse: PropTypes.bool,

  /** Additional styling for root level */
  style: PropTypes.shape(),

  /** set spacing between child components - available value - [0,1,2,3,4,5,6,7,8,9,10, 11 ,12] will be determined by the theme spacer (default 5) */
  spacing: PropTypes.oneOf([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]),

  /** Child components inside a section */
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.arrayOf(PropTypes.node)]),

  /** Set flex alignment - available values = ['start', 'center', 'end', 'stretch', 'baseline']  */
  align: PropTypes.oneOf(['start', 'center', 'end', 'stretch', 'baseline', 'inherit']),

  /** Set flex justify content - available values = ['start','center','end','between','around','evenly'] */
  justify: PropTypes.oneOf(['start', 'center', 'end', 'between', 'around', 'evenly', 'inherit']),

  /** Set the row width to fit parent */
  fullWidth: PropTypes.bool,

  /** Set the row span width - available values = [auto,1,2,3,4,5,6,7,8,9,10,11,12] */
  span: PropTypes.oneOf(['auto', 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]),

  /** specify the media query breakpoints for the row - available values = {xs,sm,md,lg,xl,xlg} */
  breakpoints: PropTypes.shape(),

  /** JSS overrides */
  jss: PropTypes.oneOfType([PropTypes.shape(), PropTypes.func])
}; // Hack for withInfo plugin

Row.displayName = 'Row';
export default Row;
