// @guestyci/foundation/legacy/ActionPopover/ActionPopoverBody.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Col = _interopRequireDefault(require("../Col/Col"));

var _RaisedButton = _interopRequireDefault(require("../RaisedButton/RaisedButton"));

var ActionPopoverBody = function ActionPopoverBody(_ref) {
  var buttonText = _ref.buttonText,
      onClick = _ref.onClick,
      body = _ref.body;
  return /*#__PURE__*/_react["default"].createElement(_Col["default"], {
    spacing: 2,
    "data-qa": "action-popover-body"
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: "text-left"
  }, body), /*#__PURE__*/_react["default"].createElement("div", {
    className: "flex-end"
  }, /*#__PURE__*/_react["default"].createElement(_RaisedButton["default"], {
    onClick: onClick
  }, buttonText)));
};

ActionPopoverBody.propTypes = {
  /** Sets the text of the button */
  buttonText: _propTypes["default"].string.isRequired,

  /** Callback function - called when clicking the tooltip action */
  onClick: _propTypes["default"].func.isRequired,

  /** Sets the body (content) of the tooltip F.E: text to be displayed */
  body: _propTypes["default"].node.isRequired
};
var _default = ActionPopoverBody;
exports["default"] = _default;
