// @guestyci/foundation/legacy/Alert/Alert.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Row = _interopRequireDefault(require("../Row/Row"));

var _SvgIcon = _interopRequireDefault(require("../SvgIcon/SvgIcon"));

var _TextField = _interopRequireDefault(require("../TextField/TextField"));

var IcoNotice = function IcoNotice(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M10.5 0C4.701 0 0 4.701 0 10.5S4.701 21 10.5 21 21 16.299 21 10.5 16.299 0 10.5 0zm-.923 5.53h1.885v3.717c0 .4-.024.795-.071 1.187-.048.392-.11.792-.189 1.199H9.837c-.078-.407-.14-.807-.189-1.2a9.873 9.873 0 0 1-.07-1.186V5.53zm1.98 8.888c-.059.137-.14.255-.245.354a1.17 1.17 0 0 1-.812.319c-.165 0-.316-.028-.455-.084a1.088 1.088 0 0 1-.601-.589c-.059-.136-.088-.282-.088-.439a1.076 1.076 0 0 1 .329-.786 1.145 1.145 0 0 1 .815-.325 1.175 1.175 0 0 1 .813.325 1.082 1.082 0 0 1 .331.786c0 .157-.03.303-.087.439z"
  }));
};

IcoNotice.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 21 21"
};

var IcoApproved = function IcoApproved(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M10.5 0C4.701 0 0 4.701 0 10.5S4.701 21 10.5 21 21 16.299 21 10.5 16.299 0 10.5 0zm4.633 6.752l-5.537 8.202c-.018.026-.044.04-.065.063-.012.014-.018.032-.032.045-.01.01-.024.013-.035.021a.597.597 0 0 1-.123.075.595.595 0 0 1-.096.04.608.608 0 0 1-.121.018.617.617 0 0 1-.12 0 .602.602 0 0 1-.103-.025.596.596 0 0 1-.13-.054c-.013-.007-.027-.008-.04-.016-.016-.01-.024-.028-.04-.04-.023-.019-.051-.03-.072-.052l-2.693-2.931c-.233-.252-.218-.648.034-.88s.647-.219.88.033l2.163 2.354 5.098-7.55a.625.625 0 0 1 .864-.168c.284.192.36.581.168.865z"
  }));
};

IcoApproved.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 21 21"
};

var IcoDeclined = function IcoDeclined(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M10.5 0C4.701 0 0 4.701 0 10.5S4.701 21 10.5 21 21 16.299 21 10.5 16.299 0 10.5 0zm4.212 13.75l-.962.962-.106-.106-3.144-3.144-3.25 3.25-.962-.962 3.25-3.25-3.25-3.25.962-.962 3.25 3.25 3.25-3.25.962.962-3.25 3.25 3.25 3.25z"
  }));
};

IcoDeclined.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 21 21"
};

var isStart = function isStart(position) {
  return position === 'start';
};

var iconForType = function iconForType(type) {
  switch (type) {
    case 'warning':
      return /*#__PURE__*/_react["default"].createElement(_SvgIcon["default"], {
        svg: IcoNotice,
        color: "yellow"
      });

    case 'success':
      return /*#__PURE__*/_react["default"].createElement(_SvgIcon["default"], {
        svg: IcoApproved,
        color: "green"
      });

    case 'error':
      return /*#__PURE__*/_react["default"].createElement(_SvgIcon["default"], {
        svg: IcoDeclined,
        color: "red"
      });

    default:
      return null;
  }
};

var Alert = function Alert(_ref) {
  var message = _ref.message,
      iconPosition = _ref.iconPosition,
      type = _ref.type;
  var isIconStart = isStart(iconPosition);
  return /*#__PURE__*/_react["default"].createElement(_react.Fragment, null, message ? /*#__PURE__*/_react["default"].createElement(_Row["default"], {
    spacing: "sm"
  }, isIconStart && iconForType(type), /*#__PURE__*/_react["default"].createElement(_TextField["default"], null, message), !isIconStart && iconForType(type)) : null);
};

Alert.defaultProps = {
  message: '',
  iconPosition: 'start',
  type: 'warning'
};
Alert.propTypes = {
  /** Message for Alert, if none provided will not render anything */
  message: _propTypes["default"].string,

  /** Position for the icon - start or end */
  iconPosition: _propTypes["default"].oneOf(['start', 'end']),

  /** Type for the icon - success, warning, danger */
  type: _propTypes["default"].oneOf(['success', 'warning', 'error'])
};
var _default = Alert;
exports["default"] = _default;
