// @guestyci/foundation/es/Input/styles.js
import { fade } from "../theme/utils";
export default {
  shared: theme => ({
    width: '100%',
    height: '100%',
    borderRadius: '2px',
    border: `solid 1px ${theme.palette.border}`,
    color: `${theme.palette.text.input} !important`,
    transition: theme.transition.create(),
    paddingLeft: theme.spacer(3),
    paddingRight: theme.spacer(3)
  }),
  focused: theme => ({
    border: `solid 1px ${theme.palette.primary.default}`,
    boxShadow: `0 1px 8px 0 ${fade(theme.palette.primary.hover, 0.5)}`
  }),
  hover: theme => ({
    border: `solid 1px ${theme.palette.primary.default}`
  }),
  error: theme => ({
    backgroundColor: theme.palette.error.background,
    border: `solid 1px ${theme.palette.error.default} !important`,
    boxShadow: 'none !important'
  }),
  disabled: theme => ({
    cursor: 'not-allowed',
    color: `${theme.palette.text.placeholder} !important`,
    backgroundColor: theme.palette.background.system,
    '&:hover': {
      border: `solid 1px ${theme.palette.border}`,
      boxShadow: 'none'
    }
  }),
  placeholder: theme => ({
    color: theme.palette.text.placeholder
  })
};
