// @guestyci/foundation/es/Dropdown/Option.js
import _objectWithoutPropertiesLoose from "@babel/runtime/helpers/esm/objectWithoutPropertiesLoose";
import _extends from "@babel/runtime/helpers/esm/extends";
import React, { useCallback, useMemo } from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import MenuItem from "../Menu/MenuItem";
import TextField from "../TextField";
import checkboxStyles from "../Checkbox/styles";
import CheckedIcon from "../Checkbox/CheckedIcon";

var CheckboxIconOff = function CheckboxIconOff(props) {
  return /*#__PURE__*/React.createElement("svg", props, /*#__PURE__*/React.createElement("path", {
    d: "M12.5 15h-10A2.503 2.503 0 0 1 0 12.5v-10C0 1.122 1.122 0 2.5 0h10C13.879 0 15 1.122 15 2.5v10c0 1.379-1.121 2.5-2.5 2.5zM2.5 1C1.673 1 1 1.673 1 2.5v10c0 .827.673 1.5 1.5 1.5h10c.827 0 1.5-.673 1.5-1.5v-10c0-.827-.673-1.5-1.5-1.5h-10z"
  }));
};

CheckboxIconOff.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 15 15"
};
import createStyles from "../createStyles";
import Row from "../Layout/Row";
export const useStyles = createStyles(theme => _extends({}, checkboxStyles(theme)), {
  name: 'Option'
});

const jssOverrides = theme => ({
  selected: {
    color: theme.palette.primary.default
  }
});

const Option = React.forwardRef(function Option(_ref, ref) {
  let {
    value,
    multiselect = false,
    children,
    selected = false,
    onClick,
    overflowEllipsis = true
  } = _ref,
      props = _objectWithoutPropertiesLoose(_ref, ["value", "multiselect", "children", "selected", "onClick", "overflowEllipsis"]);

  const classes = useStyles();
  const handleOptionClick = useCallback((e, selectedItem) => {
    onClick(e, selectedItem);
  }, [onClick]);
  const CheckboxIcon = useMemo(() => selected ? CheckedIcon : CheckboxIconOff, [selected]);
  const selectedStyles = !multiselect ? jssOverrides : undefined;
  return /*#__PURE__*/React.createElement(MenuItem, _extends({
    selected: selected,
    jss: selectedStyles,
    value: value,
    onClick: handleOptionClick
  }, props, {
    ref: ref
  }), /*#__PURE__*/React.createElement(Row, {
    spacing: 2,
    fullWidth: true,
    align: "center"
  }, multiselect && /*#__PURE__*/React.createElement(CheckboxIcon, {
    height: 16,
    width: 16,
    className: cn(classes.icon, {
      [classes.uncheckedDefaultBg]: !selected,
      [classes.checked]: selected
    })
  }), /*#__PURE__*/React.createElement(TextField, {
    overflowEllipsis: overflowEllipsis
  }, children)));
});
Option.propTypes = {
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.shape()]).isRequired,
  children: PropTypes.node.isRequired,
  overflowEllipsis: PropTypes.bool
};
Option.displayName = 'Option';
export default Option;
