// @guestyci/foundation/legacy/Table/CurrencyCell/CurrencyCell.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = require("prop-types");

var _classnames = _interopRequireDefault(require("classnames"));

var _FormattedCurrency = _interopRequireDefault(require("@guestyci/localize/FormattedCurrency"));

var _TextField = _interopRequireDefault(require("../../TextField"));

var _TextCell = _interopRequireDefault(require("../TextCell"));

function CurrencyCell(_ref) {
  var value = _ref.value,
      currency = _ref.currency,
      color = _ref.color,
      isLoading = _ref.isLoading,
      overflowEllipsis = _ref.overflowEllipsis,
      overflowEllipsisTooltip = _ref.overflowEllipsisTooltip,
      className = _ref.className,
      contentClassName = _ref.contentClassName,
      style = _ref.style,
      rest = (0, _objectWithoutProperties2["default"])(_ref, ["value", "currency", "color", "isLoading", "overflowEllipsis", "overflowEllipsisTooltip", "className", "contentClassName", "style"]);
  return /*#__PURE__*/_react["default"].createElement(_TextCell["default"], (0, _extends2["default"])({}, rest, {
    isLoading: isLoading,
    overflowEllipsis: overflowEllipsis,
    overflowEllipsisTooltip: overflowEllipsisTooltip,
    className: (0, _classnames["default"])('currency-cell', className),
    contentClassName: contentClassName,
    style: style,
    "data-qa": "currency-cell"
  }), /*#__PURE__*/_react["default"].createElement(_TextField["default"], {
    color: color,
    className: "d-inline"
  }, /*#__PURE__*/_react["default"].createElement(_FormattedCurrency["default"], {
    currency: currency,
    value: value
  })));
}

CurrencyCell.propTypes = {
  /** value is required */
  value: (0, _propTypes.oneOfType)([_propTypes.number, _propTypes.string]).isRequired,

  /** currency type (USD,EUR,...) */
  currency: _propTypes.string,

  /** color of the text as defined in design system colors */
  color: _propTypes.string,

  /** indicator that cell is yet to receive data */
  isLoading: _propTypes.bool,

  /** indicator that cell should trim cell content and show tooltip with full data */
  overflowEllipsis: _propTypes.bool,

  /** tooltip body of an overflowed cell content */
  overflowEllipsisTooltip: _propTypes.string,

  /** class name of cell */
  className: _propTypes.string,

  /** class name of cell content wrapper */
  contentClassName: _propTypes.string,

  /** styles applied to a cell */
  style: (0, _propTypes.shape)()
};
CurrencyCell.defaultProps = {
  currency: '',
  color: '',
  className: '',
  contentClassName: '',
  style: {},
  isLoading: false,
  overflowEllipsis: true,
  overflowEllipsisTooltip: ''
};
var _default = CurrencyCell;
exports["default"] = _default;
