// @guestyci/foundation/esm/legacy/RaisedButton/RaisedButton.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React from 'react';
import PropTypes from 'prop-types';
import noop from 'lodash/noop';
import cn from 'classnames';
import BaseButton from "../BaseButton/BaseButton";

var RaisedButton = function RaisedButton(_ref) {
  var className = _ref.className,
      children = _ref.children,
      onClick = _ref.onClick,
      style = _ref.style,
      disabled = _ref.disabled,
      isLoading = _ref.isLoading,
      props = _objectWithoutProperties(_ref, ["className", "children", "onClick", "style", "disabled", "isLoading"]);

  return /*#__PURE__*/React.createElement(BaseButton, _extends({
    isLoading: isLoading,
    className: cn('font-size-md', className),
    onClick: onClick,
    variant: "raised",
    color: "primary",
    style: style,
    disabled: disabled
  }, props, {
    "data-qa": "raised-button"
  }), children);
};

RaisedButton.defaultProps = {
  className: '',
  style: {},
  disabled: false,
  isLoading: false,
  onClick: noop
};
RaisedButton.propTypes = {
  /** Button click callback function */
  onClick: PropTypes.func,

  /** Button inner children */
  children: PropTypes.node.isRequired,

  /** Button indicator whether disabled or not */
  disabled: PropTypes.bool,

  /** Button additional classes */
  className: PropTypes.string,

  /** Button additional styling */
  style: PropTypes.shape(),

  /** Indicator for  button loading state  */
  isLoading: PropTypes.bool
};
export default RaisedButton;
