// @guestyci/foundation/legacy/Accordion/AccordionTitle.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _withHover = _interopRequireDefault(require("../../withHover"));

var _AccordionChevron = _interopRequireDefault(require("./AccordionChevron"));

var _TextField = _interopRequireDefault(require("../TextField/TextField"));

var AccordionTitle = function AccordionTitle(_ref) {
  var title = _ref.title,
      isHovering = _ref.isHovering,
      onHoverEnter = _ref.onHoverEnter,
      onHoverLeave = _ref.onHoverLeave,
      onClick = _ref.onClick,
      isOpen = _ref.isOpen;
  return /*#__PURE__*/_react["default"].createElement("div", {
    role: "presentation",
    className: (0, _classnames["default"])('w-fill clickable flex-space-between-center pt-3', "pb-".concat(isOpen ? 6 : 3)),
    onClick: onClick,
    onMouseEnter: onHoverEnter,
    onMouseLeave: onHoverLeave,
    "data-qa": "accordion-title"
  }, /*#__PURE__*/_react["default"].createElement(_TextField["default"], {
    size: "lg",
    className: "height-20 flex-start-center w-fill",
    color: "blue"
  }, title), /*#__PURE__*/_react["default"].createElement(_AccordionChevron["default"], {
    isOpen: isOpen,
    isActive: isHovering
  }));
};

AccordionTitle.propTypes = {
  /** Indicator whether or not collapse should be open */
  isOpen: _propTypes["default"].bool.isRequired,

  /** Title body */
  title: _propTypes["default"].node.isRequired,

  /** Callback for title click */
  onClick: _propTypes["default"].func.isRequired
};

var _default = (0, _withHover["default"])(AccordionTitle);

exports["default"] = _default;
