// @guestyci/foundation/es/legacy/Calendar/CalendarTooltipReservedItem.js
import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import SvgIcon from "../SvgIcon/SvgIcon";

var IcoNotice = function IcoNotice(props) {
  return /*#__PURE__*/React.createElement("svg", props, /*#__PURE__*/React.createElement("path", {
    d: "M10.5 0C4.701 0 0 4.701 0 10.5S4.701 21 10.5 21 21 16.299 21 10.5 16.299 0 10.5 0zm-.923 5.53h1.885v3.717c0 .4-.024.795-.071 1.187-.048.392-.11.792-.189 1.199H9.837c-.078-.407-.14-.807-.189-1.2a9.873 9.873 0 0 1-.07-1.186V5.53zm1.98 8.888c-.059.137-.14.255-.245.354a1.17 1.17 0 0 1-.812.319c-.165 0-.316-.028-.455-.084a1.088 1.088 0 0 1-.601-.589c-.059-.136-.088-.282-.088-.439a1.076 1.076 0 0 1 .329-.786 1.145 1.145 0 0 1 .815-.325 1.175 1.175 0 0 1 .813.325 1.082 1.082 0 0 1 .331.786c0 .157-.03.303-.087.439z"
  }));
};

IcoNotice.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 21 21"
};

const CalendarTooltipReservedItem = ({
  title,
  value,
  color,
  alert
}) => /*#__PURE__*/React.createElement(React.Fragment, null, value && /*#__PURE__*/React.createElement("div", {
  className: "d-flex-fill flex-start-center py-1 text-left text-white font-size-md font-weight-bold"
}, /*#__PURE__*/React.createElement("div", {
  className: "width-75"
}, title), /*#__PURE__*/React.createElement("div", {
  className: cn('flex-1 text-left', {
    [`text-${color}`]: color
  })
}, value), alert && /*#__PURE__*/React.createElement(SvgIcon, {
  color: "yellow",
  svg: IcoNotice,
  className: "ml-2"
})));

CalendarTooltipReservedItem.defaultProps = {
  alert: false,
  color: 'white',
  value: ''
};
CalendarTooltipReservedItem.propTypes = {
  title: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  alert: PropTypes.bool,
  color: PropTypes.string
};
export default CalendarTooltipReservedItem;
