// @guestyci/foundation/es/legacy/Calendar/CalendarTooltip.js
import React from 'react';
import PropTypes from 'prop-types';
import ListItem from "../List/ListItem";
import CalendarTooltipNote from "./CalendarTooltipNote";
import CalendarTooltipPrice from "./CalendarTooltipPrice";
import List from "../List/List";
import CalendarTooltipBlock from "./CalendarTooltipBlock";
import CalendarTooltipReserved from "./CalendarTooltipReserved";

const CalendarTooltip = ({
  price,
  note,
  blocks,
  isReserved,
  data
}) => /*#__PURE__*/React.createElement(List, {
  className: "calendar-tooltip-list",
  "data-qa": "calendar-tooltip"
}, !isReserved && note && /*#__PURE__*/React.createElement(ListItem, {
  dense: true,
  className: "calendar-tooltip-item"
}, /*#__PURE__*/React.createElement(CalendarTooltipNote, {
  note: note
})), isReserved && /*#__PURE__*/React.createElement(CalendarTooltipReserved, {
  data: data
}), !isReserved && blocks.map(block => /*#__PURE__*/React.createElement(ListItem, {
  key: block,
  dense: true,
  className: "calendar-tooltip-item"
}, /*#__PURE__*/React.createElement(CalendarTooltipBlock, {
  block: block
}))), !isReserved && price && /*#__PURE__*/React.createElement(ListItem, {
  dense: true,
  className: "calendar-tooltip-item"
}, /*#__PURE__*/React.createElement(CalendarTooltipPrice, {
  price: price
})));

CalendarTooltip.defaultProps = {
  price: undefined,
  note: undefined,
  blocks: [],
  isReserved: false,
  data: {}
};
CalendarTooltip.propTypes = {
  price: PropTypes.string,
  note: PropTypes.string,
  blocks: PropTypes.arrayOf(PropTypes.string),
  isReserved: PropTypes.bool,
  data: PropTypes.shape()
};
export default CalendarTooltip;
