// @guestyci/foundation/legacy/CSSTransiton/CSSTransitionGroup.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _reactTransitionGroup = require("react-transition-group");

var _CSSTransition = _interopRequireDefault(require("./CSSTransition"));

var CSSTransitionGroup = function CSSTransitionGroup(_ref) {
  var animation = _ref.animation,
      className = _ref.className,
      timeout = _ref.timeout,
      children = _ref.children,
      show = _ref.show,
      style = _ref.style;
  var transitionChildren = (0, _react.useMemo)(function () {
    return children.filter(function (child) {
      return Boolean(child);
    });
  }, [children]);
  return /*#__PURE__*/_react["default"].createElement(_reactTransitionGroup.TransitionGroup, {
    component: null,
    className: (0, _classnames["default"])('css-transition-group', className),
    style: style,
    "data-qa": "css-transition-group"
  }, _react["default"].Children.map(transitionChildren, function (child, index) {
    var id = child.key || "child-".concat(index);
    return /*#__PURE__*/_react["default"].createElement(_CSSTransition["default"], {
      key: id,
      show: show,
      animation: animation,
      timeout: timeout
    }, child);
  }));
};

CSSTransitionGroup.defaultProps = {
  className: '',
  timeout: 0,
  animation: ''
};
CSSTransitionGroup.propTypes = {
  /* Animation for css transition from animation scss */
  animation: _propTypes["default"].string,

  /* Additional class name for css transition root */
  className: _propTypes["default"].string,

  /* Child components for css transition */
  children: _propTypes["default"].arrayOf(_propTypes["default"].node).isRequired,

  /* timeout between animations */
  timeout: _propTypes["default"].oneOfType([_propTypes["default"].shape({
    enter: _propTypes["default"].number.isRequired,
    leave: _propTypes["default"].number.isRequired
  }), _propTypes["default"].number])
};
var _default = CSSTransitionGroup;
exports["default"] = _default;
