// @guestyci/foundation/legacy/CSSTransiton/TransitionPovider.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _CSSTransition = _interopRequireDefault(require("./CSSTransition"));

var _CSSTransitionGroup = _interopRequireDefault(require("./CSSTransitionGroup"));

var _utils = require("../../utils");

var TransitionProvider = function TransitionProvider(_ref) {
  var animation = _ref.animation,
      timeout = _ref.timeout,
      children = _ref.children,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["animation", "timeout", "children"]);
  var transitionTimeout = (0, _utils.isEmpty)(timeout) ? 225 : timeout;

  if ((0, _utils.isEmpty)(animation)) {
    return children;
  }

  if ((0, _utils.isEmpty)(children)) {
    return null;
  }

  return Array.isArray(children) ? /*#__PURE__*/_react["default"].createElement(_CSSTransitionGroup["default"], (0, _extends2["default"])({}, props, {
    animation: animation,
    timeout: transitionTimeout
  }), children) : /*#__PURE__*/_react["default"].createElement(_CSSTransition["default"], (0, _extends2["default"])({}, props, {
    animation: animation,
    timeout: transitionTimeout
  }), children);
};

_CSSTransition["default"].defaultProps = {
  className: '',
  children: null,
  timeout: 0,
  animation: ''
};
_CSSTransition["default"].propTypes = {
  /* Animation for css transition from animation scss */
  animation: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].shape()]),

  /* Additional class name for css transition root */
  className: _propTypes["default"].string,

  /* Child components for css transition */
  children: _propTypes["default"].oneOfType([_propTypes["default"].node, _propTypes["default"].arrayOf(_propTypes["default"].node)]),

  /* timeout between animations */
  timeout: _propTypes["default"].oneOfType([_propTypes["default"].shape({
    enter: _propTypes["default"].number.isRequired,
    exit: _propTypes["default"].number.isRequired
  }), _propTypes["default"].number])
};
var _default = TransitionProvider;
exports["default"] = _default;
