// @guestyci/foundation/esm/Tip/InlineTip.js
import React, { forwardRef } from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import createStyles from "../createStyles";

var TipIcon = function TipIcon(props) {
  return /*#__PURE__*/React.createElement("svg", props, /*#__PURE__*/React.createElement("g", {
    transform: "translate(2 2)",
    fill: "none",
    fillRule: "evenodd"
  }, /*#__PURE__*/React.createElement("circle", {
    cx: "15.5",
    cy: "15.464",
    r: "6.5",
    fill: "#FEC802"
  }), /*#__PURE__*/React.createElement("path", {
    fill: "#344773",
    d: "M15.232 6.4c2.432 0 4.736.896 6.4 2.56 1.664 1.664 2.688 3.968 2.688 6.4 0 2.688-1.152 5.248-3.328 6.912-1.152.896-1.792 2.304-1.792 3.584V28.8c0 1.024-.896 1.92-1.92 1.92h-3.84c-.896 0-1.92-.768-1.92-2.304v-2.688c0-1.28-.64-2.432-1.664-3.328C7.552 20.608 6.4 18.048 6.4 15.232c.128-4.864 4.096-8.704 8.832-8.832zm2.688 20.479H12.8v1.595c.003.297.043.966.64.966h3.84c.384 0 .64-.256.64-.64v-1.921zm6.4-3.327l1.792 1.792a.619.619 0 0 1 0 .896c0 .128-.128.128-.384.128-.128 0-.384 0-.512-.128l-1.792-1.792c-.256-.256-.256-.64 0-.896s.64-.256.896 0zm-17.152 0a.619.619 0 0 1 0 .896L5.376 26.24c-.128.128-.256.128-.384.128s-.384 0-.512-.128a.619.619 0 0 1 0-.896l1.792-1.792a.619.619 0 0 1 .896 0zM15.36 7.68c-4.096 0-7.552 3.456-7.68 7.552 0 2.432 1.024 4.736 2.944 6.144 1.372 1.122 2.136 2.609 2.174 4.223h5.129c.086-1.589.961-3.258 2.297-4.351 1.792-1.408 2.816-3.584 2.816-5.888 0-2.048-.768-3.968-2.304-5.504-1.408-1.408-3.328-2.176-5.376-2.176zm14.72 7.04c.384 0 .64.256.64.64s-.256.64-.64.64h-2.56c-.384 0-.64-.256-.64-.64s.256-.64.64-.64h2.56zm-26.88 0c.384 0 .64.256.64.64s-.256.64-.64.64H.64c-.384 0-.64-.256-.64-.64s.256-.64.64-.64H3.2zM15.36 9.6c3.2 0 5.76 2.56 5.76 5.76 0 .384-.256.64-.64.64s-.64-.256-.64-.64c0-2.432-2.048-4.48-4.48-4.48-.384 0-.64-.256-.64-.64s.256-.64.64-.64zm10.752-4.992a.619.619 0 0 1 0 .896L24.32 7.296s-.256.128-.384.128-.384 0-.512-.128a.619.619 0 0 1 0-.896l1.792-1.792a.619.619 0 0 1 .896 0zM5.376 4.48l1.792 1.792a.619.619 0 0 1 0 .896c-.128.128-.256.256-.384.256s-.384 0-.512-.128L4.48 5.376c-.256-.256-.256-.64 0-.896s.64-.256.896 0zM15.36 0c.384 0 .64.256.64.64V3.2c0 .384-.256.64-.64.64s-.64-.256-.64-.64V.64c0-.384.256-.64.64-.64z"
  })));
};

TipIcon.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  width: "35",
  height: "35",
  viewBox: "0 0 35 35"
};
import Row from "../Layout/Row";
import Icon from "../Icon";
import TextField from "../TextField";
export var useStyles = createStyles(function (theme) {
  return {
    root: {
      backgroundColor: theme.palette.alert.background
    },
    content: {
      padding: '12px'
    },
    icon: {
      margin: '8px 0 0 8px'
    }
  };
}, {
  name: 'InlineTip'
});
var InlineTip = forwardRef(function Tip(_ref, ref) {
  var className = _ref.className,
      children = _ref.children,
      _ref$jss = _ref.jss,
      jss = _ref$jss === void 0 ? {} : _ref$jss;
  var classes = useStyles({
    jss: jss
  });
  return /*#__PURE__*/React.createElement(Row, {
    className: cn(classes.root, className),
    ref: ref
  }, /*#__PURE__*/React.createElement(Icon, {
    svg: TipIcon,
    width: 24,
    height: 24,
    className: classes.icon
  }), /*#__PURE__*/React.createElement(TextField, {
    className: classes.content
  }, children));
});
InlineTip.displayName = 'InlineTip';
InlineTip.propTypes = {
  /** root level class name to add */
  className: PropTypes.string,

  /** text node element */
  children: PropTypes.node.isRequired,

  /** jss override object to customize the jss classes */
  jss: PropTypes.oneOfType([PropTypes.shape(), PropTypes.func])
};
export default InlineTip;
