// @guestyci/foundation/es/Listing/ConfirmedBadge.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _objectWithoutPropertiesLoose from "@babel/runtime/helpers/esm/objectWithoutPropertiesLoose";
import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

var BtnApprove = function BtnApprove(props) {
  return /*#__PURE__*/React.createElement("svg", props, /*#__PURE__*/React.createElement("path", {
    d: "M5.745 11.282L.935 6.933a.546.546 0 0 0-.766 0 .557.557 0 0 0 0 .799l5.191 4.881c.097.095.237.15.385.15a.543.543 0 0 0 .35-.122L15.468.944a.559.559 0 0 0 0-.8.587.587 0 0 0-.727-.034L5.745 11.282z"
  }));
};

BtnApprove.defaultProps = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 15.636672 12.7622719"
};
import createStyles from "../createStyles";
const useStyles = createStyles(theme => ({
  root: {
    height: 20,
    width: 20,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    fill: theme.palette.success.default,
    backgroundColor: 'white',
    borderRadius: '50%',
    border: `1.5px solid ${theme.palette.success.default}`
  }
}), {
  name: 'ConfirmedBadge'
});

const ConfirmedBadge = (_ref) => {
  let {
    className = '',
    style = {},
    jss = {}
  } = _ref,
      props = _objectWithoutPropertiesLoose(_ref, ["className", "style", "jss"]);

  const classes = useStyles({
    jss
  });
  return /*#__PURE__*/React.createElement("div", _extends({}, props, {
    className: cn(classes.root, className),
    style: style,
    "data-qa": "confirmed-badge"
  }), /*#__PURE__*/React.createElement(BtnApprove, {
    strokeWidth: 2,
    height: 8,
    width: 9
  }));
};

ConfirmedBadge.propTypes = {
  /** Additional class for badge */
  className: PropTypes.string,

  /** Additional style for badge */
  style: PropTypes.shape(),

  /** jss override object to customize the jss classes */
  jss: PropTypes.oneOfType([PropTypes.shape(), PropTypes.func])
};
export default ConfirmedBadge;
